diff --git a/lib/Mac/PropertyList.pm b/lib/Mac/PropertyList.pm
index a948cf6..50f2e07 100644
--- a/lib/Mac/PropertyList.pm
+++ b/lib/Mac/PropertyList.pm
@@ -194,6 +194,7 @@ sub parse_plist {
 
 	my $plist = do {
 		if( $text =~ /\A<\?xml/ ) { # XML plists
+			$text =~ s/<!--(?:[\d\D]*?)-->//g;
 			# we can handle either 0.9 or 1.0
 			$text =~ s|^<\?xml.*?>\s*<!DOC.*>\s*<plist.*?>\s*||;
 			$text =~ s|\s*</plist>\s*$||;
