diff -ru Math-Pari-2.0305_01080604\utils\Math\PariBuild.pm Math-Pari-2.0305_01080604.patched\utils\Math\PariBuild.pm
--- Math-Pari-2.0305_01080604\utils\Math\PariBuild.pm	Wed Mar 03 22:51:26 2010
+++ Math-Pari-2.0305_01080604.patched\utils\Math\PariBuild.pm	Fri Mar 05 05:55:53 2010
@@ -487,6 +487,8 @@
 			      'patches/diff_2.1.7_div',
 			      'patches/diff_2.1.6_align_power_of_2',
 			      'patches/diff_2.1.7_restart'],
+		 '2.3.5' =>  [
+			($^O eq 'MSWin32' ? 'patches/diff_2.3.5_mingw-w64' : ())], 
 		);
   print "Looking for patches for $v...\n";
   my @p = $patches{$v} ? @{$patches{$v}} : ();
