Description: spelling fix
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-07-14

--- a/lib/Finance/QuoteHist/Generic.pm
+++ b/lib/Finance/QuoteHist/Generic.pm
@@ -1327,7 +1327,7 @@
 many quotes are returned for each query. See Finance::QuoteHist::Yahoo
 for an example of how to do this.
 
-For more complicated sites, such as Yahoo, overriding additonal methods
+For more complicated sites, such as Yahoo, overriding additional methods
 might be necessary for dealing with things such as splits and dividends.
 
 =head1 METHODS
