--- lib/Kwiki/ModPerl.pm	2004-08-20 09:49:13.000000000 -0500
+++ /usr/lib/perl5/site_perl/5.8.4/Kwiki/ModPerl.pm	2004-10-11 10:22:37.000000000 -0500
@@ -22,7 +22,10 @@
     return $hub;
 }
 
-sub handler ($$) {
+BEGIN {
+
+    my $handler_code = sprintf <<'EOF', $mod_perl::VERSION >= 1.99 ? ': method' : '($$)';
+sub handler %s {
     my ($self, $r) = @_;
 
     # only handle the directory specified in the apache config.
@@ -67,6 +70,9 @@
 
     return OK;          
 }
+EOF
+    eval $handler_code;
+}    
 
 1;
 
