--- Tcl-1.05-UjbRn3/Makefile.PL	2016-06-29 00:48:16.000000000 +0800
+++ Tcl-1.05-SJ/Makefile.PL	2018-06-22 22:50:36.943227700 +0800
@@ -45,7 +45,8 @@
     $arch = "$^O-ia64" if ($Config{archname} =~ /ia64/i);
     $arch = "$^O-x86_64" if ($Config{archname} =~ /x86_64/);
 } elsif ($^O eq "cygwin") {
-    $tclconfig = '/usr/lib/tclConfig.sh';
+    $arch = "$^O-i686" if ($Config{archname} =~ /i\d86/);
+    $arch = "$^O-x86_64" if ($Config{archname} =~ /x86_64/);
 }
 
 sub _die ($) {
