Description: remove duplicate NAME sections from POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2018-07-27
Forwarded: 
Bug: 

--- a/lib/Net/Amazon/S3/Signature.pm
+++ b/lib/Net/Amazon/S3/Signature.pm
@@ -33,16 +33,12 @@
 
 =head1 NAME
 
-Net::Amazon::S3::Signature
+Net::Amazon::S3::Signature - S3 Signature implementation base class
 
 =head1 VERSION
 
 version 0.84
 
-=head1 NAME
-
-Net::Amazon::S3::Signature - S3 Signature implementation base class
-
 =head1 METHODS
 
 =over
--- a/lib/Net/Amazon/S3/Signature/V4Implementation.pm
+++ b/lib/Net/Amazon/S3/Signature/V4Implementation.pm
@@ -311,7 +311,7 @@
 
 =head1 NAME
 
-Net::Amazon::S3::Signature::V4Implementation
+Net::Amazon::Signature::V4 - Implements the Amazon Web Services signature version 4, AWS4-HMAC-SHA256
 
 =head1 VERSION
 
@@ -330,10 +330,6 @@
 
 The primary purpose of this module is to be used by Net::Amazon::Glacier.
 
-=head1 NAME
-
-Net::Amazon::Signature::V4 - Implements the Amazon Web Services signature version 4, AWS4-HMAC-SHA256
-
 =head1 VERSION
 
 Version 0.19
