--- t/readonly.old	2008-07-22 15:26:31.000000000 -0700
+++ t/readonly.t	2008-07-22 15:40:15.000000000 -0700
@@ -3,7 +3,7 @@
 # Test the Readonly function
 
 use strict;
-use Test::More tests => 19;
+use Test::More tests => 21;
 
 # Find the module (1 test)
 BEGIN {use_ok('Readonly'); }
@@ -44,7 +44,7 @@
 
 SKIP:
 {
-	skip 'Readonly $@% syntax is for perl 5.8 or later', 9  unless $] >= 5.008;
+	skip 'Readonly $@% syntax is for perl 5.8 or later', 11  unless $] >= 5.008;
 
 	eval q{Readonly my $ros => 45};
 	is $@ => '', 'Create scalar';
@@ -52,6 +52,12 @@
 	eval q{Readonly my $ros2 => 45;  $ros2 = 45};
 	like $@ => $expected, 'Modify scalar';
 
+    eval q{Readonly my $roaref => [1, 2, 3, 4]; $roaref->[2] = 3};
+    like $@ => $expected, 'Modify scalar array reference';
+
+    eval q{Readonly my $rohref => { key1 => "value", key2 => "value2" }; $rohref->{key1} = "value"};
+    like $@ => $expected, 'Modify scalar hash reference';
+
 	eval q{Readonly my @roa => (1, 2, 3, 4)};
 	is $@ => '', 'Create array';
 
