Description: fix POD whatis entry
Author: Jonathan Yu <jawnsy@cpan.org>
Origin: vendor
Forwarded: no
--- a/lib/POE/Component/Connection/Keepalive.pm
+++ b/lib/POE/Component/Connection/Keepalive.pm
@@ -99,12 +99,12 @@
 
 =head1 NAME
 
+POE::Component::Connection::Keepalive - a wheel wrapper around a
+kept-alive socket
 
 =head1 VERSION
 
 version 0.267
-POE::Component::Connection::Keepalive - a wheel wrapper around a
-kept-alive socket
 
 =head1 SYNOPSIS
 
@@ -210,4 +210,4 @@
 
 Special thanks to Rob Bloodgood.
 
-=cut
\ No newline at end of file
+=cut
