--- write.t.dist	Fri Jun 25 11:30:31 2004
+++ write.t	Wed Sep 13 19:14:03 2006
@@ -63,7 +63,7 @@
   unlink 'foo' or die "unlink 'foo' failed: $!";
 
  SKIP: {
-    skip 6, "Your configured shell, '$Config{sh}', is missing or size 0"
+    skip "Your configured shell, '$Config{sh}', is missing or size 0",7
       unless defined $sh;
     ok ((open GZ, ">$buffering:gzip", 'foo'),
         sprintf "open >$buffering:gzip [about to write %d bytes]", length $sh);
@@ -75,9 +75,9 @@
         "compare compressed copy with '$Config{sh}'");
     ok (eof (FOO), 'should be end of file');
     ok ((close GZ), 'close it');
+    unlink 'foo' or die "unlink 'foo' failed: $!";
   }
 
-  unlink 'foo' or die "unlink 'foo' failed: $!";
 
   ok ((open FOO, ">$buffering:gzip(lazy)", "empty"),
       "open empty as >$buffering:gzip(lazy)");
