diff --git a/lib/Date/Holidays/PT.pm b/lib/Date/Holidays/PT.pm
index 0aa0efc..549835a 100644
--- a/lib/Date/Holidays/PT.pm
+++ b/lib/Date/Holidays/PT.pm
@@ -6,7 +6,6 @@ use strict;
 use Date::Holidays::Super;
 use Date::Easter;
 use Time::JulianDay;
-use utf8;
 
 my @ISA = qw(Date::Holidays::Super);
 
@@ -175,8 +174,6 @@ sub holidays {
   return \%holidays;
 }
 
-=encoding utf8
-
 =head1 NATIONAL HOLIDAYS
 
 The following Portuguese holidays have fixed dates:
