Index: Maketext.pm
===================================================================
--- Maketext.pm	(revision 37506)
+++ Maketext.pm	(working copy)
@@ -510,16 +510,10 @@
    # This big scary routine compiles an entry.
    # It returns either a coderef if there's brackety bits in this, or
    #  otherwise a ref to a scalar.
-
-    my $target = ref($_[0]) || $_[0];
-
-    my(@code);
-    my(@c) = (''); # "chunks" -- scratch.
-    my $call_count = 0;
-    my $big_pile = '';
+    my ($target,$call_count,$big_pile,@c,@code) = ( (ref($_[0]) || $_[0]), 0, '', '' );
    {
-        my $in_group = 0; # start out outside a group
-        my($m, @params); # scratch
+                                          # start out outside a group
+        my($in_group, $m, @params) = (0); # scratch

        while($_[1] =~  # Iterate over chunks.
            m/\G(
