diff -ur backup-1.002004/lib/Net/XMPP/Protocol.pm callbackIQ/lib/Net/XMPP/Protocol.pm
--- backup-1.002004/lib/Net/XMPP/Protocol.pm	2011-07-19 19:58:40.000000000 +0300
+++ callbackIQ/lib/Net/XMPP/Protocol.pm	2012-01-28 01:14:50.000000000 +0200
@@ -3530,7 +3530,7 @@
     $self->{DEBUG}->Log1("callbackIQ: type($type) ns($ns)");
 
     if (exists($self->{CB}->{IQns}->{$ns})
-        && (ref($self->{CB}->{IQns}->{$ns}) eq 'HASH' )
+        && (ref($self->{CB}->{IQns}->{$ns}) eq 'CODE' )
         )
     {
         $self->{DEBUG}->Log1("callbackIQ: goto user function( $self->{CB}->{IQns}->{$ns} )");
