Description: fix some spelling mistakes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-11-16

--- a/exiftool
+++ b/exiftool
@@ -4870,7 +4870,7 @@
 
 =item exiftool '-ThumbnailImageE<lt>=thumb.jpg' dst.jpg
 
-Set the thumbnail image from specified file (Note: The quotes are neccessary
+Set the thumbnail image from specified file (Note: The quotes are necessary
 to prevent shell redirection).
 
 =item exiftool '-JpgFromRawE<lt>=%d%f_JFR.JPG' -ext CRW -r .
--- a/html/exiftool_pod.html
+++ b/html/exiftool_pod.html
@@ -1739,7 +1739,7 @@
 <dt><strong><a name="exiftool_thumbnailimage_thumb_jpg_dst_jpg" class="item">exiftool '-ThumbnailImage&lt;=thumb.jpg' dst.jpg</a></strong></dt>
 
 <dd>
-<p>Set the thumbnail image from specified file (Note: The quotes are neccessary
+<p>Set the thumbnail image from specified file (Note: The quotes are necessary
 to prevent shell redirection).</p>
 </dd>
 <dt><strong><a name="exiftool_jpgfromraw_d_f_jfr_jpg_ext_crw_r" class="item">exiftool '-JpgFromRaw&lt;=%d%f_JFR.JPG' -ext CRW -r .</a></strong></dt>
--- a/lib/Image/ExifTool/SonyIDC.pm
+++ b/lib/Image/ExifTool/SonyIDC.pm
@@ -312,7 +312,7 @@
 =head1 DESCRIPTION
 
 This module contains definitions required by Image::ExifTool to read and
-write Sony Image Data Converter verson 3.0 metadata in ARW images.
+write Sony Image Data Converter version 3.0 metadata in ARW images.
 
 =head1 AUTHOR
 
--- a/lib/Image/ExifTool.pod
+++ b/lib/Image/ExifTool.pod
@@ -552,7 +552,7 @@
 Exclude specified tags from tags extracted from an image.  The option value
 is either a tag name or reference to a list of tag names to exclude.  The
 case of tag names is not significant.  This option is ignored for
-specifically requested tags.  Tags may also be excluded by preceeding their
+specifically requested tags.  Tags may also be excluded by preceding their
 name with a '-' in the arguments to L</ImageInfo>.
 
 =item ExtractEmbedded
--- a/lib/Image/ExifTool/Sony.pm
+++ b/lib/Image/ExifTool/Sony.pm
@@ -4151,7 +4151,7 @@
     #  Mask   {  Bits     Name    Bits     Name  } Prefix flag
     # ------    ------    -----  ------    -----   -----------
     [ 0x4000, { 0x4000 => 'PZ'                   }, 1 ],
-    [ 0x0300, { 0x0100 => 'DT',  0x0300 => 'E'   }, 1 ], # (will come before preceeding prefix)
+    [ 0x0300, { 0x0100 => 'DT',  0x0300 => 'E'   }, 1 ], # (will come before preceding prefix)
     [ 0x000c, { 0x0004 => 'ZA',  0x0008 => 'G'   } ],
     [ 0x00e0, { 0x0020 => 'STF', 0x0040 => 'Reflex', 0x0060 => 'Macro', 0x0080 => 'Fisheye' } ],
     [ 0x0003, { 0x0001 => 'SSM', 0x0002 => 'SAM' } ],
--- a/lib/Image/ExifTool/MIE.pm
+++ b/lib/Image/ExifTool/MIE.pm
@@ -1860,7 +1860,7 @@
  11) Compressed meta information supported.
  12) Relocatable data elements (ie. no fixed offsets).
  13) Binary meta information (+7) with variable byte order (+3).
- 14) Mandatory tags not required (an unecessary complication).
+ 14) Mandatory tags not required (an unnecessary complication).
  15) Append information to end of file without editing.
 
                           Feature number                   Total
