From 1547e9b2bf4f9a001a35552136bbcf2f33c9ea72 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Torsten=20Sch=C3=B6nfeld?= <kaffeetisch@gmx.de>
Date: Thu, 3 Oct 2013 21:49:44 +0200
Subject: [PATCH] Fix build: ensure existence of blib/lib/Gnome2 for enums.pod

---
 Makefile.PL | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 8ef6e19..fdfdd12 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -138,7 +138,7 @@ sub MY::postamble {
 			'PERL_GNOME_VFS' => $build_reqs{'perl-Gnome2-VFS'},
 		)
           . "
-\$(INST_LIB)/\$(FULLEXT)/enums.pod : \$(BLIB_DONE) maps podifyenums.pl
+\$(INST_LIB)/\$(FULLEXT)/enums.pod : \$(BLIB_DONE) \$(INST_LIB)/\$(FULLEXT)/ maps podifyenums.pl
 	\$(PERLRUNINST) -M\$(NAME) podifyenums.pl \$(NAME) maps > \$@
 ";
 }
-- 
1.8.1.2

