Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitBooleanGrep.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitBooleanGrep.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitBooleanGrep.pm	(working copy)
@@ -29,7 +29,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_LOW          }
-sub default_themes       { return qw( core pbp performance ) }
+sub default_themes       { return qw( cert core pbp performance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringyEval.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringyEval.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringyEval.pm	(working copy)
@@ -38,7 +38,7 @@
     );
 }
 sub default_severity     { return $SEVERITY_HIGHEST   }
-sub default_themes       { return qw( core pbp bugs ) }
+sub default_themes       { return qw( cert core pbp bugs ) }
 sub applies_to           { return 'PPI::Token::Word'  }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringySplit.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringySplit.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitStringySplit.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                    }
 sub default_severity     { return $SEVERITY_LOW         }
-sub default_themes       { return qw(core pbp cosmetic) }
+sub default_themes       { return qw(cert core pbp cosmetic) }
 sub applies_to           { return 'PPI::Token::Word'    }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalCan.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalCan.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalCan.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalIsa.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalIsa.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/BuiltinFunctions/ProhibitUniversalIsa.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ClassHierarchies/ProhibitExplicitISA.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ClassHierarchies/ProhibitExplicitISA.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ClassHierarchies/ProhibitExplicitISA.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                         }
 sub default_severity     { return $SEVERITY_MEDIUM           }
-sub default_themes       { return qw( core maintenance pbp ) }
+sub default_themes       { return qw( cert core maintenance pbp ) }
 sub applies_to           { return 'PPI::Token::Symbol'       }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitMutatingListFunctions.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitMutatingListFunctions.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitMutatingListFunctions.pm	(working copy)
@@ -76,7 +76,7 @@
 }
 
 sub default_severity { return $SEVERITY_HIGHEST  }
-sub default_themes   { return qw(core bugs pbp)  }
+sub default_themes   { return qw(cert core bugs pbp)  }
 sub applies_to       { return 'PPI::Token::Word' }
 
 #-----------------------------------------------------------------------------
@@ -352,4 +352,3 @@
 #   indent-tabs-mode: nil
 # End:
 # ex: set ts=8 sts=4 sw=4 tw=78 ft=perl expandtab shiftround :
-
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitUnreachableCode.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitUnreachableCode.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ControlStructures/ProhibitUnreachableCode.pm	(working copy)
@@ -35,7 +35,7 @@
 
 sub supported_parameters { return ()                 }
 sub default_severity     { return $SEVERITY_HIGH     }
-sub default_themes       { return qw( core bugs )    }
+sub default_themes       { return qw( cert core bugs )    }
 sub applies_to           { return 'PPI::Token::Word' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ErrorHandling/RequireCarping.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ErrorHandling/RequireCarping.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ErrorHandling/RequireCarping.pm	(working copy)
@@ -44,7 +44,7 @@
 }
 
 sub default_severity  { return $SEVERITY_MEDIUM                          }
-sub default_themes    { return qw( core pbp maintenance )                }
+sub default_themes    { return qw( cert core pbp maintenance )                }
 sub applies_to        { return 'PPI::Token::Word'                        }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitBarewordFileHandles.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitBarewordFileHandles.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitBarewordFileHandles.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                  }
 sub default_severity     { return $SEVERITY_HIGHEST   }
-sub default_themes       { return qw( core pbp bugs ) }
+sub default_themes       { return qw( cert core pbp bugs ) }
 sub applies_to           { return 'PPI::Token::Word'  }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitInteractiveTest.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitInteractiveTest.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitInteractiveTest.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_HIGHEST      }
-sub default_themes       { return qw( core pbp bugs )    }
+sub default_themes       { return qw( cert core pbp bugs )    }
 sub applies_to           { return 'PPI::Token::Operator' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitOneArgSelect.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitOneArgSelect.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitOneArgSelect.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                  }
 sub default_severity     { return $SEVERITY_HIGH      }
-sub default_themes       { return qw( core bugs pbp ) }
+sub default_themes       { return qw( cert core bugs pbp ) }
 sub applies_to           { return 'PPI::Token::Word'  }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitTwoArgOpen.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitTwoArgOpen.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/ProhibitTwoArgOpen.pm	(working copy)
@@ -33,7 +33,7 @@
 
 sub supported_parameters { return ()                         }
 sub default_severity     { return $SEVERITY_HIGHEST          }
-sub default_themes       { return qw(core pbp bugs security) }
+sub default_themes       { return qw(cert core pbp bugs security) }
 sub applies_to           { return 'PPI::Token::Word'         }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedClose.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedClose.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedClose.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_LOW          }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedOpen.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedOpen.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedOpen.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedSyscalls.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedSyscalls.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/InputOutput/RequireCheckedSyscalls.pm	(working copy)
@@ -61,7 +61,7 @@
 }
 
 sub default_severity     { return $SEVERITY_LOWEST       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Miscellanea/ProhibitFormats.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Miscellanea/ProhibitFormats.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Miscellanea/ProhibitFormats.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                         }
 sub default_severity     { return $SEVERITY_MEDIUM           }
-sub default_themes       { return qw( core maintenance pbp ) }
+sub default_themes       { return qw( cert core maintenance pbp ) }
 sub applies_to           { return 'PPI::Token::Word'         }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/ProhibitEvilModules.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/ProhibitEvilModules.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/ProhibitEvilModules.pm	(working copy)
@@ -100,7 +100,7 @@
 }
 
 sub default_severity  { return $SEVERITY_HIGHEST         }
-sub default_themes    { return qw( core bugs )           }
+sub default_themes    { return qw( cert core bugs )           }
 sub applies_to        { return 'PPI::Statement::Include' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/RequireEndWithOne.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/RequireEndWithOne.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Modules/RequireEndWithOne.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                  }
 sub default_severity     { return $SEVERITY_HIGH      }
-sub default_themes       { return qw( core bugs pbp ) }
+sub default_themes       { return qw( cert core bugs pbp ) }
 sub applies_to           { return 'PPI::Document'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Objects/ProhibitIndirectSyntax.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Objects/ProhibitIndirectSyntax.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Objects/ProhibitIndirectSyntax.pm	(working copy)
@@ -45,7 +45,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGH             }
-sub default_themes       { return qw( core pbp maintenance ) }
+sub default_themes       { return qw( cert core pbp maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'         }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/RegularExpressions/ProhibitCaptureWithoutTest.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/RegularExpressions/ProhibitCaptureWithoutTest.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/RegularExpressions/ProhibitCaptureWithoutTest.pm	(working copy)
@@ -38,7 +38,7 @@
     );
 }
 sub default_severity     { return $SEVERITY_MEDIUM         }
-sub default_themes       { return qw(core pbp maintenance) }
+sub default_themes       { return qw(cert core pbp maintenance) }
 sub applies_to           { return 'PPI::Token::Magic'      }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitBuiltinHomonyms.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitBuiltinHomonyms.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitBuiltinHomonyms.pm	(working copy)
@@ -30,7 +30,7 @@
 
 sub supported_parameters { return ()                    }
 sub default_severity     { return $SEVERITY_HIGH        }
-sub default_themes       { return qw( core bugs pbp )   }
+sub default_themes       { return qw( cert core bugs pbp )   }
 sub applies_to           { return 'PPI::Statement::Sub' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitExplicitReturnUndef.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitExplicitReturnUndef.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitExplicitReturnUndef.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                 }
 sub default_severity     { return $SEVERITY_HIGHEST  }
-sub default_themes       { return qw(core pbp bugs)  }
+sub default_themes       { return qw(cert core pbp bugs)  }
 sub applies_to           { return 'PPI::Token::Word' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitReturnSort.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitReturnSort.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitReturnSort.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                 }
 sub default_severity     { return $SEVERITY_HIGHEST  }
-sub default_themes       { return qw(core bugs)      }
+sub default_themes       { return qw(cert core bugs)      }
 sub applies_to           { return 'PPI::Token::Word' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitSubroutinePrototypes.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitSubroutinePrototypes.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitSubroutinePrototypes.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                    }
 sub default_severity     { return $SEVERITY_HIGHEST     }
-sub default_themes       { return qw(core pbp bugs)     }
+sub default_themes       { return qw(cert core pbp bugs)     }
 sub applies_to           { return 'PPI::Statement::Sub' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitUnusedPrivateSubroutines.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitUnusedPrivateSubroutines.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProhibitUnusedPrivateSubroutines.pm	(working copy)
@@ -53,7 +53,7 @@
 }
 
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Statement::Sub'  }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProtectPrivateSubs.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProtectPrivateSubs.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/ProtectPrivateSubs.pm	(working copy)
@@ -91,7 +91,7 @@
 }
 
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw( core maintenance ) }
+sub default_themes       { return qw( cert core maintenance ) }
 sub applies_to           { return 'PPI::Token::Word'     }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/RequireFinalReturn.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/RequireFinalReturn.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Subroutines/RequireFinalReturn.pm	(working copy)
@@ -40,7 +40,7 @@
 }
 
 sub default_severity { return $SEVERITY_HIGH        }
-sub default_themes   { return qw( core bugs pbp )   }
+sub default_themes   { return qw( cert core bugs pbp )   }
 sub applies_to       { return 'PPI::Statement::Sub' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoStrict.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoStrict.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoStrict.pm	(working copy)
@@ -38,7 +38,7 @@
 }
 
 sub default_severity { return $SEVERITY_HIGHEST         }
-sub default_themes   { return qw( core pbp bugs )       }
+sub default_themes   { return qw( cert core pbp bugs )       }
 sub applies_to       { return 'PPI::Statement::Include' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoWarnings.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoWarnings.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitNoWarnings.pm	(working copy)
@@ -46,7 +46,7 @@
 }
 
 sub default_severity { return $SEVERITY_HIGH            }
-sub default_themes   { return qw( core bugs pbp )       }
+sub default_themes   { return qw( cert core bugs pbp )       }
 sub applies_to       { return 'PPI::Statement::Include' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitProlongedStrictureOverride.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitProlongedStrictureOverride.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/ProhibitProlongedStrictureOverride.pm	(working copy)
@@ -37,7 +37,7 @@
 }
 
 sub default_severity { return $SEVERITY_HIGH            }
-sub default_themes   { return qw( core pbp bugs )       }
+sub default_themes   { return qw( cert core pbp bugs )       }
 sub applies_to       { return 'PPI::Statement::Include' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseStrict.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseStrict.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseStrict.pm	(working copy)
@@ -44,7 +44,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGHEST   }
-sub default_themes       { return qw( core pbp bugs ) }
+sub default_themes       { return qw( cert core pbp bugs ) }
 sub applies_to           { return 'PPI::Document'     }
 
 sub default_maximum_violations_per_document { return 1; }
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseWarnings.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseWarnings.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/TestingAndDebugging/RequireUseWarnings.pm	(working copy)
@@ -44,7 +44,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGH      }
-sub default_themes       { return qw( core pbp bugs ) }
+sub default_themes       { return qw( cert core pbp bugs ) }
 sub applies_to           { return 'PPI::Document'     }
 
 sub default_maximum_violations_per_document { return 1; }
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitCommaSeparatedStatements.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitCommaSeparatedStatements.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitCommaSeparatedStatements.pm	(working copy)
@@ -39,7 +39,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGH      }
-sub default_themes       { return qw( core bugs pbp ) }
+sub default_themes       { return qw( cert core bugs pbp ) }
 sub applies_to           { return 'PPI::Statement'    }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitLeadingZeros.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitLeadingZeros.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitLeadingZeros.pm	(working copy)
@@ -38,7 +38,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGHEST           }
-sub default_themes       { return qw< core pbp bugs >         }
+sub default_themes       { return qw< cert core pbp bugs >         }
 sub applies_to           { return 'PPI::Token::Number::Octal' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMagicNumbers.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMagicNumbers.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMagicNumbers.pm	(working copy)
@@ -91,7 +91,7 @@
 }
 
 sub default_severity { return $SEVERITY_LOW          }
-sub default_themes   { return qw( core maintenance ) }
+sub default_themes   { return qw( cert core maintenance ) }
 sub applies_to       { return 'PPI::Token::Number'   }
 
 sub default_maximum_violations_per_document { return 10; }
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMismatchedOperators.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMismatchedOperators.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMismatchedOperators.pm	(working copy)
@@ -50,7 +50,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw< core bugs >        }
+sub default_themes       { return qw< cert core bugs >        }
 sub applies_to           { return 'PPI::Token::Operator' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMixedBooleanOperators.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMixedBooleanOperators.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/ValuesAndExpressions/ProhibitMixedBooleanOperators.pm	(working copy)
@@ -47,7 +47,7 @@
 
 sub supported_parameters { return ()                  }
 sub default_severity     { return $SEVERITY_HIGH      }
-sub default_themes       { return qw( core bugs pbp ) }
+sub default_themes       { return qw( cert core bugs pbp ) }
 sub applies_to           { return 'PPI::Statement'    }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitPerl4PackageNames.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitPerl4PackageNames.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitPerl4PackageNames.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                                        }
 sub default_severity     { return $SEVERITY_LOW                             }
-sub default_themes       { return qw(core maintenance)                      }
+sub default_themes       { return qw(cert core maintenance)                      }
 sub applies_to           { return qw( PPI::Token::Word PPI::Token::Symbol ) }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitUnusedVariables.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitUnusedVariables.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProhibitUnusedVariables.pm	(working copy)
@@ -30,7 +30,7 @@
 
 sub supported_parameters { return ()                     }
 sub default_severity     { return $SEVERITY_MEDIUM       }
-sub default_themes       { return qw< core maintenance > }
+sub default_themes       { return qw< cert core maintenance > }
 sub applies_to           { return qw< PPI::Document >    }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProtectPrivateVars.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProtectPrivateVars.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/ProtectPrivateVars.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                   }
 sub default_severity     { return $SEVERITY_MEDIUM     }
-sub default_themes       { return qw(core maintenance) }
+sub default_themes       { return qw(cert core maintenance) }
 sub applies_to           { return 'PPI::Token::Symbol' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireInitializationForLocalVars.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireInitializationForLocalVars.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireInitializationForLocalVars.pm	(working copy)
@@ -26,7 +26,7 @@
 
 sub supported_parameters { return ()                         }
 sub default_severity     { return $SEVERITY_MEDIUM           }
-sub default_themes       { return qw(core pbp bugs)          }
+sub default_themes       { return qw(cert core pbp bugs)          }
 sub applies_to           { return 'PPI::Statement::Variable' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLexicalLoopIterators.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLexicalLoopIterators.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLexicalLoopIterators.pm	(working copy)
@@ -29,7 +29,7 @@
 
 sub supported_parameters { return ()                         }
 sub default_severity     { return $SEVERITY_HIGHEST          }
-sub default_themes       { return qw(core pbp bugs)          }
+sub default_themes       { return qw(cert core pbp bugs)          }
 sub applies_to           { return 'PPI::Statement::Compound' }
 
 #-----------------------------------------------------------------------------
Index: distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLocalizedPunctuationVars.pm
===================================================================
--- distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLocalizedPunctuationVars.pm	(revision 4167)
+++ distributions/Perl-Critic/lib/Perl/Critic/Policy/Variables/RequireLocalizedPunctuationVars.pm	(working copy)
@@ -44,7 +44,7 @@
 }
 
 sub default_severity     { return $SEVERITY_HIGH             }
-sub default_themes       { return qw(core pbp bugs)          }
+sub default_themes       { return qw(cert core pbp bugs)          }
 sub applies_to           { return 'PPI::Token::Operator'     }
 
 #-----------------------------------------------------------------------------
