--- Zlib.xs	2017-02-16 14:32:42.000000000 +0100
+++ /usr/src/perl/blead/cperl/cpan/Compress-Raw-Zlib/Zlib.xs	2017-02-16 12:28:33.000000000 +0100
@@ -696,14 +696,14 @@
     int ret ;
     z_stream * strm = &s->stream;
 
-    strm->next_in = NULL;
-    strm->avail_in = 0;
-    
     Bytef* output = s->deflateParams_out_buffer ;
     uLong total_output = s->deflateParams_out_length;
 
     uLong have = 0;
 
+    strm->next_in = NULL;
+    strm->avail_in = 0;
+    
     do 
     {
         if (output)
@@ -1133,12 +1133,12 @@
         }
 
         RETVAL = deflate(&(s->stream), Z_NO_FLUSH);
-        if (RETVAL != Z_STREAM_ERROR) {
+        /* if (RETVAL != Z_STREAM_ERROR) {
             int done = increment -  s->stream.avail_out ;
-            /* printf("std DEFLATEr returned %d '%s'  avail in %d, out %d wrote %d\n", RETVAL,
+            printf("std DEFLATEr returned %d '%s'  avail in %d, out %d wrote %d\n", RETVAL,
             GetErrorString(RETVAL), s->stream.avail_in,
-s->stream.avail_out, done); */
-        }
+s->stream.avail_out, done);
+        } */
     
         if (trace) {
             printf("DEFLATE returned %d %s, avail in %d, out %d\n", RETVAL,
@@ -1260,12 +1260,12 @@
         }
 
         RETVAL = deflate(&(s->stream), f);
-        if (RETVAL != Z_STREAM_ERROR) {
+        /* if (RETVAL != Z_STREAM_ERROR) {
             int done = availableout -  s->stream.avail_out ;
-            /* printf("flush DEFLATEr returned %d '%s'  avail in %d, out %d wrote %d\n", RETVAL,
+            printf("flush DEFLATEr returned %d '%s'  avail in %d, out %d wrote %d\n", RETVAL,
             GetErrorString(RETVAL), s->stream.avail_in,
-s->stream.avail_out, done); */
-        }
+s->stream.avail_out, done);
+        } */
     
         if (trace) {
             printf("flush DEFLATE returned %d '%s', avail in %d, out %d\n", RETVAL,
