diff -ur Locale-Maketext-Simple-0.18.original/lib/Locale/Maketext/Simple.pm Locale-Maketext-Simple-0.18/lib/Locale/Maketext/Simple.pm
--- Locale-Maketext-Simple-0.18.original/lib/Locale/Maketext/Simple.pm	2006-09-08 14:28:01.000000000 +0000
+++ Locale-Maketext-Simple-0.18/lib/Locale/Maketext/Simple.pm	2009-07-05 07:57:26.569687333 +0000
@@ -46,6 +46,10 @@
 designed to alleviate the need of creating I<Language Classes> for
 module authors.
 
+The language used is chosen from the loc_lang call. If a lookup is not
+possible, the i-default language will be used. If the lookup is not in the 
+i-default language, then the key will be returned.
+
 If B<Locale::Maketext::Lexicon> is not present, it implements a
 minimal localization function by simply interpolating C<[_1]> with
 the first argument, C<[_2]> with the second, etc.  Interpolated
@@ -146,13 +150,13 @@
     eval "
 	package $pkg;
 	use base 'Locale::Maketext';
-        %${pkg}::Lexicon = ( '_AUTO' => 1 );
 	Locale::Maketext::Lexicon->import({
 	    'i-default' => [ 'Auto' ],
 	    '*'	=> [ Gettext => \$pattern ],
 	    _decode => \$decode,
 	    _encoding => \$encoding,
 	});
+	*${pkg}::Lexicon = \\%${pkg}::i_default::Lexicon;
 	*tense = sub { \$_[1] . ((\$_[2] eq 'present') ? 'ing' : 'ed') }
 	    unless defined &tense;
 
@@ -193,7 +197,7 @@
 
     return $Loc{$pkg}, sub {
 	$lh = $pkg->get_handle(@_);
-	$lh = $pkg->get_handle(@_);
+	#$lh = $pkg->get_handle(@_);	# Is this necessary?
     };
 }
 
