diff -uNPr Future-0.28-ej0TdH/t/12needs_all.t Future-0.28-H0UgpN/t/12needs_all.t
--- Future-0.28-ej0TdH/t/12needs_all.t	2014-06-08 22:57:28.000000000 +0100
+++ Future-0.28-H0UgpN/t/12needs_all.t	2014-06-24 14:42:16.751997673 +0100
@@ -122,6 +122,20 @@
    is( $c2, undef, '$future->cancel ignores ready subs' );
 }
 
+# cancel detection
+{
+   my $f1 = Future->new;
+   my $f2 = Future->new;
+
+   my $future = Future->needs_all( $f1, $f2 );
+
+   $f1->cancel;
+   $f2->cancel;
+
+   ok( $future->is_ready, 'marked as ready after all subs are cancelled');
+   ok( $future->is_cancelled, 'marked as cancelled after all subs are cancelled');
+}
+
 # needs_all on none
 {
    my $f = Future->needs_all( () );
