--- lib/SVN/Dump/Reader.pm 
+++ lib/SVN/Dump/Reader.pm 
@@ -62,12 +62,10 @@
         <$fh>; # chop the empty line that follows
     }
 
-    # chop empty line after the record
-    my $type = $headers->type();
-    <$fh> if $type !~ /\A(?:format|uuid)\z/;
-
-    # chop another one after a node with only a prop block
-    <$fh> if $type eq 'node' && $record->has_prop_only();
+    # chop off empty lines
+    my $old_pos = tell $fh;
+    while (<$fh> eq $NL) { $old_pos = tell $fh };
+    seek $fh, $old_pos, 0;
 
     # uuid and format record only contain headers
     return $record;
