--- a/lib/JSON/PP.pm
+++ b/lib/JSON/PP.pm
@@ -365,13 +365,6 @@ sub allow_bigint {
         my ($pre, $post) = $indent ? $self->_up_indent() : ('', '');
         my $del = ($space_before ? ' ' : '') . ':' . ($space_after ? ' ' : '');

-        if ( my $tie_class = tied %$obj ) {
-            if ( $tie_class->can('TIEHASH') ) {
-                $tie_class =~ s/=.+$//;
-                tie %res, $tie_class;
-            }
-        }
-
         # In the old Perl verions, tied hashes in bool context didn't work.
         # So, we can't use such a way (%res ? a : b)
         my $has;
@@ -405,13 +398,6 @@ sub allow_bigint {

         my ($pre, $post) = $indent ? $self->_up_indent() : ('', '');

-        if (my $tie_class = tied @$obj) {
-            if ( $tie_class->can('TIEARRAY') ) {
-                $tie_class =~ s/=.+$//;
-                tie @res, $tie_class;
-            }
-        }
-
         for my $v (@$obj){
             push @res, $self->object_to_json($v) || $self->value_to_json($v);
         }