Index: lib/Module/Build/ModuleInfo.pm
===================================================================
--- lib/Module/Build/ModuleInfo.pm	(revision 10110)
+++ lib/Module/Build/ModuleInfo.pm	(working copy)
@@ -313,7 +313,11 @@
     die "failed to build version sub for $self->{filename}";
   my $result = $vsub->();
 
-  # Bless it into our own version class
+  # Bless it into our own version class unless we already have a version 
+  # object
+  if (ref($result) eq 'version') {
+    return $result;
+  }
   $result = Module::Build::Version->new($result);
 
   return $result;
