Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-12-21

--- a/primality.c
+++ b/primality.c
@@ -590,7 +590,7 @@
   if (n < 7) return (n == 2 || n == 3 || n == 5);
   if ((n % 2) == 0 || n == UV_MAX) return 0;
   if (increment < 1 || increment > 256)
-    croak("Invalid lucas paramater increment: %"UVuf"\n", increment);
+    croak("Invalid lucas parameter increment: %"UVuf"\n", increment);
 
   P = 3;
   while (1) {
