From 8b40448ee2877feb026941f10fb465c2f56957c4 Mon Sep 17 00:00:00 2001
From: Jacob Farkas <jacobf@rkas.net>
Date: Fri, 7 Feb 2020 22:50:46 -0800
Subject: [PATCH] Adding includes to fix build on Alpine Docker images

---
 tacpluslib/tac_client.c | 2 ++
 tacpluslib/utils.c      | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/tacpluslib/tac_client.c b/tacpluslib/tac_client.c
index d26c151..9493744 100644
--- a/tacpluslib/tac_client.c
+++ b/tacpluslib/tac_client.c
@@ -11,6 +11,8 @@
 #include <sys/types.h> 
 #include <sys/time.h>  
 #include <unistd.h>  
+#include <fcntl.h>
+#include <sys/file.h>
 #include <time.h>  
 #include"tac_plus.h"
 #include "tacplus_client.h"
diff --git a/tacpluslib/utils.c b/tacpluslib/utils.c
index af39740..150106c 100644
--- a/tacpluslib/utils.c
+++ b/tacpluslib/utils.c
@@ -19,6 +19,8 @@
 
 #include "tac_plus.h"
 #include <stdlib.h>
+#include <fcntl.h>
+#include <sys/file.h>
 
 int
 tac_exit(status)
-- 
2.21.0 (Apple Git-122.2)

