--- lib/CPANPLUS/Internals/Report.pm.orig	2007-08-18 11:38:58.000000000 +0100
+++ lib/CPANPLUS/Internals/Report.pm	2007-09-07 10:04:43.000000000 +0100
@@ -57,7 +57,7 @@
         'LWP::UserAgent' => '0.0',
         'HTTP::Request'  => '0.0',
         URI              => '0.0',
-        YAML             => '0.0',
+        'YAML::Tiny'     => '0.0',
     };
 
     my $send_list = {
@@ -178,7 +178,7 @@
         return;
     }
 
-    my $aref = YAML::Load( $res->content );
+    my ($aref) = YAML::Tiny::Load( $res->content );
 
     my $dist = $mod->package_name .'-'. $mod->package_version;
 
