--- Sys-Statistics-Linux-0.66/lib/Sys/Statistics/Linux/DiskStats.pm	2019-03-21 10:12:36.584677649 +0100
+++ Sys-Statistics-Linux-0.66_patched/lib/Sys/Statistics/Linux/DiskStats.pm	2019-03-21 10:13:20.236575029 +0100
@@ -239,9 +239,9 @@
 
     if (open $fh, '<', $file_diskstats) {
         while (my $line = <$fh>) {
-            #                   --      --      --      F1     F2     F3     F4     F5     F6     F7     F8    F9    F10   F11
-            #                   $1      $2      $3      $4     --     $5     --     $6     --     $7     --    --    --    --
-            if ($line =~ /^\s+(\d+)\s+(\d+)\s+(.+?)\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+\d+\s+\d+\s+\d+$/) {
+	    #                   --      --      --      F1     F2     F3     F4     F5     F6     F7     F8    F9    F10   F11  (F12..F15)?
+            #                   $1      $2      $3      $4     --     $5     --     $6     --     $7     --    --    --    --   (-- .. --)?
+	    if ($line =~ /^\s+(\d+)\s+(\d+)\s+(.+?)\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+\d+\s+\d+\s+\d+(?:(?:\s+\d+){4})?$/) {
                 for my $x ($stats{$3}) { # $3 -> the device name
                     $x->{major}   = $1;
                     $x->{minor}   = $2;
