--- Makefile.PL	2010-05-03 09:57:49.320659544 +0300
+++ Makefile.PL.fixed	2010-05-03 09:59:32.523392723 +0300
@@ -18,7 +18,10 @@
                                                  'ExtUtils::MakeMaker' => '6.31'
                                                },
                        'AUTHOR' => 'Florian Ragwitz <rafl@debian.org>',
-                       'BUILD_REQUIRES' => {},
+                       'BUILD_REQUIRES' => {
+                            'Sub::Name'  => 0,
+                            'Test::More' => 0,
+                       },
                        'ABSTRACT' => 'Keep imports out of your namespace',
                        'EXE_FILES' => [],
                        'VERSION' => '0.10',
