Description: add encoding directives to POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2019-01-16

--- a/GRS1.pm
+++ b/GRS1.pm
@@ -258,6 +258,7 @@
 
 __END__
 
+=encoding iso8859-1
 
 =head1 NAME
 
--- a/SimpleServer.pm
+++ b/SimpleServer.pm
@@ -178,6 +178,8 @@
 __END__
 # Below is the stub of documentation for your module. You better edit it!
 
+=encoding utf8
+
 =head1 NAME
 
 Net::Z3950::SimpleServer - Simple Perl API for building Z39.50 servers. 
