diff --git a/Makefile.PL b/Makefile.PL
index 562e2a0..9d8c34d 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -5,7 +5,7 @@ WriteMakefile
 (
 
   NAME          => 'FindBin::libs',
-  VERSION       => '1.55',
+  VERSION_FROM  => 'lib/FindBin/libs.pm',
   ABSTRACT_FROM => 'lib/FindBin/libs.pm',
   PREREQ_PM     =>
   {
@@ -13,6 +13,7 @@ WriteMakefile
     # are unlikely to cause pain.
 
     strict      => 0,
+    version     => 0,
     warnings    => 0,
     Carp        => 0,
     Cwd         => 0,
diff --git a/lib/FindBin/libs.pm b/lib/FindBin/libs.pm
index f409159..1def993 100644
--- a/lib/FindBin/libs.pm
+++ b/lib/FindBin/libs.pm
@@ -78,8 +78,7 @@ BEGIN
 ########################################################################
 # package variables 
 ########################################################################
-
-our $VERSION = v1.54;
+use version; our $VERSION = version->new("v1.56");
 
 my %defaultz = 
 (
