--- a/lib/Log/Handler.pm
+++ b/lib/Log/Handler.pm
@@ -523,7 +523,7 @@
 This option is set to 1 by default.
 
 Take a look to the decription of the method C<reload> for more
-informations to this option.
+information about this option.
 
 =item B<filter_message>
 
@@ -632,7 +632,7 @@
 
 =item B<debug_trace>
 
-You can activate a debugger that writes C<caller()> informations for each
+You can activate a debugger that writes C<caller()> information about each
 active log level. The debugger is logging all defined values except C<hints>
 and C<bitmask>. Set C<debug_trace> to 1 to activate the debugger.
 The debugger is set to 0 by default.
@@ -711,7 +711,7 @@
 
 =item B<debug_skip>
 
-This option let skip the C<caller()> informations the count of C<debug_skip>.
+This option let skip the C<caller()> information the count of C<debug_skip>.
 
 =back
 
@@ -724,7 +724,7 @@
 
     $log->output($alias)->log(message => "booo");
 
-For more informations take a look to the option C<alias>.
+For more information take a look to the option C<alias>.
 
 =head2 flush()
 
@@ -806,7 +806,7 @@
 outputs. All other keys (C<error_log>, C<common_log>) are used as aliases.
 
 Take a look into the documentation of L<Log::Handler::Config> for more
-informations.
+information.
 
 =head2 reload()
 
@@ -937,7 +937,7 @@
 
 With this method it's possible to change the log level at runtime.
 
-To change the log level it's neccessary to use a alias - see option C<alias>.
+To change the log level it's necessary to use a alias - see option C<alias>.
 
     $log->set_level(
         $alias => { # option alias
--- a/lib/Log/Handler/Output/DBI.pm
+++ b/lib/Log/Handler/Output/DBI.pm
@@ -188,7 +188,7 @@
 
 =item B<debug>
 
-With this option it's possible to enable debugging. The informations can be
+With this option it's possible to enable debugging. The information can be
 intercepted with C<$SIG{__WARN__}>.
 
 =back
--- a/lib/Log/Handler/Output/Email.pm
+++ b/lib/Log/Handler/Output/Email.pm
@@ -88,7 +88,7 @@
 
 =item B<debug>
 
-With this option it's possible to enable debugging. The informations can be
+With this option it's possible to enable debugging. The information can be
 intercepted with $SIG{__WARN__}.
 
 =back
--- a/lib/Log/Handler/Output/File.pm
+++ b/lib/Log/Handler/Output/File.pm
@@ -112,7 +112,7 @@
 C<trunc> would truncate the complete log file if it exists. Please take care
 to use this option.
 
-Take a look to the documentation of C<sysopen()> to get more informations.
+Take a look to the documentation of C<sysopen()> to get more information.
 
 =item B<autoflush>
 
@@ -130,7 +130,7 @@
 got read and write permissions and users in the same group got only read
 permissions. All other users got no access.
 
-Take a look to the documentation of C<sysopen()> to get more informations.
+Take a look to the documentation of C<sysopen()> to get more information.
 
 =item B<utf8>
 
--- a/lib/Log/Handler/Plugin/Config/Properties.pm
+++ b/lib/Log/Handler/Plugin/Config/Properties.pm
@@ -14,7 +14,7 @@
 
 Expect the config file name and returns the config as a reference.
 
-The configuration will be splitted by dot.
+The configuration uses full stops "." as a delimiter.
 
 =head1 CONFIG STYLE
 
