--- /usr/lib/perl5/5.10.0/Pod/Text.pm.orig	2009-11-12 11:03:12.000000000 -0800
+++ /usr/lib/perl5/5.10.0/Pod/Text.pm	2009-11-12 11:07:11.000000000 -0800
@@ -275,6 +275,8 @@
 # Text blocks
 ##############################################################################
 
+sub strip_format { return $_[1]; }
+
 # This method is called whenever an =item command is complete (in other words,
 # we've seen its associated paragraph or know for certain that it doesn't have
 # one).  It gets the paragraph associated with the item as an argument.  If
@@ -296,7 +298,8 @@
     my $indent = $$self{INDENTS}[-1];
     $indent = $$self{opt_indent} unless defined $indent;
     my $margin = ' ' x $$self{opt_margin};
-    my $fits = ($$self{MARGIN} - $indent >= length ($tag) + 1);
+    my $tag_length = length($self->strip_format($tag));
+    my $fits = ($$self{MARGIN} - $indent >= $tag_length + 1);
 
     # If the tag doesn't fit, or if we have no associated text, print out the
     # tag separately.  Otherwise, put the tag in the margin of the paragraph.
@@ -321,7 +324,7 @@
         $space =~ s/^$margin /$margin:/ if $$self{opt_alt};
         $text = $self->reformat ($text);
         $text =~ s/^$margin /$margin:/ if ($$self{opt_alt} && $indent > 0);
-        my $tagspace = ' ' x length $tag;
+        my $tagspace = ' ' x $tag_length;
         $text =~ s/^($space)$tagspace/$1$tag/ or warn "Bizarre space in item";
         $self->output ($text);
     }
