Description: spelling fixes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-12-27

--- a/lib/Module/Reader.pm
+++ b/lib/Module/Reader.pm
@@ -421,7 +421,7 @@
 
 =head3 filename
 
-The filename that was seached for.
+The filename that was searched for.
 
 =head3 module
 
@@ -532,7 +532,7 @@
 
 =item L<Module::Locate>
 
-Innacurately searches C<@INC> for matching files.  Attempts to handle hooks, but
+Inaccurately searches C<@INC> for matching files.  Attempts to handle hooks, but
 handles most cases wrong.
 
 =item L<Module::Mapper>
