diff --git a/lib/ExtUtils/Depends.pm b/lib/ExtUtils/Depends.pm
index adc9d79..6a1737f 100644
--- a/lib/ExtUtils/Depends.pm
+++ b/lib/ExtUtils/Depends.pm
@@ -313,17 +313,15 @@ sub find_extra_libs {
 		my $lib = $mapper->($stem);
 		my $pattern = qr/$lib$/;
 
-		my $matching_dir;
 		my $matching_file;
 		find (sub {
 			if ((not $matching_file) && /$pattern/) {;
-				$matching_dir = $File::Find::dir;
 				$matching_file = $File::Find::name;
 			}
 		}, map { -d $_ ? ($_) : () } @INC); # only extant dirs
 
 		if ($matching_file && -f $matching_file) {
-			push @found_libs, ('-L' . $matching_dir, '-l' . $stem);
+			push @found_libs, $matching_file;
 			next;
 		}
 	}
