--- /home/stanton/tmp/PAR-Packer-0.976/script/par.pl    2007-07-13 09:39:25.000000000 -0700
+++ script/par.pl       2007-08-28 14:09:20.587164000 -0700
@@ -688,6 +688,7 @@
     require PAR::Heavy;
     require PAR::Dist;
     require PAR::Filter::PodStrip;
+    eval { require Cwd };
     eval { require Win32 };
     eval { require Scalar::Util };
     eval { require Archive::Unzip::Burst };
@@ -825,7 +826,7 @@
     }
 
     # XXX - hack to make PWD work
-    my $pwd = (defined &Win32::GetCwd) ? Win32::GetCwd() : $ENV{PWD};
-    $pwd = `pwd` if !defined $pwd;
+    my $pwd = (defined &Cwd::getcwd) ? Cwd::getcwd()
+                : ((defined &Win32::GetCwd) ? Win32::GetCwd() : `pwd`);
     chomp($pwd);
     $progname =~ s/^(?=\.\.?\Q$Config{_delim}\E)/$pwd$Config{_delim}/;
