--- Tiny.pm	2009-07-31 09:35:23.000000000 -0400
+++ Tiny.fixed.pm	2009-12-01 08:42:33.000000000 -0500
@@ -326,10 +326,10 @@
 		}
 
 		# Get the key
-		unless ( $lines->[0] =~ s/^\s*([^\'\" ][^\n]*?)\s*:(\s+|$)// ) {
-			if ( $lines->[0] =~ /^\s*[?\'\"]/ ) {
-				croak("YAML::Tiny does not support a feature in line '$lines->[0]'");
-			}
+                if ( $lines->[0] =~ /^\s*[?\'\"]/ ) {
+                        croak("YAML::Tiny does not support a feature in line '$lines->[0]'");
+                }
+		unless ( $lines->[0] =~ s/^\s*([^\n]*?)\s*:(\s+|$)// ) {
 			croak("YAML::Tiny failed to classify line '$lines->[0]'");
 		}
 		my $key = $1;
