Description: fix a spelling mistake in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-05-06

--- a/lib/XML/LibXML/Simple.pod
+++ b/lib/XML/LibXML/Simple.pod
@@ -156,7 +156,7 @@
 
 =item ContentKey => 'keyname' I<# seldom used>
 
-When text content is parsed to a hash value, this option let's you specify a
+When text content is parsed to a hash value, this option lets you specify a
 name for the hash key to override the default 'content'.  So for example:
 
   XMLin('<opt one="1">Two</opt>', ContentKey => 'text')
