*** HTTPServer.pm.orig	Wed Jun 13 16:56:25 2007
--- HTTPServer.pm	Wed Jun 13 16:56:12 2007
*************** sub _ServeFile
*** 1397,1402 ****
--- 1397,1404 ----
      my $fileHandle = new FileHandle($fullpath);
      return $self->_NotFound() unless defined($fileHandle);
  
+ 	$fileHandle->binmode();
+ 
      my $response = new Net::HTTPServer::Response();
  
      my ($ext) = ($fullpath =~ /\.([^\.]+?)$/);
