--- lib/Derpy/Controller/Muffin.pm	2013-06-11 14:18:43.767466001 +0200
+++ lib/Derpy/Controller/Muffin.pm	2013-06-11 14:15:27.579466003 +0200
@@ -24,10 +24,22 @@
 sub index :Path :Args(0) {
     my ( $self, $c ) = @_;
 
-    $c->response->body('Matched Derpy::Controller::Muffin in Muffin.');
+    $c->forward('muffin');
 }
 
 
+sub muffin :Local :Args(0) {
+    my ( $self, $c ) = @_;
+
+    my $body = "c.uri_for('/muffin') = " . $c->uri_for('/muffin') . "\n\n";
+
+    my $env = $c->request->env;
+    $body .= join $/, map "$_ = $env->{$_}", sort keys %$env;
+    $body = "<html><body><pre>\n$body\n</pre></body></html>\n";
+
+    $c->response->body($body);
+}
+
 =head1 AUTHOR
 
 A clever guy
