Description: Fix spelling mistakes in the POD.
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2017-01-13

--- a/lib/Parse/Dia/SQL/Output/SQLite3.pm
+++ b/lib/Parse/Dia/SQL/Output/SQLite3.pm
@@ -141,7 +141,7 @@
 
 =head2 get_schema_drop
 
-Generate drop table statments for all tables using SQLite syntax:
+Generate drop table statements for all tables using SQLite syntax:
 
   drop table {foo} if exists
 
@@ -179,7 +179,7 @@
 
 =head2 get_view_drop
 
-Generate drop view statments for all tables using SQLite syntax:
+Generate drop view statements for all tables using SQLite syntax:
 
   drop view {foo} if exists
 
--- a/lib/Parse/Dia/SQL/Output/SQLite3fk.pm
+++ b/lib/Parse/Dia/SQL/Output/SQLite3fk.pm
@@ -155,7 +155,7 @@
 
 =head2 _create_pk_string
 
-Override default functon to include foreign key clauses
+Override default function to include foreign key clauses
 
 =cut
 
@@ -197,7 +197,7 @@
 
 =head2 get_schema_drop
 
-Generate drop table statments for all tables using SQLite syntax:
+Generate drop table statements for all tables using SQLite syntax:
 
   drop table {foo} if exists
 
@@ -235,7 +235,7 @@
 
 =head2 get_view_drop
 
-Generate drop view statments for all tables using SQLite syntax:
+Generate drop view statements for all tables using SQLite syntax:
 
   drop view {foo} if exists
 
