--- XML/Filter/ExceptionLocator.pm	2013-12-12 13:13:15.389241215 -0600
+++ XML/Filter/ExceptionLocator.pm	2014-01-24 10:34:12.822907000 -0600
@@ -16,11 +16,11 @@
 
 # install handler for all SAX methods
 BEGIN {
-    for my $method qw(start_document     end_document           start_element
+    for my $method (qw(start_document     end_document           start_element
                       end_element        processing_instruction comment
                       skipped_entity     ignorable_whitespace   end_entity
                       start_entity       entity_reference
-                      start_cdata        end_cdata) {
+                      start_cdata        end_cdata)) {
         no strict 'refs';
         my $super_meth = "SUPER::$method";
         *{__PACKAGE__ . "::$method"} = 
