-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

diff -u -r MPEG-MP3Play-0.15.orig/Makefile.PL MPEG-MP3Play-0.15/Makefile.PL
- --- MPEG-MP3Play-0.15.orig/Makefile.PL	2001-01-06 06:05:31.000000000 +0800
+++ MPEG-MP3Play-0.15/Makefile.PL	2008-03-30 01:02:36.000000000 +0800
@@ -94,12 +94,10 @@
 	$sdk_dir = $SDK_DIR;
 
 	do {
- -		print "Xaudio SDK directory [$sdk_dir] : ";
- -		my $input;
- -		chomp ($input = <STDIN>);
+		my $input = prompt("Xaudio SDK directory [$sdk_dir] :");
 		$sdk_dir = $input if $input;
- -		$sdk_dir =~ s!/$!!;
 		$ok = -d "$sdk_dir/include" and -d "$sdk_dir/lib";
+		exit 0 if !$input && !$ok;
 		print "\nHmm, the directory must contain include and lib subdirectories.\n".
 		      "Try again or press Ctrl+C to abort.\n\n"
 			if not $ok;
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.9 (GNU/Linux)

iEYEARECAAYFAkfudrgACgkQi9gubzC5S1xeyACfZqmQIunFwOL7s2fXjYmm7a+l
l6cAoK3wp+sZjxdp2ZELIzJcfosfFmGu
=uiMt
-----END PGP SIGNATURE-----
