From acb5f1175c617f0be89f23648a0f3e0a4394230a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 27 Jan 2012 11:06:08 +0100
Subject: [PATCH] Do not use syslog

---
 t/client.t |    2 +-
 t/lib.pl   |    1 +
 t/server   |    2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/t/client.t b/t/client.t
index a283f2d..e19231d 100644
--- a/t/client.t
+++ b/t/client.t
@@ -30,7 +30,7 @@ my @opts = ('peeraddr' => '127.0.0.1', 'peerport' => $port, 'debug' => 1,
 my $client;
 
 # Making a first connection and closing it immediately
-Test(eval { RPC::PlClient->new(@opts) })
+Test(eval { RPC::PlClient->new(@opts, 'logfile' => 1) })
     or print "Failed to make first connection: $@\n";
 
 RunTests(@opts);
diff --git a/t/lib.pl b/t/lib.pl
index bc61b2c..559b899 100644
--- a/t/lib.pl
+++ b/t/lib.pl
@@ -18,6 +18,7 @@ sub RunTests (@) {
 	shift;
 	$key = shift;
     }
+    push @_, 'logfile', 1;
 
     # Making a new connection
     Test($client = eval { RPC::PlClient->new(@_) })
diff --git a/t/server b/t/server
index 46dcf97..c44f10a 100644
--- a/t/server
+++ b/t/server
@@ -74,7 +74,7 @@ sub Version ($) {
 
 package main;
 
-my $server = CalcServer->new({'pidfile' => 'none'}, \@ARGV);
+my $server = CalcServer->new({'pidfile' => 'none', 'logfile' => 1}, \@ARGV);
 
 $server->Bind();
 
-- 
1.7.7.6

