Description: POD fixes
 - abstract
 - spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-01-22

--- a/lib/Apache2_4/AuthCookie.pm
+++ b/lib/Apache2_4/AuthCookie.pm
@@ -52,7 +52,7 @@
 
 =head1 NAME
 
-Apache2_4::AuthCookie
+Apache2_4::AuthCookie - Perl Authentication and Authorization via cookies for Apache 2.4
 
 =head1 VERSION
 
@@ -571,7 +571,7 @@
 
 Why is my authz method called twice per request?
 
-This is normal behaviour under Apache 2.4.  This is to accomodate for
+This is normal behaviour under Apache 2.4.  This is to accommodate for
 authorization of anonymous access. You are expected to return
 C<Apache2::Const::AUTHZ_DENIED_NO_USER> IF C<< $r->user >> has not yet been set
 if you want authentication to proceed.  Your authz handler will be called a
