=== modified file 'lib/Devel/MAT/SV.pm'
--- lib/Devel/MAT/SV.pm	2014-01-22 21:25:15 +0000
+++ lib/Devel/MAT/SV.pm	2014-01-22 21:39:51 +0000
@@ -1232,6 +1232,8 @@
    $padnames->_set_padcv_at( $self->addr );
 
    foreach my $pad ( @pads ) {
+      next unless $pad;
+
       bless $pad, "Devel::MAT::SV::PAD";
       $pad->_set_padcv_at( $self->addr );
    }
@@ -1252,7 +1254,7 @@
          # Clear the obviously unused elements of lexnames and padlists
          foreach my $ix ( @$idxes ) {
             $padnames->_clear_elem( $ix );
-            $_->_clear_elem( $ix ) for @pads;
+            $_ and $_->_clear_elem( $ix ) for @pads;
          }
       }
    }

