--- Makefile.PL.dist	Mon Sep 18 22:22:12 2006
+++ Makefile.PL	Wed Sep 27 01:38:46 2006
@@ -211,6 +211,7 @@
 
 
 
+use Config qw(%Config);
 sub postamble {
     my $content;
 
@@ -222,8 +223,9 @@
 
 
 
-netdns$(LIB_EXT): netdns.o
-	$(AR) cr netdns$(LIB_EXT) netdns.o
+netdns$(LIB_EXT): netdns$(OBJ_EXT)
+	$(AR) '.($^O eq 'MSWin32' && $Config{'cc'} eq 'cl'?'/OUT:':'cr ').
+	  'netdns$(LIB_EXT) netdns$(OBJ_EXT)
 	$(RANLIB) netdns$(LIB_EXT)
 
 
