--- Time-HiRes-1.9721/Makefile.PL	2008-12-25 22:29:15.000000000 +0000
+++ Time-HiRes/Makefile.PL	2010-06-26 22:39:38.000000000 +0100
@@ -100,7 +100,7 @@
 
 	if ($ENV{PERL_CORE}) {
 	    my $updir = File::Spec->updir;
-	    $COREincdir = File::Spec->catdir(($updir) x 3);
+	    $COREincdir = File::Spec->catdir(($updir) x 2);
 	} else {
 	    $COREincdir = File::Spec->catdir($Config{'archlibexp'}, 'CORE');
 	}
