diff --git a/cpan/DB_File/t/db-btree.t b/cpan/DB_File/t/db-btree.t
index 4f2f825..29c70a1 100644
--- a/cpan/DB_File/t/db-btree.t
+++ b/cpan/DB_File/t/db-btree.t
@@ -1,12 +1,5 @@
 #!./perl -w
 
-BEGIN {
-    unless(grep /blib/, @INC) {
-        chdir 't' if -d 't';
-        @INC = '../lib' if -d '../lib';
-    }
-}
- 
 use warnings;
 use strict;
 use Config;
diff --git a/cpan/DB_File/t/db-hash.t b/cpan/DB_File/t/db-hash.t
index bb05f35..f4c8f95 100644
--- a/cpan/DB_File/t/db-hash.t
+++ b/cpan/DB_File/t/db-hash.t
@@ -1,12 +1,5 @@
 #!./perl 
 
-BEGIN {
-    unless(grep /blib/, @INC) {
-        chdir 't' if -d 't';
-        @INC = '../lib' if -d '../lib';
-    }
-}
- 
 use warnings;
 use strict;
 use Config;
diff --git a/cpan/DB_File/t/db-recno.t b/cpan/DB_File/t/db-recno.t
index 3eb6968..bd198dc 100644
--- a/cpan/DB_File/t/db-recno.t
+++ b/cpan/DB_File/t/db-recno.t
@@ -1,13 +1,5 @@
 #!./perl -w
 
-BEGIN {
-    unless(grep /blib/, @INC) {
-        chdir 't' if -d 't';
-        @INC = '../lib' if -d '../lib';
-    }
-}
- 
-use warnings;
 use strict;
 use Config;
  
