--- typemap.orig	2007-03-21 13:20:53.408981059 +0100
+++ typemap	2007-03-21 13:21:49.012456059 +0100
@@ -54,10 +54,13 @@
 GLUnurbsObj *		T_PTR
 GLUtesselator *		T_PTR
 GLuint			T_UV
+GLint			T_IV
 GLint*			T_PTR
 float *			T_PTR
 float			T_NV
 GLfloat			T_NV
+GLUquadric *		T_PTR
+GLdouble			T_NV
 Sound_DecoderInfo *	T_PTR
 const Sound_DecoderInfo *	T_PTR
 Sound_Sample *		T_PTR
--- src/OpenGL.xs.orig	2007-03-21 13:21:00.857446559 +0100
+++ src/OpenGL.xs	2007-03-21 13:23:52.864196309 +0100
@@ -1,4 +1,4 @@
-// SDL::OpenGL
+/ SDL::OpenGL
 //
 // Copyright (C) 2002,2003,2004 David J. Goehrig
 
@@ -2705,5 +2705,60 @@
 	
 #endif
 
+GLUquadric *
+gluNewQuadric ()
+	CODE:
+		RETVAL = gluNewQuadric ();
+	OUTPUT:
+		RETVAL
+
+void
+gluDeleteQuadric (quad)
+	GLUquadric	*quad
+
+void
+gluQuadricNormals ( quad, normal )
+	GLUquadric *quad
+	GLenum	normal
+
+void
+gluQuadricTexture ( quad, texture )
+	GLUquadric *quad
+	GLboolean  texture
+
+void
+gluCylinder ( quad, base, top, height, slices, stacks )
+	GLUquadric *quad
+	GLdouble  base
+	GLdouble  top
+	GLdouble  height
+	GLint  slices
+	GLint  stacks
+
+void
+gluDisk ( quad, inner, outer, slices, loops )
+	GLUquadric *quad
+	GLdouble inner
+	GLdouble outer
+	GLint slices
+	GLint loops
+
+void
+gluPartialDisk ( quad, inner, outer, slices, loops, start, sweep )
+	GLUquadric *quad
+	GLdouble inner
+	GLdouble outer
+	GLint slices
+	GLint loops
+	GLdouble start
+	GLdouble sweep
+
+void
+gluSphere ( quad, radius, slices, stacks )
+	GLUquadric *quad
+	GLdouble radius
+	GLint  slices
+	GLint  stacks
+
 #endif
 
