Description: fix a spelling mistake
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=85982
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=85982
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-05-29

--- a/lib/UR/Context/Process.pm
+++ b/lib/UR/Context/Process.pm
@@ -35,7 +35,7 @@
 
 =head1 DESCRIPTION
 
-This module provides methods to set and retreive variaous names
+This module provides methods to set and retrieve variaous names
 associated with the program and the program version number.
 
 =cut
--- a/lib/UR/DataSource/Oracle.pm
+++ b/lib/UR/DataSource/Oracle.pm
@@ -226,7 +226,7 @@
 sub get_userenv {
 
     # there are two ways to set these values but this is
-    # the only way to retreive the values after they are set
+    # the only way to retrieve the values after they are set
 
     my ($self, $dbh) = @_;
 
--- a/lib/UR/Namespace/Command/Sys/ClassBrowser.pm
+++ b/lib/UR/Namespace/Command/Sys/ClassBrowser.pm
@@ -57,7 +57,7 @@
 
 sub help_detail {
     q(The class-browser command starts an embedded web server containing an app for
-browsing throught the class structure.  After starting, it prints a URL on
+browsing through the class structure.  After starting, it prints a URL on
 STDOUT that can be copy-and-pasted into a browser to run the app.);
 }
 
@@ -647,7 +647,7 @@
 =head1 DESCRIPTION
 
 The class-browser command starts an embedded web server containing an app for
-browsing throught the class structure.  After starting, it prints a URL on
+browsing through the class structure.  After starting, it prints a URL on
 STDOUT that can be copy-and-pasted into a browser to run the app.
 
 =head1 COMMAND-LINE OPTIONS
--- a/lib/UR/Namespace/Command/Update/ClassesFromDb.pm
+++ b/lib/UR/Namespace/Command/Update/ClassesFromDb.pm
@@ -1371,7 +1371,7 @@
                 $status_message_this_update .= "A " . $class_obj->module_path;
             }
             $class_obj->rewrite_module_header() unless ($no_commit);
-            # FIXME A test of automaticly making DBIx::Class modules
+            # FIXME A test of automatically making DBIx::Class modules
             #$class_obj->dbic_rewrite_module_header() unless ($no_commit);
 
         }
--- a/lib/UR/Service/WebServer.pm
+++ b/lib/UR/Service/WebServer.pm
@@ -189,7 +189,7 @@
 =item server
 
 Holds a reference to an object that isa HTTP::Server::PSGI.  This will be
-automaticly created the first time run() is called.
+automatically created the first time run() is called.
 
 =item cb
 
