--- Makefile.PL.orig	Mon Oct 29 23:55:37 2007
+++ Makefile.PL	Mon Oct 29 23:55:18 2007
@@ -1,6 +1,5 @@
 #!perl -w
 
-BEGIN { require 5.006 }
 use strict;
 use Config qw(%Config);
 use ExtUtils::MakeMaker;
@@ -22,11 +21,16 @@
 push(@extra, 'INSTALLDIRS'  => 'perl') if $] >= 5.008;
 push(@extra, 'MAN3PODS' => {}) if $PERL_CORE; # Pods built by installman.
 
+if ($ExtUtils::MakeMaker::VERSION gt '6.30') {
+    push @extra, 'LICENSE' => 'perl';
+}
+
 WriteMakefile(
     'NAME'	   => 'Digest::MD5',
     'VERSION_FROM' => 'MD5.pm',
     'PREREQ_PM'    => { 'File::Spec' => 0,
 			'Digest::base' => '1.00',
+			'XSLoader' => 0,
 		      },
     @extra,
     'dist'         => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
