From bdc0f41b14cfea79fb03220a079eb0a4015b4a08 Mon Sep 17 00:00:00 2001
From: Jacob Farkas <jacobf@rkas.net>
Date: Sat, 8 Feb 2020 00:05:05 -0800
Subject: [PATCH] Fixing the build on Alpine Linux

---
 tacpluslib/tac_plus.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tacpluslib/tac_plus.h b/tacpluslib/tac_plus.h
index 8f44886..5044d41 100644
--- a/tacpluslib/tac_plus.h
+++ b/tacpluslib/tac_plus.h
@@ -167,6 +167,10 @@
 
 #include <unistd.h>
 
+#ifdef LINUX
+#include <fcntl.h>
+#endif
+
 #ifdef SYSV
 #include <fcntl.h>
 #define index strchr
-- 
2.21.0 (Apple Git-122.2)

