From: Matthias Schmitz <matthias@sigxcpu.org>
Date: Mon, 8 Jun 2009 15:54:38 +0200
Subject: [PATCH] fix minor POD error

---
 FileRotate.pm |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/FileRotate.pm b/FileRotate.pm
index 4036a1b..16f92b1 100644
--- a/FileRotate.pm
+++ b/FileRotate.pm
@@ -798,6 +798,8 @@ except to quote (from the man page):
 This method takes a hash of parameters.  The following options are
 valid:
 
+=over 8
+
 =item -- name ($)
 
 The name of the object (not the filename!).  Required.
@@ -866,6 +868,8 @@ will only be applied to a given message once.
 Turn on lots of warning messages to STDERR about what this module is
 doing if set to 1. Really only useful to me.
 
+=back
+
 =item * log_message( message => $ )
 
 Sends a message to the appropriate output.  Generally this shouldn't
-- 
