--- a/lib/Mail/Message/Head/Partial.pod
+++ b/lib/Mail/Message/Head/Partial.pod
@@ -28,7 +28,7 @@
 information is only useful during a short period of time in your program,
 or sometimes it is not used at all.  You then can decide to remove most
 of the header information.  However, be warned that it will be lost
-permanently: the header (and therefore the messsage) gets mutulated!
+permanently: the header (and therefore the message) gets mutulated!
 
 =head1 OVERLOADED
 
--- a/lib/Mail/Message/Head/ResentGroup.pod
+++ b/lib/Mail/Message/Head/ResentGroup.pod
@@ -146,7 +146,7 @@
 
 =over 4
 
-The header where the data is stored in.  Be dafault a
+The header where the data is stored in.  By default a
 L<Mail::Message::Head::Partial|Mail::Message::Head::Partial> is created for you.
 
 =back
--- a/lib/Mail/Transport/IMAP4.pod
+++ b/lib/Mail/Transport/IMAP4.pod
@@ -178,7 +178,7 @@
 
 Returned is a list of pairs (ref arrays) each describing one possible
 way to contact the server. Each pair contains a mechanism name and
-a challange callback (which may be C<undef>).
+a challenge callback (which may be C<undef>).
 
 The settings are used by L<login()|Mail::Transport::IMAP4/"Protocol [internals]"> to get server access.  The initial
 value origins from L<new(authenticate)|Mail::Transport::IMAP4/"METHODS">, but may be changed later.
