From 9fb4acfb0eaf7fa37fee6bf3fd70da44a416e7c6 Mon Sep 17 00:00:00 2001
From: "Stephen R. Scaffidi" <sscaffid@akamai.com>
Date: Thu, 2 Jun 2011 16:59:59 -0400
Subject: [PATCH 3/3] Fix for multi-package issue

---
 lib/CPAN/Site/Index.pm |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/CPAN/Site/Index.pm b/lib/CPAN/Site/Index.pm
index 1abda68..3576ac0 100644
--- a/lib/CPAN/Site/Index.pm
+++ b/lib/CPAN/Site/Index.pm
@@ -250,11 +250,11 @@ sub collect_package_details($$)
                 trace "pkg $package version $VERSION";
             }
         }
-    }
 
-    $VERSION = $VERSION->numify if ref $VERSION;
-    register $package, $VERSION, $dist
-        if defined $package;
+        $VERSION = $VERSION->numify if ref $VERSION;
+        register $package, $VERSION, $dist
+            if defined $package;
+    }
 }
 
 sub update_global_cpan($$)
-- 
1.7.5.2

