--- a/lib/Test/Spelling.pm
+++ b/lib/Test/Spelling.pm
@@ -217,13 +217,9 @@
 
 sub add_stopwords {
     for my $word (@_) {
-        # XXX: the processing this performs is to support "perl t/spell.t 2>>
-        # t/spell.t" which is bunk. in the near future the processing here will
-        # become more modern
-        $word =~ s/^#?\s*//;
-        $word =~ s/\s+$//;
-        next if $word =~ /\s/ or $word =~ /:/;
-        $Pod::Wordlist::Wordlist{$word} = 1;
+        if ($word =~ m{\A\#?\s*([^\s\:]+)\s*\z}xms) {
+            $Pod::Wordlist::Wordlist{$1} = 1;
+        }
     }
 }
 
