Description: Fix spelling error in manpage
Origin: vendor
Author: Salvatore Bonaccorso <carnil@debian.org>
Reviewed-by: Salvatore Bonaccorso <carnil@debian.org>
Last-Update: 2010-11-12

--- a/lib/Config/Model.pm
+++ b/lib/Config/Model.pm
@@ -312,7 +312,7 @@
 
 The "example" directory contains a configuration model example for the
 /etc/fstab file. This example includes a small program that use this
-model to show some ways to extract configuration informations.
+model to show some ways to extract configuration information.
 
 =head1 Mailing lists
 
@@ -2091,4 +2091,4 @@
 
 =back
 
-=cut
\ No newline at end of file
+=cut
