--- Utility.pm	2009-04-13 13:01:48.000000000 -0400
+++ Utility.pm.new	2009-04-13 13:02:36.000000000 -0400
@@ -151,13 +151,13 @@
     }
     elsif ( @formats == 3 ) {
         if ( $number == 0 ) {
-            $section = 3;
+            $section = 2;
         }
         elsif ( $number < 0 ) {
-            $section = 2;
+            $section = 1;
         }
         else {
-            $section = 1;
+            $section = 0;
         }
     }
     else {
