From 059147e7f73b017bd744e4daed0fd936f5371b3e Mon Sep 17 00:00:00 2001
From: Jens Rehsack <rehsack@googlemail.com>
Date: Wed, 28 Jul 2010 11:20:01 +0200
Subject: [PATCH 2/2] skip test with config when $ENV{CC} is set

---
 t/04-base.t |   18 +++++++++++-------
 1 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/t/04-base.t b/t/04-base.t
index bdc8d61..dba0af6 100644
--- a/t/04-base.t
+++ b/t/04-base.t
@@ -23,13 +23,17 @@ ok( $base, "ExtUtils::CBuilder::Base->new() returned true value" );
 isa_ok( $base, 'ExtUtils::CBuilder::Base' );
 
 $phony = 'foobar';
-$base = ExtUtils::CBuilder::Base->new(
-    config  => { cc => $phony },
-);
-ok( $base, "ExtUtils::CBuilder::Base->new() returned true value" );
-isa_ok( $base, 'ExtUtils::CBuilder::Base' );
-is( $base->{config}->{cc}, $phony,
-    "Got expected value when 'config' argument passed to new()" );
+SKIP: {
+    skip( "ExtUtils::CBuilder::Base->new() takes $ENV{CC}", 3 ) if( $ENV{CC} );
+
+    $base = ExtUtils::CBuilder::Base->new(
+	config  => { cc => $phony },
+    );
+    ok( $base, "ExtUtils::CBuilder::Base->new() returned true value" );
+    isa_ok( $base, 'ExtUtils::CBuilder::Base' );
+    is( $base->{config}->{cc}, $phony,
+	"Got expected value when 'config' argument passed to new()" );
+}
 
 {
     $phony = 'barbaz';
-- 
1.7.0.5

