<!DOCTYPE html>

<html lang="en">
  <head>
    <title>Bug #89439 for Bio-Homology-InterologWalk: bug report</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <!-- The X-UA-Compatible <meta> tag above must be very early in <head> -->

    <script>
if (window.top !== window.self) {
    document.write = "";

    window.top.location = window.self.location;

    setTimeout(function(){
        document.body.innerHTML = "";
    }, 1);

    window.self.onload = function(){
        document.body.innerHTML = "";
    };
}
</script>



<link rel="shortcut icon" href="/NoAuth/images/favicon.png" type="image/png" />
<link rel="stylesheet" href="/NoAuth/css/aileron-squished-00f758259152f8b6670ee485f462e6b8.css" type="text/css" media="all" />
<link rel="stylesheet" href="/NoAuth/css/print.css" type="text/css" media="print" />




<script type="text/javascript" src="/NoAuth/js/squished-42167f2409ce368da2d31038e5cb2a57.js"></script>

<script type="text/javascript"><!--
	jQuery( loadTitleBoxStates );

--></script>


<!--[if lt IE 8]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie.css" type="text/css" media="all" />
<![endif]-->
<!--[if lt IE 7]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie6.css" type="text/css" media="all" />
<![endif]-->
<script type="text/javascript">
jQuery(document).ready(function(){
    jQuery.fn.supersubs.defaults = {
        maxWidth: 30,
        extraWidth: 2
    };

    jQuery("#app-nav.toplevel").addClass('sf-menu sf-js-enabled sf-shadow').supersubs().superfish({ speed: 'fast' });
    jQuery("#page-menu.toplevel").addClass('sf-menu sf-js-enabled').supersubs().superfish({ dropShadows: false, speed: 'fast' }).supposition();
});
</script>


<!-- Site CSS from theme editor -->
<style type="text/css" media="all" id="sitecss">
</style>

<meta name="google-site-verification" content="kD3-uEJK1AkTqaC-rckcSt_HPkAj674DPPuEN6i-y0g" />

</head>
  <body class="aileron sidebyside" id="comp-Public-Bug-Display">

<div id="logo">
<a href="/"><img
    src="/NoAuth/images/cpan.png"
    alt="CPAN"
    width="181"
    height="38" /></a>
    <span class="rtname">RT for rt.cpan.org</span>
</div>


<div id="quickbar">
  <div id="quick-personal">
    <span class="hide"><a href="#skipnav">Skip Menu</a> | </span>
</div>

</div>
<div id="header"><h1>Bug #89439 for Bio-Homology-InterologWalk: bug report</h1></div>



<div id="page-navigation"><ul id="page-menu" class="toplevel">
 <li id="li-page-active_bugs"><a id="page-active_bugs" class="menu-item " href="/Public/Dist/Bio-Homology-InterologWalk/Active/">Active bugs</a></li>
 <li id="li-page-resolved_bugs"><a id="page-resolved_bugs" class="menu-item " href="/Public/Dist/Bio-Homology-InterologWalk/Resolved/">Resolved bugs</a></li>
 <li id="li-page-rejected_bugs"><a id="page-rejected_bugs" class="menu-item " href="/Public/Dist/Bio-Homology-InterologWalk/Rejected/">Rejected bugs</a></li>
 <li id="li-page-report"><a id="page-report" class="menu-item " href="/Public/Dist/Bio-Homology-InterologWalk/Active/"></a></li>
</ul></div>
<div id="topactions">



</div>
<div id="body">


<a name="skipnav" id="skipnav" accesskey="8"></a>





<p>This queue is for tickets about the <a href="https://metacpan.org/release/Bio-Homology-InterologWalk">Bio-Homology-InterologWalk CPAN distribution</a>.</p>





<div class="">
  <div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DUmVwb3J0IGluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Report information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------UmVwb3J0IGluZm9ybWF0aW9u---0">


<table width="100%" class="ticket-summary">
<tr>
  <td valign="top" class="boxcontainer">
    <div class="ticket-info-basics">
  <div class="titlebox ticket-info-basics " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dbasics\x2D\x2D\x2D\x2DVGhlIEJhc2ljcw\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">The Basics</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-basics----VGhlIEJhc2ljcw__---0">


        <table>
  <tr class="id">
    <td class="label">Id:</td>
    <td class="value">89439</td>
  </tr>
  <tr class="status">
    <td class="label">Status:</td>
    <td class="value">open</td>
  </tr>
  <tr class="priority">
    <td class="label">Priority:</td>
    <td class="value">0/
</td>
  </tr>
  <tr class="queue">
    <td class="label">Queue:</td>
    <td class="value"><a href="/Public/Dist/Bio-Homology-InterologWalk/Active/">Bio-Homology-InterologWalk</a>
</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>


    <div class="ticket-info-people">
  <div class="titlebox ticket-info-people " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dpeople\x2D\x2D\x2D\x2DUGVvcGxl\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">People</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-people----UGVvcGxl---0">


        <table>
  <tr>
    <td class="label">Owner:</td>
    <td class="value">Nobody in particular
    
    </td>
  </tr>
  <tr>
    <td class="labeltop">Requestors:</td>
    <td class="value">
karaltn [...] gmail.com

<br />

</td>
  </tr>
  <tr>
    <td class="labeltop">Cc:</td>
    <td class="value">

</td>
  </tr>
  <tr>
    <td class="labeltop">AdminCc:</td>
    <td class="value">

</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>



  </td>
  <td valign="top" class="boxcontainer">
    
    <div class="ticket-info-cfs">
  <div class="titlebox ticket-info-cfs " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dcfs\x2D\x2D\x2D\x2DQnVnIEluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Bug Information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-cfs----QnVnIEluZm9ybWF0aW9u---0">

 
        <table>
  <tr id="CF-1-ShowRow">
    <td class="label">Severity:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-234556-ShowRow">
    <td class="label">Broken in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-234557-ShowRow">
    <td class="label">Fixed in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
</table>


        <hr class="clear" />
  </div>
</div>




</div>


    




  </td>
</tr>
</table>

    <hr class="clear" />
  </div>
</div>




</div>


<br />


<div class="history">
<div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DSGlzdG9yeQ\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">History</span>
    <span class="right">		<a href="#" data-direction="open" onclick="return toggle_all_folds(this, 'Show\x20all\x20quoted\x20text', 'Hide\x20all\x20quoted\x20text');">Show all quoted text</a>    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------SGlzdG9yeQ__---0">


<div id="ticket-history">
<div class="ticket-transaction message Create odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Sun&nbsp;Oct&nbsp;13&nbsp;17:15:46&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Ticket created</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Sun, 13 Oct 2013 22:15:31 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi,

I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&#34;doInterologWalk.pl&#34; I am getting following two error. First error not
stopping the script but after second error its killing the process. I
copied and pasted errors from terminal and I also copied and pasted related
lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
version 73.

I hope this information will be enough for you to help me solve this
problem. If you need anything else please inform me about it.
.........
.........
........

Querying Ensembl Compara &#40;multi&#41; for orthologues----
*First error:* Use of uninitialized value in hash element at
/usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
945.

&#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
               $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
               $genome_names{$genome-&gt;name} = 1;
          }&#34;

64 genomes found in database: multi.

ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
......
......
......
Total number of interactions: 101.
Total Number of Unique NCBI taxa IDs: 7.

----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
musculus----
*Second error:*Use of uninitialized value in hash element at
/usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
1436.
get_backward_orthologies: impossible to get a protein tree  adaptor for DB:
multi
No orthologues found back in Mus musculus. Exiting..
There were errors. Stopping..


line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
               $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
               $genome_names{$genome-&gt;name} = 1;
          }&#34;

I am looking forward to hear from you.

Best regards
Altan
</div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Mon&nbsp;Oct&nbsp;14&nbsp;06:24:35&nbsp;2013</span>
    <span class="description">giuseppe.gallone [...] dpag.ox.ac.uk -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Mon, 14 Oct 2013 11:24:19 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Giuseppe Gallone &lt;giuseppe.gallone [...] dpag.ox.ac.uk&gt;</td>
  </tr>
</table>

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Dear Altan Kara

Bio::Homology::InterologWalk currently does not support Ensembl v73. At 
the moment, support exists for versions up to 64, with limited 
functionality for the v65 api. I am working on an update however this is 
not ready yet I&#39;m afraid. Apologies for any inconvenience caused.

Best regards
Giuseppe Gallone

On 10/13/13 22:15, Altan Kara via RT wrote:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; Sun Oct 13 17:15:46 2013: Request 89439 was acted upon.
&gt; Transaction: Ticket created by karaltn@gmail.com
&gt;         Queue: Bio-Homology-InterologWalk
&gt;       Subject: bug report
&gt;     Broken in: &#40;no value&#41;
&gt;      Severity: &#40;no value&#41;
&gt;         Owner: Nobody
&gt;    Requestors: karaltn@gmail.com
&gt;        Status: new
&gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt;
&gt; Hi,
&gt;
&gt; I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&gt; &#34;doInterologWalk.pl&#34; I am getting following two error. First error not
&gt; stopping the script but after second error its killing the process. I
&gt; copied and pasted errors from terminal and I also copied and pasted related
&gt; lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
&gt; version 73.
&gt;
&gt; I hope this information will be enough for you to help me solve this
&gt; problem. If you need anything else please inform me about it.
&gt; .........
&gt; .........
&gt; ........
&gt;
&gt; Querying Ensembl Compara &#40;multi&#41; for orthologues----
&gt; *First error:* Use of uninitialized value in hash element at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; 945.
&gt;
&gt; &#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt;                 $genome_names{$genome-&gt;name} = 1;
&gt;            }&#34;
&gt;
&gt; 64 genomes found in database: multi.
&gt;
&gt; ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
&gt; ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
&gt; ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
&gt; ......
&gt; ......
&gt; ......
&gt; Total number of interactions: 101.
&gt; Total Number of Unique NCBI taxa IDs: 7.
&gt;
&gt; ----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
&gt; musculus----
&gt; *Second error:*Use of uninitialized value in hash element at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; 1436.
&gt; get_backward_orthologies: impossible to get a protein tree  adaptor for DB:
&gt; multi
&gt; No orthologues found back in Mus musculus. Exiting..
&gt; There were errors. Stopping..
&gt;
&gt;
&gt; line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt;                 $genome_names{$genome-&gt;name} = 1;
&gt;            }&#34;
&gt;
&gt; I am looking forward to hear from you.
&gt;
&gt; Best regards
&gt; Altan
&gt;
</div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction basics Status odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Mon&nbsp;Oct&nbsp;14&nbsp;06:24:35&nbsp;2013</span>
    <span class="description">The RT System itself -  Status changed from &#39;new&#39; to &#39;open&#39;</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Mon&nbsp;Oct&nbsp;14&nbsp;14:40:47&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Mon, 14 Oct 2013 19:40:30 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Giuseppe,

I install the ensembl version 64 and try to run the same process&#40;
doInterologWalk.pl&#41; again and this time I got this registry error:

syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
Global symbol &#34;@dbs&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
Global symbol &#34;$type&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
Global symbol &#34;$type&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
Global symbol &#34;$species_suffix&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
Global symbol &#34;$host&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
Global symbol &#34;$user&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
Global symbol &#34;$pass&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
Global symbol &#34;$port&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
Global symbol &#34;$wait_timeout&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
Global symbol &#34;$no_cache&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
Global symbol &#34;$verbose&#34; requires explicit package name at
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
near &#34;}&#34;
/root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
Compilation failed in require at
/usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
BEGIN failed--compilation aborted at
/usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
Compilation failed in require at doInterologWalk.pl line 17.
BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
.......

how can I fix this or can you send me registry file for ensembl v64 which
wont create this error?

Best regards
Altan





On Mon, Oct 14, 2013 at 11:24 AM, Giuseppe Gallone via RT &lt;
bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Dear Altan Kara
&gt;
&gt; Bio::Homology::InterologWalk currently does not support Ensembl v73. At
&gt; the moment, support exists for versions up to 64, with limited
&gt; functionality for the v65 api. I am working on an update however this is
&gt; not ready yet I&#39;m afraid. Apologies for any inconvenience caused.
&gt;
&gt; Best regards
&gt; Giuseppe Gallone
&gt;
&gt; On 10/13/13 22:15, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt; Sun Oct 13 17:15:46 2013: Request 89439 was acted upon.
&gt; &gt; Transaction: Ticket created by karaltn@gmail.com
&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt; &gt;       Subject: bug report
&gt; &gt;     Broken in: &#40;no value&#41;
&gt; &gt;      Severity: &#40;no value&#41;
&gt; &gt;         Owner: Nobody
&gt; &gt;    Requestors: karaltn@gmail.com
&gt; &gt;        Status: new
&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt; Hi,
&gt; &gt;
&gt; &gt; I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&gt; &gt; &#34;doInterologWalk.pl&#34; I am getting following two error. First error not
&gt; &gt; stopping the script but after second error its killing the process. I
&gt; &gt; copied and pasted errors from terminal and I also copied and pasted
</div>&gt; related
<div class="message-stanza open">&gt; &gt; lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
&gt; &gt; version 73.
&gt; &gt;
&gt; &gt; I hope this information will be enough for you to help me solve this
&gt; &gt; problem. If you need anything else please inform me about it.
&gt; &gt; .........
&gt; &gt; .........
&gt; &gt; ........
&gt; &gt;
&gt; &gt; Querying Ensembl Compara &#40;multi&#41; for orthologues----
&gt; &gt; *First error:* Use of uninitialized value in hash element at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; &gt; 945.
&gt; &gt;
&gt; &gt; &#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;                 $genome_names{$genome-&gt;name} = 1;
&gt; &gt;            }&#34;
&gt; &gt;
&gt; &gt; 64 genomes found in database: multi.
&gt; &gt;
&gt; &gt; ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
&gt; &gt; ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
&gt; &gt; ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
&gt; &gt; ......
&gt; &gt; ......
&gt; &gt; ......
&gt; &gt; Total number of interactions: 101.
&gt; &gt; Total Number of Unique NCBI taxa IDs: 7.
&gt; &gt;
&gt; &gt; ----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
&gt; &gt; musculus----
&gt; &gt; *Second error:*Use of uninitialized value in hash element at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; &gt; 1436.
&gt; &gt; get_backward_orthologies: impossible to get a protein tree  adaptor for
</div>&gt; DB:
<div class="message-stanza open">&gt; &gt; multi
&gt; &gt; No orthologues found back in Mus musculus. Exiting..
&gt; &gt; There were errors. Stopping..
&gt; &gt;
&gt; &gt;
&gt; &gt; line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;                 $genome_names{$genome-&gt;name} = 1;
&gt; &gt;            }&#34;
&gt; &gt;
&gt; &gt; I am looking forward to hear from you.
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
</div>&gt;
&gt;
&gt;
</div></div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Mon&nbsp;Oct&nbsp;14&nbsp;14:52:54&nbsp;2013</span>
    <span class="description">giuseppe.gallone [...] dpag.ox.ac.uk -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Mon, 14 Oct 2013 19:52:36 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Giuseppe Gallone &lt;giuseppe.gallone [...] dpag.ox.ac.uk&gt;</td>
  </tr>
</table>

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hello Altan

This looks like a problem with the Ensembl API installation. Could you 
describe the exact steps you took in installing the Ensembl API - I will 
try to reproduce the bug. Thanks a lot,

G

On 10/14/13 19:40, Altan Kara via RT wrote:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;         Queue: Bio-Homology-InterologWalk
&gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Hi Giuseppe,
&gt;
&gt; I install the ensembl version 64 and try to run the same process&#40;
&gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt;
&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt; near &#34;}&#34;
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt; Compilation failed in require at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
&gt; BEGIN failed--compilation aborted at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
&gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt; .......
&gt;
&gt; how can I fix this or can you send me registry file for ensembl v64 which
&gt; wont create this error?
&gt;
&gt; Best regards
&gt; Altan
&gt;
&gt;
&gt;
&gt;
&gt;
&gt; On Mon, Oct 14, 2013 at 11:24 AM, Giuseppe Gallone via RT &lt;
&gt; bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:
&gt;
<div class="message-stanza open">&gt;&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt;
&gt;&gt; Dear Altan Kara
&gt;&gt;
&gt;&gt; Bio::Homology::InterologWalk currently does not support Ensembl v73. At
&gt;&gt; the moment, support exists for versions up to 64, with limited
&gt;&gt; functionality for the v65 api. I am working on an update however this is
&gt;&gt; not ready yet I&#39;m afraid. Apologies for any inconvenience caused.
&gt;&gt;
&gt;&gt; Best regards
&gt;&gt; Giuseppe Gallone
&gt;&gt;
&gt;&gt; On 10/13/13 22:15, Altan Kara via RT wrote:
<div class="message-stanza open">&gt;&gt;&gt; Sun Oct 13 17:15:46 2013: Request 89439 was acted upon.
&gt;&gt;&gt; Transaction: Ticket created by karaltn@gmail.com
&gt;&gt;&gt;          Queue: Bio-Homology-InterologWalk
&gt;&gt;&gt;        Subject: bug report
&gt;&gt;&gt;      Broken in: &#40;no value&#41;
&gt;&gt;&gt;       Severity: &#40;no value&#41;
&gt;&gt;&gt;          Owner: Nobody
&gt;&gt;&gt;     Requestors: karaltn@gmail.com
&gt;&gt;&gt;         Status: new
&gt;&gt;&gt;    Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt;&gt;
&gt;&gt;&gt;
&gt;&gt;&gt; Hi,
&gt;&gt;&gt;
&gt;&gt;&gt; I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&gt;&gt;&gt; &#34;doInterologWalk.pl&#34; I am getting following two error. First error not
&gt;&gt;&gt; stopping the script but after second error its killing the process. I
&gt;&gt;&gt; copied and pasted errors from terminal and I also copied and pasted
</div>&gt;&gt; related
<div class="message-stanza open">&gt;&gt;&gt; lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
&gt;&gt;&gt; version 73.
&gt;&gt;&gt;
&gt;&gt;&gt; I hope this information will be enough for you to help me solve this
&gt;&gt;&gt; problem. If you need anything else please inform me about it.
&gt;&gt;&gt; .........
&gt;&gt;&gt; .........
&gt;&gt;&gt; ........
&gt;&gt;&gt;
&gt;&gt;&gt; Querying Ensembl Compara &#40;multi&#41; for orthologues----
&gt;&gt;&gt; *First error:* Use of uninitialized value in hash element at
&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt;&gt;&gt; 945.
&gt;&gt;&gt;
&gt;&gt;&gt; &#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt;&gt;&gt;                  $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt;&gt;&gt;                  $genome_names{$genome-&gt;name} = 1;
&gt;&gt;&gt;             }&#34;
&gt;&gt;&gt;
&gt;&gt;&gt; 64 genomes found in database: multi.
&gt;&gt;&gt;
&gt;&gt;&gt; ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
&gt;&gt;&gt; ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
&gt;&gt;&gt; ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
&gt;&gt;&gt; ......
&gt;&gt;&gt; ......
&gt;&gt;&gt; ......
&gt;&gt;&gt; Total number of interactions: 101.
&gt;&gt;&gt; Total Number of Unique NCBI taxa IDs: 7.
&gt;&gt;&gt;
&gt;&gt;&gt; ----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
&gt;&gt;&gt; musculus----
&gt;&gt;&gt; *Second error:*Use of uninitialized value in hash element at
&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt;&gt;&gt; 1436.
&gt;&gt;&gt; get_backward_orthologies: impossible to get a protein tree  adaptor for
</div>&gt;&gt; DB:
<div class="message-stanza open">&gt;&gt;&gt; multi
&gt;&gt;&gt; No orthologues found back in Mus musculus. Exiting..
&gt;&gt;&gt; There were errors. Stopping..
&gt;&gt;&gt;
&gt;&gt;&gt;
&gt;&gt;&gt; line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt;&gt;&gt;                  $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt;&gt;&gt;                  $genome_names{$genome-&gt;name} = 1;
&gt;&gt;&gt;             }&#34;
&gt;&gt;&gt;
&gt;&gt;&gt; I am looking forward to hear from you.
&gt;&gt;&gt;
&gt;&gt;&gt; Best regards
&gt;&gt;&gt; Altan
&gt;&gt;&gt;
</div></div></div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Mon&nbsp;Oct&nbsp;14&nbsp;16:59:31&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Mon, 14 Oct 2013 21:59:09 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">I used cvs:

$ cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
CVS password: CVSUSER
$ cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r
branch-ensembl-64 ensembl

Best regards
Altan


On Mon, Oct 14, 2013 at 7:52 PM, Giuseppe Gallone via RT &lt;
bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Hello Altan
&gt;
&gt; This looks like a problem with the Ensembl API installation. Could you
&gt; describe the exact steps you took in installing the Ensembl API - I will
&gt; try to reproduce the bug. Thanks a lot,
&gt;
&gt; G
&gt;
&gt; On 10/14/13 19:40, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;
&gt; &gt; Hi Giuseppe,
&gt; &gt;
&gt; &gt; I install the ensembl version 64 and try to run the same process&#40;
&gt; &gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt; &gt;
&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt; &gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt; &gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt; &gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt; &gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt; &gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt; &gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt; &gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt; &gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt; &gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt; &gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt; &gt; near &#34;}&#34;
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt; &gt; Compilation failed in require at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
</div>&gt; 7.
<div class="message-stanza open">&gt; &gt; BEGIN failed--compilation aborted at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
</div>&gt; 7.
<div class="message-stanza open">&gt; &gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt; &gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt; &gt; .......
&gt; &gt;
&gt; &gt; how can I fix this or can you send me registry file for ensembl v64 which
&gt; &gt; wont create this error?
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt; On Mon, Oct 14, 2013 at 11:24 AM, Giuseppe Gallone via RT &lt;
&gt; &gt; bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:
&gt; &gt;
<div class="message-stanza open">&gt; &gt;&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;&gt;
&gt; &gt;&gt; Dear Altan Kara
&gt; &gt;&gt;
&gt; &gt;&gt; Bio::Homology::InterologWalk currently does not support Ensembl v73. At
&gt; &gt;&gt; the moment, support exists for versions up to 64, with limited
&gt; &gt;&gt; functionality for the v65 api. I am working on an update however this is
&gt; &gt;&gt; not ready yet I&#39;m afraid. Apologies for any inconvenience caused.
&gt; &gt;&gt;
&gt; &gt;&gt; Best regards
&gt; &gt;&gt; Giuseppe Gallone
&gt; &gt;&gt;
&gt; &gt;&gt; On 10/13/13 22:15, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt;&gt;&gt; Sun Oct 13 17:15:46 2013: Request 89439 was acted upon.
&gt; &gt;&gt;&gt; Transaction: Ticket created by karaltn@gmail.com
&gt; &gt;&gt;&gt;          Queue: Bio-Homology-InterologWalk
&gt; &gt;&gt;&gt;        Subject: bug report
&gt; &gt;&gt;&gt;      Broken in: &#40;no value&#41;
&gt; &gt;&gt;&gt;       Severity: &#40;no value&#41;
&gt; &gt;&gt;&gt;          Owner: Nobody
&gt; &gt;&gt;&gt;     Requestors: karaltn@gmail.com
&gt; &gt;&gt;&gt;         Status: new
&gt; &gt;&gt;&gt;    Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Hi,
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&gt; &gt;&gt;&gt; &#34;doInterologWalk.pl&#34; I am getting following two error. First error not
&gt; &gt;&gt;&gt; stopping the script but after second error its killing the process. I
&gt; &gt;&gt;&gt; copied and pasted errors from terminal and I also copied and pasted
</div>&gt; &gt;&gt; related
<div class="message-stanza open">&gt; &gt;&gt;&gt; lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
&gt; &gt;&gt;&gt; version 73.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; I hope this information will be enough for you to help me solve this
&gt; &gt;&gt;&gt; problem. If you need anything else please inform me about it.
&gt; &gt;&gt;&gt; .........
&gt; &gt;&gt;&gt; .........
&gt; &gt;&gt;&gt; ........
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Querying Ensembl Compara &#40;multi&#41; for orthologues----
&gt; &gt;&gt;&gt; *First error:* Use of uninitialized value in hash element at
&gt; &gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div></div></div>&gt; line
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; 945.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; &#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;&gt;&gt;                  $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;&gt;&gt;                  $genome_names{$genome-&gt;name} = 1;
&gt; &gt;&gt;&gt;             }&#34;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 64 genomes found in database: multi.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
&gt; &gt;&gt;&gt; ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
&gt; &gt;&gt;&gt; ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
&gt; &gt;&gt;&gt; ......
&gt; &gt;&gt;&gt; ......
&gt; &gt;&gt;&gt; ......
&gt; &gt;&gt;&gt; Total number of interactions: 101.
&gt; &gt;&gt;&gt; Total Number of Unique NCBI taxa IDs: 7.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; ----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
&gt; &gt;&gt;&gt; musculus----
&gt; &gt;&gt;&gt; *Second error:*Use of uninitialized value in hash element at
&gt; &gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div></div>&gt; line
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; 1436.
&gt; &gt;&gt;&gt; get_backward_orthologies: impossible to get a protein tree  adaptor for
</div>&gt; &gt;&gt; DB:
<div class="message-stanza open">&gt; &gt;&gt;&gt; multi
&gt; &gt;&gt;&gt; No orthologues found back in Mus musculus. Exiting..
&gt; &gt;&gt;&gt; There were errors. Stopping..
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;&gt;&gt;                  $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;&gt;&gt;                  $genome_names{$genome-&gt;name} = 1;
&gt; &gt;&gt;&gt;             }&#34;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; I am looking forward to hear from you.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Best regards
&gt; &gt;&gt;&gt; Altan
&gt; &gt;&gt;&gt;
</div></div>&gt;
&gt;
</div></div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Tue&nbsp;Oct&nbsp;15&nbsp;16:50:32&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Tue, 15 Oct 2013 21:50:14 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Giuseppe

Can you give any idea about when update of Bio::Homology::InterologWalk
will be completed for Ensemblv73?

Best regards
Altan


On Mon, Oct 14, 2013 at 11:24 AM, Giuseppe Gallone via RT &lt;
bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Dear Altan Kara
&gt;
&gt; Bio::Homology::InterologWalk currently does not support Ensembl v73. At
&gt; the moment, support exists for versions up to 64, with limited
&gt; functionality for the v65 api. I am working on an update however this is
&gt; not ready yet I&#39;m afraid. Apologies for any inconvenience caused.
&gt;
&gt; Best regards
&gt; Giuseppe Gallone
&gt;
&gt; On 10/13/13 22:15, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt; Sun Oct 13 17:15:46 2013: Request 89439 was acted upon.
&gt; &gt; Transaction: Ticket created by karaltn@gmail.com
&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt; &gt;       Subject: bug report
&gt; &gt;     Broken in: &#40;no value&#41;
&gt; &gt;      Severity: &#40;no value&#41;
&gt; &gt;         Owner: Nobody
&gt; &gt;    Requestors: karaltn@gmail.com
&gt; &gt;        Status: new
&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt; Hi,
&gt; &gt;
&gt; &gt; I recently installed &#34;Bio-Homology-InterologWalk-0.6b&#34; and when I run
&gt; &gt; &#34;doInterologWalk.pl&#34; I am getting following two error. First error not
&gt; &gt; stopping the script but after second error its killing the process. I
&gt; &gt; copied and pasted errors from terminal and I also copied and pasted
</div>&gt; related
<div class="message-stanza open">&gt; &gt; lines with error from my script &#40;InterologWalk.pm&#41;. I am using ensemble
&gt; &gt; version 73.
&gt; &gt;
&gt; &gt; I hope this information will be enough for you to help me solve this
&gt; &gt; problem. If you need anything else please inform me about it.
&gt; &gt; .........
&gt; &gt; .........
&gt; &gt; ........
&gt; &gt;
&gt; &gt; Querying Ensembl Compara &#40;multi&#41; for orthologues----
&gt; &gt; *First error:* Use of uninitialized value in hash element at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; &gt; 945.
&gt; &gt;
&gt; &gt; &#34;line945 : foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;                 $genome_names{$genome-&gt;name} = 1;
&gt; &gt;            }&#34;
&gt; &gt;
&gt; &gt; 64 genomes found in database: multi.
&gt; &gt;
&gt; &gt; ENSMUSG00000000326 Comt ..69 orthologue&#40;s&#41;.
&gt; &gt; ENSMUSG00000001323 Srr ..64 orthologue&#40;s&#41;.
&gt; &gt; ENSMUSG00000001632 Brpf1 ..61 orthologue&#40;s&#41;.
&gt; &gt; ......
&gt; &gt; ......
&gt; &gt; ......
&gt; &gt; Total number of interactions: 101.
&gt; &gt; Total Number of Unique NCBI taxa IDs: 7.
&gt; &gt;
&gt; &gt; ----Querying Ensembl Compara &#40;multi&#41; for orthologues back in Mus
&gt; &gt; musculus----
&gt; &gt; *Second error:*Use of uninitialized value in hash element at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
&gt; &gt; 1436.
&gt; &gt; get_backward_orthologies: impossible to get a protein tree  adaptor for
</div>&gt; DB:
<div class="message-stanza open">&gt; &gt; multi
&gt; &gt; No orthologues found back in Mus musculus. Exiting..
&gt; &gt; There were errors. Stopping..
&gt; &gt;
&gt; &gt;
&gt; &gt; line 1436 : &#34;foreach my $genome &#40;@{$all_genome_dbs}&#41;{
&gt; &gt;                 $genome_taxon_ids{$genome-&gt;taxon_id} = 1;
&gt; &gt;                 $genome_names{$genome-&gt;name} = 1;
&gt; &gt;            }&#34;
&gt; &gt;
&gt; &gt; I am looking forward to hear from you.
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
</div>&gt;
&gt;
&gt;
</div></div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Wed&nbsp;Oct&nbsp;16&nbsp;07:49:51&nbsp;2013</span>
    <span class="description">giuseppe.gallone [...] dpag.ox.ac.uk -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 16 Oct 2013 12:49:29 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Giuseppe Gallone &lt;giuseppe.gallone [...] dpag.ox.ac.uk&gt;</td>
  </tr>
</table>

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hello Altan

I am testing Bio::Homology::InterologWalk v 0.6b with Ensembl v64 and it 
seems to work on my platform.

In order to make it easier for me to debug the errors you&#39;re seeing, 
could you please test it once more, following these steps &#40;I&#39;m assuming 
you&#39;re on a linux machine&#41;:

1-Please delete any instances of the ensembl API you might have 
previously installed. In your home directory, create a src directory and 
cd into it.

#------------
cd ~
mkdir src
cd src
#------------

2-Install Bioperl 1.2.3 as described here

<span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_cvs.html">http://www.ensembl.org/info/docs/api/api_cvs.html</a></span>

3-Install the full Ensembl-64 api typing the following commands:

#-------------------------------------
cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
CVS password: CVSUSER

cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r 
branch-ensembl-64 ensembl-api
#-------------------------------------
The second command is particularly important - you will need ALL the 
APIs for the module to work correctly.

4-At this stage, if all has gone smoothly, you should have the following 
four directories under /home/YOURNAME/src:

bioperl-live
ensembl
ensembl-compara
ensembl-external
ensembl-functgenomics
ensembl-tools
ensembl-variation

5-Set up your environment as described here
<span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_installation.html">http://www.ensembl.org/info/docs/api/api_installation.html</a></span>

For example, for a fedora/RH linux machine, you&#39;d open your .bashrc file 
and type in the paths to those directories in point &#40;4&#41;, as follows &#40;I&#39;m 
assuming the vim editor is installed on your system, otherwise maybe try 
gedit&#41;:

#-------------
vim ~/.bashrc
#-------------
and type the following at the bottom of the file:
#--------------
PERL5LIB=${PERL5LIB}:${HOME}/src/bioperl-1.2.3
PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl/modules
PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-compara/modules
PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-variation/modules
PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-functgenomics/modules
export PERL5LIB
#-----------------
#&#40;no need to export paths to ensembl-tools and ensembl-external&#41;

Now save the file, and exit. Log off the session and log in again &#40;This 
is very important for the changes you&#39;ve made to .bashrc to have effect&#41;

6-At this time your Ensembl/Bioperl configuration should be complete. 
You may now attempt the installation of Bio::Homology::InterologWalk. 
Download it from here:

#----------
wget 
<span class="clickylink"><a target="new" rel="nofollow" href="http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz">http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz</a></span>
#----------

then

#----------
tar xzvf Bio-Homology-InterologWalk-0.6b.tar.gz
#----------

and

#----------
cd Bio-Homology-InterologWalk-0.6b
#----------

now you can follow the included README for installation. Basically you&#39;ll do

#----------
perl Makefile.PL
#----------

the output should be the following &#40;provided you have all the 
prerequisite modules installed&#41;

#----------
Checking for recommended module Algorithm::Combinatorics &gt;= 0.25 ... ok.
Checking for required module Bio::EnsEMBL::Registry ... ok.
Checking for required module Bio::Perl ... ok.
Checking for required module Carp ... ok.
Checking for required module DBD::CSV ... ok.
Checking for recommended module Data::PowerSet &gt;= 0.05 ... ok.
Checking for required module File::Glob ... ok.
Checking for required module GO::Parser ... ok.
Checking for recommended module Getopt::Long &gt;= 2.38 ... ok.
Checking for recommended module Graph &gt;= 0.94 ... ok.
Checking for required module List::Util ... ok.
Checking for required module REST::Client ... ok.
Checking for required module String::Approx ... ok.
Checking for recommended module URI::Escape &gt;= 1.53 ... ok.

Checking if your kit is complete...
Looks good
Writing Makefile for Bio::Homology::InterologWalk
#-----------

Then you should do

#----------
make
#----------

which should give you the following output

#------------------
cp lib/Bio/Homology/InterologWalk.pm blib/lib/Bio/Homology/InterologWalk.pm
Manifying blib/man3/Bio::Homology::InterologWalk.3pm
#------------------

next do a

#------------
make test
#------------
This will take a while, and it will need a internet connection. It will 
fail if the Ensembl/Intact services are down. A typical successful run 
would look like this:

#----------------------------
PERL_DL_NONLAZY=1 /usr/bin/perl &#34;-MExtUtils::Command::MM&#34; &#34;-e&#34; 
&#34;test_harness&#40;0, &#39;blib/lib&#39;, &#39;blib/arch&#39;&#41;&#34; t/*.t
t/Bio-Homology-InterologWalk.t .. 1/19 #
#
# #######PLEASE NOTE######
# ########################
# In order for some of the following tests to succeed,
# remote services provided by Ensembl and Intact MUST BE ACTIVE.
# in case of multiple test failure,
# please check status of Ensembl services on announce@ensembl.org
# and status of EBI Intact PSICQUIC service on
# www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS
# ########################
# ########################
#
# Testing interfaces to remote services..Please be patient..
#
t/Bio-Homology-InterologWalk.t .. 13/19 add method for created_by at 
/usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
add method for creation_date at 
/usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
t/Bio-Homology-InterologWalk.t .. ok
All tests successful.
Files=1, Tests=19, 111 wallclock secs &#40; 0.05 usr  0.00 sys +  6.36 cusr 
  0.26 csys =  6.67 CPU&#41;
Result: PASS
#----------------------------------

After this you&#39;d just do a

make install

Let me know how it goes. I&#39;m in the process of simplifying the 
documentation for a new release - apologies for the complexity.

Cheers, G



On 10/14/13 19:40, Altan Kara via RT wrote:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;         Queue: Bio-Homology-InterologWalk
&gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Hi Giuseppe,
&gt;
&gt; I install the ensembl version 64 and try to run the same process&#40;
&gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt;
&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt; near &#34;}&#34;
&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt; Compilation failed in require at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
&gt; BEGIN failed--compilation aborted at
&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line 7.
&gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt; .......
&gt;
&gt; how can I fix this or can you send me registry file for ensembl v64 which
&gt; wont create this error?
&gt;
&gt; Best regards
&gt; Altan
&gt;
</div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Wed&nbsp;Oct&nbsp;16&nbsp;07:54:26&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 16 Oct 2013 12:54:08 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi,

Honestly I did all these steps but I will do it again and inform you about
result and I hope I can have the new release soon which supports ensembl
v73.

Best regards
A


On Wed, Oct 16, 2013 at 12:49 PM, Giuseppe Gallone via RT &lt;
bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Hello Altan
&gt;
&gt; I am testing Bio::Homology::InterologWalk v 0.6b with Ensembl v64 and it
&gt; seems to work on my platform.
&gt;
&gt; In order to make it easier for me to debug the errors you&#39;re seeing,
&gt; could you please test it once more, following these steps &#40;I&#39;m assuming
&gt; you&#39;re on a linux machine&#41;:
&gt;
&gt; 1-Please delete any instances of the ensembl API you might have
&gt; previously installed. In your home directory, create a src directory and
&gt; cd into it.
&gt;
&gt; #------------
&gt; cd ~
&gt; mkdir src
&gt; cd src
&gt; #------------
&gt;
&gt; 2-Install Bioperl 1.2.3 as described here
&gt;
&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_cvs.html">http://www.ensembl.org/info/docs/api/api_cvs.html</a></span>
&gt;
&gt; 3-Install the full Ensembl-64 api typing the following commands:
&gt;
&gt; #-------------------------------------
&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
&gt; Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
&gt; CVS password: CVSUSER
&gt;
&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r
&gt; branch-ensembl-64 ensembl-api
&gt; #-------------------------------------
&gt; The second command is particularly important - you will need ALL the
&gt; APIs for the module to work correctly.
&gt;
&gt; 4-At this stage, if all has gone smoothly, you should have the following
&gt; four directories under /home/YOURNAME/src:
&gt;
&gt; bioperl-live
&gt; ensembl
&gt; ensembl-compara
&gt; ensembl-external
&gt; ensembl-functgenomics
&gt; ensembl-tools
&gt; ensembl-variation
&gt;
&gt; 5-Set up your environment as described here
&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_installation.html">http://www.ensembl.org/info/docs/api/api_installation.html</a></span>
&gt;
&gt; For example, for a fedora/RH linux machine, you&#39;d open your .bashrc file
&gt; and type in the paths to those directories in point &#40;4&#41;, as follows &#40;I&#39;m
&gt; assuming the vim editor is installed on your system, otherwise maybe try
&gt; gedit&#41;:
&gt;
&gt; #-------------
&gt; vim ~/.bashrc
&gt; #-------------
&gt; and type the following at the bottom of the file:
&gt; #--------------
&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/bioperl-1.2.3
&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl/modules
&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-compara/modules
&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-variation/modules
&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-functgenomics/modules
&gt; export PERL5LIB
&gt; #-----------------
&gt; #&#40;no need to export paths to ensembl-tools and ensembl-external&#41;
&gt;
&gt; Now save the file, and exit. Log off the session and log in again &#40;This
&gt; is very important for the changes you&#39;ve made to .bashrc to have effect&#41;
&gt;
&gt; 6-At this time your Ensembl/Bioperl configuration should be complete.
&gt; You may now attempt the installation of Bio::Homology::InterologWalk.
&gt; Download it from here:
&gt;
&gt; #----------
&gt; wget
&gt;
&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz">http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz</a></span>
&gt; #----------
&gt;
&gt; then
&gt;
&gt; #----------
&gt; tar xzvf Bio-Homology-InterologWalk-0.6b.tar.gz
&gt; #----------
&gt;
&gt; and
&gt;
&gt; #----------
&gt; cd Bio-Homology-InterologWalk-0.6b
&gt; #----------
&gt;
&gt; now you can follow the included README for installation. Basically you&#39;ll
&gt; do
&gt;
&gt; #----------
&gt; perl Makefile.PL
&gt; #----------
&gt;
&gt; the output should be the following &#40;provided you have all the
&gt; prerequisite modules installed&#41;
&gt;
&gt; #----------
&gt; Checking for recommended module Algorithm::Combinatorics &gt;= 0.25 ... ok.
&gt; Checking for required module Bio::EnsEMBL::Registry ... ok.
&gt; Checking for required module Bio::Perl ... ok.
&gt; Checking for required module Carp ... ok.
&gt; Checking for required module DBD::CSV ... ok.
&gt; Checking for recommended module Data::PowerSet &gt;= 0.05 ... ok.
&gt; Checking for required module File::Glob ... ok.
&gt; Checking for required module GO::Parser ... ok.
&gt; Checking for recommended module Getopt::Long &gt;= 2.38 ... ok.
&gt; Checking for recommended module Graph &gt;= 0.94 ... ok.
&gt; Checking for required module List::Util ... ok.
&gt; Checking for required module REST::Client ... ok.
&gt; Checking for required module String::Approx ... ok.
&gt; Checking for recommended module URI::Escape &gt;= 1.53 ... ok.
&gt;
&gt; Checking if your kit is complete...
&gt; Looks good
&gt; Writing Makefile for Bio::Homology::InterologWalk
&gt; #-----------
&gt;
&gt; Then you should do
&gt;
&gt; #----------
&gt; make
&gt; #----------
&gt;
&gt; which should give you the following output
&gt;
&gt; #------------------
&gt; cp lib/Bio/Homology/InterologWalk.pm blib/lib/Bio/Homology/InterologWalk.pm
&gt; Manifying blib/man3/Bio::Homology::InterologWalk.3pm
&gt; #------------------
&gt;
&gt; next do a
&gt;
&gt; #------------
&gt; make test
&gt; #------------
&gt; This will take a while, and it will need a internet connection. It will
&gt; fail if the Ensembl/Intact services are down. A typical successful run
&gt; would look like this:
&gt;
&gt; #----------------------------
&gt; PERL_DL_NONLAZY=1 /usr/bin/perl &#34;-MExtUtils::Command::MM&#34; &#34;-e&#34;
&gt; &#34;test_harness&#40;0, &#39;blib/lib&#39;, &#39;blib/arch&#39;&#41;&#34; t/*.t
&gt; t/Bio-Homology-InterologWalk.t .. 1/19 #
&gt; #
&gt; # #######PLEASE NOTE######
&gt; # ########################
&gt; # In order for some of the following tests to succeed,
&gt; # remote services provided by Ensembl and Intact MUST BE ACTIVE.
&gt; # in case of multiple test failure,
&gt; # please check status of Ensembl services on announce@ensembl.org
&gt; # and status of EBI Intact PSICQUIC service on
&gt; # www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS
&gt; # ########################
&gt; # ########################
&gt; #
&gt; # Testing interfaces to remote services..Please be patient..
&gt; #
&gt; t/Bio-Homology-InterologWalk.t .. 13/19 add method for created_by at
&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt; add method for creation_date at
&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt; t/Bio-Homology-InterologWalk.t .. ok
&gt; All tests successful.
&gt; Files=1, Tests=19, 111 wallclock secs &#40; 0.05 usr  0.00 sys +  6.36 cusr
&gt;   0.26 csys =  6.67 CPU&#41;
&gt; Result: PASS
&gt; #----------------------------------
&gt;
&gt; After this you&#39;d just do a
&gt;
&gt; make install
&gt;
&gt; Let me know how it goes. I&#39;m in the process of simplifying the
&gt; documentation for a new release - apologies for the complexity.
&gt;
&gt; Cheers, G
&gt;
&gt;
&gt;
&gt; On 10/14/13 19:40, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;
&gt; &gt; Hi Giuseppe,
&gt; &gt;
&gt; &gt; I install the ensembl version 64 and try to run the same process&#40;
&gt; &gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt; &gt;
&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt; &gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt; &gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt; &gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt; &gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt; &gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt; &gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt; &gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt; &gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt; &gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt; &gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt; &gt; near &#34;}&#34;
&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt; &gt; Compilation failed in require at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
</div>&gt; 7.
<div class="message-stanza open">&gt; &gt; BEGIN failed--compilation aborted at
&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm line
</div>&gt; 7.
<div class="message-stanza open">&gt; &gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt; &gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt; &gt; .......
&gt; &gt;
&gt; &gt; how can I fix this or can you send me registry file for ensembl v64 which
&gt; &gt; wont create this error?
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
</div>&gt;
&gt;
</div></div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Wed&nbsp;Oct&nbsp;16&nbsp;18:51:37&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 16 Oct 2013 23:51:20 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Giuseppe,

I checked all of my installation steps and everything looks fine, as you
know if I made an installation error at any step other Ensembl Apis also
wont work but only v64  is not working.
Can you just send me your &#34;Registry.pm&#34; for Ensembl v64?

Best regards
Altan

Best regards
Altan




On Wed, Oct 16, 2013 at 12:54 PM, Altan Kara &lt;karaltn@gmail.com&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; Hi,
&gt;
&gt; Honestly I did all these steps but I will do it again and inform you about
&gt; result and I hope I can have the new release soon which supports ensembl
&gt; v73.
&gt;
&gt; Best regards
&gt; A
&gt;
&gt;
&gt; On Wed, Oct 16, 2013 at 12:49 PM, Giuseppe Gallone via RT &lt;
&gt; bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:
&gt;
<div class="message-stanza open">&gt;&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt;
&gt;&gt; Hello Altan
&gt;&gt;
&gt;&gt; I am testing Bio::Homology::InterologWalk v 0.6b with Ensembl v64 and it
&gt;&gt; seems to work on my platform.
&gt;&gt;
&gt;&gt; In order to make it easier for me to debug the errors you&#39;re seeing,
&gt;&gt; could you please test it once more, following these steps &#40;I&#39;m assuming
&gt;&gt; you&#39;re on a linux machine&#41;:
&gt;&gt;
&gt;&gt; 1-Please delete any instances of the ensembl API you might have
&gt;&gt; previously installed. In your home directory, create a src directory and
&gt;&gt; cd into it.
&gt;&gt;
&gt;&gt; #------------
&gt;&gt; cd ~
&gt;&gt; mkdir src
&gt;&gt; cd src
&gt;&gt; #------------
&gt;&gt;
&gt;&gt; 2-Install Bioperl 1.2.3 as described here
&gt;&gt;
&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_cvs.html">http://www.ensembl.org/info/docs/api/api_cvs.html</a></span>
&gt;&gt;
&gt;&gt; 3-Install the full Ensembl-64 api typing the following commands:
&gt;&gt;
&gt;&gt; #-------------------------------------
&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
&gt;&gt; Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
&gt;&gt; CVS password: CVSUSER
&gt;&gt;
&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r
&gt;&gt; branch-ensembl-64 ensembl-api
&gt;&gt; #-------------------------------------
&gt;&gt; The second command is particularly important - you will need ALL the
&gt;&gt; APIs for the module to work correctly.
&gt;&gt;
&gt;&gt; 4-At this stage, if all has gone smoothly, you should have the following
&gt;&gt; four directories under /home/YOURNAME/src:
&gt;&gt;
&gt;&gt; bioperl-live
&gt;&gt; ensembl
&gt;&gt; ensembl-compara
&gt;&gt; ensembl-external
&gt;&gt; ensembl-functgenomics
&gt;&gt; ensembl-tools
&gt;&gt; ensembl-variation
&gt;&gt;
&gt;&gt; 5-Set up your environment as described here
&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_installation.html">http://www.ensembl.org/info/docs/api/api_installation.html</a></span>
&gt;&gt;
&gt;&gt; For example, for a fedora/RH linux machine, you&#39;d open your .bashrc file
&gt;&gt; and type in the paths to those directories in point &#40;4&#41;, as follows &#40;I&#39;m
&gt;&gt; assuming the vim editor is installed on your system, otherwise maybe try
&gt;&gt; gedit&#41;:
&gt;&gt;
&gt;&gt; #-------------
&gt;&gt; vim ~/.bashrc
&gt;&gt; #-------------
&gt;&gt; and type the following at the bottom of the file:
&gt;&gt; #--------------
&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/bioperl-1.2.3
&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl/modules
&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-compara/modules
&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-variation/modules
&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-functgenomics/modules
&gt;&gt; export PERL5LIB
&gt;&gt; #-----------------
&gt;&gt; #&#40;no need to export paths to ensembl-tools and ensembl-external&#41;
&gt;&gt;
&gt;&gt; Now save the file, and exit. Log off the session and log in again &#40;This
&gt;&gt; is very important for the changes you&#39;ve made to .bashrc to have effect&#41;
&gt;&gt;
&gt;&gt; 6-At this time your Ensembl/Bioperl configuration should be complete.
&gt;&gt; You may now attempt the installation of Bio::Homology::InterologWalk.
&gt;&gt; Download it from here:
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; wget
&gt;&gt;
&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz">http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz</a></span>
&gt;&gt; #----------
&gt;&gt;
&gt;&gt; then
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; tar xzvf Bio-Homology-InterologWalk-0.6b.tar.gz
&gt;&gt; #----------
&gt;&gt;
&gt;&gt; and
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; cd Bio-Homology-InterologWalk-0.6b
&gt;&gt; #----------
&gt;&gt;
&gt;&gt; now you can follow the included README for installation. Basically you&#39;ll
&gt;&gt; do
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; perl Makefile.PL
&gt;&gt; #----------
&gt;&gt;
&gt;&gt; the output should be the following &#40;provided you have all the
&gt;&gt; prerequisite modules installed&#41;
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; Checking for recommended module Algorithm::Combinatorics &gt;= 0.25 ... ok.
&gt;&gt; Checking for required module Bio::EnsEMBL::Registry ... ok.
&gt;&gt; Checking for required module Bio::Perl ... ok.
&gt;&gt; Checking for required module Carp ... ok.
&gt;&gt; Checking for required module DBD::CSV ... ok.
&gt;&gt; Checking for recommended module Data::PowerSet &gt;= 0.05 ... ok.
&gt;&gt; Checking for required module File::Glob ... ok.
&gt;&gt; Checking for required module GO::Parser ... ok.
&gt;&gt; Checking for recommended module Getopt::Long &gt;= 2.38 ... ok.
&gt;&gt; Checking for recommended module Graph &gt;= 0.94 ... ok.
&gt;&gt; Checking for required module List::Util ... ok.
&gt;&gt; Checking for required module REST::Client ... ok.
&gt;&gt; Checking for required module String::Approx ... ok.
&gt;&gt; Checking for recommended module URI::Escape &gt;= 1.53 ... ok.
&gt;&gt;
&gt;&gt; Checking if your kit is complete...
&gt;&gt; Looks good
&gt;&gt; Writing Makefile for Bio::Homology::InterologWalk
&gt;&gt; #-----------
&gt;&gt;
&gt;&gt; Then you should do
&gt;&gt;
&gt;&gt; #----------
&gt;&gt; make
&gt;&gt; #----------
&gt;&gt;
&gt;&gt; which should give you the following output
&gt;&gt;
&gt;&gt; #------------------
&gt;&gt; cp lib/Bio/Homology/InterologWalk.pm
&gt;&gt; blib/lib/Bio/Homology/InterologWalk.pm
&gt;&gt; Manifying blib/man3/Bio::Homology::InterologWalk.3pm
&gt;&gt; #------------------
&gt;&gt;
&gt;&gt; next do a
&gt;&gt;
&gt;&gt; #------------
&gt;&gt; make test
&gt;&gt; #------------
&gt;&gt; This will take a while, and it will need a internet connection. It will
&gt;&gt; fail if the Ensembl/Intact services are down. A typical successful run
&gt;&gt; would look like this:
&gt;&gt;
&gt;&gt; #----------------------------
&gt;&gt; PERL_DL_NONLAZY=1 /usr/bin/perl &#34;-MExtUtils::Command::MM&#34; &#34;-e&#34;
&gt;&gt; &#34;test_harness&#40;0, &#39;blib/lib&#39;, &#39;blib/arch&#39;&#41;&#34; t/*.t
&gt;&gt; t/Bio-Homology-InterologWalk.t .. 1/19 #
&gt;&gt; #
&gt;&gt; # #######PLEASE NOTE######
&gt;&gt; # ########################
&gt;&gt; # In order for some of the following tests to succeed,
&gt;&gt; # remote services provided by Ensembl and Intact MUST BE ACTIVE.
&gt;&gt; # in case of multiple test failure,
&gt;&gt; # please check status of Ensembl services on announce@ensembl.org
&gt;&gt; # and status of EBI Intact PSICQUIC service on
&gt;&gt; #
&gt;&gt; www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS
&gt;&gt; # ########################
&gt;&gt; # ########################
&gt;&gt; #
&gt;&gt; # Testing interfaces to remote services..Please be patient..
&gt;&gt; #
&gt;&gt; t/Bio-Homology-InterologWalk.t .. 13/19 add method for created_by at
&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt;&gt; add method for creation_date at
&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt;&gt; t/Bio-Homology-InterologWalk.t .. ok
&gt;&gt; All tests successful.
&gt;&gt; Files=1, Tests=19, 111 wallclock secs &#40; 0.05 usr  0.00 sys +  6.36 cusr
&gt;&gt;   0.26 csys =  6.67 CPU&#41;
&gt;&gt; Result: PASS
&gt;&gt; #----------------------------------
&gt;&gt;
&gt;&gt; After this you&#39;d just do a
&gt;&gt;
&gt;&gt; make install
&gt;&gt;
&gt;&gt; Let me know how it goes. I&#39;m in the process of simplifying the
&gt;&gt; documentation for a new release - apologies for the complexity.
&gt;&gt;
&gt;&gt; Cheers, G
&gt;&gt;
&gt;&gt;
&gt;&gt;
&gt;&gt; On 10/14/13 19:40, Altan Kara via RT wrote:
<div class="message-stanza open">&gt;&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt;&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt; &gt;
&gt;&gt; &gt; Hi Giuseppe,
&gt;&gt; &gt;
&gt;&gt; &gt; I install the ensembl version 64 and try to run the same process&#40;
&gt;&gt; &gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt;&gt; &gt;
&gt;&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt;&gt; &gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt;&gt; &gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt;&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt;&gt; &gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt;&gt; &gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt;&gt; &gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt;&gt; &gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt;&gt; &gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt;&gt; &gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt;&gt; &gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt;&gt; &gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt;&gt; &gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt;&gt; &gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt;&gt; &gt; near &#34;}&#34;
&gt;&gt; &gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt;&gt; &gt; Compilation failed in require at
&gt;&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt;&gt; line 7.
<div class="message-stanza open">&gt;&gt; &gt; BEGIN failed--compilation aborted at
&gt;&gt; &gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt;&gt; line 7.
<div class="message-stanza open">&gt;&gt; &gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt;&gt; &gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt;&gt; &gt; .......
&gt;&gt; &gt;
&gt;&gt; &gt; how can I fix this or can you send me registry file for ensembl v64
</div>&gt;&gt; which
<div class="message-stanza open">&gt;&gt; &gt; wont create this error?
&gt;&gt; &gt;
&gt;&gt; &gt; Best regards
&gt;&gt; &gt; Altan
&gt;&gt; &gt;
</div>&gt;&gt;
&gt;&gt;
</div>&gt;
</div></div></div>

<div class="messagebody">
</div>
</div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Thu&nbsp;Oct&nbsp;17&nbsp;06:20:04&nbsp;2013</span>
    <span class="description">giuseppe.gallone [...] dpag.ox.ac.uk -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Thu, 17 Oct 2013 11:19:45 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Giuseppe Gallone &lt;giuseppe.gallone [...] dpag.ox.ac.uk&gt;</td>
  </tr>
</table>

<div class="messagebody">
<div class="message-stanza plain-text-white-space">Dear Altan

The &#34;Registry.pm&#34; file is part of the Ensembl API v.64 distribution.

Best
Giuseppe

On 10/16/13 23:51, Altan Kara via RT wrote:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;         Queue: Bio-Homology-InterologWalk
&gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Hi Giuseppe,
&gt;
&gt; I checked all of my installation steps and everything looks fine, as you
&gt; know if I made an installation error at any step other Ensembl Apis also
&gt; wont work but only v64  is not working.
&gt; Can you just send me your &#34;Registry.pm&#34; for Ensembl v64?
&gt;
&gt; Best regards
&gt; Altan
&gt;
&gt; Best regards
&gt; Altan
&gt;
&gt;
&gt;
&gt;
&gt; On Wed, Oct 16, 2013 at 12:54 PM, Altan Kara &lt;karaltn@gmail.com&gt; wrote:
&gt;
<div class="message-stanza open">&gt;&gt; Hi,
&gt;&gt;
&gt;&gt; Honestly I did all these steps but I will do it again and inform you about
&gt;&gt; result and I hope I can have the new release soon which supports ensembl
&gt;&gt; v73.
&gt;&gt;
&gt;&gt; Best regards
&gt;&gt; A
&gt;&gt;
&gt;&gt;
&gt;&gt; On Wed, Oct 16, 2013 at 12:49 PM, Giuseppe Gallone via RT &lt;
&gt;&gt; bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:
&gt;&gt;
<div class="message-stanza open">&gt;&gt;&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt;&gt;
&gt;&gt;&gt; Hello Altan
&gt;&gt;&gt;
&gt;&gt;&gt; I am testing Bio::Homology::InterologWalk v 0.6b with Ensembl v64 and it
&gt;&gt;&gt; seems to work on my platform.
&gt;&gt;&gt;
&gt;&gt;&gt; In order to make it easier for me to debug the errors you&#39;re seeing,
&gt;&gt;&gt; could you please test it once more, following these steps &#40;I&#39;m assuming
&gt;&gt;&gt; you&#39;re on a linux machine&#41;:
&gt;&gt;&gt;
&gt;&gt;&gt; 1-Please delete any instances of the ensembl API you might have
&gt;&gt;&gt; previously installed. In your home directory, create a src directory and
&gt;&gt;&gt; cd into it.
&gt;&gt;&gt;
&gt;&gt;&gt; #------------
&gt;&gt;&gt; cd ~
&gt;&gt;&gt; mkdir src
&gt;&gt;&gt; cd src
&gt;&gt;&gt; #------------
&gt;&gt;&gt;
&gt;&gt;&gt; 2-Install Bioperl 1.2.3 as described here
&gt;&gt;&gt;
&gt;&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_cvs.html">http://www.ensembl.org/info/docs/api/api_cvs.html</a></span>
&gt;&gt;&gt;
&gt;&gt;&gt; 3-Install the full Ensembl-64 api typing the following commands:
&gt;&gt;&gt;
&gt;&gt;&gt; #-------------------------------------
&gt;&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
&gt;&gt;&gt; Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
&gt;&gt;&gt; CVS password: CVSUSER
&gt;&gt;&gt;
&gt;&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r
&gt;&gt;&gt; branch-ensembl-64 ensembl-api
&gt;&gt;&gt; #-------------------------------------
&gt;&gt;&gt; The second command is particularly important - you will need ALL the
&gt;&gt;&gt; APIs for the module to work correctly.
&gt;&gt;&gt;
&gt;&gt;&gt; 4-At this stage, if all has gone smoothly, you should have the following
&gt;&gt;&gt; four directories under /home/YOURNAME/src:
&gt;&gt;&gt;
&gt;&gt;&gt; bioperl-live
&gt;&gt;&gt; ensembl
&gt;&gt;&gt; ensembl-compara
&gt;&gt;&gt; ensembl-external
&gt;&gt;&gt; ensembl-functgenomics
&gt;&gt;&gt; ensembl-tools
&gt;&gt;&gt; ensembl-variation
&gt;&gt;&gt;
&gt;&gt;&gt; 5-Set up your environment as described here
&gt;&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_installation.html">http://www.ensembl.org/info/docs/api/api_installation.html</a></span>
&gt;&gt;&gt;
&gt;&gt;&gt; For example, for a fedora/RH linux machine, you&#39;d open your .bashrc file
&gt;&gt;&gt; and type in the paths to those directories in point &#40;4&#41;, as follows &#40;I&#39;m
&gt;&gt;&gt; assuming the vim editor is installed on your system, otherwise maybe try
&gt;&gt;&gt; gedit&#41;:
&gt;&gt;&gt;
&gt;&gt;&gt; #-------------
&gt;&gt;&gt; vim ~/.bashrc
&gt;&gt;&gt; #-------------
&gt;&gt;&gt; and type the following at the bottom of the file:
&gt;&gt;&gt; #--------------
&gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/bioperl-1.2.3
&gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl/modules
&gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-compara/modules
&gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-variation/modules
&gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-functgenomics/modules
&gt;&gt;&gt; export PERL5LIB
&gt;&gt;&gt; #-----------------
&gt;&gt;&gt; #&#40;no need to export paths to ensembl-tools and ensembl-external&#41;
&gt;&gt;&gt;
&gt;&gt;&gt; Now save the file, and exit. Log off the session and log in again &#40;This
&gt;&gt;&gt; is very important for the changes you&#39;ve made to .bashrc to have effect&#41;
&gt;&gt;&gt;
&gt;&gt;&gt; 6-At this time your Ensembl/Bioperl configuration should be complete.
&gt;&gt;&gt; You may now attempt the installation of Bio::Homology::InterologWalk.
&gt;&gt;&gt; Download it from here:
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; wget
&gt;&gt;&gt;
&gt;&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz">http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz</a></span>
&gt;&gt;&gt; #----------
&gt;&gt;&gt;
&gt;&gt;&gt; then
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; tar xzvf Bio-Homology-InterologWalk-0.6b.tar.gz
&gt;&gt;&gt; #----------
&gt;&gt;&gt;
&gt;&gt;&gt; and
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; cd Bio-Homology-InterologWalk-0.6b
&gt;&gt;&gt; #----------
&gt;&gt;&gt;
&gt;&gt;&gt; now you can follow the included README for installation. Basically you&#39;ll
&gt;&gt;&gt; do
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; perl Makefile.PL
&gt;&gt;&gt; #----------
&gt;&gt;&gt;
&gt;&gt;&gt; the output should be the following &#40;provided you have all the
&gt;&gt;&gt; prerequisite modules installed&#41;
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; Checking for recommended module Algorithm::Combinatorics &gt;= 0.25 ... ok.
&gt;&gt;&gt; Checking for required module Bio::EnsEMBL::Registry ... ok.
&gt;&gt;&gt; Checking for required module Bio::Perl ... ok.
&gt;&gt;&gt; Checking for required module Carp ... ok.
&gt;&gt;&gt; Checking for required module DBD::CSV ... ok.
&gt;&gt;&gt; Checking for recommended module Data::PowerSet &gt;= 0.05 ... ok.
&gt;&gt;&gt; Checking for required module File::Glob ... ok.
&gt;&gt;&gt; Checking for required module GO::Parser ... ok.
&gt;&gt;&gt; Checking for recommended module Getopt::Long &gt;= 2.38 ... ok.
&gt;&gt;&gt; Checking for recommended module Graph &gt;= 0.94 ... ok.
&gt;&gt;&gt; Checking for required module List::Util ... ok.
&gt;&gt;&gt; Checking for required module REST::Client ... ok.
&gt;&gt;&gt; Checking for required module String::Approx ... ok.
&gt;&gt;&gt; Checking for recommended module URI::Escape &gt;= 1.53 ... ok.
&gt;&gt;&gt;
&gt;&gt;&gt; Checking if your kit is complete...
&gt;&gt;&gt; Looks good
&gt;&gt;&gt; Writing Makefile for Bio::Homology::InterologWalk
&gt;&gt;&gt; #-----------
&gt;&gt;&gt;
&gt;&gt;&gt; Then you should do
&gt;&gt;&gt;
&gt;&gt;&gt; #----------
&gt;&gt;&gt; make
&gt;&gt;&gt; #----------
&gt;&gt;&gt;
&gt;&gt;&gt; which should give you the following output
&gt;&gt;&gt;
&gt;&gt;&gt; #------------------
&gt;&gt;&gt; cp lib/Bio/Homology/InterologWalk.pm
&gt;&gt;&gt; blib/lib/Bio/Homology/InterologWalk.pm
&gt;&gt;&gt; Manifying blib/man3/Bio::Homology::InterologWalk.3pm
&gt;&gt;&gt; #------------------
&gt;&gt;&gt;
&gt;&gt;&gt; next do a
&gt;&gt;&gt;
&gt;&gt;&gt; #------------
&gt;&gt;&gt; make test
&gt;&gt;&gt; #------------
&gt;&gt;&gt; This will take a while, and it will need a internet connection. It will
&gt;&gt;&gt; fail if the Ensembl/Intact services are down. A typical successful run
&gt;&gt;&gt; would look like this:
&gt;&gt;&gt;
&gt;&gt;&gt; #----------------------------
&gt;&gt;&gt; PERL_DL_NONLAZY=1 /usr/bin/perl &#34;-MExtUtils::Command::MM&#34; &#34;-e&#34;
&gt;&gt;&gt; &#34;test_harness&#40;0, &#39;blib/lib&#39;, &#39;blib/arch&#39;&#41;&#34; t/*.t
&gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. 1/19 #
&gt;&gt;&gt; #
&gt;&gt;&gt; # #######PLEASE NOTE######
&gt;&gt;&gt; # ########################
&gt;&gt;&gt; # In order for some of the following tests to succeed,
&gt;&gt;&gt; # remote services provided by Ensembl and Intact MUST BE ACTIVE.
&gt;&gt;&gt; # in case of multiple test failure,
&gt;&gt;&gt; # please check status of Ensembl services on announce@ensembl.org
&gt;&gt;&gt; # and status of EBI Intact PSICQUIC service on
&gt;&gt;&gt; #
&gt;&gt;&gt; www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS
&gt;&gt;&gt; # ########################
&gt;&gt;&gt; # ########################
&gt;&gt;&gt; #
&gt;&gt;&gt; # Testing interfaces to remote services..Please be patient..
&gt;&gt;&gt; #
&gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. 13/19 add method for created_by at
&gt;&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt;&gt;&gt; add method for creation_date at
&gt;&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line 7644.
&gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. ok
&gt;&gt;&gt; All tests successful.
&gt;&gt;&gt; Files=1, Tests=19, 111 wallclock secs &#40; 0.05 usr  0.00 sys +  6.36 cusr
&gt;&gt;&gt;    0.26 csys =  6.67 CPU&#41;
&gt;&gt;&gt; Result: PASS
&gt;&gt;&gt; #----------------------------------
&gt;&gt;&gt;
&gt;&gt;&gt; After this you&#39;d just do a
&gt;&gt;&gt;
&gt;&gt;&gt; make install
&gt;&gt;&gt;
&gt;&gt;&gt; Let me know how it goes. I&#39;m in the process of simplifying the
&gt;&gt;&gt; documentation for a new release - apologies for the complexity.
&gt;&gt;&gt;
&gt;&gt;&gt; Cheers, G
&gt;&gt;&gt;
&gt;&gt;&gt;
&gt;&gt;&gt;
&gt;&gt;&gt; On 10/14/13 19:40, Altan Kara via RT wrote:
<div class="message-stanza open">&gt;&gt;&gt;&gt;          Queue: Bio-Homology-InterologWalk
&gt;&gt;&gt;&gt;    Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; Hi Giuseppe,
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; I install the ensembl version 64 and try to run the same process&#40;
&gt;&gt;&gt;&gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1564,
&gt;&gt;&gt;&gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt;&gt;&gt;&gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt;&gt;&gt;&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt;&gt;&gt;&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt;&gt;&gt;&gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt;&gt;&gt;&gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt;&gt;&gt;&gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt;&gt;&gt;&gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt;&gt;&gt;&gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt;&gt;&gt;&gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt;&gt;&gt;&gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt;&gt;&gt;&gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt;&gt;&gt;&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1613,
&gt;&gt;&gt;&gt; near &#34;}&#34;
&gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt;&gt;&gt;&gt; Compilation failed in require at
&gt;&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt;&gt;&gt; line 7.
<div class="message-stanza open">&gt;&gt;&gt;&gt; BEGIN failed--compilation aborted at
&gt;&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt;&gt;&gt; line 7.
<div class="message-stanza open">&gt;&gt;&gt;&gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt;&gt;&gt;&gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt;&gt;&gt;&gt; .......
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; how can I fix this or can you send me registry file for ensembl v64
</div>&gt;&gt;&gt; which
<div class="message-stanza open">&gt;&gt;&gt;&gt; wont create this error?
&gt;&gt;&gt;&gt;
&gt;&gt;&gt;&gt; Best regards
&gt;&gt;&gt;&gt; Altan
&gt;&gt;&gt;&gt;
</div>&gt;&gt;&gt;
&gt;&gt;&gt;
</div>&gt;&gt;
</div>&gt;
</div>
-- 
Giuseppe Gallone, PhD
MRC career development fellow
MRC Functional Genomics Unit - DPAG
University of Oxford, UK
</div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      &nbsp;      
    </span>
    <span class="date">Thu&nbsp;Oct&nbsp;17&nbsp;07:15:56&nbsp;2013</span>
    <span class="description">karaltn [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #89439] bug report</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Thu, 17 Oct 2013 12:15:30 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-Homology-InterologWalk [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Altan Kara &lt;karaltn [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="messageattachments">

<div class="messagebody">
<div class="message-stanza plain-text-white-space">I know but which comes with my cvs installation not working fine, I mean
the source of the error is the Registry.pm which comes with Ensembl API v64
and you said that this API works just fine when you tried it so maybe I can
try with your registration file ?
Cheers
A


On Thu, Oct 17, 2013 at 11:20 AM, Giuseppe Gallone via RT &lt;
bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt;
&gt; Dear Altan
&gt;
&gt; The &#34;Registry.pm&#34; file is part of the Ensembl API v.64 distribution.
&gt;
&gt; Best
&gt; Giuseppe
&gt;
&gt; On 10/16/13 23:51, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt;         Queue: Bio-Homology-InterologWalk
&gt; &gt;   Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;
&gt; &gt; Hi Giuseppe,
&gt; &gt;
&gt; &gt; I checked all of my installation steps and everything looks fine, as you
&gt; &gt; know if I made an installation error at any step other Ensembl Apis also
&gt; &gt; wont work but only v64  is not working.
&gt; &gt; Can you just send me your &#34;Registry.pm&#34; for Ensembl v64?
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
&gt; &gt; Best regards
&gt; &gt; Altan
&gt; &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt;
&gt; &gt; On Wed, Oct 16, 2013 at 12:54 PM, Altan Kara &lt;karaltn@gmail.com&gt; wrote:
&gt; &gt;
<div class="message-stanza open">&gt; &gt;&gt; Hi,
&gt; &gt;&gt;
&gt; &gt;&gt; Honestly I did all these steps but I will do it again and inform you
</div></div>&gt; about
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt; result and I hope I can have the new release soon which supports ensembl
&gt; &gt;&gt; v73.
&gt; &gt;&gt;
&gt; &gt;&gt; Best regards
&gt; &gt;&gt; A
&gt; &gt;&gt;
&gt; &gt;&gt;
&gt; &gt;&gt; On Wed, Oct 16, 2013 at 12:49 PM, Giuseppe Gallone via RT &lt;
&gt; &gt;&gt; bug-Bio-Homology-InterologWalk@rt.cpan.org&gt; wrote:
&gt; &gt;&gt;
<div class="message-stanza open">&gt; &gt;&gt;&gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Hello Altan
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; I am testing Bio::Homology::InterologWalk v 0.6b with Ensembl v64 and
</div></div></div>&gt; it
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; seems to work on my platform.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; In order to make it easier for me to debug the errors you&#39;re seeing,
&gt; &gt;&gt;&gt; could you please test it once more, following these steps &#40;I&#39;m assuming
&gt; &gt;&gt;&gt; you&#39;re on a linux machine&#41;:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 1-Please delete any instances of the ensembl API you might have
&gt; &gt;&gt;&gt; previously installed. In your home directory, create a src directory
</div></div>&gt; and
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; cd into it.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #------------
&gt; &gt;&gt;&gt; cd ~
&gt; &gt;&gt;&gt; mkdir src
&gt; &gt;&gt;&gt; cd src
&gt; &gt;&gt;&gt; #------------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 2-Install Bioperl 1.2.3 as described here
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_cvs.html">http://www.ensembl.org/info/docs/api/api_cvs.html</a></span>
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 3-Install the full Ensembl-64 api typing the following commands:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #-------------------------------------
&gt; &gt;&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl login
&gt; &gt;&gt;&gt; Logging in to :pserver:cvsuser@cvs.sanger.ac.uk:2401/cvsroot/ensembl
&gt; &gt;&gt;&gt; CVS password: CVSUSER
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; cvs -d :pserver:cvsuser@cvs.sanger.ac.uk:/cvsroot/ensembl checkout -r
&gt; &gt;&gt;&gt; branch-ensembl-64 ensembl-api
&gt; &gt;&gt;&gt; #-------------------------------------
&gt; &gt;&gt;&gt; The second command is particularly important - you will need ALL the
&gt; &gt;&gt;&gt; APIs for the module to work correctly.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 4-At this stage, if all has gone smoothly, you should have the
</div></div>&gt; following
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; four directories under /home/YOURNAME/src:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; bioperl-live
&gt; &gt;&gt;&gt; ensembl
&gt; &gt;&gt;&gt; ensembl-compara
&gt; &gt;&gt;&gt; ensembl-external
&gt; &gt;&gt;&gt; ensembl-functgenomics
&gt; &gt;&gt;&gt; ensembl-tools
&gt; &gt;&gt;&gt; ensembl-variation
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 5-Set up your environment as described here
&gt; &gt;&gt;&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://www.ensembl.org/info/docs/api/api_installation.html">http://www.ensembl.org/info/docs/api/api_installation.html</a></span>
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; For example, for a fedora/RH linux machine, you&#39;d open your .bashrc
</div></div>&gt; file
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; and type in the paths to those directories in point &#40;4&#41;, as follows
</div></div>&gt; &#40;I&#39;m
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; assuming the vim editor is installed on your system, otherwise maybe
</div></div>&gt; try
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; gedit&#41;:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #-------------
&gt; &gt;&gt;&gt; vim ~/.bashrc
&gt; &gt;&gt;&gt; #-------------
&gt; &gt;&gt;&gt; and type the following at the bottom of the file:
&gt; &gt;&gt;&gt; #--------------
&gt; &gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/bioperl-1.2.3
&gt; &gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl/modules
&gt; &gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-compara/modules
&gt; &gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-variation/modules
&gt; &gt;&gt;&gt; PERL5LIB=${PERL5LIB}:${HOME}/src/ensembl-functgenomics/modules
&gt; &gt;&gt;&gt; export PERL5LIB
&gt; &gt;&gt;&gt; #-----------------
&gt; &gt;&gt;&gt; #&#40;no need to export paths to ensembl-tools and ensembl-external&#41;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Now save the file, and exit. Log off the session and log in again &#40;This
&gt; &gt;&gt;&gt; is very important for the changes you&#39;ve made to .bashrc to have
</div></div>&gt; effect&#41;
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; 6-At this time your Ensembl/Bioperl configuration should be complete.
&gt; &gt;&gt;&gt; You may now attempt the installation of Bio::Homology::InterologWalk.
&gt; &gt;&gt;&gt; Download it from here:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; wget
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
</div></div>&gt; <span class="clickylink"><a target="new" rel="nofollow" href="http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz">http://search.cpan.org/CPAN/authors/id/G/GG/GGALLONE/Bio-Homology-InterologWalk-0.6b.tar.gz</a></span>
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; then
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; tar xzvf Bio-Homology-InterologWalk-0.6b.tar.gz
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; and
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; cd Bio-Homology-InterologWalk-0.6b
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; now you can follow the included README for installation. Basically
</div></div>&gt; you&#39;ll
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; do
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; perl Makefile.PL
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; the output should be the following &#40;provided you have all the
&gt; &gt;&gt;&gt; prerequisite modules installed&#41;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; Checking for recommended module Algorithm::Combinatorics &gt;= 0.25 ...
</div></div>&gt; ok.
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; Checking for required module Bio::EnsEMBL::Registry ... ok.
&gt; &gt;&gt;&gt; Checking for required module Bio::Perl ... ok.
&gt; &gt;&gt;&gt; Checking for required module Carp ... ok.
&gt; &gt;&gt;&gt; Checking for required module DBD::CSV ... ok.
&gt; &gt;&gt;&gt; Checking for recommended module Data::PowerSet &gt;= 0.05 ... ok.
&gt; &gt;&gt;&gt; Checking for required module File::Glob ... ok.
&gt; &gt;&gt;&gt; Checking for required module GO::Parser ... ok.
&gt; &gt;&gt;&gt; Checking for recommended module Getopt::Long &gt;= 2.38 ... ok.
&gt; &gt;&gt;&gt; Checking for recommended module Graph &gt;= 0.94 ... ok.
&gt; &gt;&gt;&gt; Checking for required module List::Util ... ok.
&gt; &gt;&gt;&gt; Checking for required module REST::Client ... ok.
&gt; &gt;&gt;&gt; Checking for required module String::Approx ... ok.
&gt; &gt;&gt;&gt; Checking for recommended module URI::Escape &gt;= 1.53 ... ok.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Checking if your kit is complete...
&gt; &gt;&gt;&gt; Looks good
&gt; &gt;&gt;&gt; Writing Makefile for Bio::Homology::InterologWalk
&gt; &gt;&gt;&gt; #-----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Then you should do
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt; make
&gt; &gt;&gt;&gt; #----------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; which should give you the following output
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #------------------
&gt; &gt;&gt;&gt; cp lib/Bio/Homology/InterologWalk.pm
&gt; &gt;&gt;&gt; blib/lib/Bio/Homology/InterologWalk.pm
&gt; &gt;&gt;&gt; Manifying blib/man3/Bio::Homology::InterologWalk.3pm
&gt; &gt;&gt;&gt; #------------------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; next do a
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #------------
&gt; &gt;&gt;&gt; make test
&gt; &gt;&gt;&gt; #------------
&gt; &gt;&gt;&gt; This will take a while, and it will need a internet connection. It will
&gt; &gt;&gt;&gt; fail if the Ensembl/Intact services are down. A typical successful run
&gt; &gt;&gt;&gt; would look like this:
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; #----------------------------
&gt; &gt;&gt;&gt; PERL_DL_NONLAZY=1 /usr/bin/perl &#34;-MExtUtils::Command::MM&#34; &#34;-e&#34;
&gt; &gt;&gt;&gt; &#34;test_harness&#40;0, &#39;blib/lib&#39;, &#39;blib/arch&#39;&#41;&#34; t/*.t
&gt; &gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. 1/19 #
&gt; &gt;&gt;&gt; #
&gt; &gt;&gt;&gt; # #######PLEASE NOTE######
&gt; &gt;&gt;&gt; # ########################
&gt; &gt;&gt;&gt; # In order for some of the following tests to succeed,
&gt; &gt;&gt;&gt; # remote services provided by Ensembl and Intact MUST BE ACTIVE.
&gt; &gt;&gt;&gt; # in case of multiple test failure,
&gt; &gt;&gt;&gt; # please check status of Ensembl services on announce@ensembl.org
&gt; &gt;&gt;&gt; # and status of EBI Intact PSICQUIC service on
&gt; &gt;&gt;&gt; #
&gt; &gt;&gt;&gt;
</div></div>&gt; www.ebi.ac.uk/Tools/webservices/psicquic/registry/registry?action=STATUS
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; # ########################
&gt; &gt;&gt;&gt; # ########################
&gt; &gt;&gt;&gt; #
&gt; &gt;&gt;&gt; # Testing interfaces to remote services..Please be patient..
&gt; &gt;&gt;&gt; #
&gt; &gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. 13/19 add method for created_by at
&gt; &gt;&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line
</div></div>&gt; 7644.
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; add method for creation_date at
&gt; &gt;&gt;&gt; /usr/local/share/perl5/GO/Handlers/obj.pm line 289, &lt;GEN103&gt; line
</div></div>&gt; 7644.
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt; t/Bio-Homology-InterologWalk.t .. ok
&gt; &gt;&gt;&gt; All tests successful.
&gt; &gt;&gt;&gt; Files=1, Tests=19, 111 wallclock secs &#40; 0.05 usr  0.00 sys +  6.36 cusr
&gt; &gt;&gt;&gt;    0.26 csys =  6.67 CPU&#41;
&gt; &gt;&gt;&gt; Result: PASS
&gt; &gt;&gt;&gt; #----------------------------------
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; After this you&#39;d just do a
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; make install
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Let me know how it goes. I&#39;m in the process of simplifying the
&gt; &gt;&gt;&gt; documentation for a new release - apologies for the complexity.
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Cheers, G
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; On 10/14/13 19:40, Altan Kara via RT wrote:
<div class="message-stanza open">&gt; &gt;&gt;&gt;&gt;          Queue: Bio-Homology-InterologWalk
&gt; &gt;&gt;&gt;&gt;    Ticket &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=89439">https://rt.cpan.org/Ticket/Display.html?id=89439</a></span> &gt;
&gt; &gt;&gt;&gt;&gt;
&gt; &gt;&gt;&gt;&gt; Hi Giuseppe,
&gt; &gt;&gt;&gt;&gt;
&gt; &gt;&gt;&gt;&gt; I install the ensembl version 64 and try to run the same process&#40;
&gt; &gt;&gt;&gt;&gt; doInterologWalk.pl&#41; again and this time I got this registry error:
&gt; &gt;&gt;&gt;&gt;
&gt; &gt;&gt;&gt;&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line
</div></div></div>&gt; 1564,
<div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt;&gt; near &#34;$type qw &#40;core cdna vega otherfeatures rnaseq&#41;&#34;
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;@dbs&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1575.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1583.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$type&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1598.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$species_suffix&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1599.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$host&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1600.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$user&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1601.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$pass&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1602.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$port&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1603.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$wait_timeout&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1605.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$no_cache&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1606.
&gt; &gt;&gt;&gt;&gt; Global symbol &#34;$verbose&#34; requires explicit package name at
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line 1608.
&gt; &gt;&gt;&gt;&gt; syntax error at /root/ensembl/modules/Bio/EnsEMBL/Registry.pm line
</div></div>&gt; 1613,
<div class="message-stanza open"><div class="message-stanza open"><div class="message-stanza open"><div class="message-stanza open">&gt; &gt;&gt;&gt;&gt; near &#34;}&#34;
&gt; &gt;&gt;&gt;&gt; /root/ensembl/modules/Bio/EnsEMBL/Registry.pm has too many errors.
&gt; &gt;&gt;&gt;&gt; Compilation failed in require at
&gt; &gt;&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt; &gt;&gt;&gt; line 7.
<div class="message-stanza open">&gt; &gt;&gt;&gt;&gt; BEGIN failed--compilation aborted at
&gt; &gt;&gt;&gt;&gt; /usr/local/lib/perl5/site_perl/5.18.1/Bio/Homology/InterologWalk.pm
</div>&gt; &gt;&gt;&gt; line 7.
<div class="message-stanza open">&gt; &gt;&gt;&gt;&gt; Compilation failed in require at doInterologWalk.pl line 17.
&gt; &gt;&gt;&gt;&gt; BEGIN failed--compilation aborted at doInterologWalk.pl line 17.
&gt; &gt;&gt;&gt;&gt; .......
&gt; &gt;&gt;&gt;&gt;
&gt; &gt;&gt;&gt;&gt; how can I fix this or can you send me registry file for ensembl v64
</div>&gt; &gt;&gt;&gt; which
<div class="message-stanza open">&gt; &gt;&gt;&gt;&gt; wont create this error?
&gt; &gt;&gt;&gt;&gt;
&gt; &gt;&gt;&gt;&gt; Best regards
&gt; &gt;&gt;&gt;&gt; Altan
&gt; &gt;&gt;&gt;&gt;
</div>&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt;
</div>&gt; &gt;&gt;
</div>&gt; &gt;
</div>&gt;
&gt; --
&gt; Giuseppe Gallone, PhD
&gt; MRC career development fellow
&gt; MRC Functional Genomics Unit - DPAG
&gt; University of Oxford, UK
&gt;
&gt;
</div></div></div>

<div class="messagebody">
<p>Message body is not shown because it is too large.</p></div>
</div>
    </div>
</div>
</div>
</div>
    <hr class="clear" />
  </div>
</div>




</div>
 


  <hr class="clear" />
</div>
</body>
</html>
