Description: Fix spelling errors in manpage
Origin: vendor
Last-Update: 2011-06-27

--- a/Simple.pm
+++ b/Simple.pm
@@ -406,7 +406,7 @@
 
 =item redirect_output
 
-This allows to redirect the stdout and/or stderr output to a file.
+This allows one to redirect the stdout and/or stderr output to a file.
 Specify undef to leave the stderr/stdout handles of the process alone.
 
   # stdout to a file, left stderr unchanged
