From 73c3c54ec404f8ee0653fcb62ba34705562a2219 Mon Sep 17 00:00:00 2001
From: Gisle Aas <gisle@activestate.com>
Date: Wed, 7 Aug 2013 14:30:22 -0700
Subject: [PATCH] Workaround for perl-5.10.1 brokenness

---
 lib/Probe/Perl.pm |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/lib/Probe/Perl.pm b/lib/Probe/Perl.pm
index 3d8b404..64a0a12 100644
--- a/lib/Probe/Perl.pm
+++ b/lib/Probe/Perl.pm
@@ -56,7 +56,8 @@ sub _backticks {
   eval {open $fh, '-|', $perl, @_ or die $!};
   if (!$@) {
     return <$fh> if wantarray;
-    return do {local $/=undef; <$fh>};
+    my $tmp = do {local $/=undef; <$fh>};
+    return $tmp;
   }
 
   # Quoting only happens on the path to perl - I control the rest of
-- 
1.7.0.5

