--- ProhibitOneArgSelect.old	2008-09-02 14:40:01.000000000 -0400
+++ ProhibitOneArgSelect.pm	2008-12-24 09:07:13.000000000 -0500
@@ -37,7 +37,8 @@
     return if $elem ne 'select';
     return if ! is_function_call($elem);
 
-    if( scalar parse_arg_list($elem) == 1 ) {
+    my @args = parse_arg_list($elem);
+    if( scalar @args == 1 ) {
         return $self->violation( $DESC, $EXPL, $elem );
     }
     return; #ok!
