--- PDF/API2/Util.pm	2011-03-10 15:46:12.000000000 -0800
+++ PDF/API2/Util.pm	2012-06-06 16:17:19.000000000 -0700
@@ -126,10 +126,20 @@
     } else { return 0; }
 }
 
+sub detaint_float {
+    my $match = $_[0] =~ /^([-+]?\d+[.]?\d*)$/s;
+    $_[0] = $match ? $1 : undef;
+    return (defined($_[0]));
+}
+
+
 sub float {
     my $f=shift @_;
     my $mxd=shift @_||4;
     $f=0 if(abs($f)<0.0000000000000001);
+
+    detaint_float($f);
+
     my $ad=floor(xlog10($f)-$mxd);
     if(abs($f-int($f)) < (10**(-$mxd))) {
         # just in case we have an integer
