--- IMAPClient.pm.orig	2003-10-02 15:49:41.000000000 +0200
+++ IMAPClient.pm	2003-10-02 15:49:58.000000000 +0200
@@ -302,9 +302,9 @@
 
 	my $id   = $self->User;
 	my $has_quotes = $id =~ /^".*"$/ ? 1 : 0;
-	my $string = 	"Login " . ( $has_quotes ? $id : qq("$id") ) . " " . 
-			"{" . length($self->Password) . 
-			"}\r\n".$self->Password."\r\n";
+
+	my $string = "Login " . ( $has_quotes ? $id : qq("$id") ) . " " . $self->Password . "\r\n";
+
 	$self->_imap_command($string) 
 		and $self->State(Authenticated);
 	# $self->folders and $self->separator unless $self->NoAutoList;
