Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-02-07

--- a/lib/Net/OpenID/Extension/SimpleRegistration.pm
+++ b/lib/Net/OpenID/Extension/SimpleRegistration.pm
@@ -238,7 +238,7 @@
         email => $email,
     ));
 
-And finally, when back in Consumer recieving the response:
+And finally, when back in Consumer receiving the response:
 
     my $sreg_res = $verified_identity->get_extension_response('Net::OpenID::Extension::SimpleRegistration');
     my $nickname = $sreg_res->nickname;
