--- Clone.xs.orig       2008-07-12 18:49:01.000000000 +0200
+++ Clone.xs    2008-11-15 14:49:10.000000000 +0100
@@ -52,6 +52,7 @@
   while (next = hv_iternext (self))
     {
       SV *key = hv_iterkeysv (next);
+      TRACEME(("clone item %s\n", SvPV_nolen(key) ));
       hv_store_ent (clone, key, 
                 sv_clone (hv_iterval (self, next), recur), 0);
     }
@@ -251,6 +252,9 @@
             case '<':  /* PERL_MAGIC_backref */
              continue;
               break;
+            case '@':  /* PERL_MAGIC_arylen_p */
+             continue;
+              break;
             default:
               obj = sv_clone(mg->mg_obj, -1); 
           }

