Index: config-edit
===================================================================
--- config-edit	(revision 70521)
+++ config-edit	(working copy)
@@ -117,7 +117,7 @@
 
 =head2 upgrade mode
 
-  config-edit -ui none -application popcon -saved
+  config-edit -ui none -application popcon -save
 
 Loads configuration date, perform validation and migration of old data 
 (if specified by the model) and save the new configuration file.
