Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-01-23

--- a/lib/GraphViz.pm
+++ b/lib/GraphViz.pm
@@ -392,7 +392,7 @@
 node, edge and graph attributes (in addition to those controlled by
 the special attributes described above). The value should be a hash
 reference containing the corresponding key-value pairs. For example,
-to make all nodes box-shaped (unless explicity given another shape):
+to make all nodes box-shaped (unless explicitly given another shape):
 
   my $g = GraphViz->new(node => {shape => 'box'});
 
