Index: lib/ExtUtils/MM_Unix.pm
===================================================================
--- lib/ExtUtils/MM_Unix.pm	(revision 2196)
+++ lib/ExtUtils/MM_Unix.pm	(working copy)
@@ -2579,7 +2579,7 @@
 $(MAP_TARGET) :: static $(MAKE_APERL_FILE)
 	$(MAKE) $(USEMAKEFILE) $(MAKE_APERL_FILE) $@
 
-$(MAKE_APERL_FILE) : $(FIRST_MAKEFILE)
+$(MAKE_APERL_FILE) : $(FIRST_MAKEFILE) pm_to_blib.ts
 	$(NOECHO) $(ECHO) Writing \"$(MAKE_APERL_FILE)\" for this $(MAP_TARGET)
 	$(NOECHO) $(PERLRUNINST) \
 		Makefile.PL DIR=}, $dir, q{ \
@@ -3342,7 +3342,7 @@
 all :: %s
 	$(NOECHO) $(NOOP)
 
-%s :: %s
+%s :: %s pm_to_blib.ts
 	$(PERLRUNINST) %s
 MAKE_FRAG
 
@@ -4028,3 +4028,4 @@
 =cut
 
 __END__
+# vim: ts=8 sw=4 noet
