--- PAR-0.956/lib/PAR.pm        2006-10-03 08:50:00.000000000 -0400
+++ /Users/jesse/build/PAR-0.952/lib/PAR.pm     2006-10-03 18:44:15.000000000 -0400
@@ -302,6 +302,20 @@
             "script/main.pl",
             "main.pl",
         );
+    
+        if ($progname and !$member) {
+            require File::Spec;
+            my @path = File::Spec->splitdir($progname);
+            my $filename = pop @path;
+            $member = _first_member( $zip,
+                "script/".$filename,
+                "script/".$filename.".pl",
+                $filename,
+                $filename.".pl",
+            )
+
+        }
+
 
         # finally take $ARGV[0] as the hint for file to run
         if (defined $ARGV[0] and !$member) {
@@ -313,7 +327,10 @@
             ) or die qq(PAR.pm: Can't open perl script "$ARGV[0]": No such file or directory);
             shift @ARGV;
         }
-        elsif (!$member) {
+        
+        
+        
+        if (!$member) {
             die "Usage: $0 script_file_name.\n";
