--- Nameserver.pm.orig	2005-10-25 16:03:00.758328000 -0700
+++ Nameserver.pm	2005-10-25 16:03:55.908095000 -0700
@@ -15,11 +15,9 @@
  	    @DEFAULT_ADDR       
  	    $DEFAULT_PORT
  	    );
-use constant {
-	STATE_ACCEPTED	=> 1,
-	STATE_GOT_LENGTH=> 2,
-	STATE_SENDING	=> 3
-    };
+use constant STATE_ACCEPTED	=> 1;
+use constant STATE_GOT_LENGTH	=> 2;
+use constant STATE_SENDING	=> 3;
 
 $VERSION = (qw$LastChangedRevision: 502 $)[1];
 
