Description: add encoding to POD, required by newer perls
Origin: vendor
Bug-Debian: http://bugs.debian.org/710995
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-05

--- a/bin/chartex
+++ b/bin/chartex
@@ -214,6 +214,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/BIFFwriter.pm
+++ b/lib/Spreadsheet/WriteExcel/BIFFwriter.pm
@@ -292,6 +292,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Big.pm
+++ b/lib/Spreadsheet/WriteExcel/Big.pm
@@ -52,7 +52,7 @@
 
 __END__
 
-
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Examples.pm
+++ b/lib/Spreadsheet/WriteExcel/Examples.pm
@@ -24,6 +24,8 @@
 
 =pod
 
+=encoding latin1
+
 =head1 NAME
 
 Examples - Spreadsheet::WriteExcel example programs.
--- a/lib/Spreadsheet/WriteExcel/Format.pm
+++ b/lib/Spreadsheet/WriteExcel/Format.pm
@@ -789,6 +789,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Formula.pm
+++ b/lib/Spreadsheet/WriteExcel/Formula.pm
@@ -1397,6 +1397,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/OLEwriter.pm
+++ b/lib/Spreadsheet/WriteExcel/OLEwriter.pm
@@ -425,6 +425,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Properties.pm
+++ b/lib/Spreadsheet/WriteExcel/Properties.pm
@@ -328,6 +328,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Utility.pm
+++ b/lib/Spreadsheet/WriteExcel/Utility.pm
@@ -52,6 +52,7 @@
 
 
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Workbook.pm
+++ b/lib/Spreadsheet/WriteExcel/Workbook.pm
@@ -3614,6 +3614,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
--- a/lib/Spreadsheet/WriteExcel/Worksheet.pm
+++ b/lib/Spreadsheet/WriteExcel/Worksheet.pm
@@ -7619,6 +7619,7 @@
 
 __END__
 
+=encoding latin1
 
 =head1 NAME
 
