diff -r -u DBIx-Class-EncodedColumn-0.00002/Changes DBIx-Class-EncodedColumn-0.00003/Changes
--- DBIx-Class-EncodedColumn-0.00002/Changes	2008-07-29 00:16:02.000000000 +0200
+++ DBIx-Class-EncodedColumn-0.00003/Changes	2009-07-27 10:13:26.000000000 +0200
@@ -1,3 +1,6 @@
+0.00003        2009-07-27
+        - Win32 compatibility patch for 02digest.t
+
 0.00002        2008-07-27
         - Support for Crypt::OpenPGP
 
diff -r -u DBIx-Class-EncodedColumn-0.00002/lib/DBIx/Class/EncodedColumn.pm DBIx-Class-EncodedColumn-0.00003/lib/DBIx/Class/EncodedColumn.pm
--- DBIx-Class-EncodedColumn-0.00002/lib/DBIx/Class/EncodedColumn.pm	2008-07-29 00:40:02.000000000 +0200
+++ DBIx-Class-EncodedColumn-0.00003/lib/DBIx/Class/EncodedColumn.pm	2009-07-27 10:13:26.000000000 +0200
@@ -9,7 +9,7 @@
 
 __PACKAGE__->mk_classdata( _column_encoders => {} );
 
-our $VERSION = '0.00002';
+our $VERSION = '0.00003';
 
 sub register_column {
   my $self = shift;
diff -r -u DBIx-Class-EncodedColumn-0.00002/META.yml DBIx-Class-EncodedColumn-0.00003/META.yml
--- DBIx-Class-EncodedColumn-0.00002/META.yml	2008-07-29 00:42:44.000000000 +0200
+++ DBIx-Class-EncodedColumn-0.00003/META.yml	2009-07-27 10:13:26.000000000 +0200
@@ -24,4 +24,4 @@
   Sub::Name: 0.04
 resources:
   license: http://dev.perl.org/licenses/
-version: 0.00002
+version: 0.00003
diff -r -u DBIx-Class-EncodedColumn-0.00002/t/02digest.t DBIx-Class-EncodedColumn-0.00003/t/02digest.t
--- DBIx-Class-EncodedColumn-0.00002/t/02digest.t	2008-07-29 00:29:12.000000000 +0200
+++ DBIx-Class-EncodedColumn-0.00003/t/02digest.t	2009-07-27 10:13:26.000000000 +0200
@@ -178,6 +178,7 @@
 
 }
 
+$schema->storage()->disconnect();
 DigestTest->clear;
 
 #TODO
