--- Pluggable.pm.orig	Sat Jul 09 08:20:38 2005
+++ Pluggable.pm	Wed Jul 27 08:33:18 2005
@@ -417,11 +417,32 @@
               return $opts{'search_path'};
     };
 
+    my $onlysub = sub {
+        my ($self, $only) = @_;
+
+        if (defined $only) {
+            $opts{'only'} = $only;
+        };
+        
+        return $opts{'only'};
+    };
+
+    my $exceptsub = sub {
+        my ($self, $except) = @_;
+
+        if (defined $except) {
+            $opts{'except'} = $except;
+        };
+        
+        return $opts{'except'};
+    };
 
     no strict 'refs';
     no warnings 'redefine';
     *{"$pkg\::$sub"} = $subroutine;
     *{"$pkg\::search_path"} = $searchsub;
+    *{"$pkg\::only"} = $onlysub;
+    *{"$pkg\::except"} = $exceptsub;
 }
 
 
