--- MM.pm.0	2011-08-10 21:33:13.107912381 +0200
+++ MM.pm	2011-08-10 22:09:38.343916883 +0200
@@ -14,6 +14,8 @@
 
 my $Is_VMS = $^O eq 'VMS';
 
+use Time::HiRes qw(stat);
+sub mtime($) { (stat($_[0]))[9] }
 
 =head1 NAME
 
@@ -131,8 +133,8 @@
         my ($pod, $man) = splice(@ARGV, 0, 2);
 
         next if ((-e $man) &&
-                 (-M $man < -M $pod) &&
-                 (-M $man < -M "Makefile"));
+                 (mtime $man > mtime $pod) &&
+                 (mtime $man > mtime "Makefile"));
 
         print "Manifying $man\n";
 
