diff -urN Text-Column-0.05/lib/Text/Column.pm Text-Column-0.05-patch/lib/Text/Column.pm
--- Text-Column-0.05/lib/Text/Column.pm	2012-03-26 11:03:18.000046000 +0200
+++ Text-Column-0.05-patch/lib/Text/Column.pm	2012-03-26 11:37:04.000011000 +0200
@@ -139,6 +139,7 @@
 
         $r_total = 0;     
         foreach $r (@r) {
+            $r='' if not defined($r);
             $r_total += length( $r);   
         }
 
@@ -184,7 +185,7 @@
 
             $r_total = 0;     
             foreach $r (@r) {
-                $r_total = length( $r);   
+                $r_total += length( $r);   
             }
         }
 
@@ -447,4 +448,4 @@
 
 =cut
 
-### end of file ###
\ No newline at end of file
+### end of file ###
