--- Filter-Crypto-1.14/inc/Module/Install/PRIVATE/Filter/Crypto.pm	2005-06-01 12:02:30.000000000 +0200
+++ Filter-Crypto-1.14_cosimo/inc/Module/Install/PRIVATE/Filter/Crypto.pm	2005-09-01 10:05:25.301147947 +0200
@@ -465,7 +465,13 @@
     # or out/ (up to and including 0.8.1b).
     my $prefix_dir = $self->_prefix_dir();
     my($dir, $lib_dir);
-    if (-d ($dir = catdir($prefix_dir, 'lib'))) {
+
+    # Detect 64bit libraries (x86_64 machines)
+    if (-d ($dir = catdir($prefix_dir, 'lib64'))) {
+        $lib_dir = $dir;
+    }
+    # Or plain `lib'...
+    elsif (-d ($dir = catdir($prefix_dir, 'lib'))) {
         $lib_dir = $dir;
     }
     elsif ($self->is_win32()) {
