--- Net-DNS-1.06_05/lib/Net/DNS/Packet.pm	2016-11-12 11:23:47.000000000 +0800
+++ Net-DNS-1.06_05.fix/lib/Net/DNS/Packet.pm	2016-12-28 16:13:02.577195268 +0800
@@ -237,10 +237,12 @@
 =cut
 
 sub edns {
-	my $self = shift;
-	my $link = \$self->{xedns};
-	($$link) = grep $_->isa(qw(Net::DNS::RR::OPT)), @{$self->{additional}} unless $$link;
-	$$link = new Net::DNS::RR( type => 'OPT' ) unless $$link;
+    my $self = shift;
+    my $link ;
+    $link = \$self->{xedns} if(exists $self->{xedns});
+    ($$link) = grep $_->isa(qw(Net::DNS::RR::OPT)), @{$self->{additional}} unless ($link and $$link);
+    $$link = new Net::DNS::RR( type => 'OPT' ) unless ($link and $$link);
+    return $$link;
 }
 
 
