From 1695e23193d9bba4c230ab6a77a26ac37d352f66 Mon Sep 17 00:00:00 2001
From: moltar <moltar@moltar.net>
Date: Fri, 27 Dec 2013 22:01:15 +0200
Subject: [PATCH] Recursively copy from temp dir to target dir.

---
 lib/DBIx/Class/Fixtures.pm | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/lib/DBIx/Class/Fixtures.pm b/lib/DBIx/Class/Fixtures.pm
index 87ff12c..704f8f4 100644
--- a/lib/DBIx/Class/Fixtures.pm
+++ b/lib/DBIx/Class/Fixtures.pm
@@ -718,8 +718,7 @@ sub dump {
   }
 
   $self->msg("- moving temp dir to $output_dir");
-  move($_, dir($output_dir, $_->relative($_->parent)->stringify)) 
-    for $tmp_output_dir->children;
+  dircopy($tmp_output_dir, $output_dir);
 
   if (-e $output_dir) {
     $self->msg("- clearing tmp dir $tmp_output_dir");
-- 
1.8.3.4 (Apple Git-47)

