--- /Users/mblom/perl/5.10/lib/Image/TIFF.pm	2009-09-09 09:45:50.000000000 +0200
+++ lib/Image/TIFF.pm	2009-09-09 11:24:29.000000000 +0200
@@ -9,7 +9,7 @@
 use strict;
 use vars qw($VERSION);
 
-$VERSION = '1.06';
+$VERSION = '1.06_01';
 
 my @types = (
   [ "BYTE",      "C1", 1],
@@ -970,7 +970,7 @@
 		    #print STDERR "# Decoding Makernotes from $maker\n";
 
                     $self->{tag_prefix} = $tag_prefix;
-	            if ($ifd_off == -1) {
+	            if ($ifd_off == -1 && length($val) >= 12) {
                		# fuji kludge -  http://www.butaman.ne.jp/~tsuruzoh/Computer/Digicams/exif-e.html#APP4
                 	my $save_endian = $self->{little_endian};
                 	$self->{little_endian} = 1;
