From 9d9cce744e8cb9c6573578beb18cb25089cd6b86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Tue, 10 Feb 2015 11:44:18 +0100
Subject: [PATCH] Install mce_grep
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

CPAN RT#102040

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Makefile.PL | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 9508423..6fdf236 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -33,7 +33,9 @@ WriteMakefile(
 
    VERSION        => '1.600',
 
-   EXE_FILES      => [],
+   EXE_FILES      => [
+       'bin/mce_grep'
+   ],
 
    (($ExtUtils::MakeMaker::VERSION lt '6.25') ?
       (PL_FILES => { }) : ()),
-- 
1.9.3

