--- lib/POE/Component/SSLify.pm	2008-08-08 02:03:18.000000000 +0100
+++ /usr/lib/perl5/site_perl/5.8.8/POE/Component/SSLify.pm	2009-02-04 10:20:46.000000000 +0000
@@ -117,13 +117,15 @@
 	# Get the socket!
 	my $socket = shift;
 
+	my $custom_ctx = shift;
+
 	# Validation...
 	if ( ! defined $socket ) {
 		die "Did not get a defined socket";
 	}
 
 	# If we don't have a ctx ready, we can't do anything...
-	if ( ! defined $ctx ) {
+	if ( !$custom_ctx and ! defined $ctx ) {
 		die 'Please do SSLify_Options() first';
 	}
 
@@ -132,7 +134,7 @@
 
 	# Now, we create the new socket and bind it to our subclass of Net::SSLeay::Handle
 	my $newsock = gensym();
-	tie( *$newsock, 'POE::Component::SSLify::ServerHandle', $socket, $ctx ) or die "Unable to tie to our subclass: $!";
+	tie( *$newsock, 'POE::Component::SSLify::ServerHandle', $socket, ( $custom_ctx || $ctx ) ) or die "Unable to tie to our subclass: $!";
 
 	# All done!
 	return $newsock;
