--- Size.xs	2008-08-24 11:24:04.000000000 +0200
+++ /ms/dev/perl5/Devel-Size/0.71/src/distro/Size.xs	2009-04-30 16:16:21.000000000 +0200
@@ -767,10 +767,11 @@
 
 	case SVt_PVAV:
 	  {
+	    AV *tempAV;
+	    SV **tempSV;
 	    dbg_printf(("# Found type AV\n"));
 	    /* Quick alias to cut down on casting */
-	    AV *tempAV = (AV *)thing;
-	    SV **tempSV;
+	    tempAV = (AV *)thing;
 	    
 	    /* Any elements? */
 	    if (av_len(tempAV) != -1) {
