Description: Fixing failed test.
Author: Fabrizio Regalli <fabreg@fabreg.it>
Last-Update: 2011-06-19
Forwarded: 

--- a/t/00-compile.t
+++ b/t/00-compile.t
@@ -7,7 +7,7 @@
 
 
 
-  use File::Find;
+use File::Find;
 use File::Temp qw{ tempdir };
 
 my @modules;
@@ -19,6 +19,7 @@
         $found =~ s{[/\\]}{::}g;
         $found =~ s/\.pm$//;
         # nothing to skip push @modules, $found;
+	push @modules, $found;
     },
     'lib',
 );
@@ -30,6 +31,7 @@
             return unless -f;
             my $found = $File::Find::name;
             # nothing to skip push @scripts, $found;
+	    push @scripts, $found;
         },
         'bin',
     );
@@ -42,6 +44,7 @@
 
     # fake home for cpan-testers
     # no fake requested ## local $ENV{HOME} = tempdir( CLEANUP => 1 );
+    local $ENV{HOME} = tempdir( CLEANUP => 1 );
 
     like( qx{ $^X -Ilib -e "require $_; print '$_ ok'" },
         qr/^\s*$_ ok/s, "$_ loaded ok" )
