--- axestype.pm.bk.phil	2006-09-22 00:37:48.000000000 -0700
+++ axestype.pm	2006-09-22 00:58:14.000000000 -0700
@@ -1902,9 +1902,12 @@
     my $y_min = $self->{y_min}[$axis];
     my $y_max = $self->{y_max}[$axis];
 
+    my $y_diff = $y_max - $y_min;
+    $y_diff = 1 if ($y_diff == 0);
+
     my $y_step = $self->{rotate_chart} ?
-        abs(($self->{right} - $self->{left})/($y_max - $y_min)) :
-        abs(($self->{bottom} - $self->{top})/($y_max - $y_min));
+        abs(($self->{right} - $self->{left})/$y_diff) :
+        abs(($self->{bottom} - $self->{top})/$y_diff);
 
     my $ret_x;
     my $origin = $self->{rotate_chart} ? $self->{top} : $self->{left};
