--- old/Math-Pari-2.01080603/utils/Math/PariBuild.pm	2009-12-11 02:45:28.000000000 +0100
+++ Math-Pari-2.01080603/utils/Math/PariBuild.pm	2010-02-27 09:38:41.000000000 +0100
@@ -1029,6 +1029,7 @@ sub inline_headers_arr {     # These fil
 	       # ppc is not done yet (2.0.15)
 	($pari_version > 2002007
 		? (ppc		     => ['ppc/asm0.h', 'none/divll.h'],
+                   x86_64            => ['x86_64/asm0.h','none/level1.h'],
 		   ia64		     => ['ia64/asm0.h','ia64/asm1.h'])
 		: ()),
 	       sparcv7	      => ['none/asm0.h','none/level1.h'],
@@ -1125,7 +1126,7 @@ sub choose_and_report_assembler {
 			       ? 'hppa' : 'none'),
 	     );
   my $asmarch = $asmarch{$machine} || $machine; # Temporary only
-  my %skip64 = (alpha => 1, none => 1, ia64 => 1);
+  my %skip64 = (alpha => 1, none => 1, ia64 => 1, x86_64 => 1,);
   if (not $skip64{$asmarch}
       and ($Config{longsize} || 0) == 8) {
     $asmarch .= '_64';
@@ -1601,7 +1602,7 @@ sub code_C_translator {
   my @t = ExtUtils::Constant::constant_types(); # macro defs
   my @tt =
     ExtUtils::Constant::C_constant(
-	  'Math::Pari::func_type', 
+	  'Math::Pari::func_type',
           'func_ord_by_type', undef, undef, undef, undef,
           map {{name => $_, value => $codes{$_}, macro => 1}} keys %codes);
 
