Index: libmoose-perl/lib/Moose/Meta/Role/Application/RoleSummation.pm
===================================================================
--- libmoose-perl.orig/lib/Moose/Meta/Role/Application/RoleSummation.pm	2008-02-08 11:35:26.000000000 +0200
+++ libmoose-perl/lib/Moose/Meta/Role/Application/RoleSummation.pm	2008-02-08 11:40:50.000000000 +0200
@@ -230,7 +230,7 @@
 
 =head1 NAME
 
-Moose::Meta::Role::Application::RoleSummation
+Moose::Meta::Role::Application::RoleSummation - unite non-conflicting bindings
 
 =head1 DESCRIPTION
 
Index: libmoose-perl/lib/Moose/Meta/Role/Application/ToClass.pm
===================================================================
--- libmoose-perl.orig/lib/Moose/Meta/Role/Application/ToClass.pm	2008-02-08 11:35:26.000000000 +0200
+++ libmoose-perl/lib/Moose/Meta/Role/Application/ToClass.pm	2008-02-08 11:40:50.000000000 +0200
@@ -179,7 +179,7 @@
 
 =head1 NAME
 
-Moose::Meta::Role::Application::ToClass
+Moose::Meta::Role::Application::ToClass - convert to a class
 
 =head1 DESCRIPTION
 
Index: libmoose-perl/lib/Moose/Meta/Role/Application/ToInstance.pm
===================================================================
--- libmoose-perl.orig/lib/Moose/Meta/Role/Application/ToInstance.pm	2008-02-08 11:35:26.000000000 +0200
+++ libmoose-perl/lib/Moose/Meta/Role/Application/ToInstance.pm	2008-02-08 11:40:50.000000000 +0200
@@ -42,7 +42,7 @@
 
 =head1 NAME
 
-Moose::Meta::Role::Application::ToInstance
+Moose::Meta::Role::Application::ToInstance - apply to an instance
 
 =head1 DESCRIPTION
 
Index: libmoose-perl/lib/Moose/Meta/Role/Application/ToRole.pm
===================================================================
--- libmoose-perl.orig/lib/Moose/Meta/Role/Application/ToRole.pm	2008-02-08 11:35:26.000000000 +0200
+++ libmoose-perl/lib/Moose/Meta/Role/Application/ToRole.pm	2008-02-08 11:40:50.000000000 +0200
@@ -160,7 +160,7 @@
 
 =head1 NAME
 
-Moose::Meta::Role::Application::ToRole
+Moose::Meta::Role::Application::ToRole - apply to a role
 
 =head1 DESCRIPTION
 
Index: libmoose-perl/lib/Moose/Meta/Role/Application.pm
===================================================================
--- libmoose-perl.orig/lib/Moose/Meta/Role/Application.pm	2008-02-08 11:40:55.000000000 +0200
+++ libmoose-perl/lib/Moose/Meta/Role/Application.pm	2008-02-08 11:41:25.000000000 +0200
@@ -89,7 +89,7 @@
 
 =head1 NAME
 
-Moose::Meta::Role::Application
+Moose::Meta::Role::Application - base class for role applications
 
 =head1 DESCRIPTION
 
