diff --git a/README b/README
index 7079aee..b7c96d6 100644
--- a/README
+++ b/README
@@ -56,7 +56,8 @@ The LDAP attribute can also be a subroutine reference
 that returns either an arrayref or a list of attributes
 
 By default users are created as Unprivileged, but you can change this by
-setting $LDAPCreatePrivileged to 1.
+setting $LDAPCreatePrivileged to 1.  If the privilege status for users is 
+required to be updated then also set $LDAPUpdateSetsPrivileged to 1.
 
 For more information on these see the import_users documentation
 in RT::Extension::LDAPImport
diff --git a/lib/RT/Extension/LDAPImport.pm b/lib/RT/Extension/LDAPImport.pm
index c6022d7..938456f 100644
--- a/lib/RT/Extension/LDAPImport.pm
+++ b/lib/RT/Extension/LDAPImport.pm
@@ -146,7 +146,8 @@ If it is an arrayref, the values will be concatenated
 together with a single space.
 
 By default users are created as Unprivileged, but you can change this by
-setting $LDAPCreatePrivileged to 1.
+setting $LDAPCreatePrivileged to 1.  If the privilege status for users is 
+required to be updated then also set $LDAPUpdateSetsPrivileged to 1.
 
 =cut
 
@@ -350,6 +351,7 @@ sub create_rt_user {
         if ($RT::LDAPUpdateUsers || $RT::LDAPUpdateOnly) {
             $self->_debug("$message, updating their data");
             if ($args{import}) {
+		$user->{'Privileged'} = $RT::LDAPCreatePrivileged if $RT::LDAPUpdateSetsPrivileged;
                 my @results = $user_obj->Update( ARGSRef => $user, AttributesRef => [keys %$user] );
                 $self->_debug(join("\n",@results)||'no change');
             } else {
