*** InsideOut.original	Wed Mar 14 23:17:30 2007
--- InsideOut.new	Wed Mar 14 23:19:38 2007
***************
*** 211,217 ****
          }
  
          # Restricted class
!         if ($pkg =~ /^:RESTRICT(?:ED)?\((.*)\)/i) {
              *{$class.'::new'}
                  = wrap_RESTRICTED($class, 'new',
                                    sub { goto &Object::InsideOut::new },
--- 211,217 ----
          }
  
          # Restricted class
!         if ($pkg =~ /^:RESTRICT(?:ED)?(?:\((.*)\))?/i) {
              *{$class.'::new'}
                  = wrap_RESTRICTED($class, 'new',
                                    sub { goto &Object::InsideOut::new },
***************
*** 223,229 ****
          }
  
          # Private class
!         if ($pkg =~ /^:PRIV(?:ATE)?\((.*)\)/i) {
              *{$class.'::new'}
                  = wrap_PRIVATE($class, 'new',
                                 sub { goto &Object::InsideOut::new },
--- 223,229 ----
          }
  
          # Private class
!         if ($pkg =~ /^:PRIV(?:ATE)?(?:\((.*)\))?/i) {
              *{$class.'::new'}
                  = wrap_PRIVATE($class, 'new',
                                 sub { goto &Object::InsideOut::new },
