Only in .: blib
diff -ru ../Net-HTTP-6.01-0orig/lib/Net/HTTP/Methods.pm ./lib/Net/HTTP/Methods.pm
--- ../Net-HTTP-6.01-0orig/lib/Net/HTTP/Methods.pm	2011-03-20 04:36:18.000000000 -0700
+++ ./lib/Net/HTTP/Methods.pm	2011-10-18 12:24:35.000000000 -0700
@@ -135,6 +135,7 @@
     push(@{${*$self}{'http_request_method'}}, $method);
     my $ver = ${*$self}{'http_version'};
     my $peer_ver = ${*$self}{'http_peer_http_version'} || "1.0";
+    $ver = '1.0' if $method eq 'CONNECT';
 
     my @h;
     my @connection;
@@ -162,7 +163,7 @@
     if ($given{te}) {
 	push(@connection, "TE") unless grep lc($_) eq "te", @connection;
     }
-    elsif ($self->send_te && gunzip_ok()) {
+    elsif ($self->send_te && gunzip_ok() && $method ne 'CONNECT') {
 	# gzip is less wanted since the IO::Uncompress::Gunzip interface for
 	# it does not really allow chunked decoding to take place easily.
 	push(@h2, "TE: deflate,gzip;q=0.3");
Only in ./lib/Net/HTTP: Methods.pm~
Only in .: Makefile
Only in .: pm_to_blib
