diff -u ../vvanilla/vperl/vpp.pm vperl/vpp.pm
--- ../vvanilla/vperl/vpp.pm	2009-09-07 19:25:38.000000000 -0400
+++ vperl/vpp.pm	2009-10-08 22:37:43.000000000 -0400
@@ -496,10 +496,11 @@
 sub _un_vstring {
     my $value = shift;
     # may be a v-string
-    if ( $] >= 5.006_000 && length($value) == 3 && $value !~ /[._]/
+    if ( $] >= 5.006_000 && length($value) >= 3 && $value !~ /[._]/
 	&& (ord($value) < ord('0') || ord($value) > ord('9')) ) {
 	my $tvalue = sprintf("v%vd",$value);
-	if ( $tvalue =~ /^v\d+\.\d+\.\d+$/ ) {
+	#if ( $tvalue =~ /^v\d+\.\d+\.[\d.]*\d$/ ) {
+	if ( $tvalue =~ /^v\d+(\.\d+){2,}$/ ) {
 	    # must be a v-string
 	    $value = $tvalue;
 	}
