Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject: Spelling
Forwarded: no
Last-Update: 2011-06-18
--- a/AutoRunmode/FileDelegate.pm
+++ b/AutoRunmode/FileDelegate.pm
@@ -108,7 +108,7 @@
 =head1 SEE ALSO
 
 If you like the idea of moving everything outside of Perl modules
-into seperate files, you should also have a look at
+into separate files, you should also have a look at
 L<CGI::Application::Plugin::TemplateRunner>, which does a similar
 thing for HTML templates and the Perl code needed to provide them with 
 data.
