diff -up Net-DNS-0.63/lib/Net/DNS/Resolver/Base.pm.BAD Net-DNS-0.63/lib/Net/DNS/Resolver/Base.pm
--- Net-DNS-0.63/lib/Net/DNS/Resolver/Base.pm.BAD	2008-04-09 10:19:50.000000000 -0400
+++ Net-DNS-0.63/lib/Net/DNS/Resolver/Base.pm	2008-04-09 10:20:15.000000000 -0400
@@ -62,7 +62,7 @@ BEGIN {
 	 # INET6 prior to 2.01 will not work; sorry.
 	 eval {require IO::Socket::INET6; IO::Socket::INET6->VERSION("2.00");}
 	 ) {
- 	import Socket6;
+ 	Socket6->import(qw(getaddrinfo));
  	$has_inet6=1;
     }else{
  	$has_inet6=0;
diff -up Net-DNS-0.63/lib/Net/DNS/Nameserver.pm.BAD Net-DNS-0.63/lib/Net/DNS/Nameserver.pm
--- Net-DNS-0.63/lib/Net/DNS/Nameserver.pm.BAD	2008-04-09 10:23:49.000000000 -0400
+++ Net-DNS-0.63/lib/Net/DNS/Nameserver.pm	2008-04-09 10:31:14.000000000 -0400
@@ -33,7 +33,7 @@ BEGIN {
 	} elsif ( eval {require Socket6;} &&
 			# INET6 earlier than V2.01 will not work; sorry.
 			eval {require IO::Socket::INET6; IO::Socket::INET6->VERSION("2.01");} ) {
- 		import Socket6;
+ 		import IO::Socket::INET6;
 		$has_inet6 = 1;
 	} else {
 		$has_inet6=0;
