--- lib/CPANPLUS/Dist/Fedora.pm-orig	2015-11-17 03:57:19.991994346 -0500
+++ lib/CPANPLUS/Dist/Fedora.pm	2015-11-17 16:07:03.369983805 -0500
@@ -17,6 +17,7 @@
 use CPANPLUS::Error; # imported subs: error(), msg()
 use File::Basename;
 use File::Copy      qw[ copy ];
+use File::Slurp;
 use IPC::Cmd        qw[ run can_run ];
 use List::Util      qw[ first ];
 use Pod::POM;
@@ -274,6 +275,12 @@
         $status->specpath,
     );
 
+    if ( $intern->_callbacks->munge_dist_metafile ) {
+        my $contents = read_file( $status->specpath );
+        $contents = $intern->_callbacks->munge_dist_metafile->($intern, $contents);
+        write_file( $status->specpath , $contents );
+    }
+
     # copy package.
     my $tarball = "$dir/" . basename $module->status->fetch;
     copy $module->status->fetch, $tarball;
