Index: t/local/33_x509_create_cert.t
===================================================================
--- t/local/33_x509_create_cert.t	(revision 327)
+++ t/local/33_x509_create_cert.t	(working copy)
@@ -48,9 +48,9 @@
   
   ##let us do some ASN1_INTEGER related testing
   #test big integer via P_ASN1_INTEGER_set_dec
-  Net::SSLeay::P_ASN1_INTEGER_set_dec($sn, '123456789123456789');    
+  Net::SSLeay::P_ASN1_INTEGER_set_dec($sn, '123456789123456789123456789123456789123456789');
   is(Net::SSLeay::ASN1_INTEGER_get(Net::SSLeay::X509_get_serialNumber($x509)), -1, "ASN1_INTEGER_get");
-  is(Net::SSLeay::P_ASN1_INTEGER_get_hex(Net::SSLeay::X509_get_serialNumber($x509)), '01B69B4BACD05F15', "P_ASN1_INTEGER_get_hex");
+  is(Net::SSLeay::P_ASN1_INTEGER_get_hex(Net::SSLeay::X509_get_serialNumber($x509)), '058936E53D139AFEFABB2683F150B684045F15', "P_ASN1_INTEGER_get_hex");
   #test short integer via P_ASN1_INTEGER_set_hex
   Net::SSLeay::P_ASN1_INTEGER_set_hex($sn, 'D05F14');    
   is(Net::SSLeay::ASN1_INTEGER_get(Net::SSLeay::X509_get_serialNumber($x509)), 13655828, "ASN1_INTEGER_get");
