Description: fix a couple of spelling mistakes in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-02-06

--- a/lib/Geo/Coordinates/OSGB.pm
+++ b/lib/Geo/Coordinates/OSGB.pm
@@ -623,7 +623,7 @@
 The default ellipsoid shape used for conversion to and from latitude and longitude
 is normally `WGS84' as used in the international GPS system.  But you can use this 
 function to set the default shape to `OSGB36' if you want to process or produce latitude and longitude
-coordinates in the British Ordnance Survey sytem (as printed round the edges of OS Landranger maps).
+coordinates in the British Ordnance Survey system (as printed round the edges of OS Landranger maps).
 You can also use this function to set the shape back to `WGS84' again when finished.
 
 =head3 C<ll_to_grid_helmert(lat, lon)>
--- a/lib/Geo/Coordinates/OSGB/Background.pod
+++ b/lib/Geo/Coordinates/OSGB/Background.pod
@@ -253,7 +253,7 @@
 line, and is read using the magic C<< <DATA> >> file handle. In my tests
 this proved to be the fastest way to load all that data, by a long way. 
 
-There are 1229 rows of data, and each row contains upto 701 pairs of shift data
+There are 1229 rows of data, and each row contains up to 701 pairs of shift data
 encoded as pairs of hexadecimal integers representing the shift in mm.  Leading
 and trailing zeros are omitted, and the number of leading zeros omitted is
 stored in the first three characters of each row.
@@ -341,7 +341,7 @@
 Wales, this software produces WGS lat/lon coordinates from the given
 grid references that are within a few mm of the OSGB data.  However
 in Scotland, and especially for those stations that are further
-north and west the latitude coordinates produced are upto 12cm out, 
+north and west the latitude coordinates produced are up to 12cm out,
 while the longitude coordinates are about 1cm out.  This may be
 because the OS test data has some bias, or it may be a feature of
 the mathematics.  Either way that's what's achievable with the OS
@@ -351,10 +351,10 @@
 Outside the area covered by OSTN02 this module uses the small
 Helmert transformation recommended by the OS.  The OS state, that
 with the parameters they provide, this transformation will be
-accurate upto about 5m within the immediate vicinity of the British
+accurate up to about 5m within the immediate vicinity of the British
 Isles.  
 
-You can also use this tranformation within the OSTN02 polygon by calling
+You can also use this transformation within the OSTN02 polygon by calling
 the C<grid_to_ll_helmert> and C<ll_to_grid_helmert> routines.  If you 
 compare the output from these routines to the output from the more
 accurate routines that use OSTN02 you will find that the differences
--- a/lib/Geo/Coordinates/OSGB/Grid.pm
+++ b/lib/Geo/Coordinates/OSGB/Grid.pm
@@ -598,7 +598,7 @@
 removed in this version, because it's not always obvious where the SW corner of a 
 sheet is (for an example look at the inset on Landranger sheet 107).
 
-If you need access to the postion of the sheets in this version, you should work directly with
+If you need access to the position of the sheets in this version, you should work directly with
 the data in L<Geo::Coordinates::OSGB::Maps>.
 
 =back  
