Description: spelling fix
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-10-31

--- a/lib/Perinci/CmdLine/Manual/Examples.pod
+++ b/lib/Perinci/CmdLine/Manual/Examples.pod
@@ -798,7 +798,7 @@
 Perinci::CmdLine will take care of encoding this data to network server when you
 specify C<riap_version> attribute to 1.2. So this process is transparent to you.
 
-When outputing binary result, in the C<text> output formats, Perinci::CmdLine
+When outputting binary result, in the C<text> output formats, Perinci::CmdLine
 will also print the binary result from server as-is without any newline added.
 So you can pipe binary data to files/processes unmodified.
 
