--- Test/CheckManifest.pm.orig	Tue Nov 21 16:22:10 2006
+++ Test/CheckManifest.pm	Wed Nov 22 13:19:42 2006
@@ -56,6 +56,7 @@
           wanted   => sub{ my $file = $File::Find::name;
                            push(@dir_files,Cwd::realpath($file)) if -f $File::Find::name 
                                                                      and $File::Find::name !~ m!/blib/!
+                                                                     and $File::Find::name !~ m!/\.svn/!
                                                                      and !_is_excluded($_,$arref)}},$home);
 
         #print STDERR Dumper(\@files,\@dir_files);
