--- 01live.t	2008-11-22 19:33:02.000000000 +1100
+++ 01live.t_new	2008-11-22 19:33:26.000000000 +1100
@@ -34,10 +34,12 @@
 }
 
 
-TODO: { 
-    local $TODO = "We don't currently wait for 'server is running' responses from the client";
-    run_server_tests('SlowServer');
-
+ TODO: { 
+     local $TODO = "We don't currently wait for 'server is running' responses from the client";
+   SKIP: {
+       skip "SlowServer tests don't work on win32 because the CORE::kill() is subtly broken", 4 if $^O = 'MSWin32';
+       run_server_tests('SlowServer');
+     }
 }
 
 
