Description: Fixing man page
Author: Fabrizio Regalli
Last-Update: 2011-08-11
Forwarded: no

--- a/lib/Tie/iCal.pm
+++ b/lib/Tie/iCal.pm
@@ -64,7 +64,7 @@
 =head1 FILE LOCKING
 
 The Tie::iCal object returned by tie can also be used to access the underlying Tie::File object.
-This is accessable via the 'A' class variable.
+This is accessible via the 'A' class variable.
 This may be useful for file locking.
 
 	my $ical = tie %events, 'Tie::iCal', "mycalendar.ics";
@@ -602,4 +602,4 @@
 
 =cut
 
-1;
\ No newline at end of file
+1;
