--- perl5/perlbrew/bin/perlbrew~	2016-02-12 05:13:34.000000000 -0500
+++ perl5/perlbrew/bin/perlbrew	2016-02-12 10:20:56.000000000 -0500
@@ -1995,11 +1995,12 @@
       for my $i ( @exec_with ) {
           next if -l $self->root . '/perls/' . $i->{name}; # Skip Aliases
           my %env = $self->perlbrew_env($i->{name});
           next if !$env{PERLBREW_PERL};
   
-          local @ENV{ keys %env } = values %env;
+          local $ENV{$_} = $env{$_} foreach ( grep defined $env{$_}, keys %env );
+          local $ENV{$_} foreach ( grep !defined $env{$_}, keys %env );
           local $ENV{PATH}    = join(':', $env{PERLBREW_PATH}, $ENV{PATH});
           local $ENV{MANPATH} = join(':', $env{PERLBREW_MANPATH}, $ENV{MANPATH}||"");
           local $ENV{PERL5LIB} = $env{PERL5LIB} || "";
   
           print "$i->{name}\n==========\n" unless $self->{quiet};
