--- Session.pm	Thu Dec  4 08:06:22 2003
+++ Session.pm.orig	Thu Dec  4 08:06:00 2003
@@ -1260,7 +1260,7 @@
 
 In the following code snippets, Servlet is a session that acts like a
 tiny daemon.  It receives requests from "client" sessions, performs
-some long-running task, and eventually posts responses back.  Client
+som long-running task, and eventually posts responses back.  Client
 sends requests to Servlet and eventually receives its responses.
 
   # Aliases are a common way for daemon sessions to advertise
