--- nslookup.t.dist	Wed Apr  2 18:07:46 2008
+++ nslookup.t	Tue Apr 15 14:37:51 2008
@@ -6,6 +6,7 @@
 my @windows_hosts_file_locations = qw(
     c:/windows/hosts
     c:/winnt/system32/hosts
+    C:/WINDOWS/system32/drivers/etc/hosts
 );
 
 use Net::Nslookup;
@@ -24,7 +25,7 @@
 
 # Populate %hosts
 unless (open HOSTS, $hostfile) {
-    plan skip_all => "Can't open /etc/hosts: $!";
+    plan skip_all => "Can't open $hostfile: $!";
 }
 
 while (<HOSTS>) {
