--- xml-unavailable.t.old	Mon Mar 09 14:40:24 2009
+++ xml-unavailable.t	Mon Mar 09 14:47:05 2009
@@ -4,11 +4,15 @@
 use File::Spec::Functions;
 
 ### whitebox test, disabling XML::Simple
-{   package XML::Simple;
-    $INC{'XML/Simple.pm'} = $0;
-    
-    sub import { die };
-}
+use lib 't';
+
+### This has been moved to a sample XML::Simple file in the inc folder
+### Due to the new way XML::Simple is loaded
+#{   package XML::Simple;
+#    $INC{'XML/Simple.pm'} = $0;
+#    
+#    sub import { die };
+#}
 
 use_ok('Config::Auto');
 
