diff --git i/Makefile.PL w/Makefile.PL
index 8534170..246b4a6 100644
--- i/Makefile.PL
+++ w/Makefile.PL
@@ -10,34 +10,7 @@ use ExtUtils::MakeMaker;
 my $version = '2.17.8'; 
 my $mm_v    = ExtUtils::MakeMaker->VERSION;
 
-#my @merge_spec
-#= do
-#{
-#    if( $mm_v < 6.46 )
-#    {
-#        ()
-#    }
-#    else
-#    {
-#        META_MERGE => 
-#        {
-#            'meta-spec' =>
-#            {
-#                version => 2
-#            },
-#
-#            dynamic_config => 1,
-#
-#            configure_requires =>
-#            {
-#                qw
-#                (
-#                    Module::FromPerlVer 0.003,
-#                )
-#            },
-#        }
-#    }
-#};
+my @add_opts = $mm_v >= 6.46 ? ('CONFIGURE_REQUIRES' => { 'ExtUtils::MakeMaker' => 0, 'Module::FromPerlVer' => 0.003 }) : ();
 
 WriteMakefile
 (
@@ -73,7 +46,7 @@ WriteMakefile
         )
     },
 
-    @merge_spec,
+    @add_opts,
 );
 
 __END__
