--- C:\strawberry\perl\lib\Pod\Simple\XHTML.pm	Thu Dec 17 10:52:00 2009 UTC
+++ C:\strawberry\perl\lib\Pod\Simple\XHTML_patch.pm	Tue Aug 10 23:54:06 2010 UTC
@@ -505,6 +505,11 @@
             : $type eq 'man' ? $self->resolve_man_page_link($to, $section)
             :                  undef;

+    # Escape any XML entities in the URL. Mainly for &.
+    if (defined $url) {
+        $url = encode_entities($url);
+    }
+
     # If it's an unknown type, use an attribute-less <a> like HTML.pm.
     $self->{'scratch'} .= '<a' . ($url ? ' href="'. $url . '">' : '>');
 }
