--- /usr/share/perl5/IO/InnerFile.pm.dist	2011-11-25 15:08:04.048188626 +1100
+++ /usr/share/perl5/IO/InnerFile.pm	2011-11-25 15:56:18.947465726 +1100
@@ -230,6 +230,12 @@
     my ($self) = @_;
     return undef if ($self->{CRPOS} >= $self->{LG});
 
+    # Handle slurp mode
+    if (! defined $/) {
+        $self->READ(my $text, $self->{LG} - $self->{CRPOS});
+        return $text;
+    }
+
     ### Save and seek...
     my $old_pos = $self->{FH}->tell;
     $self->{FH}->seek($self->{CRPOS}+$self->{START}, 0);
