Author: Dan Rumney <reneeb@cpan.org>
Subject: 5.14 deprecation
Last-Update: 2011-11-26
Bug: http://rt.cpan.org/Tickt/Display.html?id=67664
--- a/lib/CGI/Application/Dispatch.pm
+++ b/lib/CGI/Application/Dispatch.pm
@@ -792,7 +792,7 @@
     # setup our args to dispatch()
     my %args;
     my $config_args = $r->dir_config();
-    for my $var qw(DEFAULT PREFIX ERROR_DOCUMENT) {
+    for my $var (qw(DEFAULT PREFIX ERROR_DOCUMENT)) {
         my $dir_var = "CGIAPP_DISPATCH_$var";
         $args{lc($var)} = $config_args->{$dir_var}
           if($config_args->{$dir_var});
