From d334dc8365831c246cbce569fc312b7c7077911e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 28 Jul 2017 13:36:53 +0200
Subject: [PATCH] LibTIFF-4.0.8 calls pkgconfig file libtiff-4
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 Makefile.PL | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.PL b/Makefile.PL
index 648bc99..0c3a178 100644
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -20,7 +20,7 @@ my %pkgcfg;
 if (
     eval {
         %pkgcfg =
-          ExtUtils::PkgConfig->find( 'libtiff >= ' . $build_reqs{libtiff} );
+          ExtUtils::PkgConfig->find( 'libtiff-4 >= ' . $build_reqs{libtiff} );
     }
   )
 {
-- 
2.9.4

