From 7243c7acfa7731697dfd75e930906817588c9c2f Mon Sep 17 00:00:00 2001
From: Leon Timmermans <fawaka@gmail.com>
Date: Mon, 1 Sep 2014 11:53:23 +0200
Subject: [PATCH] Raise instead of kill the signal

---
 t/basic.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/basic.t b/t/basic.t
index 4519941..95fbb90 100644
--- a/t/basic.t
+++ b/t/basic.t
@@ -18,7 +18,7 @@ local $ENV{PERL5LIB} = join $Config::Config{path_sep} => @INC;
 for my $signal (@signals) {
     next unless __PACKAGE__->can($signal);
     my $signum = __PACKAGE__->can($signal)->();
-    my @cmd = ($^X, qw(-d:bt -e), "kill $signum, \$\$");
+    my @cmd = ($^X, qw(-d:bt -MPOSIX=raise -e), "raise($signum)");
 
     use Capture::Tiny 'capture';
     my ($stdout) = capture { system @cmd };
-- 
1.9.1

