--- Packages.pm.orig	2008-09-16 20:09:31.000000000 +0100
+++ Packages.pm	2008-09-16 20:11:03.000000000 +0100
@@ -52,10 +52,9 @@
 
 sub parse {
     my $self    = shift;
-    my $details = $self->_slurp_details(shift);
 
     # read the preamble
-    my @details = split "\n", $details;
+    my @details = split "\n", $self->_slurp_details(shift);
     while (@details) {
         local $_ = shift @details;
         last if /^\s*$/;
