--- Socket6-0.27/aclocal.m4.orig        2016-06-18 12:07:06.000000000 -0400
+++ Socket6-0.27/aclocal.m4     2016-06-18 11:56:14.000000000 -0400
@@ -203,7 +203,7 @@
 
   addr.s6_addr[15] = 0x21;
   inet_ntop(AF_INET6, &addr, str, sizeof(str));
-  if (strcmp(str,"::21"))
+  if (strcmp(str,"::21")&&strcmp(str,"::0.0.0.33"))
     exit(1);
 }
 ]], [ipv6_cv_can_inet_ntop=yes], [ipv6_cv_can_inet_ntop=no])])dnl
--- Socket6-0.27/configure.orig 2016-06-18 12:07:17.000000000 -0400
+++ Socket6-0.27/configure      2016-06-18 11:56:37.000000000 -0400
@@ -4271,7 +4271,7 @@
 
   addr.s6_addr[15] = 0x21;
   inet_ntop(AF_INET6, &addr, str, sizeof(str));
-  if (strcmp(str,"::21"))
+  if (strcmp(str,"::21")&&strcmp(str,"::0.0.0.33"))
     exit(1);
 }
 
