--- /usr/local/bin/wsdl2perl.pl.orig	Tue Nov  3 10:15:37 2009
+++ /usr/local/bin/wsdl2perl.pl			Tue Sep 13 09:44:37 2011
@@ -51,6 +51,17 @@ my %opt = (
         };
         return ($user, $password);
     };
+
+    use SOAP::WSDL::Base;
+    *SOAP::WSDL::Base::expand_orig = \&SOAP::WSDL::Base::expand;
+    *SOAP::WSDL::Base::expand = sub { 
+        my ($self, $qname) = @_;
+        my ($prefix, $localname) = $qname =~ m/^(.+?):(.+)$/;
+        if(defined($prefix) && defined($localname) && $prefix eq 'tns' && $localname eq 'lookupKey') {    
+            return ('http://localhost/CommonTypes', $localname);
+        }
+        return SOAP::WSDL::Base::expand_orig($self, $qname);
+    };
 }
 
 GetOptions(\%opt,
