Description: spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-10-17

--- a/lib/ExtUtils/LibBuilder.pm
+++ b/lib/ExtUtils/LibBuilder.pm
@@ -29,7 +29,7 @@
 
 This method creates a new ExtUtils::LibBuilder object. While it
 supports all C<ExtUtils::CBuilder> methods some might work slightly
-differently (namely the two bellow).
+differently (namely the two below).
 
 You can supply to the constructor any option recognized by
 C<ExtUtils::CBuilder> constructor. None of them will be used by
