--- lib/DBIx/Class/ResultSet/RecursiveUpdate.pm	2011-02-09 14:42:55.000000000 +0100
+++ lib/DBIx/Class/ResultSet/RecursiveUpdate.pm.patched	2011-02-09 14:42:37.000000000 +0100
@@ -332,7 +332,7 @@
                 ->{is_nullable};
     }
 
-    $if_not_submitted = $all_fks_nullable ? 'nullify' : 'delete'
+    $if_not_submitted = $all_fks_nullable ? 'set_to_null' : 'delete'
         unless defined $if_not_submitted;
 
     #warn "\tNULLABLE: $all_fks_nullable ACTION: $if_not_submitted\n";
