--- repro.pl.old	2011-01-27 17:31:29.000000000 +0300
+++ repro.pl	2011-01-27 17:33:19.000000000 +0300
@@ -1,9 +1,7 @@
 use Data::Dumper;
 use strict;
 use DBI;
-use utf8;
 use Encode;
-binmode STDOUT, ":utf8";
 binmode STDIN, ":utf8";
 
 
@@ -33,11 +31,10 @@
       foreach my $val (@$str)
       {
         print ("Value: $val, it is ".(utf8::is_utf8($val)?"":"non-")."utf8 string, and ".(utf8::valid($str)?"":"not")." valid<br>\n");
-        Encode::_utf8_on($val);
+        utf8::upgrade( $val );
         print ("Value: $val, it is ".(utf8::is_utf8($val)?"":"non-")."utf8 string, and ".(utf8::valid($str)?"":"not")." valid<br>\n");
       }
     }
     $query->finish();
     $dbh->disconnect();
-    return $data;
 
