--- /home/ccm_root/lib/perl5/site_perl/5.8.8/Spreadsheet/WriteExcel/Worksheet.pm.orig	2009-04-01 15:42:02.000000000 +0200
+++ /home/ccm_root/lib/perl5/site_perl/5.8.8/Spreadsheet/WriteExcel/Worksheet.pm	2009-04-01 15:42:46.000000000 +0200
@@ -7542,7 +7542,7 @@
     }
 
     # Force 2d ranges to be a reference class.
-    s/_range2d/_range2dR/ for @tokens;
+    s/_range([23])d/_range${1}dR/ for @tokens;
 
     # Parse the tokens into a formula string.
     $formula = $parser->parse_tokens(@tokens);
