From 00a0558d0fce633a9fe79c2612e961128c73fba7 Mon Sep 17 00:00:00 2001
From: Curtis Jewell <perl@csjewell.fastmail.us>
Date: Fri, 16 Jul 2010 14:05:08 -0600
Subject: [PATCH] Wrap the requirement for Test::Requires in $ENV{RELEASE_TESTING}.

This fixes RT#59460.
---
 Makefile.PL |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile.PL b/Makefile.PL
index 17cdb7e..dff623f 100755
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -1,7 +1,7 @@
 # for developpers:
 BEGIN{
     if(grep{ $_ eq '--author' } @ARGV){
-        print "^X -S cpanm < author/requires.cpanm\n";
+        print "$^X -S cpanm < author/requires.cpanm\n";
         system "$^X -S cpanm < author/requires.cpanm";
     }
 }
@@ -26,7 +26,9 @@ requires 'Scalar::Util' => 1.14;
 
 test_requires 'Test::More'      => 0.88;
 test_requires 'Test::Exception' => 0.29;
-test_requires 'Test::Requires'  => 0.03;
+if ( $ENV{RELEASE_TESTING} || (grep { $_ eq '--author' } @ARGV) ) {
+    test_requires 'Test::Requires'  => 0.03;
+}
 # test_requires 'Test::Output'    => 0.16; # too many dependencies!
 
 if($] < 5.010) {
-- 
1.6.5.1.1367.gcd48

