--- t/agent.t	2013-05-04 13:27:57.000000000 +0200
+++ t/agent.t	2014-07-16 17:12:12.197429338 +0200
@@ -10,6 +10,8 @@
 my $KEY = "EFEA4EAD"; # 49539D60EFEA4EAD
 my $WHO = "Mail::GnuPG Test Key <mail\@gnupg.dom>";
 
+$ENV{'LC_ALL'}="C"; # avoid other locales for test
+
 unless ( 0 == system("gpg --version 2>&1 >/dev/null") &&
        0 == system("gpg-agent --version 2>&1 >/dev/null")) {
   plan skip_all => "gpg, gpg-agent in path required for testing agent";
@@ -75,43 +77,43 @@
 
 $copy = $me->dup;
 
-is( 0, $mg->mime_sign( $copy ) );
+is( $mg->mime_sign( $copy ), 0 );
 
 my ($verify,$key,$who) = $mg->verify($copy);
-is( 0, $verify );
-is( $KEY, $key );
-is( $WHO, $who );
+is( $verify, 0 );
+is( $key, $KEY );
+is( $who, $WHO );
 
-is( 1, $mg->is_signed($copy) );
-is( 0, $mg->is_encrypted($copy) );
+is( $mg->is_signed($copy), 1 );
+is( $mg->is_encrypted($copy), 0 );
 
 # Test Clear Signing Round Trip
 
 $copy = $me->dup;
 
-is( 0, $mg->clear_sign( $copy ) );
+is( $mg->clear_sign( $copy ), 0 );
 
 { my ($verify,$key,$who) = $mg->verify($copy);
-is( 0, $verify );
-is( $KEY, $key );
-is( $WHO, $who );
+is( $verify, 0 );
+is( $key, $KEY );
+is( $who, $WHO );
 
-is( 1, $mg->is_signed($copy) );
-is( 0, $mg->is_encrypted($copy) );
+is( $mg->is_signed($copy), 1 );
+is( $mg->is_encrypted($copy), 0 );
 }
 # Test MIME Encryption Round Trip
 
 $copy = $me->dup;
 
-is( 0, $mg->ascii_encrypt( $copy, $KEY ));
-is( 0, $mg->is_signed($copy) );
-is( 1, $mg->is_encrypted($copy) );
+is( $mg->ascii_encrypt( $copy, $KEY ), 0 );
+is( $mg->is_signed($copy), 0);
+is( $mg->is_encrypted($copy), 1 );
 
 ($verify,$key,$who) = $mg->decrypt($copy);
 
-is( 0, $verify );
-is( undef, $key );
-is( undef, $who );
+is( $verify, 0 );
+is( $key, undef );
+is( $who, undef );
 
 is_deeply($mg->{decrypted}->body,$me->body);
 
