Description: fix spelling mistakes in POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2018-07-27

--- a/bin/s3cl
+++ b/bin/s3cl
@@ -343,7 +343,7 @@
 service. It does not limit the number of requests (which may cost
 you more money than if you did it a different way!) and each
 request costs Money (although some costs from EC2 may be $0.0,
-check latest from Amazon costs page) - we take NO reponsibility
+check latest from Amazon costs page) - we take NO responsibility
 for your bill.
 
 =head1 AUTHOR
--- a/lib/Net/Amazon/S3.pm
+++ b/lib/Net/Amazon/S3.pm
@@ -613,7 +613,7 @@
 =item use_virtual_host
 
 Use the virtual host method ('bucketname.s3.amazonaws.com') instead of specifying the
-bucket at the first part of the path. This is particularily useful if you want to access
+bucket at the first part of the path. This is particularly useful if you want to access
 buckets not located in the US-Standard region (such as EU, Asia Pacific or South America).
 See L<http://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html> for the pros and cons.
 
@@ -653,7 +653,7 @@
 
 =item acl_short (optional)
 
-See the set_acl subroutine for documenation on the acl_short options
+See the set_acl subroutine for documentation on the acl_short options
 
 =item location_constraint (option)
 
