Description: spelling fix
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-07-31

--- a/lib/GraphViz/Data/Grapher.pm
+++ b/lib/GraphViz/Data/Grapher.pm
@@ -28,7 +28,7 @@
 quite how the structure fits together.
 
 Data::Dumper can help by representing the structure as a text
-heirarchy, but GraphViz::Data::Grapher goes a step further and
+hierarchy, but GraphViz::Data::Grapher goes a step further and
 visualises the structure by drawing a graph which represents the data
 structure.
 
