From 4186de94870eeb603dec2bd2be36b0b7fb83a8c5 Mon Sep 17 00:00:00 2001
From: Gavin Carr <gavin@openfusion.com.au>
Date: Sat, 5 May 2012 16:56:39 +1000
Subject: [PATCH] Set RootEnable on all root domains, not just top-level.

---
 lib/Domain/PublicSuffix.pm |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/lib/Domain/PublicSuffix.pm b/lib/Domain/PublicSuffix.pm
index 42ba4f3..4bce8f4 100755
--- a/lib/Domain/PublicSuffix.pm
+++ b/lib/Domain/PublicSuffix.pm
@@ -269,6 +269,7 @@ sub _parse_data_file {
 			next if (!$sub);
 			
 			$last->{$sub} = {} unless ( defined $last->{$sub} );
+			$last->{$sub}->{'RootEnable'} = 1 if scalar @domain_array == 0;
 			$last = $last->{$sub};
 		}
 	}
-- 
1.7.1

