--- List-MoreUtils/lib/List/MoreUtils.pm	2007-11-20 17:24:32.000000000 +0100
+++ List-MoreUtils_perl5.6/lib/List/MoreUtils.pm	2008-05-26 17:46:54.000000000 +0200
@@ -84,7 +84,7 @@
 sub firstidx (&@) {
     my $f = shift;
     for my $i (0 .. $#_) {
-	local *_ = \$_[$i];	
+	local $_ = $_[$i];
 	return $i if $f->();
     }
     return -1;
@@ -93,7 +93,7 @@
 sub lastidx (&@) {
     my $f = shift;
     for my $i (reverse 0 .. $#_) {
-	local *_ = \$_[$i];
+	local $_ = $_[$i];
 	return $i if $f->();
     }
     return -1;
@@ -102,7 +102,7 @@
 sub insert_after (&$\@) {
     my ($code, $val, $list) = @_;
     my $c = -1;
-    local *_;
+    local $_;
     for my $i (0 .. $#$list) {
 	$_ = $list->[$i];
 	$c = $i, last if $code->();
@@ -161,7 +161,7 @@
 sub indexes (&@)
 {
     my $test = shift;
-    grep {local *_=\$_[$_]; $test->()} 0..$#_;
+    grep {local $_ = $_[$_]; $test->()} 0..$#_;
 }
 
 sub lastval (&@)
@@ -170,7 +170,7 @@
     my $ix;
     for ($ix=$#_; $ix>=0; $ix--)
     {
-        local *_ = \$_[$ix];
+        local $_ = $_[$ix];
         my $testval = $test->();
         $_[$ix] = $_;    # simulate $_ as alias
         return $_ if $testval;
