Description: wording improvement
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-05-16

--- a/CSV_XS.pm
+++ b/CSV_XS.pm
@@ -2609,7 +2609,7 @@
 1004 "INI - callbacks should be undef or a hashref"
 X<1004>
 
-The C<callbacks> attribute only allows to be C<undef> or a hash reference.
+The C<callbacks> attribute only allows one to be C<undef> or a hash reference.
 
 =item *
 2010 "ECR - QUO char inside quotes followed by CR not part of EOL"
