diff --git a/lib/Test/Base.pm b/lib/Test/Base.pm
index fbd1cc2..7dc5ae7 100644
--- a/lib/Test/Base.pm
+++ b/lib/Test/Base.pm
@@ -290,7 +290,8 @@ sub _assert_plan {
 }
 
 sub END {
-    run_compare() unless $Have_Plan or $DIED or not $import_called;
+	my $Test = Test::Builder->new;
+    run_compare() unless $Test->current_test() != 0 or $Have_Plan or $DIED or not $import_called;
 }
 
 sub run_compare() {
