Index: lib/Cache/FileBackend.pm
===================================================================
--- lib/Cache/FileBackend.pm	(revision 73321)
+++ lib/Cache/FileBackend.pm	(working copy)
@@ -12,7 +12,7 @@
 
 use strict;
 use Cache::CacheUtils qw( Assert_Defined Build_Path Freeze_Data Thaw_Data );
-use Digest::SHA1 qw( sha1_hex );
+use Digest::SHA qw( sha1_hex );
 use Error;
 use File::Path qw( mkpath );
 use File::Temp qw( tempfile );
Index: META.yml
===================================================================
--- META.yml	(revision 73321)
+++ META.yml	(working copy)
@@ -5,7 +5,6 @@
 version_from: lib/Cache/Cache.pm
 installdirs:  site
 requires:
-    Digest::SHA1:                  2.02
     Error:                         0.15
     File::Spec:                    0.82
     Storable:                      1.014
Index: Makefile.PL
===================================================================
--- Makefile.PL	(revision 73321)
+++ Makefile.PL	(working copy)
@@ -10,7 +10,6 @@
 my $COMPRESS = 'gzip';
 my $SUFFIX = '.gz';
 my $DEFAULT_PREREQ_PM = {
-                         Digest::SHA1 => '2.02',
                          File::Spec => '0.82',
                          Storable => '1.014',
                          IPC::ShareLite => '0.09',
