diff -ru Math-Pari-2.01080603.orig/Pari.xs Math-Pari-2.01080603_patched/Pari.xs
--- Math-Pari-2.01080603.orig/Pari.xs	2009-12-11 03:18:24.000000000 +0100
+++ Math-Pari-2.01080603_patched/Pari.xs	2010-01-29 16:06:22.351979000 +0100
@@ -3795,7 +3795,7 @@
 		SvREFCNT_dec(tmp);
 	    }
 	    if(GIMME_V == G_VOID) {
-		fputs(SvPV_nolen(ret),stdout);
+		fputs(stdout,SvPV_nolen(ret));
 		SvREFCNT_dec(ret);
 		XSRETURN(0);
 	    } else {
Pouze v Math-Pari-2.01080603_patched/patches: diff_2.1.7_mingw-w64
diff -ru Math-Pari-2.01080603.orig/utils/Math/PariBuild.pm Math-Pari-2.01080603_patched/utils/Math/PariBuild.pm
--- Math-Pari-2.01080603.orig/utils/Math/PariBuild.pm	2009-12-11 02:45:28.000000000 +0100
+++ Math-Pari-2.01080603_patched/utils/Math/PariBuild.pm	2010-01-29 15:54:08.843275700 +0100
@@ -481,6 +481,7 @@
 			      'patches/diff_2.1.6_no-common'],
 		 '2.1.7' =>  [
 			($^O =~ /darwin/i ? 'patches/diff_2.1.6_no-common' : ()),
+			($^O eq 'MSWin32' ? 'patches/diff_2.1.7_mingw-w64' : ()),
 			      'patches/patch-pari-unnormalized-float',
 			      'patches/diff_2.1.7_-O',
 			      'patches/diff_2.1.7_div',
@@ -507,7 +508,7 @@
   return '/' unless $^O =~ /win32/i;
   my($patch, $p) = (shift, 'utils/inc_h.diff');
   $p =~ s,/,\\,g;
-  system "$patch --binary < $p"
+  system "$patch -p0 --binary < $p"
     or warn("... Apparently, your patch takes flag --binary...\n"),
        return ('\\', '--binary');
   return '\\';
@@ -887,6 +888,8 @@
 	   or $os eq 'freebsd' or $os =~ /^cygwin/) {
     chomp($machine = `uname -m`);
     $machine ||= 'ix86';
+  } elsif ($Config{archname} eq 'MSWin32-x86-multi-thread' or $Config{cc} =~ /gcc/) {
+    $machine = 'ix86';
   } elsif (0 and $os =~ /win32/i and not $Config{gccversion}) {
     # Not needed with rename of kernel1.s to kernel1.c?
     $machine = 'port'; # Win32 compilers would not understand the assmebler anyway
