--- Client.pm.old	2011-11-19 14:49:43.000000000 -0600
+++ Client.pm	2011-11-21 11:27:52.946685770 -0600
@@ -490,9 +490,9 @@
 
     # Due to a bug in JIRA we have to substitute the names of some fields.
     foreach my $field (keys %JRA12300) {
-	if (my $value = delete $params->{$field}) {
-	    $params->{$JRA12300{$field}} = $value;
-	}
+        if (my $value = $params->{$field}) {
+            $params->{$JRA12300{$field}} = $value;
+        }
     }
 
     # Expand the custom_fields hash into the custom fields themselves.
@@ -502,7 +502,14 @@
         }
     }
 
-    return $self->updateIssue($key, $params);
+    $issue = $self->updateIssue($key, $params);
+
+    # Scrub renamed fields out of the input hash after update
+    foreach my $key ( keys %JRA12300 ) {
+        delete $params->{ $JRA12300{$key} };
+    }
+
+    return $issue;
 }
 
 =item B<get_issue_types>
