<!DOCTYPE html>

<html lang="en">
  <head>
    <title>Bug #100681 for Bio-LITE-Taxonomy-NCBI-Gi2taxid: makefile of Bio-LITE-Taxonomy-NCBI-Gi2taxid</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <!-- The X-UA-Compatible <meta> tag above must be very early in <head> -->

    <script>
if (window.top !== window.self) {
    document.write = "";

    window.top.location = window.self.location;

    setTimeout(function(){
        document.body.innerHTML = "";
    }, 1);

    window.self.onload = function(){
        document.body.innerHTML = "";
    };
}
</script>



<link rel="shortcut icon" href="/NoAuth/images/favicon.png" type="image/png" />
<link rel="stylesheet" href="/NoAuth/css/aileron-squished-00f758259152f8b6670ee485f462e6b8.css" type="text/css" media="all" />
<link rel="stylesheet" href="/NoAuth/css/print.css" type="text/css" media="print" />




<script type="text/javascript" src="/NoAuth/js/squished-42167f2409ce368da2d31038e5cb2a57.js"></script>

<script type="text/javascript"><!--
	jQuery( loadTitleBoxStates );

--></script>


<!--[if lt IE 8]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie.css" type="text/css" media="all" />
<![endif]-->
<!--[if lt IE 7]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie6.css" type="text/css" media="all" />
<![endif]-->
<script type="text/javascript">
jQuery(document).ready(function(){
    jQuery.fn.supersubs.defaults = {
        maxWidth: 30,
        extraWidth: 2
    };

    jQuery("#app-nav.toplevel").addClass('sf-menu sf-js-enabled sf-shadow').supersubs().superfish({ speed: 'fast' });
    jQuery("#page-menu.toplevel").addClass('sf-menu sf-js-enabled').supersubs().superfish({ dropShadows: false, speed: 'fast' }).supposition();
});
</script>


<!-- Site CSS from theme editor -->
<style type="text/css" media="all" id="sitecss">
</style>

<meta name="google-site-verification" content="kD3-uEJK1AkTqaC-rckcSt_HPkAj674DPPuEN6i-y0g" />

</head>
  <body class="aileron sidebyside" id="comp-Public-Bug-Display">

<div id="logo">
<a href="https://rt.cpan.org/"><img
    src="/NoAuth/images/cpan.png"
    alt="CPAN"
    width="181"
    height="38" /></a>
    <span class="rtname">RT for rt.cpan.org</span>
</div>


<div id="quickbar">
  <div id="quick-personal">
    <span class="hide"><a href="#skipnav">Skip Menu</a> | </span>
</div>

</div>
<div id="header"><h1>Bug #100681 for Bio-LITE-Taxonomy-NCBI-Gi2taxid: makefile of Bio-LITE-Taxonomy-NCBI-Gi2taxid</h1></div>


<div id="main-navigation"><ul id="app-nav" class="toplevel">
 <li id="li-search_dist"><a id="search_dist" class="menu-item " href="/Public/">Search Distributions</a></li>
 <li id="li-browse_dist"><a id="browse_dist" class="menu-item " href="/Public/Dist/Browse.html">Browse Distributions</a></li>
 <li id="li-about" class="has-children"><a id="about" class="menu-item " href="/NoAuth/RT/CPAN/About.html">About rt.cpan.org</a>
 <ul>
  <li id="li-about-usage"><a id="about-usage" class="menu-item " href="/NoAuth/RT/CPAN/Usage.html">How do Iâ€¦?</a></li>
  <li id="li-about-about"><a id="about-about" class="menu-item " href="/NoAuth/RT/CPAN/About.html">About</a></li>
 </ul>
 </li>
 <li id="li-preferences"><a id="preferences" class="menu-item ">Welcome <span class="current-user">anonymous guest</span>.</a></li>
 <li id="li-login"><a id="login" class="menu-item " href="/NoAuth/Logout.html">Login as another user</a></li>
</ul></div>
<div id="page-navigation"><ul id="page-menu" class="toplevel">
 <li id="li-page-active_bugs"><a id="page-active_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Active;Name=Bio-LITE-Taxonomy-NCBI-Gi2taxid">Active bugs</a></li>
 <li id="li-page-resolved_bugs"><a id="page-resolved_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Resolved;Name=Bio-LITE-Taxonomy-NCBI-Gi2taxid">Resolved bugs</a></li>
 <li id="li-page-rejected_bugs"><a id="page-rejected_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Rejected;Name=Bio-LITE-Taxonomy-NCBI-Gi2taxid">Rejected bugs</a></li>
 <li id="li-page-report"><a id="page-report" class="menu-item " href="/Public/Bug/Report.html?Queue=Bio-LITE-Taxonomy-NCBI-Gi2taxid">Report a new bug</a></li>
</ul></div>
<div id="topactions"><form action="/Public/Search/Simple.html" id="simple-search">
  <input size="12" name="q" accesskey="0" class="field" value="" placeholder="Search..." />
</form>

<form action="/Public/Bug/Report.html" name="CreateTicketInQueue" id="CreateTicketInQueue" >
<input type="submit" class="button" value="New ticket in" />&nbsp;<label accesskey="9">
  <input name="Queue" size="25" placeholder="Module or distribution" value="" class="select-queue" data-autocomplete="Queues" data-autocomplete-autosubmit=1 data-autocomplete-params="right=CreateTicket" />

</label>

</form>

</div>
<div id="body">
<div id="announcement" style="padding: 1em; margin: 1em; border: solid black 1px; background: #eee">
<a href="https://log.perl.org/2020/12/rtcpanorg-sunset.html">rt.cpan.org will be shut down on March 1st, 2021.</a>
</div>

<a name="skipnav" id="skipnav" accesskey="8"></a>





<p>This queue is for tickets about the <a href="https://metacpan.org/release/Bio-LITE-Taxonomy-NCBI-Gi2taxid">Bio-LITE-Taxonomy-NCBI-Gi2taxid CPAN distribution</a>.</p>





<div class="">
  <div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DUmVwb3J0IGluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Report information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------UmVwb3J0IGluZm9ybWF0aW9u---0">


<table width="100%" class="ticket-summary">
<tr>
  <td valign="top" class="boxcontainer">
    <div class="ticket-info-basics">
  <div class="titlebox ticket-info-basics " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dbasics\x2D\x2D\x2D\x2DVGhlIEJhc2ljcw\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">The Basics</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-basics----VGhlIEJhc2ljcw__---0">


        <table>
  <tr class="id">
    <td class="label">Id:</td>
    <td class="value">100681</td>
  </tr>
  <tr class="status">
    <td class="label">Status:</td>
    <td class="value">resolved</td>
  </tr>
  <tr class="priority">
    <td class="label">Priority:</td>
    <td class="value">0/
</td>
  </tr>
  <tr class="queue">
    <td class="label">Queue:</td>
    <td class="value"><a href="/Dist/Display.html?Queue=Bio-LITE-Taxonomy-NCBI-Gi2taxid">Bio-LITE-Taxonomy-NCBI-Gi2taxid</a>
</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>


    <div class="ticket-info-people">
  <div class="titlebox ticket-info-people " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dpeople\x2D\x2D\x2D\x2DUGVvcGxl\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">People</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-people----UGVvcGxl---0">


        <table>
  <tr>
    <td class="label">Owner:</td>
    <td class="value">Nobody in particular
    
    </td>
  </tr>
  <tr>
    <td class="labeltop">Requestors:</td>
    <td class="value">
denis.baurain [...] ulg.ac.be

<br />

</td>
  </tr>
  <tr>
    <td class="labeltop">Cc:</td>
    <td class="value">

</td>
  </tr>
  <tr>
    <td class="labeltop">AdminCc:</td>
    <td class="value">

</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>



  </td>
  <td valign="top" class="boxcontainer">
    
    <div class="ticket-info-cfs">
  <div class="titlebox ticket-info-cfs " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dcfs\x2D\x2D\x2D\x2DQnVnIEluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Bug Information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-cfs----QnVnIEluZm9ybWF0aW9u---0">

 
        <table>
  <tr id="CF-1-ShowRow">
    <td class="label">Severity:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-235552-ShowRow">
    <td class="label">Broken in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-235553-ShowRow">
    <td class="label">Fixed in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
</table>


        <hr class="clear" />
  </div>
</div>




</div>


    




  </td>
</tr>
</table>

    <hr class="clear" />
  </div>
</div>




</div>


<br />


<div class="history">
<div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DSGlzdG9yeQ\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">History</span>
    <span class="right">		<a href="#" data-direction="open" onclick="return toggle_all_folds(this, 'Show\x20all\x20quoted\x20text', 'Hide\x20all\x20quoted\x20text');">Show all quoted text</a> &mdash; <a href="/Public/Bug/Display.html?ShowHeaders=1;id=100681">Show full headers</a>    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------SGlzdG9yeQ__---0">


<div id="ticket-history">
<div class="ticket-transaction message Create odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438477" href="/Public/Bug/Display.html?id=100681#txn-1438477">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;02:43:23&nbsp;2014</span>
    <span class="description">denis.baurain [...] ulg.ac.be -  Ticket created</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> makefile of Bio-LITE-Taxonomy-NCBI-Gi2taxid</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 3 Dec 2014 08:43:09 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> &lt;bug-Bio-LITE-Taxonomy-NCBI-Gi2taxid [...] rt.cpan.org&gt;</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> &#34;Denis BAURAIN&#34; &lt;denis.baurain [...] ulg.ac.be&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1438477/764685/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/764685">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 3.5k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Miguel,

I&#39;ve noticed a minor issue with the current &#40;0.12&#41; Makefile.PL of your module Bio-LITE-Taxonomy-NCBI-Gi2taxid. You request :
    &#39;File::Tail&#39; =&gt; 0.96
but this should read:
    &#39;File::Tail&#39; =&gt; 0.096
Otherwise  your module fails to install without forcing.

All the best,
Denis

--

Prof. Denis BAURAIN
Eukaryotic Phylogenomics

Department of Life Sciences
University of LiÃ¨ge
Sart Tilman, BÃ¢t. B22
B-4000 LiÃ¨ge, Belgium


At Fri, 30 May 2014 16:22:48 +0200, denis.baurain@ulg.ac.be wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;Thank you, Miguel.
&gt;This is great! :-&#41;
&gt;
&gt;All the best,
&gt;Denis
&gt;
&gt;At Fri, 30 May 2014 10:08:31 +0100, mp@ebi.ac.uk wrote:
&gt;
<div class="message-stanza open">&gt;&gt;Hi Denis,
&gt;&gt;
&gt;&gt;Thanks a lot for your email.
&gt;&gt;Your request makes a lot of sense. I have included your patch, added a
&gt;&gt;couple of tests and corrected the indentation :-&#41;
&gt;&gt;
&gt;&gt;I have uploaded the new version to cpan &#40;v0.09&#41;. It may take some time
&gt;&gt;to reach all the mirrors but it should be available during the morning.
&gt;&gt;
&gt;&gt;Please, let me know if you have any other problem with the module.
&gt;&gt;
&gt;&gt;Best regards,
&gt;&gt;
&gt;&gt;M;
&gt;&gt;
&gt;&gt;
&gt;&gt;On 29/05/14 22:05, Denis BAURAIN wrote:
<div class="message-stanza open">&gt;&gt;&gt; Dear Miguel,
&gt;&gt;&gt;
&gt;&gt;&gt; During the last few years, I have been developing an extensive suite
</div>&gt;&gt;of Perl modules for phylogenomics. My modules do not use BioPerl but
&gt;&gt;rest on Bio::Phylo and on your Bio::LITE::Taxonomy distribution. Note
&gt;&gt;that you will not find them on CPAN yet because there are still in
&gt;&gt;active development.
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; Anyway, I make extensive use of NCBI Taxonomy in my work and the fact
</div>&gt;&gt;that Bio::LITE::Taxonomy does not handle synonyms drives me crazy
&gt;&gt;because NCBI often renames organisms: e.g., Canis familiaris does not
&gt;&gt;work &#40;should be Canis lupus familiaris&#41; ; Xenopus tropicalis neither
&gt;&gt;&#40;should be Xenopus &#40;Silurana&#41; tropicalis&#41; etc.
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; Therefore, I propose you a very small patch so as to handle them in
</div>&gt;&gt;your module. The idea is simply to include the synonyms as additional
&gt;&gt;keys for a given taxon id. This would increase the size of the taxon id
&gt;&gt;lookup by about 15% considering the current version of the NCBI Taxonomy
&gt;&gt;database :
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; $ grep -c &#39;scientific name&#39; taxdump/names.dmp
&gt;&gt;&gt; 1160242
&gt;&gt;&gt; grep -c &#39;synonym&#39; taxdump/names.dmp
&gt;&gt;&gt; 187443
&gt;&gt;&gt;
&gt;&gt;&gt;  From my understanding of your module, this should not have adverse
</div>&gt;&gt;effects since there will still be only one name by taxon id when going
&gt;&gt;from taxon id to name. This is only in the opposite direction that
&gt;&gt;synonyms will come to play.
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; I attach a patch file. Here&#39;s the patch in context &#40;sorry for the
</div>&gt;&gt;funny indentation; it appears like this on my system&#41;:
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; sub _name_nodes
&gt;&gt;&gt;      {
&gt;&gt;&gt;        my &#40;$self&#41; = @_;
&gt;&gt;&gt;        my $namesFile = $self-&gt;{namesFile};
&gt;&gt;&gt;        my $nodesNames;
&gt;&gt;&gt;        if &#40;&#40;UNIVERSAL::isa&#40;$namesFile, &#39;GLOB&#39;&#41;&#41; or &#40;ref \$namesFile eq
</div>&gt;&gt;&#39;GLOB&#39;&#41;&#41; {
<div class="message-stanza open">&gt;&gt;&gt;      $nodesNames = $namesFile;
&gt;&gt;&gt;        } else {
&gt;&gt;&gt;      open $nodesNames, &#34;&lt;&#34;, $namesFile or croak $!;
&gt;&gt;&gt;        }
&gt;&gt;&gt;        while &#40;&lt;$nodesNames&gt;&#41;{
&gt;&gt;&gt;      chomp;
&gt;&gt;&gt;      my &#40;$taxId,$taxName,$comment&#41; = _process_tax_name &#40;$_&#41;;
&gt;&gt;&gt;      if &#40;$comment eq &#34;scientific name&#34;&#41;{
&gt;&gt;&gt;        ${$self-&gt;{nodes}-&gt;{$taxId}}{name} = $taxName;
&gt;&gt;&gt;            $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt;&gt;      }
&gt;&gt;&gt;      elsif &#40;$comment eq &#34;synonym&#34;&#41; {
&gt;&gt;&gt;          $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt;&gt;      }
&gt;&gt;&gt;        }
&gt;&gt;&gt;        close $nodesNames;
&gt;&gt;&gt;      }
&gt;&gt;&gt;
&gt;&gt;&gt; Would you like to consider this change for your next release? This
</div>&gt;&gt;would help me a lot. If you need help for writing tests for this
&gt;&gt;feature, please let me know.
<div class="message-stanza open">&gt;&gt;&gt;
&gt;&gt;&gt; Thank you very much for your time!
&gt;&gt;&gt;
&gt;&gt;&gt; Best regards,
&gt;&gt;&gt; Denis
&gt;&gt;&gt;
</div>&gt;&gt;
</div></div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438536" href="/Public/Bug/Display.html?id=100681#txn-1438536">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;04:57:49&nbsp;2014</span>
    <span class="description">MOTIF [...] cpan.org -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<div class="downloadattachment">
<a href="/Ticket/Attachment/1438536/764725/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/764725">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 4.4k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Denis,

I can&#39;t reproduce this issue. I have installed version 0.99.3 &#40;current&#41; of File::Tail and I&#39;m able to build &#38; test Bio-LITE-Taxonomy-NCBI-Gi2taxid without problem.
File::Tail version 0.99.3 is from Sept 2005, so it is very likely that you have that version installed too. I don&#39;t see how requiring a version &gt; 0.96 is failing for you.
I haven&#39;t changed that dependency in the latest version of the module &#40;I don&#39;t remember changing it ever really&#41; and the code that depends on File::Tail has not been changed lately &#40;AFAICT&#41;.

What version of File::Tail do you have installed?
In what system is this failing for you? 
What version of Perl are you using?

Thanks,

M;


El MiÃ© Dic 03 02:43:23 2014, denis.baurain@ulg.ac.be escribiÃ³:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; Hi Miguel,
&gt; 
&gt; I&#39;ve noticed a minor issue with the current &#40;0.12&#41; Makefile.PL of your
&gt; module Bio-LITE-Taxonomy-NCBI-Gi2taxid. You request :
&gt;     &#39;File::Tail&#39; =&gt; 0.96
&gt; but this should read:
&gt;     &#39;File::Tail&#39; =&gt; 0.096
&gt; Otherwise  your module fails to install without forcing.
&gt; 
&gt; All the best,
&gt; Denis
&gt; 
&gt; --
&gt; 
&gt; Prof. Denis BAURAIN
&gt; Eukaryotic Phylogenomics
&gt; 
&gt; Department of Life Sciences
&gt; University of LiÃ¨ge
&gt; Sart Tilman, BÃ¢t. B22
&gt; B-4000 LiÃ¨ge, Belgium
&gt; 
&gt; 
&gt; At Fri, 30 May 2014 16:22:48 +0200, denis.baurain@ulg.ac.be wrote:
&gt; 
<div class="message-stanza open">&gt; &gt; Thank you, Miguel.
&gt; &gt; This is great! :-&#41;
&gt; &gt;
&gt; &gt; All the best,
&gt; &gt; Denis
&gt; &gt;
&gt; &gt; At Fri, 30 May 2014 10:08:31 +0100, mp@ebi.ac.uk wrote:
&gt; &gt;
<div class="message-stanza open">&gt; &gt;&gt; Hi Denis,
&gt; &gt;&gt;
&gt; &gt;&gt; Thanks a lot for your email.
&gt; &gt;&gt; Your request makes a lot of sense. I have included your patch, added
&gt; &gt;&gt; a
&gt; &gt;&gt; couple of tests and corrected the indentation :-&#41;
&gt; &gt;&gt;
&gt; &gt;&gt; I have uploaded the new version to cpan &#40;v0.09&#41;. It may take some
&gt; &gt;&gt; time
&gt; &gt;&gt; to reach all the mirrors but it should be available during the
&gt; &gt;&gt; morning.
&gt; &gt;&gt;
&gt; &gt;&gt; Please, let me know if you have any other problem with the module.
&gt; &gt;&gt;
&gt; &gt;&gt; Best regards,
&gt; &gt;&gt;
&gt; &gt;&gt; M;
&gt; &gt;&gt;
&gt; &gt;&gt;
&gt; &gt;&gt; On 29/05/14 22:05, Denis BAURAIN wrote:
<div class="message-stanza open">&gt; &gt;&gt;&gt; Dear Miguel,
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; During the last few years, I have been developing an extensive
&gt; &gt;&gt;&gt; suite
</div>&gt; &gt;&gt; of Perl modules for phylogenomics. My modules do not use BioPerl but
&gt; &gt;&gt; rest on Bio::Phylo and on your Bio::LITE::Taxonomy distribution.
&gt; &gt;&gt; Note
&gt; &gt;&gt; that you will not find them on CPAN yet because there are still in
&gt; &gt;&gt; active development.
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Anyway, I make extensive use of NCBI Taxonomy in my work and the
&gt; &gt;&gt;&gt; fact
</div>&gt; &gt;&gt; that Bio::LITE::Taxonomy does not handle synonyms drives me crazy
&gt; &gt;&gt; because NCBI often renames organisms: e.g., Canis familiaris does
&gt; &gt;&gt; not
&gt; &gt;&gt; work &#40;should be Canis lupus familiaris&#41; ; Xenopus tropicalis neither
&gt; &gt;&gt; &#40;should be Xenopus &#40;Silurana&#41; tropicalis&#41; etc.
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Therefore, I propose you a very small patch so as to handle them in
</div>&gt; &gt;&gt; your module. The idea is simply to include the synonyms as
&gt; &gt;&gt; additional
&gt; &gt;&gt; keys for a given taxon id. This would increase the size of the taxon
&gt; &gt;&gt; id
&gt; &gt;&gt; lookup by about 15% considering the current version of the NCBI
&gt; &gt;&gt; Taxonomy
&gt; &gt;&gt; database :
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; $ grep -c &#39;scientific name&#39; taxdump/names.dmp
&gt; &gt;&gt;&gt; 1160242
&gt; &gt;&gt;&gt; grep -c &#39;synonym&#39; taxdump/names.dmp
&gt; &gt;&gt;&gt; 187443
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; From my understanding of your module, this should not have adverse
</div>&gt; &gt;&gt; effects since there will still be only one name by taxon id when
&gt; &gt;&gt; going
&gt; &gt;&gt; from taxon id to name. This is only in the opposite direction that
&gt; &gt;&gt; synonyms will come to play.
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; I attach a patch file. Here&#39;s the patch in context &#40;sorry for the
</div>&gt; &gt;&gt; funny indentation; it appears like this on my system&#41;:
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; sub _name_nodes
&gt; &gt;&gt;&gt;      {
&gt; &gt;&gt;&gt;        my &#40;$self&#41; = @_;
&gt; &gt;&gt;&gt;        my $namesFile = $self-&gt;{namesFile};
&gt; &gt;&gt;&gt;        my $nodesNames;
&gt; &gt;&gt;&gt;        if &#40;&#40;UNIVERSAL::isa&#40;$namesFile, &#39;GLOB&#39;&#41;&#41; or &#40;ref \$namesFile
&gt; &gt;&gt;&gt; eq
</div>&gt; &gt;&gt; &#39;GLOB&#39;&#41;&#41; {
<div class="message-stanza open">&gt; &gt;&gt;&gt; $nodesNames = $namesFile;
&gt; &gt;&gt;&gt;   } else {
&gt; &gt;&gt;&gt; open $nodesNames, &#34;&lt;&#34;, $namesFile or croak $!;
&gt; &gt;&gt;&gt;   }
&gt; &gt;&gt;&gt;   while &#40;&lt;$nodesNames&gt;&#41;{
&gt; &gt;&gt;&gt; chomp;
&gt; &gt;&gt;&gt; my &#40;$taxId,$taxName,$comment&#41; = _process_tax_name &#40;$_&#41;;
&gt; &gt;&gt;&gt; if &#40;$comment eq &#34;scientific name&#34;&#41;{
&gt; &gt;&gt;&gt;   ${$self-&gt;{nodes}-&gt;{$taxId}}{name} = $taxName;
&gt; &gt;&gt;&gt;       $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt; &gt;&gt;&gt; }
&gt; &gt;&gt;&gt; elsif &#40;$comment eq &#34;synonym&#34;&#41; {
&gt; &gt;&gt;&gt;     $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt; &gt;&gt;&gt; }
&gt; &gt;&gt;&gt;   }
&gt; &gt;&gt;&gt;   close $nodesNames;
&gt; &gt;&gt;&gt; }
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Would you like to consider this change for your next release? This
</div>&gt; &gt;&gt; would help me a lot. If you need help for writing tests for this
&gt; &gt;&gt; feature, please let me know.
<div class="message-stanza open">&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Thank you very much for your time!
&gt; &gt;&gt;&gt;
&gt; &gt;&gt;&gt; Best regards,
&gt; &gt;&gt;&gt; Denis
&gt; &gt;&gt;&gt;
</div>&gt; &gt;&gt;
</div></div></div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction basics Status odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438537" href="/Public/Bug/Display.html?id=100681#txn-1438537">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;04:57:50&nbsp;2014</span>
    <span class="description">The RT System itself -  Status changed from &#39;new&#39; to &#39;open&#39;</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438588" href="/Public/Bug/Display.html?id=100681#txn-1438588">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;06:49:42&nbsp;2014</span>
    <span class="description">denis.baurain [...] ulg.ac.be -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #100681] makefile of Bio-LITE-Taxonomy-NCBI-Gi2taxid</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 3 Dec 2014 12:49:28 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> &lt;bug-Bio-LITE-Taxonomy-NCBI-Gi2taxid [...] rt.cpan.org&gt;</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> &#34;Denis BAURAIN&#34; &lt;denis.baurain [...] ulg.ac.be&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1438588/764771/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/764771">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 5.4k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Thank you for your fast reply, Miguel.

I use cpanm in conjunction with Perlbrew. I did not have the issue on all combinations of perl and OS, but here&#39;s an example of error from Linux with perl 5.20.1:

Configuring Bio-LITE-Taxonomy-NCBI-Gi2taxid-0.12 ... OK
==&gt; Found dependencies: File::Tail
Found File::Tail 0.099003 which doesn&#39;t satisfy 0.96.
! Installing the dependencies failed: Module &#39;File::Tail&#39; is not installed
! Bailing out the installation for Bio-LITE-Taxonomy-NCBI-Gi2taxid-0.12.

My understanding is that 0.96 should be spelled 0.096. Otherwise, I fully agree with your analysis. Do you think that either cpanm or something related has changed the way version numbers are evaluated?

Best,
Denis

At Wed, 3 Dec 2014 04:57:50 -0500, bug-Bio-LITE-Taxonomy-NCBI-Gi2taxid@rt.cpan.org wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;&lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=100681">https://rt.cpan.org/Ticket/Display.html?id=100681</a></span> &gt;
&gt;
&gt;Hi Denis,
&gt;
&gt;I can&#39;t reproduce this issue. I have installed version 0.99.3 &#40;current&#41;
&gt;of File::Tail and I&#39;m able to build &#38; test Bio-LITE-Taxonomy-NCBI-
&gt;Gi2taxid without problem.
&gt;File::Tail version 0.99.3 is from Sept 2005, so it is very likely that
&gt;you have that version installed too. I don&#39;t see how requiring a version
<div class="message-stanza open">&gt;&gt; 0.96 is failing for you.
</div>&gt;I haven&#39;t changed that dependency in the latest version of the module &#40;I
&gt;don&#39;t remember changing it ever really&#41; and the code that depends on
&gt;File::Tail has not been changed lately &#40;AFAICT&#41;.
&gt;
&gt;What version of File::Tail do you have installed?
&gt;In what system is this failing for you?
&gt;What version of Perl are you using?
&gt;
&gt;Thanks,
&gt;
&gt;M;
&gt;
&gt;
&gt;El MiÃ© Dic 03 02:43:23 2014, denis.baurain@ulg.ac.be escribiÃ³:
<div class="message-stanza open">&gt;&gt; Hi Miguel,
&gt;&gt;
&gt;&gt; I&#39;ve noticed a minor issue with the current &#40;0.12&#41; Makefile.PL of your
&gt;&gt; module Bio-LITE-Taxonomy-NCBI-Gi2taxid. You request :
&gt;&gt;     &#39;File::Tail&#39; =&gt; 0.96
&gt;&gt; but this should read:
&gt;&gt;     &#39;File::Tail&#39; =&gt; 0.096
&gt;&gt; Otherwise  your module fails to install without forcing.
&gt;&gt;
&gt;&gt; All the best,
&gt;&gt; Denis
&gt;&gt;
&gt;&gt; --
&gt;&gt;
&gt;&gt; Prof. Denis BAURAIN
&gt;&gt; Eukaryotic Phylogenomics
&gt;&gt;
&gt;&gt; Department of Life Sciences
&gt;&gt; University of LiÃ¨ge
&gt;&gt; Sart Tilman, BÃ¢t. B22
&gt;&gt; B-4000 LiÃ¨ge, Belgium
&gt;&gt;
&gt;&gt;
&gt;&gt; At Fri, 30 May 2014 16:22:48 +0200, denis.baurain@ulg.ac.be wrote:
&gt;&gt;
<div class="message-stanza open">&gt;&gt; &gt; Thank you, Miguel.
&gt;&gt; &gt; This is great! :-&#41;
&gt;&gt; &gt;
&gt;&gt; &gt; All the best,
&gt;&gt; &gt; Denis
&gt;&gt; &gt;
&gt;&gt; &gt; At Fri, 30 May 2014 10:08:31 +0100, mp@ebi.ac.uk wrote:
&gt;&gt; &gt;
<div class="message-stanza open">&gt;&gt; &gt;&gt; Hi Denis,
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; Thanks a lot for your email.
&gt;&gt; &gt;&gt; Your request makes a lot of sense. I have included your patch, added
&gt;&gt; &gt;&gt; a
&gt;&gt; &gt;&gt; couple of tests and corrected the indentation :-&#41;
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; I have uploaded the new version to cpan &#40;v0.09&#41;. It may take some
&gt;&gt; &gt;&gt; time
&gt;&gt; &gt;&gt; to reach all the mirrors but it should be available during the
&gt;&gt; &gt;&gt; morning.
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; Please, let me know if you have any other problem with the module.
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; Best regards,
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; M;
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; On 29/05/14 22:05, Denis BAURAIN wrote:
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt; Dear Miguel,
&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; During the last few years, I have been developing an extensive
&gt;&gt; &gt;&gt;&gt; suite
</div>&gt;&gt; &gt;&gt; of Perl modules for phylogenomics. My modules do not use BioPerl but
&gt;&gt; &gt;&gt; rest on Bio::Phylo and on your Bio::LITE::Taxonomy distribution.
&gt;&gt; &gt;&gt; Note
&gt;&gt; &gt;&gt; that you will not find them on CPAN yet because there are still in
&gt;&gt; &gt;&gt; active development.
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; Anyway, I make extensive use of NCBI Taxonomy in my work and the
&gt;&gt; &gt;&gt;&gt; fact
</div>&gt;&gt; &gt;&gt; that Bio::LITE::Taxonomy does not handle synonyms drives me crazy
&gt;&gt; &gt;&gt; because NCBI often renames organisms: e.g., Canis familiaris does
&gt;&gt; &gt;&gt; not
&gt;&gt; &gt;&gt; work &#40;should be Canis lupus familiaris&#41; ; Xenopus tropicalis neither
&gt;&gt; &gt;&gt; &#40;should be Xenopus &#40;Silurana&#41; tropicalis&#41; etc.
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; Therefore, I propose you a very small patch so as to handle them in
</div>&gt;&gt; &gt;&gt; your module. The idea is simply to include the synonyms as
&gt;&gt; &gt;&gt; additional
&gt;&gt; &gt;&gt; keys for a given taxon id. This would increase the size of the taxon
&gt;&gt; &gt;&gt; id
&gt;&gt; &gt;&gt; lookup by about 15% considering the current version of the NCBI
&gt;&gt; &gt;&gt; Taxonomy
&gt;&gt; &gt;&gt; database :
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; $ grep -c &#39;scientific name&#39; taxdump/names.dmp
&gt;&gt; &gt;&gt;&gt; 1160242
&gt;&gt; &gt;&gt;&gt; grep -c &#39;synonym&#39; taxdump/names.dmp
&gt;&gt; &gt;&gt;&gt; 187443
&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; From my understanding of your module, this should not have adverse
</div>&gt;&gt; &gt;&gt; effects since there will still be only one name by taxon id when
&gt;&gt; &gt;&gt; going
&gt;&gt; &gt;&gt; from taxon id to name. This is only in the opposite direction that
&gt;&gt; &gt;&gt; synonyms will come to play.
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; I attach a patch file. Here&#39;s the patch in context &#40;sorry for the
</div>&gt;&gt; &gt;&gt; funny indentation; it appears like this on my system&#41;:
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; sub _name_nodes
&gt;&gt; &gt;&gt;&gt;      {
&gt;&gt; &gt;&gt;&gt;        my &#40;$self&#41; = @_;
&gt;&gt; &gt;&gt;&gt;        my $namesFile = $self-&gt;{namesFile};
&gt;&gt; &gt;&gt;&gt;        my $nodesNames;
&gt;&gt; &gt;&gt;&gt;        if &#40;&#40;UNIVERSAL::isa&#40;$namesFile, &#39;GLOB&#39;&#41;&#41; or &#40;ref \$namesFile
&gt;&gt; &gt;&gt;&gt; eq
</div>&gt;&gt; &gt;&gt; &#39;GLOB&#39;&#41;&#41; {
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt; $nodesNames = $namesFile;
&gt;&gt; &gt;&gt;&gt;   } else {
&gt;&gt; &gt;&gt;&gt; open $nodesNames, &#34;&lt;&#34;, $namesFile or croak $!;
&gt;&gt; &gt;&gt;&gt;   }
&gt;&gt; &gt;&gt;&gt;   while &#40;&lt;$nodesNames&gt;&#41;{
&gt;&gt; &gt;&gt;&gt; chomp;
&gt;&gt; &gt;&gt;&gt; my &#40;$taxId,$taxName,$comment&#41; = _process_tax_name &#40;$_&#41;;
&gt;&gt; &gt;&gt;&gt; if &#40;$comment eq &#34;scientific name&#34;&#41;{
&gt;&gt; &gt;&gt;&gt;   ${$self-&gt;{nodes}-&gt;{$taxId}}{name} = $taxName;
&gt;&gt; &gt;&gt;&gt;       $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt;&gt; elsif &#40;$comment eq &#34;synonym&#34;&#41; {
&gt;&gt; &gt;&gt;&gt;     $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt;&gt;   }
&gt;&gt; &gt;&gt;&gt;   close $nodesNames;
&gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; Would you like to consider this change for your next release? This
</div>&gt;&gt; &gt;&gt; would help me a lot. If you need help for writing tests for this
&gt;&gt; &gt;&gt; feature, please let me know.
<div class="message-stanza open">&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; Thank you very much for your time!
&gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt;&gt; Best regards,
&gt;&gt; &gt;&gt;&gt; Denis
&gt;&gt; &gt;&gt;&gt;
</div>&gt;&gt; &gt;&gt;
</div></div></div>&gt;
&gt;
&gt;
</div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438593" href="/Public/Bug/Display.html?id=100681#txn-1438593">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;07:11:33&nbsp;2014</span>
    <span class="description">MOTIF [...] cpan.org -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<div class="downloadattachment">
<a href="/Ticket/Attachment/1438593/764776/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/764776">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 6.2k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Hi Denis,

Ok, I see. I suppose it is fine to remove the 0.96 version dependency &#40;it shouldn&#39;t break with previous versions anyway&#41;.
I have updated a new version to cpan.
Let me know if you still have problems with this.

Thanks for reporting!


El MiÃ© Dic 03 06:49:42 2014, denis.baurain@ulg.ac.be escribiÃ³:
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; Thank you for your fast reply, Miguel.
&gt; 
&gt; I use cpanm in conjunction with Perlbrew. I did not have the issue on
&gt; all combinations of perl and OS, but here&#39;s an example of error from
&gt; Linux with perl 5.20.1:
&gt; 
&gt; Configuring Bio-LITE-Taxonomy-NCBI-Gi2taxid-0.12 ... OK
&gt; ==&gt; Found dependencies: File::Tail
&gt; Found File::Tail 0.099003 which doesn&#39;t satisfy 0.96.
&gt; ! Installing the dependencies failed: Module &#39;File::Tail&#39; is not
&gt; installed
&gt; ! Bailing out the installation for Bio-LITE-Taxonomy-NCBI-Gi2taxid-
&gt; 0.12.
&gt; 
&gt; My understanding is that 0.96 should be spelled 0.096. Otherwise, I
&gt; fully agree with your analysis. Do you think that either cpanm or
&gt; something related has changed the way version numbers are evaluated?
&gt; 
&gt; Best,
&gt; Denis
&gt; 
&gt; At Wed, 3 Dec 2014 04:57:50 -0500, bug-Bio-LITE-Taxonomy-NCBI-
&gt; Gi2taxid@rt.cpan.org wrote:
&gt; 
<div class="message-stanza open">&gt; &gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=100681">https://rt.cpan.org/Ticket/Display.html?id=100681</a></span> &gt;
&gt; &gt;
&gt; &gt; Hi Denis,
&gt; &gt;
&gt; &gt; I can&#39;t reproduce this issue. I have installed version 0.99.3
&gt; &gt; &#40;current&#41;
&gt; &gt; of File::Tail and I&#39;m able to build &#38; test Bio-LITE-Taxonomy-NCBI-
&gt; &gt; Gi2taxid without problem.
&gt; &gt; File::Tail version 0.99.3 is from Sept 2005, so it is very likely
&gt; &gt; that
&gt; &gt; you have that version installed too. I don&#39;t see how requiring a
&gt; &gt; version
<div class="message-stanza open">&gt; &gt;&gt; 0.96 is failing for you.
</div>&gt; &gt; I haven&#39;t changed that dependency in the latest version of the module
&gt; &gt; &#40;I
&gt; &gt; don&#39;t remember changing it ever really&#41; and the code that depends on
&gt; &gt; File::Tail has not been changed lately &#40;AFAICT&#41;.
&gt; &gt;
&gt; &gt; What version of File::Tail do you have installed?
&gt; &gt; In what system is this failing for you?
&gt; &gt; What version of Perl are you using?
&gt; &gt;
&gt; &gt; Thanks,
&gt; &gt;
&gt; &gt; M;
&gt; &gt;
&gt; &gt;
&gt; &gt; El MiÃ© Dic 03 02:43:23 2014, denis.baurain@ulg.ac.be escribiÃ³:
<div class="message-stanza open">&gt; &gt;&gt; Hi Miguel,
&gt; &gt;&gt;
&gt; &gt;&gt; I&#39;ve noticed a minor issue with the current &#40;0.12&#41; Makefile.PL of
&gt; &gt;&gt; your
&gt; &gt;&gt; module Bio-LITE-Taxonomy-NCBI-Gi2taxid. You request :
&gt; &gt;&gt;     &#39;File::Tail&#39; =&gt; 0.96
&gt; &gt;&gt; but this should read:
&gt; &gt;&gt;     &#39;File::Tail&#39; =&gt; 0.096
&gt; &gt;&gt; Otherwise  your module fails to install without forcing.
&gt; &gt;&gt;
&gt; &gt;&gt; All the best,
&gt; &gt;&gt; Denis
&gt; &gt;&gt;
&gt; &gt;&gt; --
&gt; &gt;&gt;
&gt; &gt;&gt; Prof. Denis BAURAIN
&gt; &gt;&gt; Eukaryotic Phylogenomics
&gt; &gt;&gt;
&gt; &gt;&gt; Department of Life Sciences
&gt; &gt;&gt; University of LiÃ¨ge
&gt; &gt;&gt; Sart Tilman, BÃ¢t. B22
&gt; &gt;&gt; B-4000 LiÃ¨ge, Belgium
&gt; &gt;&gt;
&gt; &gt;&gt;
&gt; &gt;&gt; At Fri, 30 May 2014 16:22:48 +0200, denis.baurain@ulg.ac.be wrote:
&gt; &gt;&gt;
<div class="message-stanza open">&gt; &gt;&gt; &gt; Thank you, Miguel.
&gt; &gt;&gt; &gt; This is great! :-&#41;
&gt; &gt;&gt; &gt;
&gt; &gt;&gt; &gt; All the best,
&gt; &gt;&gt; &gt; Denis
&gt; &gt;&gt; &gt;
&gt; &gt;&gt; &gt; At Fri, 30 May 2014 10:08:31 +0100, mp@ebi.ac.uk wrote:
&gt; &gt;&gt; &gt;
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt; Hi Denis,
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; Thanks a lot for your email.
&gt; &gt;&gt; &gt;&gt; Your request makes a lot of sense. I have included your patch,
&gt; &gt;&gt; &gt;&gt; added
&gt; &gt;&gt; &gt;&gt; a
&gt; &gt;&gt; &gt;&gt; couple of tests and corrected the indentation :-&#41;
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; I have uploaded the new version to cpan &#40;v0.09&#41;. It may take some
&gt; &gt;&gt; &gt;&gt; time
&gt; &gt;&gt; &gt;&gt; to reach all the mirrors but it should be available during the
&gt; &gt;&gt; &gt;&gt; morning.
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; Please, let me know if you have any other problem with the
&gt; &gt;&gt; &gt;&gt; module.
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; Best regards,
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; M;
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt;
&gt; &gt;&gt; &gt;&gt; On 29/05/14 22:05, Denis BAURAIN wrote:
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt; Dear Miguel,
&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; During the last few years, I have been developing an extensive
&gt; &gt;&gt; &gt;&gt;&gt; suite
</div>&gt; &gt;&gt; &gt;&gt; of Perl modules for phylogenomics. My modules do not use BioPerl
&gt; &gt;&gt; &gt;&gt; but
&gt; &gt;&gt; &gt;&gt; rest on Bio::Phylo and on your Bio::LITE::Taxonomy distribution.
&gt; &gt;&gt; &gt;&gt; Note
&gt; &gt;&gt; &gt;&gt; that you will not find them on CPAN yet because there are still
&gt; &gt;&gt; &gt;&gt; in
&gt; &gt;&gt; &gt;&gt; active development.
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; Anyway, I make extensive use of NCBI Taxonomy in my work and the
&gt; &gt;&gt; &gt;&gt;&gt; fact
</div>&gt; &gt;&gt; &gt;&gt; that Bio::LITE::Taxonomy does not handle synonyms drives me crazy
&gt; &gt;&gt; &gt;&gt; because NCBI often renames organisms: e.g., Canis familiaris does
&gt; &gt;&gt; &gt;&gt; not
&gt; &gt;&gt; &gt;&gt; work &#40;should be Canis lupus familiaris&#41; ; Xenopus tropicalis
&gt; &gt;&gt; &gt;&gt; neither
&gt; &gt;&gt; &gt;&gt; &#40;should be Xenopus &#40;Silurana&#41; tropicalis&#41; etc.
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; Therefore, I propose you a very small patch so as to handle them
&gt; &gt;&gt; &gt;&gt;&gt; in
</div>&gt; &gt;&gt; &gt;&gt; your module. The idea is simply to include the synonyms as
&gt; &gt;&gt; &gt;&gt; additional
&gt; &gt;&gt; &gt;&gt; keys for a given taxon id. This would increase the size of the
&gt; &gt;&gt; &gt;&gt; taxon
&gt; &gt;&gt; &gt;&gt; id
&gt; &gt;&gt; &gt;&gt; lookup by about 15% considering the current version of the NCBI
&gt; &gt;&gt; &gt;&gt; Taxonomy
&gt; &gt;&gt; &gt;&gt; database :
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; $ grep -c &#39;scientific name&#39; taxdump/names.dmp
&gt; &gt;&gt; &gt;&gt;&gt; 1160242
&gt; &gt;&gt; &gt;&gt;&gt; grep -c &#39;synonym&#39; taxdump/names.dmp
&gt; &gt;&gt; &gt;&gt;&gt; 187443
&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; From my understanding of your module, this should not have
&gt; &gt;&gt; &gt;&gt;&gt; adverse
</div>&gt; &gt;&gt; &gt;&gt; effects since there will still be only one name by taxon id when
&gt; &gt;&gt; &gt;&gt; going
&gt; &gt;&gt; &gt;&gt; from taxon id to name. This is only in the opposite direction
&gt; &gt;&gt; &gt;&gt; that
&gt; &gt;&gt; &gt;&gt; synonyms will come to play.
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; I attach a patch file. Here&#39;s the patch in context &#40;sorry for
&gt; &gt;&gt; &gt;&gt;&gt; the
</div>&gt; &gt;&gt; &gt;&gt; funny indentation; it appears like this on my system&#41;:
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; sub _name_nodes
&gt; &gt;&gt; &gt;&gt;&gt;      {
&gt; &gt;&gt; &gt;&gt;&gt;        my &#40;$self&#41; = @_;
&gt; &gt;&gt; &gt;&gt;&gt;        my $namesFile = $self-&gt;{namesFile};
&gt; &gt;&gt; &gt;&gt;&gt;        my $nodesNames;
&gt; &gt;&gt; &gt;&gt;&gt;        if &#40;&#40;UNIVERSAL::isa&#40;$namesFile, &#39;GLOB&#39;&#41;&#41; or &#40;ref
&gt; &gt;&gt; &gt;&gt;&gt; \$namesFile
&gt; &gt;&gt; &gt;&gt;&gt; eq
</div>&gt; &gt;&gt; &gt;&gt; &#39;GLOB&#39;&#41;&#41; {
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt; $nodesNames = $namesFile;
&gt; &gt;&gt; &gt;&gt;&gt;   } else {
&gt; &gt;&gt; &gt;&gt;&gt; open $nodesNames, &#34;&lt;&#34;, $namesFile or croak $!;
&gt; &gt;&gt; &gt;&gt;&gt;   }
&gt; &gt;&gt; &gt;&gt;&gt;   while &#40;&lt;$nodesNames&gt;&#41;{
&gt; &gt;&gt; &gt;&gt;&gt; chomp;
&gt; &gt;&gt; &gt;&gt;&gt; my &#40;$taxId,$taxName,$comment&#41; = _process_tax_name &#40;$_&#41;;
&gt; &gt;&gt; &gt;&gt;&gt; if &#40;$comment eq &#34;scientific name&#34;&#41;{
&gt; &gt;&gt; &gt;&gt;&gt;   ${$self-&gt;{nodes}-&gt;{$taxId}}{name} = $taxName;
&gt; &gt;&gt; &gt;&gt;&gt;       $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt; &gt;&gt; &gt;&gt;&gt; }
&gt; &gt;&gt; &gt;&gt;&gt; elsif &#40;$comment eq &#34;synonym&#34;&#41; {
&gt; &gt;&gt; &gt;&gt;&gt;     $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt; &gt;&gt; &gt;&gt;&gt; }
&gt; &gt;&gt; &gt;&gt;&gt;   }
&gt; &gt;&gt; &gt;&gt;&gt;   close $nodesNames;
&gt; &gt;&gt; &gt;&gt;&gt; }
&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; Would you like to consider this change for your next release?
&gt; &gt;&gt; &gt;&gt;&gt; This
</div>&gt; &gt;&gt; &gt;&gt; would help me a lot. If you need help for writing tests for this
&gt; &gt;&gt; &gt;&gt; feature, please let me know.
<div class="message-stanza open">&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; Thank you very much for your time!
&gt; &gt;&gt; &gt;&gt;&gt;
&gt; &gt;&gt; &gt;&gt;&gt; Best regards,
&gt; &gt;&gt; &gt;&gt;&gt; Denis
&gt; &gt;&gt; &gt;&gt;&gt;
</div>&gt; &gt;&gt; &gt;&gt;
</div></div></div>&gt; &gt;
&gt; &gt;
&gt; &gt;
</div></div></div></div>
    </div>
</div>
</div>
<div class="ticket-transaction basics Status even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438595" href="/Public/Bug/Display.html?id=100681#txn-1438595">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;07:13:04&nbsp;2014</span>
    <span class="description">MOTIF [...] cpan.org -  Status changed from &#39;open&#39; to &#39;resolved&#39;</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1438666" href="/Public/Bug/Display.html?id=100681#txn-1438666">#</a>      
    </span>
    <span class="date">Wed&nbsp;Dec&nbsp;03&nbsp;07:54:08&nbsp;2014</span>
    <span class="description">denis.baurain [...] ulg.ac.be -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #100681] makefile of Bio-LITE-Taxonomy-NCBI-Gi2taxid</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 3 Dec 2014 13:53:51 +0100</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> &lt;bug-Bio-LITE-Taxonomy-NCBI-Gi2taxid [...] rt.cpan.org&gt;</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> &#34;Denis BAURAIN&#34; &lt;denis.baurain [...] ulg.ac.be&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1438666/764794/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/764794">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 6.6k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Great, it works now.
Thank you,
Denis

At Wed, 3 Dec 2014 07:11:33 -0500, bug-Bio-LITE-Taxonomy-NCBI-Gi2taxid@rt.cpan.org wrote:

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt;&lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=100681">https://rt.cpan.org/Ticket/Display.html?id=100681</a></span> &gt;
&gt;
&gt;Hi Denis,
&gt;
&gt;Ok, I see. I suppose it is fine to remove the 0.96 version dependency
&gt;&#40;it shouldn&#39;t break with previous versions anyway&#41;.
&gt;I have updated a new version to cpan.
&gt;Let me know if you still have problems with this.
&gt;
&gt;Thanks for reporting!
&gt;
&gt;
&gt;El MiÃ© Dic 03 06:49:42 2014, denis.baurain@ulg.ac.be escribiÃ³:
<div class="message-stanza open">&gt;&gt; Thank you for your fast reply, Miguel.
&gt;&gt;
&gt;&gt; I use cpanm in conjunction with Perlbrew. I did not have the issue on
&gt;&gt; all combinations of perl and OS, but here&#39;s an example of error from
&gt;&gt; Linux with perl 5.20.1:
&gt;&gt;
&gt;&gt; Configuring Bio-LITE-Taxonomy-NCBI-Gi2taxid-0.12 ... OK
&gt;&gt; ==&gt; Found dependencies: File::Tail
&gt;&gt; Found File::Tail 0.099003 which doesn&#39;t satisfy 0.96.
&gt;&gt; ! Installing the dependencies failed: Module &#39;File::Tail&#39; is not
&gt;&gt; installed
&gt;&gt; ! Bailing out the installation for Bio-LITE-Taxonomy-NCBI-Gi2taxid-
&gt;&gt; 0.12.
&gt;&gt;
&gt;&gt; My understanding is that 0.96 should be spelled 0.096. Otherwise, I
&gt;&gt; fully agree with your analysis. Do you think that either cpanm or
&gt;&gt; something related has changed the way version numbers are evaluated?
&gt;&gt;
&gt;&gt; Best,
&gt;&gt; Denis
&gt;&gt;
&gt;&gt; At Wed, 3 Dec 2014 04:57:50 -0500, bug-Bio-LITE-Taxonomy-NCBI-
&gt;&gt; Gi2taxid@rt.cpan.org wrote:
&gt;&gt;
<div class="message-stanza open">&gt;&gt; &gt; &lt;URL: <span class="clickylink"><a target="new" rel="nofollow" href="https://rt.cpan.org/Ticket/Display.html?id=100681">https://rt.cpan.org/Ticket/Display.html?id=100681</a></span> &gt;
&gt;&gt; &gt;
&gt;&gt; &gt; Hi Denis,
&gt;&gt; &gt;
&gt;&gt; &gt; I can&#39;t reproduce this issue. I have installed version 0.99.3
&gt;&gt; &gt; &#40;current&#41;
&gt;&gt; &gt; of File::Tail and I&#39;m able to build &#38; test Bio-LITE-Taxonomy-NCBI-
&gt;&gt; &gt; Gi2taxid without problem.
&gt;&gt; &gt; File::Tail version 0.99.3 is from Sept 2005, so it is very likely
&gt;&gt; &gt; that
&gt;&gt; &gt; you have that version installed too. I don&#39;t see how requiring a
&gt;&gt; &gt; version
<div class="message-stanza open">&gt;&gt; &gt;&gt; 0.96 is failing for you.
</div>&gt;&gt; &gt; I haven&#39;t changed that dependency in the latest version of the module
&gt;&gt; &gt; &#40;I
&gt;&gt; &gt; don&#39;t remember changing it ever really&#41; and the code that depends on
&gt;&gt; &gt; File::Tail has not been changed lately &#40;AFAICT&#41;.
&gt;&gt; &gt;
&gt;&gt; &gt; What version of File::Tail do you have installed?
&gt;&gt; &gt; In what system is this failing for you?
&gt;&gt; &gt; What version of Perl are you using?
&gt;&gt; &gt;
&gt;&gt; &gt; Thanks,
&gt;&gt; &gt;
&gt;&gt; &gt; M;
&gt;&gt; &gt;
&gt;&gt; &gt;
&gt;&gt; &gt; El MiÃ© Dic 03 02:43:23 2014, denis.baurain@ulg.ac.be escribiÃ³:
<div class="message-stanza open">&gt;&gt; &gt;&gt; Hi Miguel,
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; I&#39;ve noticed a minor issue with the current &#40;0.12&#41; Makefile.PL of
&gt;&gt; &gt;&gt; your
&gt;&gt; &gt;&gt; module Bio-LITE-Taxonomy-NCBI-Gi2taxid. You request :
&gt;&gt; &gt;&gt;     &#39;File::Tail&#39; =&gt; 0.96
&gt;&gt; &gt;&gt; but this should read:
&gt;&gt; &gt;&gt;     &#39;File::Tail&#39; =&gt; 0.096
&gt;&gt; &gt;&gt; Otherwise  your module fails to install without forcing.
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; All the best,
&gt;&gt; &gt;&gt; Denis
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; --
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; Prof. Denis BAURAIN
&gt;&gt; &gt;&gt; Eukaryotic Phylogenomics
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; Department of Life Sciences
&gt;&gt; &gt;&gt; University of LiÃ¨ge
&gt;&gt; &gt;&gt; Sart Tilman, BÃ¢t. B22
&gt;&gt; &gt;&gt; B-4000 LiÃ¨ge, Belgium
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; At Fri, 30 May 2014 16:22:48 +0200, denis.baurain@ulg.ac.be wrote:
&gt;&gt; &gt;&gt;
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt; Thank you, Miguel.
&gt;&gt; &gt;&gt; &gt; This is great! :-&#41;
&gt;&gt; &gt;&gt; &gt;
&gt;&gt; &gt;&gt; &gt; All the best,
&gt;&gt; &gt;&gt; &gt; Denis
&gt;&gt; &gt;&gt; &gt;
&gt;&gt; &gt;&gt; &gt; At Fri, 30 May 2014 10:08:31 +0100, mp@ebi.ac.uk wrote:
&gt;&gt; &gt;&gt; &gt;
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt; Hi Denis,
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; Thanks a lot for your email.
&gt;&gt; &gt;&gt; &gt;&gt; Your request makes a lot of sense. I have included your patch,
&gt;&gt; &gt;&gt; &gt;&gt; added
&gt;&gt; &gt;&gt; &gt;&gt; a
&gt;&gt; &gt;&gt; &gt;&gt; couple of tests and corrected the indentation :-&#41;
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; I have uploaded the new version to cpan &#40;v0.09&#41;. It may take some
&gt;&gt; &gt;&gt; &gt;&gt; time
&gt;&gt; &gt;&gt; &gt;&gt; to reach all the mirrors but it should be available during the
&gt;&gt; &gt;&gt; &gt;&gt; morning.
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; Please, let me know if you have any other problem with the
&gt;&gt; &gt;&gt; &gt;&gt; module.
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; Best regards,
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; M;
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt; On 29/05/14 22:05, Denis BAURAIN wrote:
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt; Dear Miguel,
&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; During the last few years, I have been developing an extensive
&gt;&gt; &gt;&gt; &gt;&gt;&gt; suite
</div>&gt;&gt; &gt;&gt; &gt;&gt; of Perl modules for phylogenomics. My modules do not use BioPerl
&gt;&gt; &gt;&gt; &gt;&gt; but
&gt;&gt; &gt;&gt; &gt;&gt; rest on Bio::Phylo and on your Bio::LITE::Taxonomy distribution.
&gt;&gt; &gt;&gt; &gt;&gt; Note
&gt;&gt; &gt;&gt; &gt;&gt; that you will not find them on CPAN yet because there are still
&gt;&gt; &gt;&gt; &gt;&gt; in
&gt;&gt; &gt;&gt; &gt;&gt; active development.
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Anyway, I make extensive use of NCBI Taxonomy in my work and the
&gt;&gt; &gt;&gt; &gt;&gt;&gt; fact
</div>&gt;&gt; &gt;&gt; &gt;&gt; that Bio::LITE::Taxonomy does not handle synonyms drives me crazy
&gt;&gt; &gt;&gt; &gt;&gt; because NCBI often renames organisms: e.g., Canis familiaris does
&gt;&gt; &gt;&gt; &gt;&gt; not
&gt;&gt; &gt;&gt; &gt;&gt; work &#40;should be Canis lupus familiaris&#41; ; Xenopus tropicalis
&gt;&gt; &gt;&gt; &gt;&gt; neither
&gt;&gt; &gt;&gt; &gt;&gt; &#40;should be Xenopus &#40;Silurana&#41; tropicalis&#41; etc.
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Therefore, I propose you a very small patch so as to handle them
&gt;&gt; &gt;&gt; &gt;&gt;&gt; in
</div>&gt;&gt; &gt;&gt; &gt;&gt; your module. The idea is simply to include the synonyms as
&gt;&gt; &gt;&gt; &gt;&gt; additional
&gt;&gt; &gt;&gt; &gt;&gt; keys for a given taxon id. This would increase the size of the
&gt;&gt; &gt;&gt; &gt;&gt; taxon
&gt;&gt; &gt;&gt; &gt;&gt; id
&gt;&gt; &gt;&gt; &gt;&gt; lookup by about 15% considering the current version of the NCBI
&gt;&gt; &gt;&gt; &gt;&gt; Taxonomy
&gt;&gt; &gt;&gt; &gt;&gt; database :
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; $ grep -c &#39;scientific name&#39; taxdump/names.dmp
&gt;&gt; &gt;&gt; &gt;&gt;&gt; 1160242
&gt;&gt; &gt;&gt; &gt;&gt;&gt; grep -c &#39;synonym&#39; taxdump/names.dmp
&gt;&gt; &gt;&gt; &gt;&gt;&gt; 187443
&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; From my understanding of your module, this should not have
&gt;&gt; &gt;&gt; &gt;&gt;&gt; adverse
</div>&gt;&gt; &gt;&gt; &gt;&gt; effects since there will still be only one name by taxon id when
&gt;&gt; &gt;&gt; &gt;&gt; going
&gt;&gt; &gt;&gt; &gt;&gt; from taxon id to name. This is only in the opposite direction
&gt;&gt; &gt;&gt; &gt;&gt; that
&gt;&gt; &gt;&gt; &gt;&gt; synonyms will come to play.
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; I attach a patch file. Here&#39;s the patch in context &#40;sorry for
&gt;&gt; &gt;&gt; &gt;&gt;&gt; the
</div>&gt;&gt; &gt;&gt; &gt;&gt; funny indentation; it appears like this on my system&#41;:
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; sub _name_nodes
&gt;&gt; &gt;&gt; &gt;&gt;&gt;      {
&gt;&gt; &gt;&gt; &gt;&gt;&gt;        my &#40;$self&#41; = @_;
&gt;&gt; &gt;&gt; &gt;&gt;&gt;        my $namesFile = $self-&gt;{namesFile};
&gt;&gt; &gt;&gt; &gt;&gt;&gt;        my $nodesNames;
&gt;&gt; &gt;&gt; &gt;&gt;&gt;        if &#40;&#40;UNIVERSAL::isa&#40;$namesFile, &#39;GLOB&#39;&#41;&#41; or &#40;ref
&gt;&gt; &gt;&gt; &gt;&gt;&gt; \$namesFile
&gt;&gt; &gt;&gt; &gt;&gt;&gt; eq
</div>&gt;&gt; &gt;&gt; &gt;&gt; &#39;GLOB&#39;&#41;&#41; {
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt; $nodesNames = $namesFile;
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   } else {
&gt;&gt; &gt;&gt; &gt;&gt;&gt; open $nodesNames, &#34;&lt;&#34;, $namesFile or croak $!;
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   }
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   while &#40;&lt;$nodesNames&gt;&#41;{
&gt;&gt; &gt;&gt; &gt;&gt;&gt; chomp;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; my &#40;$taxId,$taxName,$comment&#41; = _process_tax_name &#40;$_&#41;;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; if &#40;$comment eq &#34;scientific name&#34;&#41;{
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   ${$self-&gt;{nodes}-&gt;{$taxId}}{name} = $taxName;
&gt;&gt; &gt;&gt; &gt;&gt;&gt;       $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt; &gt;&gt;&gt; elsif &#40;$comment eq &#34;synonym&#34;&#41; {
&gt;&gt; &gt;&gt; &gt;&gt;&gt;     $self-&gt;{names}-&gt;{$taxName} = $taxId;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   }
&gt;&gt; &gt;&gt; &gt;&gt;&gt;   close $nodesNames;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; }
&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Would you like to consider this change for your next release?
&gt;&gt; &gt;&gt; &gt;&gt;&gt; This
</div>&gt;&gt; &gt;&gt; &gt;&gt; would help me a lot. If you need help for writing tests for this
&gt;&gt; &gt;&gt; &gt;&gt; feature, please let me know.
<div class="message-stanza open">&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Thank you very much for your time!
&gt;&gt; &gt;&gt; &gt;&gt;&gt;
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Best regards,
&gt;&gt; &gt;&gt; &gt;&gt;&gt; Denis
&gt;&gt; &gt;&gt; &gt;&gt;&gt;
</div>&gt;&gt; &gt;&gt; &gt;&gt;
</div></div></div>&gt;&gt; &gt;
&gt;&gt; &gt;
&gt;&gt; &gt;
</div></div>&gt;
&gt;
&gt;
</div></div></div>
    </div>
</div>
</div>
</div>
    <hr class="clear" />
  </div>
</div>




</div>
 


  <hr class="clear" />
</div>
<p id="sponsors">
  This service is sponsored and maintained by
  <a href="http://bestpractical.com">Best Practical Solutions</a> and runs on
  <a href="http://noc.perl.org">Perl.org</a> infrastructure.
</p>

<p id="contact-info">
  Please report any issues with rt.cpan.org to <a href="mailto:rt-cpan-admin@bestpractical.com">rt-cpan-admin@bestpractical.com</a>.
</p>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA\x2D937849\x2D12');
  ga('send', 'pageview');
</script>

<div id="footer">
  <p id="time"><span>Time to display: 0.441584</span></p>
  <p id="bpscredits"><span>&#187;&#124;&#171; RT 4.0.18 Copyright 1996-2013 <a href="http://www.bestpractical.com?rt=4.0.18">Best Practical Solutions, LLC</a>.
</span></p>
</div>
  </body>
</html>
