--- lib/Locale/Maketext/Extract/Plugin/FormFu.pm	2008-12-28 17:57:47.000000000 +0100
+++ /opt/perl/lib/site_perl/5.8.8/Locale/Maketext/Extract/Plugin/FormFu.pm	2008-09-21 20:21:54.000000000 +0200
@@ -47,14 +47,14 @@
 
 sub extract {
     my $self = shift;
-    my $content = shift;
-    my $lno = 0;
-    foreach my $line (split /\n/, $content) {
-        $lno++;
-        if (my ($str) = $line =~ /.*?_loc[:]*\s+(.*?)$/) {
-            $self->add_entry($str, $lno);
-        }
+    local $_ = shift;
+    my$line = 1;
+    while (m/\G(.*?_loc:\s+(.*))/sg) {
+        my ($str) = $2;
+        $line += ( () = ($1 =~ /\n/g) ); # cryptocontext!
+        $self->add_entry($str, $line );
     }
+
 }
 
 =head1 SEE ALSO
