--- netdns.c	2009-01-26 09:31:43.000000000 -0800
+++ netdns.c.new	2009-04-08 16:19:45.000000000 -0700
@@ -95,6 +95,10 @@
   cp = comp_dn;
   eom = exp_dn + length;
 
+  if (!*cp) {
+      /* the while will fail */
+      *dn++ = '.';
+  }
 
   /*
    * fetch next label in domain name
@@ -163,12 +167,13 @@
       return (-1);/* flag error */
     }
   }
-  *dn = '\0';
 
+  if (len < 0) {
+      len = cp - comp_dn;
+  }
 
+  *dn = '\0';
 
-  if (len < 0)
-    len = cp - comp_dn;
   return (len);
 }
 
