cd c:\Perl\site\lib\Net\
diff -u IP.pm.orig IP.pm
Compilation started at Thu Nov 04 22:38:54 2004 +0100 (W. Europe Standard Time)
--- IP.pm.orig	2003-04-09 14:42:25.000000000 +0200
+++ IP.pm	2004-11-04 22:38:46.000000000 +0100
@@ -1533,10 +1533,8 @@
 	while (ip_bincomp ($binip,'le',$endbinip) == 1)
 	{
 		# Find all 0s at the end
-		$binip =~ m/(0+)$/;
-		# nbits = nb of 0 bits
-		
-		if ($1) {
+		if ($binip =~ m/(0+)$/) {
+			# nbits = nb of 0 bits
 			$nbits = length ($1);
 		}
 		else {

Compilation exited abnormally with code 1 at Thu Nov 04 22:38:54
