diff --git a/MANIFEST b/MANIFEST
index 2f9d199..17d481f 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -51,6 +51,7 @@ t/15_decrypt.t
 t/16_decrypt.t
 t/17_101092.t
 t/18_bug_92205.t
+t/19_bug_101240.t
 t/badjpeg/expected.jpg
 t/badjpeg/source.zip
 t/common.pm
diff --git a/lib/Archive/Zip/Member.pm b/lib/Archive/Zip/Member.pm
index f440a36..b2ae11a 100644
--- a/lib/Archive/Zip/Member.pm
+++ b/lib/Archive/Zip/Member.pm
@@ -496,11 +496,7 @@ sub extractToFileNamed {
         if ($^O eq 'MSWin32' && $Archive::Zip::UNICODE) {
             $name = decode_utf8(Win32::GetFullPathName($name));
             mkpath_win32($name);
-            if ($self->{'compressedSize'} == 0) {
-                return;
-            } else {
-                Win32::CreateFile($name);
-            }
+            Win32::CreateFile($name);
             ($status, $fh) = _newFileHandle(Win32::GetANSIPathName($name), 'w');
         } else {
             mkpath(dirname($name));    # croaks on error
