--- Tcl-1.05/Tcl.xs	2016-06-29 00:48:16.000000000 +0800
+++ Tcl-1.05-patched/Tcl.xs	2018-05-28 00:57:49.000000000 +0800
@@ -370,8 +370,10 @@
     /*
      * We want the Tcl_InitStubs func static to ourselves - before Tcl
      * is loaded dyanmically and possibly changes it.
+     * Variable initstubs have to be declared as volatile to prevent
+     * compiler optimizing it out.
      */
-    static CONST char *(*initstubs)(Tcl_Interp *, CONST char *, int)
+    static CONST char *(*volatile initstubs)(Tcl_Interp *, CONST char *, int)
 	= Tcl_InitStubs;
     char dllFilename[MAX_PATH];
     dllFilename[0] = '\0';
