Index: lib/GnuPG/Interface.pm
===================================================================
--- lib/GnuPG/Interface.pm	(révision 66357)
+++ lib/GnuPG/Interface.pm	(copie de travail)
@@ -552,10 +552,12 @@
             ) = @fields[ 1 .. 11 ];
 
             my $expiration_date_string;
-            if ($expiration_date eq '') {
-              $expiration_date = undef;
-            } else {
-              $expiration_date_string = $self->_downrez_date($expiration_date);
+            if (defined $expiration_date) {
+                if ($expiration_date eq '') {
+                    $expiration_date = undef;
+                } else {
+                    $expiration_date_string = $self->_downrez_date($expiration_date);
+                }
             }
             my $creation_date_string = $self->_downrez_date($creation_date);
 
