diff -ruN Module-Build-0.31011.orig/t/tilde.t Module-Build-0.31011/t/tilde.t
--- Module-Build-0.31011.orig/t/tilde.t	2009-01-13 06:04:58.000000000 +0000
+++ Module-Build-0.31011/t/tilde.t	2009-01-13 09:31:54.726252400 +0000
@@ -47,7 +47,7 @@
 
     unless (defined $home) {
       my @info = eval { getpwuid $> };
-      skip "No home directory for tilde-expansion tests", 14 if $@;
+      skip "No home directory for tilde-expansion tests", 15 if $@;
       $home = $info[7];
     }
 
