--- SSLeay.old	2019-05-05 23:26:44.000000000 -0400
+++ SSLeay.xs	2019-05-05 23:26:01.000000000 -0400
@@ -5681,8 +5681,10 @@
 	int md_size = -1;
 	c = s->enc_read_ctx->cipher;
 #if OPENSSL_VERSION_NUMBER >= 0x10001000L
+#if LIBRESSL_VERSION_NUMBER < 0x20900000L
 	if (s->s3)
 	    md_size = s->s3->tmp.new_mac_secret_size;
+#endif
 #elif OPENSSL_VERSION_NUMBER >= 0x00909000L
 	h = EVP_MD_CTX_md(s->read_hash);
 	md_size = EVP_MD_size(h);
