--- lib/Getopt/Long.old	2016-04-13 13:19:47.000000000 -0400
+++ lib/Getopt/Long.pm	2016-04-13 13:56:03.000000000 -0400
@@ -1451,7 +1451,7 @@
 
     my $v = $main::VERSION;
     my $fh = $pa->{-output} ||
-      ($pa->{-exitval} eq "NOEXIT" || $pa->{-exitval} < 2) ? \*STDOUT : \*STDERR;
+      ( ($pa->{-exitval} eq "NOEXIT" || $pa->{-exitval} < 2) ? \*STDOUT : \*STDERR );
 
     print $fh (defined($pa->{-message}) ? $pa->{-message} : (),
 	       $0, defined $v ? " version $v" : (),
