--- Look.pm.orig	Mon Feb 27 23:48:47 2012
+++ Look.pm	Mon Mar 05 11:33:45 2012
@@ -14,7 +14,8 @@
 
 BEGIN {
     my $path = $ENV{PATH} || '';
-    open my $svnlook, '-|', 'svnlook', '--version'
+    my $cmdline = 'svnlook --version';
+    open my $svnlook, "$cmdline |"
 	or die "Aborting because I couldn't find the 'svnlook' executable in PATH='$path'.\n";
     $_ = <$svnlook>;
     if (my ($major, $minor, $patch) = (/(\d+)\.(\d+)\.(\d+)/)) {
@@ -43,7 +44,8 @@
     my ($self, $cmd, @args) = @_;
     my @cmd = ('svnlook', $cmd, $self->{repo});
     push @cmd, @{$self->{opts}} unless $cmd =~ /^(?:youngest|uuid|lock)$/;
-    open my $fd, '-|', @cmd, @args
+    my $cmdline = "@cmd @args";
+    open my $fd, "$cmdline |"
         or die "Can't exec svnlook $cmd: $!\n";
     if (wantarray) {
         my @lines = <$fd>;
