Description: fix manpage-has-bad-whatis-entry issue
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-01-13

--- a/lib/Perinci/Sub/Util/PropertyModule.pm
+++ b/lib/Perinci/Sub/Util/PropertyModule.pm
@@ -69,7 +69,7 @@
 
 =head1 NAME
 
-Perinci::Sub::Util::PropertyModule
+Perinci::Sub::Util::PropertyModule - Helper for detecting additional property modules
 
 =head1 VERSION
 
