Description: Fix unknown directive =head on POD testing
Origin: vendor
Author: Salvatore Bonaccorso <carnil@debian.org>
Last-Update: 2012-02-16

--- a/lib/Contextual/Return.pm
+++ b/lib/Contextual/Return.pm
@@ -2257,7 +2257,7 @@
 (if it is also specified).
 
 
-=head 3 Preventing fallbacks
+=head3 Preventing fallbacks
 
 Sometimes fallbacks can be too helpful. Or sometimes you want to impose
 strict type checking on a return value.
