--- Apache.pm.orig	2008-01-08 08:45:00.000000000 +0100
+++ Apache.pm.without_locationmatch_mapping	2008-01-08 09:27:00.000000000 +0100
@@ -133,7 +133,7 @@
     
     # Check if $base_path appears to be a regex (contains invalid characters),
     # meaning we're in a LocationMatch block
-    if ( $base_path =~ m/[^$URI::uric]/o ) {
+    if ( $base_path =~ m/[^$URI::uric]/o AND NOT APACHEVARIABLE('CATALYST_DISABLE_LOCATIONMATCH_MAPPING') ) {
         # Find out what part of the URI path matches the LocationMatch regex,
         # that will become our base
         my $match = qr/($base_path)/;
