*** axestype-orig.pm	Wed Apr 25 20:16:10 2007
--- axestype.pm	Mon Nov 08 16:59:41 2010
***************
*** 1515,1526 ****
      # First, calculate some decent values
      if ( $self->{two_axes} ) 
      {   # XXX this is almost certainly a bug: this key is not set anywhere
!         my $min_range_1 = defined($self->{min_range_1})
!                 ? $self->{min_range_1}
!                 : $self->{min_range};
!         my $min_range_2 = defined($self->{min_range_2})
!                 ? $self->{min_range_2}
!                 : $self->{min_range};
  
          my(@y_min, @y_max);
          for my $nd (1 .. $self->{_data}->num_sets)
--- 1515,1526 ----
      # First, calculate some decent values
      if ( $self->{two_axes} ) 
      {   # XXX this is almost certainly a bug: this key is not set anywhere
!         my $min_range_1 = defined($self->{y1_min_range})
!                 ? $self->{y1_min_range}
!                 : $self->{y_min_range};
!         my $min_range_2 = defined($self->{y2_min_range})
!                 ? $self->{y2_min_range}
!                 : $self->{y_min_range};
  
          my(@y_min, @y_max);
          for my $nd (1 .. $self->{_data}->num_sets)
