diff -ur MooseX-Daemonize-0.12.old/lib/MooseX/Daemonize.pm MooseX-Daemonize-0.12/lib/MooseX/Daemonize.pm
--- MooseX-Daemonize-0.12.old/lib/MooseX/Daemonize.pm	2011-06-06 08:12:44.497535239 +0200
+++ MooseX-Daemonize-0.12/lib/MooseX/Daemonize.pm	2011-06-06 09:48:04.937460270 +0200
@@ -523,6 +523,28 @@
 
 L<Moose>, L<MooseX::Getopt>, L<MooseX::Types::Path::Class> and L<POSIX>
 
+=head1 CAVEATS
+
+When going into background MooseX::Daemonize closes all open file
+handles. This may interfere with you logging because it may also close the log
+file handle you want to write to. To prevent this you can either defer opening
+the log file until after start. Alternatively, use can use the
+'dont_close_all_files' option either from the command line or in your .sh
+script.
+
+Assuming you want to use Log::Log4perl for example you could expand the
+MooseX::Daemonize example above like this.
+
+    after start => sub {
+        my $self = shift;
+        return unless $self->is_daemon;
+        Log::Log4perl->init(\$log4perl_config);
+        my $logger = Log::Log4perl->get_logger();
+        $logger->info("Daemon started");
+        # your daemon code here ...
+    };
+
+
 =head1 INCOMPATIBILITIES
 
 None reported. Although obviously this will not work on Windows.
