--- Symdump.pm~	2007-06-25 15:07:38.000000000 +0900
+++ Symdump.pm	2009-07-03 15:02:55.451240000 +0900
@@ -49,8 +49,10 @@
     my($key,$val,$num,$pack,@todo,$tmp);
     my $result = {};
     foreach $pack (@packages){
-	no strict;
-	while (($key,$val) = each(%{*{"$pack\::"}})) {
+	no strict 'refs';
+	my $stash = \%{$pack .'::'};
+	while (($key,$val) = each(%{$stash})) {
+		next if ref(\$val) ne 'GLOB'; # stubs and constants are not type-globs
 	    my $gotone = 0;
 	    local(*ENTRY) = $val;
 	    #### SCALAR ####
