diff -uprN Module-ScanDeps-0.82/lib/Module/ScanDeps.pm Module-ScanDeps-0.82.az/lib/Module/ScanDeps.pm
--- Module-ScanDeps-0.82/lib/Module/ScanDeps.pm	2008-01-28 18:28:03.000000000 +0200
+++ Module-ScanDeps-0.82.az/lib/Module/ScanDeps.pm	2008-01-31 20:57:34.000000000 +0200
@@ -914,6 +914,7 @@ sub add_deps {
             my ($path, $basename) = ($1, $2);
 
             foreach (_glob_in_inc("auto/$path")) {
+				next if $_->{file} =~ m{(Glib)|(Cairo)|(Gtk2)|(GladeXML)} and $^O eq 'MSWin32';
                 next if $_->{file} =~ m{\bauto/$path/.*/};  # weed out subdirs
                 next if $_->{name} =~ m/(?:^|\/)\.(?:exists|packlist)$/;
                 my $ext = lc($1) if $_->{name} =~ /(\.[^.]+)$/;
