diff --git i/lib/Net/Stomp/MooseHelpers/ReadTrace.pm w/lib/Net/Stomp/MooseHelpers/ReadTrace.pm
index 3c08fb0..bf915b6 100644
--- i/lib/Net/Stomp/MooseHelpers/ReadTrace.pm
+++ w/lib/Net/Stomp/MooseHelpers/ReadTrace.pm
@@ -37,7 +37,9 @@ sub read_frame_from_fh {
     my ($self,$fh) = @_;
 
     local $/="\x0A";
-    my $command=<$fh>;chomp $command;
+    my $command=<$fh>;
+    return unless $command;
+    chomp $command;
     my %headers;
     while (defined(my $header_line=<$fh>)) {
         chomp $header_line;
@@ -51,6 +53,7 @@ sub read_frame_from_fh {
 
     my $body=<$fh>;
 
+    return unless $body;
     return unless $body =~ s{\x00$}{}; # 0 marks the end of the frame
 
     return Net::Stomp::Frame->new({
