From f195ee259a0247f382b961fa38cdd20f41c25c54 Mon Sep 17 00:00:00 2001
From: Reini Urban <rurban@cpanel.net>
Date: Fri, 4 Apr 2014 10:31:41 -0500
Subject: [PATCH 1/5] Makefile.PL: simplify 5.6 PREREQ_PM handling

---
 Makefile.PL | 24 ++++++++----------------
 1 file changed, 8 insertions(+), 16 deletions(-)

diff --git Makefile.PL Makefile.PL
index 8ea8cfb..158bafa 100644
--- Makefile.PL
+++ Makefile.PL
@@ -40,22 +40,14 @@ unless($md5_check) {
 
 my $twv = $] > 5.013 ? 0.23 : 0.21;
 
-my $prereq_pm = $] > 5.007 ?
-                             {
-                              Data::Dumper => 2.09,
-                              Digest::MD5 => 2.09,
-	                        Parse::RecDescent => 1.80,
-                              File::Spec => 0.8,
-                              Test::Warn => $twv,
-                             }
-
-                           :
-                             {
-                              Data::Dumper => 2.09,
-                              Digest::MD5 => 2.09,
-	                        Parse::RecDescent => 1.80,
-                              File::Spec => 0.8,
-                             };
+my $prereq_pm = {
+    Data::Dumper => 2.09,
+    Digest::MD5 => 2.09,
+    Parse::RecDescent => 1.80,
+    File::Spec => 0.8,
+    Test::Warn => $twv,
+};
+delete $prereq_pm->{'Test::Warn'} if $] <= 5.007;
 
 WriteMakefile(
     NAME => 'Inline',
-- 
1.9.1

