--- Math-Pari-2.010702.orig/utils/Math/PariBuild.pm	2005-11-11 05:49:34.000000000 -0500
+++ Math-Pari-2.010702/utils/Math/PariBuild.pm	2006-01-18 17:45:59.000000000 -0500
@@ -717,7 +717,9 @@ sub find_machine_architecture () {
   } elsif ($os eq 'linux') {
     chomp($machine = `uname -m`);
     $machine = 'sparcv9' if $machine eq 'sparc64';
-    if (-e '/proc/cpuinfo') {
+	$machine = 'hppa' if $machine eq 'parisc';
+	$machine = 'hppa' if $machine eq 'parisc64';
+	if (-e '/proc/cpuinfo' && -R '/proc/cpuinfo') {
       open IN, '/proc/cpuinfo' or die "open /proc/cpuinfo: $!";
       local $/ = undef;		# Needed?
       my $info = <IN>;
