Description: add =encoding to make newer Pod::Simple happy
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-08

--- a/lib/String/ToIdentifier/EN/Unicode.pm
+++ b/lib/String/ToIdentifier/EN/Unicode.pm
@@ -5,6 +5,8 @@
 use base 'String::ToIdentifier::EN';
 use Exporter 'import';
 
+=encoding utf8
+
 =head1 NAME
 
 String::ToIdentifier::EN::Unicode - Convert Strings to Unicode English Program
