Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-08

--- a/lib/UR/Context/Process.pm
+++ b/lib/UR/Context/Process.pm
@@ -35,7 +35,7 @@
 
 =head1 DESCRIPTION
 
-This module provides methods to set and retreive variaous names
+This module provides methods to set and retrieve variaous names
 associated with the program and the program version number.
 
 =cut
--- a/lib/UR/DataSource/Oracle.pm
+++ b/lib/UR/DataSource/Oracle.pm
@@ -222,7 +222,7 @@
 sub get_userenv {
 
     # there are two ways to set these values but this is
-    # the only way to retreive the values after they are set
+    # the only way to retrieve the values after they are set
 
     my ($self, $dbh) = @_;
 
