diff -ru SVN-Dump-0.03/lib/SVN/Dump/Reader.pm SVN-Dump-0.03.blanklines/lib/SVN/Dump/Reader.pm
--- SVN-Dump-0.03/lib/SVN/Dump/Reader.pm        2006-11-20 19:15:21.000000000 -0500
+++ SVN-Dump-0.03.blanklines/lib/SVN/Dump/Reader.pm     2007-03-15 12:32:01.000000000 -0400
@@ -60,16 +60,8 @@
     {
         my $included = $fh->read_record();
         $record->set_included_record( $included );
-        <$fh>; # chop the empty line that follows
     }

-    # chop empty line after the record
-    my $type = $headers->type();
-    <$fh> if $type !~ /\A(?:format|uuid)\z/;
-
-    # chop another one after a node with only a prop block
-    <$fh> if $type eq 'node' && $record->has_prop_only();
-
     # uuid and format record only contain headers
     return $record;
 }
@@ -83,7 +75,8 @@
         my $line = <$fh>;
         croak _eof() if !defined $line;
         chop $line;
-        last if $line eq ''; # stop on empty line
+        next if $line eq '' && ! keys %$headers; # ignore leading empty line
+        last if $line eq '' &&   keys %$headers; # stop on trailing empty line

         my ($key, $value) = split /: /, $line, 2;
         $headers->{$key} = $value;

