--- a/lib/DBD/SQLite.pm	2008-03-24 23:12:03.000000000 +0000
+++ b/lib/DBD/SQLite.pm	2008-03-24 23:12:42.000000000 +0000
@@ -208,7 +208,7 @@
 	my @pk = split /\s*,\s*/, $2 || '';
 	unless (@pk) {
 	    my $prefix = $1;
-	    $prefix =~ s/.*create\s+table\s+.*?\(\s*//i;
+	    $prefix =~ s/.*create\s+table\s+.*?\(\s*//si;
 	    $prefix = (split /\s*,\s*/, $prefix)[-1];
 	    @pk = (split /\s+/, $prefix)[0]; # take first word as name
 	}
