--- Module-Build-0.3601/lib/Module/Build/Compat.pm.orig	2010-01-05 08:41:07.000000000 -0500
+++ Module-Build-0.3601/lib/Module/Build/Compat.pm	2010-01-05 15:22:59.000000000 -0500
@@ -76,6 +76,7 @@
   # validate formats
   for my $p ( $req, $breq ) {
     for my $k (keys %$p) {
+      next if $k eq 'perl';
       die "Prereq '$p->{$k}' for '$k' is not supported by Module::Build::Compat\n"
         unless _simple_prereq($p->{$k});
     }
