--- old/DateTime.pm	2014-02-12 13:36:07.536401887 -0500
+++ new/DateTime.pm	2014-02-12 13:40:01.587274581 -0500
@@ -980,10 +980,10 @@
 }
 
 # RD 1 is JD 1,721,424.5 - a simple offset
-sub jd {
+sub mjd {
     my $self = shift;
 
-    my $jd = $self->{utc_rd_days} + 1_721_424.5;
+    my $jd = $self->{utc_rd_days} - 678576;
 
     my $day_length = $self->_day_length( $self->{utc_rd_days} );
 
@@ -992,7 +992,7 @@
             + ( $self->{rd_nanosecs} / $day_length / MAX_NANOSECONDS ) );
 }
 
-sub mjd { $_[0]->jd - 2_400_000.5 }
+sub jd { $_[0]->mjd + 2_400_000.5 }
 
 {
     my %strftime_patterns = (
