diff -ur Test-WWW-Mechanize-1.24/Makefile.PL Test-WWW-Mechanize-1.25/Makefile.PL
--- Test-WWW-Mechanize-1.24/Makefile.PL	2008-12-22 16:20:57.000000000 -0200
+++ Test-WWW-Mechanize-1.25/Makefile.PL	2010-02-14 23:37:53.869207309 -0200
@@ -13,7 +13,7 @@
         'HTTP::Server::Simple'      => '0.35',
         'HTTP::Server::Simple::CGI' => 0,
         'Test::Builder::Tester'     => '1.09',
-        'Test::LongString'          => '0.07',
+        'Test::LongString'          => '0.12',
         'Test::More'                => 0,
         'URI::file'                 => 0,
         'WWW::Mechanize'            => '1.24',
diff -ur Test-WWW-Mechanize-1.24/META.yml Test-WWW-Mechanize-1.25/META.yml
--- Test-WWW-Mechanize-1.24/META.yml	2009-01-17 17:41:50.000000000 -0200
+++ Test-WWW-Mechanize-1.25/META.yml	2010-02-14 23:42:53.373210892 -0200
@@ -12,7 +12,7 @@
     HTTP::Server::Simple:          0.35
     HTTP::Server::Simple::CGI:     0
     Test::Builder::Tester:         1.09
-    Test::LongString:              0.07
+    Test::LongString:              0.12
     Test::More:                    0
     URI::file:                     0
     WWW::Mechanize:                1.24
Only in Test-WWW-Mechanize-1.25: pm_to_blib
diff -ur Test-WWW-Mechanize-1.24/t/content_contains.t Test-WWW-Mechanize-1.25/t/content_contains.t
--- Test-WWW-Mechanize-1.24/t/content_contains.t	2008-12-22 21:44:52.000000000 -0200
+++ Test-WWW-Mechanize-1.25/t/content_contains.t	2010-02-14 23:29:49.437204797 -0200
@@ -32,9 +32,11 @@
 test_test( 'Finds the contains - default desc' );
 
 test_out( 'not ok 1 - Where is Mungo?' );
-test_fail(+3);
+test_fail(+5);
 test_diag(q(    searched: "<html>\x{0a}  <head>\x{0a}    <title>Test Page</title>\x{0a}  </h"...) );
 test_diag(q(  can't find: "Mungo") );
+test_diag(q(        LCSS: "go"));
+test_diag(q(LCSS context: "ead>\x{0a}  <body>\x{0a}  <h1>Test Page</h1>\x{0a}    <a href='go"));
 $mech->content_contains( 'Mungo', 'Where is Mungo?' );
 test_test( 'Handles not finding it' );
 
