diff -Nur Email-MessageID-1.31/t/defined.t Email-MessageID-1.32/t/defined.t
--- Email-MessageID-1.31/t/defined.t	2004-07-03 13:00:57.000000000 -0400
+++ Email-MessageID-1.32/t/defined.t	2006-07-03 20:08:55.000000000 -0400
@@ -1,4 +1,4 @@
-use Test::More qw[no_plan];
+use Test::More tests => 8;
 
 use_ok 'Email::MessageID';
 
diff -Nur Email-MessageID-1.31/t/unique.t Email-MessageID-1.32/t/unique.t
--- Email-MessageID-1.31/t/unique.t	2004-12-22 20:13:52.000000000 -0500
+++ Email-MessageID-1.32/t/unique.t	2006-07-03 20:09:22.000000000 -0400
@@ -1,4 +1,4 @@
-use Test::More qw[no_plan];
+use Test::More tests => 2001;
 
 use_ok 'Email::MessageID';
 
