--- a/lib/Apache/SizeLimit/Core.pm	2011-12-22 03:46:44.000000000 +0000
+++ b/lib/Apache/SizeLimit/Core.pm	2014-03-11 16:52:11.795119567 +0000
@@ -184,10 +184,10 @@
 sub _linux_size_check {
     my $class = shift;
 
-    my ($size, $share) = (0, 0);
+    my ($size, $data) = (0, 0);
 
     if (open my $fh, '<', '/proc/self/statm') {
-        ($size, $share) = (split /\s/, scalar <$fh>)[0,2];
+        ($size, $data) = (split /\s/, scalar <$fh>)[0,5];
         close $fh;
     }
     else {
@@ -195,7 +195,9 @@
     }
 
     # linux on intel x86 has 4KB page size...
-    return ($size * 4, $share * 4);
+    $size <<= 2;
+    $data <<= 2;
+    return ($size, $size - $data, $data);
 }
 
 sub _solaris_2_6_size_check {
