Description: fix a typo
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-05-10

--- a/bin/efa
+++ b/bin/efa
@@ -70,7 +70,7 @@
 		exit 2;
 	}
 	if ( $e->isa('Travel::Routing::DE::EFA::Exception::NoData') ) {
-		printf STDERR ( "Backend rror: %s\n", $e->description );
+		printf STDERR ( "Backend error: %s\n", $e->description );
 		exit 3;
 	}
 	if ( $e->isa('Travel::Routing::DE::EFA::Exception::Ambiguous') ) {
