--- Perl.pm-orig	2015-02-16 14:30:39.023740041 +0100
+++ Perl.pm	2015-02-16 14:35:12.243733195 +0100
@@ -685,7 +685,7 @@
     $logger->debug("Cleanline: $cleanline");
     
     # Remove any comments even those inline with code but not if the hash mark "#" is in a pattern match 
-    unless ($cleanline =~ /=~/) { $cleanline =~ s/#.*$//; }
+    unless ($cleanline =~ /=~/) { $cleanline =~ s/([^\$])#.*$/$1/; }
     $logger->debug("Cleanline: $cleanline");
     # Need to remove braces from counting when they are in a pattern match but not when they are supposed to be 
     # there as in the second use case listed below.  Below the use cases is some ideas on how to do this.
