--- 0.18/00.Locale-Maketext-Utils.t	2010-03-02 17:53:32.000000000 +0100
+++ HR/00.Locale-Maketext-Utils.t	2010-03-02 18:25:43.000000000 +0100
@@ -67,7 +67,7 @@
     local $ENV{'maketext_obj_skip_env'} = 1;
     local $ENV{'maketext_obj'} = 'CURRENT VALUE';
     my $noarg = TestApp::Localize->get_handle();
-    # depending on their Locales::Base may not have one of these
+    # depending on their Locales may not have one of these
     ok($noarg->language_tag() eq 'en' || $noarg->language_tag() eq 'en-us', 'get_handle no arg');
 
     my $first_lex = (@{ $noarg->_lex_refs() })[0];
@@ -79,7 +79,7 @@
 }
 
 my $no_arg = TestApp::Localize->get_handle();
-ok(ref($no_arg) eq 'TestApp::Localize::en_us', 'no argument has highest level langtag NS');
+ok(ref($no_arg) eq 'TestApp::Localize::en', 'no argument has highest level langtag NS');
 
 my $en = TestApp::Localize->get_handle('en');
 ok($ENV{'maketext_obj'} eq $en, 'ENV maketext_obj_skip_env false');
@@ -250,10 +250,10 @@
 ok($fr_hr->{'xxyyzz'} eq 'xxyyzz', 'names fake');
 
 my $sig_warn = exists $SIG{__WARN__} && defined $SIG{__WARN__} ? $SIG{__WARN__} : 'no exists/defined';
-my $base_sig_warn = exists $Locales::Base::SIG{__WARN__} && defined $Locales::Base::SIG{__WARN__} ? $Locales::Base::SIG{__WARN__} : 'no exists/defined';
+my $base_sig_warn = exists $Locales::SIG{__WARN__} && defined $Locales::SIG{__WARN__} ? $Locales::SIG{__WARN__} : 'no exists/defined';
 my ($loc_hr, $nat_hr) = $fr->lang_names_hashref('en-uk', 'it', 'xxyyzz');
 my $sig_warn_aft = exists $SIG{__WARN__} && defined $SIG{__WARN__} ? $SIG{__WARN__} : 'no exists/defined';
-my $base_sig_warn_aft = exists $Locales::Base::SIG{__WARN__} && defined $Locales::Base::SIG{__WARN__} ? $Locales::Base::SIG{__WARN__} : 'no exists/defined';
+my $base_sig_warn_aft = exists $Locales::SIG{__WARN__} && defined $Locales::SIG{__WARN__} ? $Locales::SIG{__WARN__} : 'no exists/defined';
 ok($sig_warn eq $sig_warn_aft, 'main sig warn unchanged by lang_names_hashref()');
 ok($base_sig_warn eq $base_sig_warn_aft, 'locale::base sig warn unchanged by lang_names_hashref()');
 
@@ -264,7 +264,7 @@
 
 ok($nat_hr->{'en'} eq 'English', 'array context native names default');
 ok($nat_hr->{'en-uk'} eq 'English (UK)', 'array context native names suffix');
-ok($nat_hr->{'it'} eq 'Italian', 'array context native names normal');
+ok($nat_hr->{'it'} eq 'Italiano', 'array context native names normal');
 ok($nat_hr->{'xxyyzz'} eq 'xxyyzz', 'array context native names fake');
 
 my $loadable_hr = $fr->loadable_lang_names_hashref('en-uk', 'it', 'xxyyzz', 'fr');
