--- Mail-Bulkmail-3.12/Bulkmail.pm	2003-12-22 18:54:10.000000000 +0000
+++ Mail/Bulkmail.pm	2004-11-15 19:00:14.000000000 +0000
@@ -485,18 +485,18 @@
 		};
 	};
 
-	#first, wipe out any previously set headers_from_message
-	if (defined $self->_previous_headers_from_message) {
-		foreach my $header (@{$self->_previous_headers_from_message}){
-			$self->header($header, undef);
+	#then, if we're setting new headers, we should set them.
+	if ($self->headers_from_message && ! $self->_extracted_headers_from_message) {
+		#first, wipe out any previously set headers_from_message
+		if (defined $self->_previous_headers_from_message) {
+			foreach my $header (@{$self->_previous_headers_from_message}){
+				$self->header($header, undef);
+			};
 		};
-	};
 
-	#wipe out the list of previously set headers
-	$self->_previous_headers_from_message([]);
+		#wipe out the list of previously set headers
+		$self->_previous_headers_from_message([]);
 
-	#then, if we're setting new headers, we should set them.
-	if ($self->headers_from_message && ! $self->_extracted_headers_from_message) {
 		$self->_extracted_headers_from_message(1);
 		$passed[0] ||= $self->_Message();	#We'll sometimes call this method after setting the message
 		#sendmail-ify our messages newlines
