Description: fix spelling mistakes in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-05-15

--- a/lib/Net/CLI/Interact/Manual/Cookbook.pod
+++ b/lib/Net/CLI/Interact/Manual/Cookbook.pod
@@ -92,7 +92,7 @@
 
  $s->set_prompt('name');
 
-If you want to test whether the current prompt matches a diffrent named Prompt
+If you want to test whether the current prompt matches a different named Prompt
 from the phrasebook, this method can be used:
 
  $s->prompt_looks_like('name');
--- a/lib/Net/CLI/Interact/Transport/Base.pm
+++ b/lib/Net/CLI/Interact/Transport/Base.pm
@@ -127,7 +127,7 @@
 
 =head2 irs_re
 
-Returns the Regular Expression reference used to split lines of reponse from
+Returns the Regular Expression reference used to split lines of response from
 the connected device. In the end, you will only receive data from this module
 separated by the C<ors> value (by default a newline character). The C<irs_re>
 is used internally by the module and is:
