--- HiRes.xs~	Sat Nov  5 15:45:22 2005
+++ HiRes.xs	Sun Nov  6 00:54:20 2005
@@ -77,8 +77,6 @@
 
 #endif /* #if defined(TIME_HIRES_CLOCK_GETTIME) && defined(_STRUCT_ITIMERSPEC) */
 
-#include "const-c.inc"
-
 #if defined(WIN32) || defined(CYGWIN_WITH_W32API)
 
 #ifndef HAS_GETTIMEOFDAY
@@ -689,6 +687,8 @@
 }
 
 #endif /* #ifdef HAS_GETTIMEOFDAY */
+
+#include "const-c.inc"
 
 MODULE = Time::HiRes            PACKAGE = Time::HiRes
 
