diff --git a/Build.PL b/Build.PL
index 7f4a15b..c716288 100644
--- a/Build.PL
+++ b/Build.PL
@@ -42,14 +42,14 @@ my @build_argz  =
     {
         qw
         (
-            strict          >0
+            strict          0
 
-            Carp            >0
-            Cwd             >0
-            FindBin         >0
-            Symbol          >0
-            File::Spec      >0
-            Test::More      >0
+            Carp            0
+            Cwd             0
+            FindBin         0
+            Symbol          0
+            File::Spec      0
+            Test::More      0
 
         )
     },
