--- 02file_ok.t.dist	Mon Apr 30 19:28:36 2007
+++ 02file_ok.t	Mon Apr 30 19:23:08 2007
@@ -6,8 +6,11 @@
 use Test::Files;
 
 my $test_file    = File::Spec->catfile( 't', '02file_ok.t' );
+$test_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $missing_file = File::Spec->catfile( 't', 'missing' );
+$missing_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $ok_pass_file = File::Spec->catfile( 't', 'ok_pass.dat' );
+$ok_pass_file=~s#\\#/#g if $^O eq 'MSWin32';
 
 #-----------------------------------------------------------------
 # Compare text to a file with same text.
