--- myldr/Makefile.PL.orig	2011-04-11 15:51:02.000000000 +0200
+++ myldr/Makefile.PL	2011-04-11 16:03:58.000000000 +0200
@@ -207,10 +207,11 @@
 if ($dynperl and $^O eq 'MSWin32'
     and defined $Config{gccversion}     # gcc version 4.x or above was used
     and $Config{gccversion} =~ m{\A(\d+)}ms && $1 >= 4) {
-    # look for a libgcc_*.dll in the same directory
-    # as the perl executable itself
-    ($libgcc) = glob(File::Spec->catfile(File::Basename::dirname($^X),
-                                         "libgcc_*.$Config{so}"));
+    # look for a libgcc_*.dll: 
+    # - in the same directory as the perl executable itself
+    # - in the same directory as gcc
+    ($libgcc) = map { glob(File::Spec->catfile(File::Basename::dirname($_), "libgcc_*.$Config{so}")) }
+                    $^X, $cc;
 }
 $libgcc ||= "";
 
