--- MakeMaker.pm.dist	Fri Feb 29 02:06:55 2008
+++ MakeMaker.pm	Wed Apr 16 14:11:58 2008
@@ -236,7 +236,7 @@
     LINKTYPE MAKE MAKEAPERL MAKEFILE MAKEFILE_OLD MAN1PODS MAN3PODS MAP_TARGET 
     MYEXTLIB NAME NEEDS_LINKING NOECHO NO_META NORECURS NO_VC OBJECT OPTIMIZE 
     PERL_MALLOC_OK PERL PERLMAINCC PERLRUN PERLRUNINST PERL_CORE
-    PERL_SRC PERM_RW PERM_RWX
+    PERL_SRC PERM_RW PERM_RWX MIN_PERL_VERSION
     PL_FILES PM PM_FILTER PMLIBDIRS PMLIBPARENTDIRS POLLUTE PPM_INSTALL_EXEC
     PPM_INSTALL_SCRIPT PREREQ_FATAL PREREQ_PM PREREQ_PRINT PRINT_PREREQ
     SIGN SKIP TYPEMAPS VERSION VERSION_FROM XS XSOPT XSPROTOARG
@@ -389,6 +389,10 @@
 
     check_hints($self);
 
+    if ($self->{MIN_PERL_VERSION} and $self->{MIN_PERL_VERSION} > $]) {
+        die "This distribution won't work without a higher version of Perl.\n";
+    }
+    
     my %configure_att;         # record &{$self->{CONFIGURE}} attributes
     my(%initial_att) = %$self; # record initial attributes
 
