diff --git a/t/1.t b/t/1.t
index 50b83d5..a9b79d1 100644
--- a/t/1.t
+++ b/t/1.t
@@ -239,7 +239,7 @@ SKIP: {
 				];
 				my $d = $_[ARG0]->{result};
 				for my $i ( 0 .. $#{$r} ) {
-					foreach my $k (keys %{$r->[i]}) {
+					foreach my $k (keys %{$r->[$i]}) {
 						unless ($r->[$i]->{$k} eq $d->[$i]->{$k}) {
 							$_[ARG0]->{error} = "incorrect data in $i ($k)";
 						}
diff --git a/t/2.t b/t/2.t
index fc55a43..4c8cc56 100644
--- a/t/2.t
+++ b/t/2.t
@@ -263,7 +263,7 @@ SKIP: {
 				];
 				my $d = $_[ARG0]->{result};
 				for my $i ( 0 .. $#{$r} ) {
-					foreach my $k (keys %{$r->[i]}) {
+					foreach my $k (keys %{$r->[$i]}) {
 						unless ($r->[$i]->{$k} eq $d->[$i]->{$k}) {
 							$_[ARG0]->{error} = "incorrect data in $i ($k)";
 						}
diff --git a/t/3.t b/t/3.t
index 9e11c46..d942a0b 100644
--- a/t/3.t
+++ b/t/3.t
@@ -265,7 +265,7 @@ SKIP: {
 				];
 				my $d = $_[ARG0]->{result};
 				for my $i ( 0 .. $#{$r} ) {
-					foreach my $k (keys %{$r->[i]}) {
+					foreach my $k (keys %{$r->[$i]}) {
 						unless ($r->[$i]->{$k} eq $d->[$i]->{$k}) {
 							$_[ARG0]->{error} = "incorrect data in $i ($k)";
 						}
