Index: lib/CPANPLUS/Dist.pm
===================================================================
--- lib/CPANPLUS/Dist.pm	(revision 2507)
+++ lib/CPANPLUS/Dist.pm	(working copy)
@@ -455,6 +455,8 @@
                             $cb->_perl_version( perl => $^X ) ) );
                 return;                            
             }
+
+            next;
         }
         
         my $modobj  = $cb->module_tree($mod);
