diff -u -r a/lib/Net/SFTP/Foreign/Backend/Unix.pm b/lib/Net/SFTP/Foreign/Backend/Unix.pm
--- a/lib/Net/SFTP/Foreign/Backend/Unix.pm	2010-09-29 09:17:22.004245800 +0200
+++ b/lib/Net/SFTP/Foreign/Backend/Unix.pm	2010-09-29 13:39:17.517121800 +0200
@@ -61,11 +61,11 @@
 	    return undef;
 	}
 	local ($@, $SIG{__DIE__}, $SIG{__WARN__});
-	return eval { open3(@_[1,0], ">&SSHERR", @_[3..$#_]) }
+	return eval { untie *STDERR; open3(@_[1,0], ">&SSHERR", @_[3..$#_]) }
     }
     else {
 	local ($@, $SIG{__DIE__}, $SIG{__WARN__});
-	return eval { open2(@_[0,1], @_[3..$#_]) };
+	return eval { untie *STDERR; open2(@_[0,1], @_[3..$#_]) };
     }
 }
 
