Description: Fix HTML by closing head correctly
Author: Thomas Poindessous <thomas@poindessous.com>
Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?msg=5;filename=libhtml-template-perl.diff;att=1;bug=204075
Bug-Debian: http://bugs.debian.org/204075
Reviewed-By: Florian Ernst <florian@debian.org>
Last-Update: 2010-08-12

--- a/Template.pm
+++ b/Template.pm
@@ -14,7 +14,7 @@ extra tags, the simplest being <TMPL_VAR
 For example, test.tmpl:
 
   <html>
-  <head><title>Test Template</title>
+  <head><title>Test Template</title></head>
   <body>
   My Home Directory is <TMPL_VAR NAME=HOME>
   <p>
