From: Ansgar Burchardt <ansgar@43-1.org>
Date: Thu, 05 Aug 2010 20:20:58 +0900
Origin: vendor
Subject: Fix spelling error in documentation

--- libwww-mechanize-shell-perl.orig/lib/WWW/Mechanize/Shell.pm
+++ libwww-mechanize-shell-perl/lib/WWW/Mechanize/Shell.pm
@@ -1967,7 +1967,7 @@
 =head1 ONLINE HELP
 
 The online help feature is currently a bit broken in C<Term::Shell>,
-but a fix is in the works. Until then, you can reenable the
+but a fix is in the works. Until then, you can re-enable the
 dynamic online help by patching C<Term::Shell> :
 
 Remove the three lines
