*** Log/Dispatch/FileShared.pm.orig	Fri Feb 17 13:11:55 2012
--- Log/Dispatch/FileShared.pm	Fri Feb 17 13:12:21 2012
***************
*** 240,246 ****
     			};
     			alarm 0;
     			if ($@) {
!    				close($h);
     				if ($@ eq __PACKAGE__ . ".ALRM\n") {
     					# This is a dilemma too.
  					warn(sprintf("Timeout waiting for lock on '%s'.", $self->{'filename'}));
--- 240,246 ----
     			};
     			alarm 0;
     			if ($@) {
!    				$self->_close_handle();
     				if ($@ eq __PACKAGE__ . ".ALRM\n") {
     					# This is a dilemma too.
  					warn(sprintf("Timeout waiting for lock on '%s'.", $self->{'filename'}));
