Description: fix pod(2man) glitches
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-05-30

--- a/lib/Pod/PseudoPod/Tutorial.pod
+++ b/lib/Pod/PseudoPod/Tutorial.pod
@@ -1,8 +1,10 @@
 =pod
 
-=head1 PseudoPod
+=encoding ISO8859-1
 
-A Pod Extension for Authoring Large Documents
+=head1 NAME
+
+PseudoPod - A Pod Extension for Authoring Large Documents
 
 =head2 Introduction
 
