--- File.xs~	2006-06-30 15:47:43.000000000 +0100
+++ File.xs	2010-10-28 16:15:33.731196800 +0100
@@ -645,3 +645,23 @@
     OUTPUT:
 	RETVAL
 	ouBytesWritten
+
+void
+GetStdHandle(fd)
+    DWORD fd
+PPCODE:
+#ifdef _WIN64
+    XSRETURN_IV((DWORD_PTR)GetStdHandle(fd));
+#else
+    XSRETURN_IV((DWORD)GetStdHandle(fd));
+#endif
+
+void
+SetStdHandle(fd,handle)
+    DWORD fd
+    HANDLE handle
+PPCODE:
+    if (SetStdHandle(fd, handle))
+	XSRETURN_YES;
+    else
+	XSRETURN_NO;
