--- Catalyst/Plugin/Log/Handler.pm	2009-11-26 18:08:37.000000000 +0000
+++ Catalyst/Plugin/Log/Handler.pm	2009-11-26 18:14:08.000000000 +0000
@@ -8,13 +8,13 @@
 
 =head1 VERSION
 
-Version 0.07
+Version 0.08
 
 =cut
 
-our $VERSION = '0.07';
+our $VERSION = '0.08';
 
-use NEXT;
+use MRO::Compat;
 
 sub setup {
     my $c = shift;
@@ -23,7 +23,7 @@
 
     $c->log((__PACKAGE__ . '::Backend')->new($config));
 
-    return $c->NEXT::setup(@_);
+    return $c->next::method(@_);
 }
 
 
@@ -31,7 +31,8 @@
 use strict;
 use warnings;
 use base qw(Class::Accessor::Fast);
-use Log::Handler 0.33;
+use Log::Handler 0.63;
+use MRO::Compat;
 
 __PACKAGE__->mk_accessors(qw(handler));
 
@@ -68,7 +69,7 @@
 
 sub new {
     my $class = shift;
-    my $self  = $class->SUPER::new;
+    my $self  = $class->next::method();
 
     my ($config) = @_;
 
@@ -80,11 +81,11 @@
     # Log::Handler->new will fail if there's no filename in the conf.  But let's
     # try it anyway to convince the user.
 
-    $self->handler(Log::Handler->new(
+    $self->handler(Log::Handler->new( file => {
 	minlevel => 0,
 	maxlevel => 7,
 	%$config,
-    ));
+    } ));
     return $self;
 }
 
@@ -100,10 +101,11 @@
 Catalyst configuration (e. g. in YAML format):
 
     Log::Handler:
-        filename: /var/log/myapp.log
-        fileopen: 1
-        mode: append
-        newline: 1
+        file:
+            filename: /var/log/myapp.log
+            fileopen: 1
+            mode: append
+            newline: 1
 
 To log a message:
 
