--- /usr/local/share/perl/5.10.0/CAM/PDF.pm	2010-01-15 01:31:25.000000000 -0600
+++ CAM/PDF.pm	2010-01-15 01:54:56.126502736 -0600
@@ -2825,9 +2825,7 @@ sub getFormField
       {
          my $root = $self->getRootDict()->{AcroForm};
          return if (!$root);
-         $parent = $self->dereference($root->{value});
-         return if (!$parent);
-         my $dict = $self->getValue($parent);
+         my $dict = $self->getValue($root);
          return if (!exists $dict->{Fields});
          $kidlist = $self->getValue($dict->{Fields});
       }
@@ -4576,15 +4574,16 @@ sub fillFormFields  ## no critic(Subrout
             # Search out font resources.  This is a total kluge.
             # TODO: the right way to do this is to look for the DR
             # attribute in the form element or it's ancestors.
-            for my $font (@rsrcs)
-            {
-               my $fobj = $self->dereference("/$font", 'All');
-               if (!$fobj)
-               {
-                  die "Could not find resource /$font while preparing form field $key\n";
-               }
-               $fdict->{$font} = CAM::PDF::Node->new('reference', $fobj->{objnum}, $formonum, $formgnum);
-            }
+# this code doesn't work with the fix for bug #53698
+#            for my $font (@rsrcs)
+#            {
+#               my $fobj = $self->dereference("/$font", 'All');
+#               if (!$fobj)
+#               {
+#                  die "Could not find resource /$font while preparing form field $key\n";
+#               }
+#               $fdict->{$font} = CAM::PDF::Node->new('reference', $fobj->{objnum}, $formonum, $formgnum);
+#            }
          }
       }
       $filled++;
