From 203ea7adfd66a00ebb104af882c4d96faf244230 Mon Sep 17 00:00:00 2001
From: Sebastian Podjasek <sebastian@podjasek.pl>
Date: Tue, 13 May 2014 08:43:07 +0200
Subject: [PATCH] Fix indentation of wide-characters

---
 lib/Perl/Tidy.pm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Perl/Tidy.pm b/lib/Perl/Tidy.pm
index c326a8f..19b92bd 100644
--- a/lib/Perl/Tidy.pm
+++ b/lib/Perl/Tidy.pm
@@ -165,6 +165,7 @@ EOM
     }
     $fh = $New->( $filename, $mode )
       or Warn("Couldn't open file:$filename in mode:$mode : $!\n");
+    $fh->binmode(':utf8');
     return $fh, ( $ref or $filename );
 }
 
-- 
1.9.1

