diff -ruN Image-Info-1.20-beta2/Changes Image-Info-1.20/Changes
--- Image-Info-1.20-beta2/Changes	2006-03-13 18:36:52.000000000 +0000
+++ Image-Info-1.20/Changes	2006-03-13 19:01:04.000000000 +0000
@@ -8,7 +8,8 @@
     + read only 11 bytes to determine file format (tiny.pgm is 11 bytes long)
     + fix provided patch to really work
   * fix bug #18147: 
-    + new TIFF code failed on little-endian machines - thanx jpo!
+    + new TIFF code failed if the system endianness didn't match the TIFF image
+      endianness  - thanx jpo!
   * replace the link to the TIFF spec with a public link - thanx jpo!
 
 2006-03-05 v1.19 Tels 74 tests
diff -ruN Image-Info-1.20-beta2/CREDITS Image-Info-1.20/CREDITS
--- Image-Info-1.20-beta2/CREDITS	2006-03-10 10:30:36.000000000 +0000
+++ Image-Info-1.20/CREDITS	2006-03-13 19:01:20.000000000 +0000
@@ -19,7 +19,7 @@
 Gisle Aas				Main author and contributor
 Jay Soffian <jay@loudcloud.com>		Exif MakerNote decoding
 Jerrad Pierce <belg4mit@mit.edu>	ASCII, BMP SVG, XPM and XBM support
-jpo <jpo@di.uminho.pt>			XBM/XPM bug reports and patches
+Jose Pedro Oliveira <jpo@di.uminho.pt>	XBM/XPM/TIFF bug reports and patches
 Phil Harvey <phil@ owl.phy.queensu.ca>	Panasonic Exif MakerNotes
 Ralf Steines <metamonk@yahoo.com> 	GIF fixes
 <srezic@cpan.org>			bug report #8522
diff -ruN Image-Info-1.20-beta2/lib/Image/Info/TIFF.pm Image-Info-1.20/lib/Image/Info/TIFF.pm
--- Image-Info-1.20-beta2/lib/Image/Info/TIFF.pm	2006-03-13 18:33:19.000000000 +0000
+++ Image-Info-1.20/lib/Image/Info/TIFF.pm	2006-03-13 19:01:20.000000000 +0000
@@ -256,7 +256,7 @@
 MAGIC: /^MM\x00\x2a/
 MAGIC: /^II\x2a\x00/
 
-The C<TIFF> spec can be found at (requires a login):
+The C<TIFF> spec can be found at:
 L<http://partners.adobe.com/public/developer/tiff/>
 
 The EXIF spec can be found at:
