--- Stash.xs.dist	Mon Nov  6 13:47:48 2006
+++ Stash.xs	Mon Nov  6 12:49:04 2006
@@ -45,12 +45,22 @@
 }
 #endif
 
+#if defined(__BORLANDC__) || defined(_MSC_VER)
+#define snprintf _snprintf
+#endif
+
 #if 0
 #define debug(format, ...) fprintf (stderr, format, ## __VA_ARGS__)
 #else
+
+#if defined(_MSC_VER)
+#define debug()
+#else
 #define debug(format, ...)
 #endif
 
+#endif
+
 #define TT_STASH_PKG	"Template::Stash::XS"
 #define TT_LIST_OPS	    "Template::Stash::LIST_OPS"
 #define TT_HASH_OPS	    "Template::Stash::HASH_OPS"
@@ -499,7 +509,7 @@
         }
         
         /* drop-through if not an object or method not found  */
-        switch SvTYPE(SvRV(root)) {
+        switch (SvTYPE(SvRV(root))) {
             
         case SVt_PVHV:				    /* HASH */
             roothv = (HV *) SvRV(root);
