--- /usr/local/lib/perl5/site_perl/5.8.6/PAR.pm~	2005-01-07 00:15:37.000000000 -0500
+++ /usr/local/lib/perl5/site_perl/5.8.6/PAR.pm	2005-01-07 00:15:57.000000000 -0500
@@ -525,7 +525,7 @@
         return ($fh, 0, $filename);
     }
 
-    ($filename) = $filename =~ /^([\x20-\xff]+)$
+    ($filename) = $filename =~ /^([\x20-\xff]+)$/;
     open my $fh, '+>', $filename or die $!;
     binmode($fh);
     return ($fh, 1, $filename);
