--- c:\Perl\site\lib\ExtUtils\Manifest.pm       Mon Nov 17 20:56:06 2008
+++ Manifest.pm.NEW     Mon Apr 13 18:11:05 2009
@@ -378,11 +378,12 @@
     while (<M>){
        chomp;
        s/\r//;
-       next if /^#/;
-       next if /^\s*$/;
-        s/^'//;
-        s/'$//;
-       push @skip, _macify($_);
+       $_ =~ qr{^\s*(?:(?:'([^\\']*(?:\\.[^\\']*)*)')|([^#\s]\S*))?(?:(?:\s*)|(?:\s+(.*?)\s*))$};
+       #my $comment = $3;
+       my $filename = $2;
+       if ( defined($1) ) { $filename = $1; $filename =~ s/\\(['\\])/$1/g; }
+       next if (not defined($filename) or not $filename);
+       push @skip, _macify($filename);
     }
     close M;
     return sub {0} unless (scalar @skip > 0);
