--- DBD-mysql-4.033/dbdimp.c	2016-02-16 04:39:12.852079166 -0800
+++ DBD-mysql-4.033/dbdimp.c	2016-02-16 04:38:42.856000275 -0800
@@ -4879,7 +4879,7 @@ int dbd_bind_ph(SV *sth, imp_sth_t *imp_
           if (DBIc_TRACE_LEVEL(imp_xxh) >= 2)
             PerlIO_printf(DBIc_LOGPIO(imp_xxh),
                           "   SCALAR type %d ->%ld<- IS A INT NUMBER\n",
-                          (int) sql_type, (long) (*buffer));
+                          (int) sql_type, *((int32_t *)buffer));
           break;
 
         case MYSQL_TYPE_DOUBLE:
--- DBD-mysql-4.033/dbdimp.h	2016-02-16 04:39:55.405194264 -0800
+++ DBD-mysql-4.033/dbdimp.h	2016-02-16 04:39:48.165925859 -0800
@@ -212,7 +212,7 @@ typedef struct imp_sth_ph_st {
 typedef struct imp_sth_phb_st {
     union
     {
-      long lval;
+      int32_t lval;
       double dval;
     } numeric_val;
     unsigned long   length;
@@ -233,7 +233,7 @@ typedef struct imp_sth_fbh_st {
     char           *data;
     int            charsetnr;
     double         ddata;
-    long           ldata;
+    int32_t        ldata;
 #if MYSQL_VERSION_ID < FIELD_CHARSETNR_VERSION 
     unsigned int   flags;
 #endif
