--- a/lib/Locale/Maketext/Extract.pm
+++ b/lib/Locale/Maketext/Extract.pm
@@ -255,7 +255,7 @@ sub extract {
 
     # Template Toolkit
     $line = 1; pos($_) = 0;
-    while (m!\G(.*?\[%-?\s*\|l(?:oc)?(.*?)\s*(-?)%\](.*?)\[%(-?)\s*END\s*-?%\])!sg) { 
+    while (m!\G(.*?\[%[-=~+]?\s*\|l(?:oc)?(.*?)\s*([-=~+]?)%\](.*?)\[%([-=~+]?)\s*END\s*[-=~+]?%\])!sg) {
         my ($vars, $trim_start, $str, $trim_end) = ($2, $3, $4, $5);
         $line += ( () = ($1 =~ /\n/g) ); # cryptocontext!
         $vars =~ s/^\s*\(//;
