=== modified file 'SQL/Output.pm'
--- SQL/Output.pm	2009-12-16 08:57:49 +0000
+++ SQL/Output.pm	2009-12-16 09:43:21 +0000
@@ -779,6 +779,7 @@
     . $self->{newline}
     . $self->{indent}
     . join($self->{newline}, @columns)
+    . $self->get_smallpackage_column_sql()
     . $self->{newline} . ")"
     . $self->{indent}
     . join(
@@ -1041,7 +1042,17 @@
 # SQL clauses to add at the end of the named table column definitions
 sub get_smallpackage_column_sql  {
   my $self = shift;
-  return $self->{log}->logdie("NOTIMPL");
+  my $clause = $self->_get_smallpackage_sql(q{columns});
+  if ( $clause ne '' )
+  {
+    $clause =~ s/\n(.*?)/\n$self->{indent}$1/g;
+    $clause = ','
+      . $self->{newline}
+      . $self->{indent}
+      . $clause;
+    return $clause;
+  }
+  return '';
 }
 
 # SQL clauses to add at the end of the named table index definitions

