--- ./lib/Net/Amazon/EC2_patch.pm	2015-03-15 19:33:52.000000000 +0100
+++ ./lib/Net/Amazon/EC2.pm	2014-08-05 04:13:42.000000000 +0200
@@ -1836,18 +1836,6 @@
 			}
 			$item->{description} = undef if ref ($item->{description});
 
-			my $tag_sets;
-			foreach my $tag_arr (@{$item->{tagSet}{item}}) {
-                if ( ref $tag_arr->{value} eq "HASH" ) {
-                    $tag_arr->{value} = "";
-                }
-				my $tag = Net::Amazon::EC2::TagSet->new(
-					key => $tag_arr->{key},
-					value => $tag_arr->{value},
-				);
-				push @$tag_sets, $tag;
-			}
-
 			my $image = Net::Amazon::EC2::DescribeImagesResponse->new(
 				image_id				=> $item->{imageId},
 				image_owner_id			=> $item->{imageOwnerId},
@@ -1866,7 +1854,6 @@
 				root_device_type		=> $item->{rootDeviceType},
 				root_device_name		=> $item->{rootDeviceName},
 				block_device_mapping	=> $block_device_mappings,
-				tag_set			        => $tag_sets,
 			);
 			
 			if (grep { defined && length } $item->{productCodes} ) {
@@ -2885,18 +2872,6 @@
 				$snap->{progress} = undef;
 			}
 
-			my $tag_sets;
-			foreach my $tag_arr (@{$snap->{tagSet}{item}}) {
-                if ( ref $tag_arr->{value} eq "HASH" ) {
-                    $tag_arr->{value} = "";
-                }
-				my $tag = Net::Amazon::EC2::TagSet->new(
-					key => $tag_arr->{key},
-					value => $tag_arr->{value},
-				);
-				push @$tag_sets, $tag;
-			}
-
  			my $snapshot = Net::Amazon::EC2::Snapshot->new(
  				snapshot_id		=> $snap->{snapshotId},
  				status			=> $snap->{status},
@@ -2907,7 +2882,6 @@
  				volume_size		=> $snap->{volumeSize},
  				description		=> $snap->{description},
  				owner_alias		=> $snap->{ownerAlias},
-				tag_set			=> $tag_sets,
  			);
  			
  			push @$snapshots, $snapshot;
