diff -urN libcrypt-hcesha-perl-0.70.orig//HCE_SHA.pm libcrypt-hcesha-perl-0.70/HCE_SHA.pm
--- libcrypt-hcesha-perl-0.70.orig//HCE_SHA.pm	2004-03-26 16:27:58.000000000 +0100
+++ libcrypt-hcesha-perl-0.70/HCE_SHA.pm	2011-04-29 20:42:11.309252526 +0200
@@ -10,7 +10,7 @@
 use strict;
 use vars qw($VERSION @ISA @EXPORT @EXPORT_OK);
 
-use Digest::SHA1;
+use Digest::SHA;
 use MIME::Base64;
 use Carp;
 
@@ -46,7 +46,7 @@
     my $self = shift;
     my ($rnd) = @_;
  
-    my $context = new Digest::SHA1;
+    my $context = new Digest::SHA;
     $context->add($self->{SKEY}, $rnd);
     my $digest = $context->digest();
     my @e_block = unpack('C*', $digest);
diff -urN libcrypt-hcesha-perl-0.70.orig//META.yml libcrypt-hcesha-perl-0.70/META.yml
--- libcrypt-hcesha-perl-0.70.orig//META.yml	2004-03-26 16:28:13.000000000 +0100
+++ libcrypt-hcesha-perl-0.70/META.yml	2011-04-29 20:42:17.622499154 +0200
@@ -5,7 +5,7 @@
 version_from: HCE_SHA.pm
 installdirs:  site
 requires:
-    Digest::SHA1:                  0
+    Digest::SHA:                  0
     MIME::Base64:                  2
 
 distribution_type: module
diff -urN libcrypt-hcesha-perl-0.70.orig//Makefile.PL libcrypt-hcesha-perl-0.70/Makefile.PL
--- libcrypt-hcesha-perl-0.70.orig//Makefile.PL	2003-11-13 00:24:46.000000000 +0100
+++ libcrypt-hcesha-perl-0.70/Makefile.PL	2011-04-29 20:42:25.417251532 +0200
@@ -4,6 +4,6 @@
 WriteMakefile(
     'NAME'	=> 'Crypt::HCE_SHA',
     'VERSION_FROM' => 'HCE_SHA.pm', # finds $VERSION
-    'PREREQ_PM' => {'Digest::SHA1' => 0,
+    'PREREQ_PM' => {'Digest::SHA' => 0,
 		    'MIME::Base64' => 2}
 );
