From b5d838f494dcc6a0852050e565450e68b6587dd9 Mon Sep 17 00:00:00 2001
From: Jens Rehsack <rehsack@googlemail.com>
Date: Wed, 28 Jul 2010 11:19:07 +0200
Subject: [PATCH 1/2] adding cxx as possible c++ compiler and a possible improvement to find_perl_interpreter

---
 lib/ExtUtils/CBuilder/Base.pm |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/ExtUtils/CBuilder/Base.pm b/lib/ExtUtils/CBuilder/Base.pm
index f823df4..3edcff0 100644
--- a/lib/ExtUtils/CBuilder/Base.pm
+++ b/lib/ExtUtils/CBuilder/Base.pm
@@ -22,7 +22,7 @@ $VERSION = '0.27_04';
 
 my %cc2cxx = (
     # first line order is important to support wrappers like in pkgsrc
-    cc => [ 'c++', 'CC', 'aCC', ], # Sun Studio, HP ANSI C/C++ Compilers
+    cc => [ 'c++', 'CC', 'aCC', 'cxx', ], # Sun Studio, HP ANSI C/C++ Compilers
     gcc => [ 'g++' ], # GNU Compiler Collection
     xlc => [ 'xlC' ], # IBM C/C++ Set, xlc without thread-safety
     xlc_r => [ 'xlC_r' ], # IBM C/C++ Set, xlc with thread-safety
@@ -77,7 +77,7 @@ sub find_perl_interpreter {
   my $perl;
   File::Spec->file_name_is_absolute($perl = $^X)
     or -f ($perl = $Config::Config{perlpath})
-    or ($perl = $^X);
+    or ($perl = $^X); # XXX how about using IPC::Cmd::can_run here?
   return $perl;
 }
 
-- 
1.7.0.5

