Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-05-06

--- a/README
+++ b/README
@@ -157,7 +157,7 @@
       $sub->($thing, @import_args);
 
     And you need to switch between these implementations depending on if you
-    are targetting a specific package, or something in your call stack.
+    are targeting a specific package, or something in your call stack.
 
     Remembering all this, however, is excessively irritating. So I wrote a
     module so I didn't have to anymore. Loading Import::Into creates a
--- a/lib/Import/Into.pm
+++ b/lib/Import/Into.pm
@@ -214,7 +214,7 @@
   $sub->($thing, @import_args);
 
 And you need to switch between these implementations depending on if you are
-targetting a specific package, or something in your call stack.
+targeting a specific package, or something in your call stack.
 
 Remembering all this, however, is excessively irritating. So I wrote a module
 so I didn't have to anymore. Loading L<Import::Into> creates a global method
