--- Makefile.PL.dist	Mon Nov  6 14:28:53 2006
+++ Makefile.PL	Mon Nov  6 14:20:09 2006
@@ -22,19 +22,18 @@
 # check O/S to set sensible defaults
 
 my ($WIN32, $FLAVOUR, $PREFIX, $IMAGES, $MAKE);
+$MAKE=$Config{'make'};
 if ($^O eq 'MSWin32') {  # any others also?
     $WIN32   = 1;
     $FLAVOUR = 'Win32';
     $PREFIX  = 'C:/Program Files/Template Toolkit 2';
     $IMAGES  = '/tt2/images';
-    $MAKE    = 'nmake';
 }
 else {
     $WIN32   = 0;
     $FLAVOUR = 'Unix';
     $PREFIX  = '/usr/local/tt2';
     $IMAGES  = '/tt2/images';
-    $MAKE    = 'make';
 }
 
 
@@ -266,6 +265,9 @@
     $opts{ ABSTRACT } = 'comprehensive template processing system',
 }
 
+if ($ExtUtils::MakeMaker::VERSION ge '6.30_00') {
+    $opts{'LICENSE' } = 'perl';
+}
 
 WriteMakefile( %opts );
 
