--- List-MoreUtils-0.33/t/lib/Test.pm	2011-08-04 02:39:36.000000000 -0700
+++ fixed/t/lib/Test.pm	2012-10-25 07:52:19.000000000 -0700
@@ -698,6 +698,10 @@
     ok( arrayeq($part[1], [ 2, 5, 8, 11 ]) );
     ok( arrayeq($part[2], [ 3, 6, 9, 12 ]) );
 
+    $list[2] = 0;
+    is( $part[2][0], 3, 'Values are not aliases' );
+    @list = 1 .. 12;
+
     @part = part { 3 } @list;
     is( $part[0], undef );
     is( $part[1], undef );
