Description: fix a spelling mistake
Origin: vendor
Forwarded:
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-08-29

--- a/lib/DBD/Mock.pm
+++ b/lib/DBD/Mock.pm
@@ -846,7 +846,7 @@
 
 Basically this feature allows you to alias attributes to other attributes. So for instance, you can alias a commonly expected attribute like 'mysql_insertid' to something DBD::Mock already has like 'mock_last_insert_id'. While you can also just set 'mysql_insertid' yourself, this functionality allows it to take advantage of things like the autoincrementing of the 'mock_last_insert_id' attribute.
 
-Right now this feature is highly experimental, and has been added as a first attempt to automatically handle some of the DBD specific attributes which are commonly used/accessed in DBI programming. The functionality is off by default so as to not cause any issues with backwards compatability, but can easily be turned on and off like this:
+Right now this feature is highly experimental, and has been added as a first attempt to automatically handle some of the DBD specific attributes which are commonly used/accessed in DBI programming. The functionality is off by default so as to not cause any issues with backwards compatibility, but can easily be turned on and off like this:
 
   # turn it on
   $DBD::Mock::AttributeAliasing++;
