Description: fix spelling
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-10-22

--- a/Org.pm
+++ b/Org.pm
@@ -428,7 +428,7 @@
 Translation direction. Please note that dict.leo.org always translates
 either to or from german.
 
-The following languages are supported: english, polish, spanish, portugese
+The following languages are supported: english, polish, spanish, portuguese
 russian and chinese.
 
 You can  specify only the country  code, or append B<de2>  in order to
@@ -446,7 +446,7 @@
  en    english
  es    spanish
  ru    russian
- pt    portugese
+ pt    portuguese
  pl    polish
  ch    chinese
 
--- a/leo
+++ b/leo
@@ -346,7 +346,7 @@
 en    english
 es    spanish
 ru    russian
-pt    portugese
+pt    portuguese
 pl    polish
 ch    chinese
 
@@ -433,7 +433,7 @@
 Translation direction. Please note that dict.leo.org always translates
 either to or from german.
 
-The following languages are supported: english, polish, spanish, portugese
+The following languages are supported: english, polish, spanish, portuguese
 russian and chinese.
 
 You can  specify only the country  code, or append B<de2>  in order to
@@ -451,7 +451,7 @@
 en    english
 es    spanish
 ru    russian
-pt    portugese
+pt    portuguese
 pl    polish
 ch    chinese
 
