--- dbdimp.c.orig	2011-10-12 11:52:09.000000000 +0200
+++ dbdimp.c	2011-10-12 11:52:14.000000000 +0200
@@ -3455,8 +3455,9 @@ int dbd_st_execute(SV* sth, imp_sth_t* i
                                                   &imp_sth->has_been_bound
                                                  );
   }
-  else {
+  else
 #endif
+  {
     imp_sth->row_num= mysql_st_internal_execute(
                                                 sth,
                                                 *statement,
