Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-28

--- a/lib/ExtUtils/Config.pm
+++ b/lib/ExtUtils/Config.pm
@@ -90,7 +90,7 @@
 
 =head2 get($key)
 
-Get the value of C<$key>. If not overriden it will return the value in %Config.
+Get the value of C<$key>. If not overridden it will return the value in %Config.
 
 =head2 exists($key)
 
