--- myldr/Makefile.PL.orig	2011-04-26 08:40:32.000000000 +0200
+++ myldr/Makefile.PL	2011-04-26 08:44:42.000000000 +0200
@@ -203,10 +203,14 @@
 if ($dynperl and $^O eq 'MSWin32'
     and defined $Config{gccversion}     # gcc version 4.x or above was used
     and $Config{gccversion} =~ m{\A(\d+)}ms && $1 >= 4) {
-    # look for a libgcc_*.dll in the same directory
-    # as the perl executable itself
-    ($libgcc) = glob(File::Spec->catfile(File::Basename::dirname($^X),
-                                         "libgcc_*.$Config{so}"));
+    # look for a libgcc_*.dll 
+    # - in PATH
+    # - in the same directory as the perl executable itself
+    # - in the same directory as gcc
+    ($libgcc) = map { glob(File::Spec->catfile($_, "libgcc_*.$Config{so}")) }
+                    File::Spec->path(),
+                    File::Basename::dirname($^X),
+                    File::Basename::dirname($cc);
 }
 $libgcc ||= "";
 
