Description: spelling fix in the documentation
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-04-29

--- a/lib/Dancer/Logger/LogReport.pod
+++ b/lib/Dancer/Logger/LogReport.pod
@@ -57,7 +57,7 @@
 in messages depends on the dispatcher 'mode' (f.i. 'DEBUG').
 
 You also want to set the log level to C<debug>, because level filtering
-is controled per dispatcher (as well)
+is controlled per dispatcher (as well)
 
 =head1 SEE ALSO
 
