-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

diff -u -r libwww-perl-5.805.orig/lib/LWP/Protocol.pm libwww-perl-5.805/lib/LWP/Protocol.pm
- --- libwww-perl-5.805.orig/lib/LWP/Protocol.pm	2004-11-12 21:34:10.000000000 +0800
+++ libwww-perl-5.805/lib/LWP/Protocol.pm	2006-07-05 00:45:05.000000000 +0800
@@ -104,6 +104,7 @@
     if ($parse_head && $response->content_type eq 'text/html') {
 	require HTML::HeadParser;
 	$parser = HTML::HeadParser->new($response->{'_headers'});
+	$parser->utf8_mode(1);
     }
     my $content_size = 0;
 
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1.4.3 (GNU/Linux)

iD8DBQFEqptii9gubzC5S1wRAvHqAJ4zsxBTvoVFV+9MVX9cDK1rz0SgRgCeLbOM
LWBd8tpfdtF/yELWyPAsTQo=
=sfar
-----END PGP SIGNATURE-----
