Description: spelling fixes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2017-03-03

--- a/lib/Net/Prometheus/Counter.pm
+++ b/lib/Net/Prometheus/Counter.pm
@@ -41,7 +41,7 @@
 =head1 DESCRIPTION
 
 This class provides a counter metric - a value that monotonically increases,
-usually used to represent occurrances of some event that happens within the
+usually used to represent occurrences of some event that happens within the
 instrumented program. It is a subclass of L<Net::Prometheus::Metric>.
 
 =cut
--- a/lib/Net/Prometheus/Metric.pm
+++ b/lib/Net/Prometheus/Metric.pm
@@ -303,7 +303,7 @@
 
    @samples = $metric->samples
 
-An abstract method in this class, this method is intended to be overriden by
+An abstract method in this class, this method is intended to be overridden by
 subclasses.
 
 Called during the value collection process, this method should return a list
