diff -urN DBIx-Class-Cursor-Cached-1.001001.orig/META.yml DBIx-Class-Cursor-Cached-1.001001/META.yml
--- DBIx-Class-Cursor-Cached-1.001001.orig/META.yml	2011-03-30 18:23:17.000000000 +0200
+++ DBIx-Class-Cursor-Cached-1.001001/META.yml	2011-08-05 20:13:16.133600921 +0200
@@ -23,7 +23,7 @@
 requires:
   Carp::Clan: 6.0
   DBIx::Class: 0.08124
-  Digest::SHA1: 0
+  Digest::SHA: 0
   perl: 5.8.1
 resources:
   license: http://dev.perl.org/licenses/
diff -urN DBIx-Class-Cursor-Cached-1.001001.orig/Makefile.PL DBIx-Class-Cursor-Cached-1.001001/Makefile.PL
--- DBIx-Class-Cursor-Cached-1.001001.orig/Makefile.PL	2011-03-30 18:21:30.000000000 +0200
+++ DBIx-Class-Cursor-Cached-1.001001/Makefile.PL	2011-08-05 20:13:23.237601493 +0200
@@ -6,7 +6,7 @@
 all_from 'lib/DBIx/Class/Cursor/Cached.pm';
 
 requires 'DBIx::Class' => '0.08124';
-requires 'Digest::SHA1';
+requires 'Digest::SHA';
 requires 'Carp::Clan' => '6.0';
 
 build_requires 'DBD::SQLite' => '1.25';
diff -urN DBIx-Class-Cursor-Cached-1.001001.orig/lib/DBIx/Class/Cursor/Cached.pm DBIx-Class-Cursor-Cached-1.001001/lib/DBIx/Class/Cursor/Cached.pm
--- DBIx-Class-Cursor-Cached-1.001001.orig/lib/DBIx/Class/Cursor/Cached.pm	2011-03-30 18:21:51.000000000 +0200
+++ DBIx-Class-Cursor-Cached-1.001001/lib/DBIx/Class/Cursor/Cached.pm	2011-08-05 20:13:53.677601210 +0200
@@ -4,7 +4,7 @@
 use warnings;
 use 5.008001;
 use Storable ();
-use Digest::SHA1 ();
+use Digest::SHA ();
 use Carp::Clan qw/^DBIx::Class/;
 
 use vars qw($VERSION);
@@ -66,7 +66,7 @@
   }
   
   local $Storable::canonical = 1;
-  return Digest::SHA1::sha1_hex(Storable::nfreeze( [ $ref, $conn->{Name}, $conn->{Username} || '' ] ));
+  return Digest::SHA::sha1_hex(Storable::nfreeze( [ $ref, $conn->{Name}, $conn->{Username} || '' ] ));
 
 }
 
