diff -ru ../JavaScript-Minifier-1.05/Changes ./Changes
--- ../JavaScript-Minifier-1.05/Changes	Tue Mar 18 10:27:45 2008
+++ ./Changes	Wed Jun  5 11:20:49 2013
@@ -1,5 +1,11 @@
 Revision history for Perl extension JavaScript::Minifier.
 
+1.06  Sat Sep 22, 10:47:41 PDT 2012
+	reduce perl requirement to 5.6.1 (tested) and possibly 
+	older (untested)
+        for platforms running Apache 1.x and mod_perl 1
+	miker@cpan.org
+
 0.01  Sat May 19 15:28:19 2007
 	- original version; created by h2xs 1.23 with options
 		-X -n JavaScript::Minifier
diff -ru ../JavaScript-Minifier-1.05/MANIFEST ./MANIFEST
--- ../JavaScript-Minifier-1.05/MANIFEST	Tue Mar 18 10:37:45 2008
+++ ./MANIFEST	Wed Jun  5 11:07:35 2013
@@ -1,6 +1,7 @@
 Changes
 Makefile.PL
 MANIFEST
+MANIFEST.skip
 README
 lib/JavaScript/Minifier.pm
 t/JavaScript-Minifier.t
Only in ./: MANIFEST.skip
diff -ru ../JavaScript-Minifier-1.05/Makefile.PL ./Makefile.PL
--- ../JavaScript-Minifier-1.05/Makefile.PL	Tue Mar 18 10:27:45 2008
+++ ./Makefile.PL	Wed Jun  5 11:12:13 2013
@@ -1,10 +1,9 @@
-use 5.8.0;
 use ExtUtils::MakeMaker;
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 WriteMakefile(
     NAME              => 'JavaScript::Minifier',
-    VERSION_FROM      => 'lib/JavaScript/Minifier.pm', # finds $VERSION
+    VERSION           => 'lib/JavaScript/Minifier.pm', # finds $VERSION
     PREREQ_PM         => {}, # e.g., Module::Name => 1.1
     ($] >= 5.005 ?     ## Add these new keywords supported since 5.005
       (ABSTRACT_FROM  => 'lib/JavaScript/Minifier.pm', # retrieve abstract from module
