--- Filter-Crypto-1.15/inc/Module/Install/PRIVATE/Filter/Crypto.pm	2005-09-01 14:16:50.000000000 +0200
+++ Filter-Crypto-1.15_cosimo/inc/Module/Install/PRIVATE/Filter/Crypto.pm	2005-09-01 20:08:36.000000000 +0200
@@ -564,6 +564,11 @@
             $lib_file = $file;
             $lib_name = 'crypto';
         }
+        else
+        {
+            # No library found!
+            return;
+        }
     }
 
     return $lib_file, $lib_name;
