=== Build.PL
==================================================================
--- Build.PL    (revision 513)
+++ Build.PL    (local)
@@ -20,6 +20,8 @@
     dynamic_config  => 1,
 );

+mkdir $t->config_dir(); # must already exist for have_c_compiler()
+
 if ( $perl_only or not $t->have_c_compiler() ) {
     $build_arguments{module_name} = 'version::vpp';
     if ( $] > 5.008001 ) {

