--- timeout.pl~	2020-09-22 12:13:19.131749592 -0400
+++ timeout.pl	2020-09-22 13:23:03.308904610 -0400
@@ -81,6 +81,10 @@ $telnet = new Net::Telnet
 $telnet->input_log($log_file) if defined $log_file;
 $telnet->open;
 
+$_ = sub { $telnet->close; exit }
+    foreach @SIG{qw(INT PWR QUIT TERM)};
+$SIG{__DIE__} = sub { $telnet->close; die @_ };
+
 # Log in.
 print "Waiting for login prompt\n";
 $telnet->waitfor('/\rlogin: $/');
