diff -ur Text-Restructured-0.003024/lib/Text/Restructured/Writer/html.wrt hacked-Restructured/lib/Text/Restructured/Writer/html.wrt
--- Text-Restructured-0.003024/lib/Text/Restructured/Writer/html.wrt	2007-02-21 14:23:10.000000000 -0500
+++ hacked-Restructured/lib/Text/Restructured/Writer/html.wrt	2007-02-26 13:16:52.000000000 -0500
@@ -1257,6 +1257,9 @@
 	# Handle the body.
         my $battr = $body_attr ? " $body_attr" : '';
 	my $body = [["<body$battr>\n"], ["</body>\n"]];
+	if (defined $writer->{opt}{D}{text_only}){
+	    $body = [[], []];
+	}
 	unshift @{$body->[1]}, $FOOTER if defined $FOOTER;
 	push @{$doc->[0]}, $body;
 	push @{$body->[0]}, $HEADER if defined $HEADER;
@@ -1267,8 +1270,8 @@
 	push (@{$body->[0]},
 	      qq(<div class="document") .
 	      ($dom->{attr}{ids} ? qq( id="$dom->{attr}{ids}[0]") : "")
-	      . qq(>\n));
-	unshift (@{$body->[1]}, qq(</div>\n));
+	      . qq(>\n)) unless defined $writer->{opt}{D}{text_only};
+	unshift (@{$body->[1]}, qq(</div>\n)) unless defined $writer->{opt}{D}{text_only};
 	push (@{$body->[0]}, qq(<h1 class="title">$title</h1>\n))
 	    if $title ne '' && ! $writer->{opt}{D}{keep_title_section};
 	my $id = $dom->num_contents() > 1 &&
@@ -1290,7 +1293,14 @@
 	    push (@{$body->[0]}, $content->{val});
 	}
 
-	my @list = Flatten($doc);
+	my @list;
+	if (defined $writer->{opt}{D}{text_only}){
+	    @list = Flatten($body);
+	}
+	else {
+	    @list = Flatten($doc);
+	}
+
 	return join '',@list;
 
 	# This subroutine takes an array of items which may
