diff --git a/lib/SQL/Translator/Producer/PostgreSQL.pm b/lib/SQL/Translator/Producer/PostgreSQL.pm
index 4b883c7..f08deda 100644
--- a/lib/SQL/Translator/Producer/PostgreSQL.pm
+++ b/lib/SQL/Translator/Producer/PostgreSQL.pm
@@ -34,6 +34,22 @@ producer.
 
 Now handles PostGIS Geometry and Geography data types on table definitions.
 Does not yet support PostGIS Views.
+
+=head1 producer_args
+
+=over
+
+=item quote_field_names
+
+Sets the quote character to use around column names in generated DDL.
+Defaults to none.
+
+=item quote_table_names
+
+Sets the quote character used around table, sequence and trigger names in
+generated DDL.  Defaults to none.
+
+=back
 	
 =cut
 
