=== modified file 'lib/Devel/MAT/Dumper.xs'
--- lib/Devel/MAT/Dumper.xs	2014-01-22 21:25:15 +0000
+++ lib/Devel/MAT/Dumper.xs	2015-04-16 18:39:49 +0000
@@ -830,6 +830,8 @@
   for(sp = PL_stack_base; sp <= PL_stack_sp; sp++)
     write_svptr(fh, *sp);
 
+  bool seen_defstash = false;
+
   // Heap
   SV *arena;
   for(arena = PL_sv_arenaroot; arena; arena = (SV *)SvANY(arena)) {
@@ -844,11 +846,15 @@
       }
 
       write_sv(fh, sv);
+
+      if(sv == (const SV *)PL_defstash)
+        seen_defstash = true;
     }
   }
 
   // and a few other things that don't actually appear in the arena
-  write_sv(fh, (const SV *)PL_defstash); 
+  if(!seen_defstash)
+    write_sv(fh, (const SV *)PL_defstash);
 
   write_u8(fh, 0);
 

