Description: add back whatis entry
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=89724
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=89724
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-10-23

--- a/lib/Return/Value.pm
+++ b/lib/Return/Value.pm
@@ -108,7 +108,7 @@
 
 =head1 NAME
 
-Return::Value
+Return::Value - polymorphic return values (deprecated)
 
 =head1 VERSION
 
