--- Client.pm.orig	2011-11-23 10:36:32.356135982 -0600
+++ Client.pm	2011-11-23 10:41:46.666134080 -0600
@@ -565,9 +565,9 @@
 =cut
 
 sub get_security_levels {
-    my ($self) = @_;
-    $self->{cache}{seclevels} ||= {map {$_->{name} => $_} @{$self->getSecurityLevels()}};
-    return $self->{cache}{seclevels};
+    my ($self, $project_key) = @_;
+    $self->{cache}{seclevels}{$project_key} ||= {map {$_->{name} => $_} @{$self->getSecurityLevels($project_key)}};
+    return $self->{cache}{seclevels}{$project_key};
 }
 
 =item B<get_custom_fields>
