diff -urN Atompub-0.3.4.orig//lib/Atompub/Client.pm Atompub-0.3.4//lib/Atompub/Client.pm
--- Atompub-0.3.4.orig//lib/Atompub/Client.pm	2010-05-17 15:10:22.000000000 +0200
+++ Atompub-0.3.4//lib/Atompub/Client.pm	2011-05-13 22:05:16.880949961 +0200
@@ -7,7 +7,7 @@
 use Atompub::DateTime qw(datetime);
 use Atompub::MediaType qw(media_type);
 use Atompub::Util qw(is_acceptable_media_type is_allowed_category);
-use Digest::SHA1 qw(sha1);
+use Digest::SHA qw(sha1);
 use Encode qw(encode_utf8);
 use File::Slurp;
 use HTTP::Status;
diff -urN Atompub-0.3.4.orig//lib/Atompub/Server.pm Atompub-0.3.4//lib/Atompub/Server.pm
--- Atompub-0.3.4.orig//lib/Atompub/Server.pm	2010-10-12 11:48:54.000000000 +0200
+++ Atompub-0.3.4//lib/Atompub/Server.pm	2011-05-13 22:05:16.880949961 +0200
@@ -4,7 +4,7 @@
 use warnings;
 
 use Atompub;
-use Digest::SHA1 qw(sha1);
+use Digest::SHA qw(sha1);
 use MIME::Base64 qw(encode_base64 decode_base64);
 use HTTP::Status;
 use XML::Atom;
diff -urN Atompub-0.3.4.orig//Makefile.PL Atompub-0.3.4//Makefile.PL
--- Atompub-0.3.4.orig//Makefile.PL	2010-05-17 15:05:54.000000000 +0200
+++ Atompub-0.3.4//Makefile.PL	2011-05-13 22:05:16.880949961 +0200
@@ -11,7 +11,7 @@
 requires('DateTime');
 requires('DateTime::Format::W3CDTF');
 requires('DateTime::TimeZone');
-requires('Digest::SHA1');
+requires('Digest::SHA');
 requires('File::Slurp');
 requires('HTTP::Date');
 requires('HTTP::Headers');
