diff -NurbB a/Makefile-Parser-0.211/lib/Makefile/Parser.pm b/Makefile-Parser-0.211/lib/Makefile/Parser.pm
--- a/lib/Makefile/Parser.pm	2008-03-16 10:04:59.000000000 +0000
+++ b/lib/Makefile/Parser.pm	2011-08-17 12:14:49.568281708 +0100
@@ -69,7 +69,6 @@
     my ($var, $value, $tar_name, $tar, $colon_type, $depends, $cmd);
     my @cmds;
     my %tars;
-    %$rvars = ();
     my $first_tar = 1;
     while (<$in>) {
         next if /^\s*#/;
