Index: lib/Test/Builder.pm
===================================================================
RCS file: /usr/local/cvsroot/Test-Simple/lib/Test/Builder.pm,v
retrieving revision 1.22
diff -u -r1.22 Builder.pm
--- lib/Test/Builder.pm	25 Apr 2002 04:28:20 -0000	1.22
+++ lib/Test/Builder.pm	3 May 2002 06:06:59 -0000
@@ -131,6 +131,11 @@
 
     return unless $cmd;
 
+    if( $Have_Plan ) {
+        die sprintf "You tried to plan twice!  Second plan at %s line %d\n",
+          ($self->caller)[1,2];
+    }
+
     if( $cmd eq 'no_plan' ) {
         $self->no_plan;
     }
