--- Image.pm.bk	2008-11-14 19:58:45.000000000 -0800
+++ Image.pm	2008-11-14 20:00:02.000000000 -0800
@@ -120,6 +120,14 @@
         return $fh;
     }
 
+    if (    $max_width  && ($ow <= $max_width)
+         && $max_height && ($oh <= $max_height)
+       ) {
+        #warn "Image does not need to be resized";
+        seek( $fh, 0, 0 );
+        return $fh;
+    }
+
     if ( $max_width && $nw > $max_width ) {
         $nw = $max_width;
         $nh = $oh * ( $max_width / $ow );
