--- a/lib/String/MkPasswd.pm
+++ b/lib/String/MkPasswd.pm
@@ -279,7 +279,7 @@
 
 By default, C<mkpasswd()> will return C<undef> if it cannot generate a
 password.  Some people are inclined to exception handling, so
-B<String::MkPasswd> does its best to accomodate them.  If the variable
+B<String::MkPasswd> does its best to accommodate them.  If the variable
 C<$String::MkPasswd::FATAL> is set to a true value, C<mkpasswd()> will
 L<Carp::croak()> with an error instead of returning C<undef>.
 
