--- /root/.cpan/build/Mail-GnuPG-0.19-vA0Gug/blib/lib/Mail/GnuPG.pm	2014-07-16 16:24:24.619319414 +0200
+++ /usr/lib/perl5/site_perl/5.16.2/Mail/GnuPG.pm	2014-07-16 17:01:06.583403823 +0200
@@ -555,7 +555,7 @@
   my $handles = GnuPG::Handles->new( stdin      => $input,
 				     stdout     => $output,
 				     stderr     => $error,
-				     passphrase => $passphrase_fh,
+				     $self->{use_agent} ? () : (passphrase => $passphrase_fh),
 				     status     => $status_fh,
 				   );
   my $pid = $gnupg->detach_sign( handles => $handles );
@@ -581,9 +581,9 @@
 #  print STDERR $plaintext;
 #  print "<----\n";
   my $read = _communicate([$output, $error, $status_fh],
-                        [$input, $passphrase_fh],
+                        [$input, $self->{use_agent} ? () : ($passphrase_fh)],
                         { $input => $plaintext,
-                          $passphrase_fh => $self->{passphrase}}
+                          $self->{use_agent} ? () : ($passphrase_fh => $self->{passphrase})}
              );
 
   my @signature  = split(/^/m, $read->{$output});
@@ -854,7 +854,7 @@
   my $handles = GnuPG::Handles->new( stdin      => $input,
 				     stdout     => $output,
 				     stderr     => $error,
-				     passphrase => $passphrase_fh,
+				     $self->{use_agent} ? () : (passphrase => $passphrase_fh),
 				     status     => $status_fh,
 				   );
 
@@ -885,9 +885,9 @@
 
   die "NO PASSPHRASE" unless defined $passphrase_fh;
   my $read = _communicate([$output, $error, $status_fh],
-                        [$input, $passphrase_fh],
+                        [$input, $self->{use_agent} ? () : ($passphrase_fh)],
                         { $input => $plaintext,
-                          $passphrase_fh => $self->{passphrase}}
+                          $self->{use_agent} ? () : ($passphrase_fh => $self->{passphrase})}
              );
 
   my @plaintext    = split(/^/m, $read->{$output});
