--- Net-DNS-0.56/t/11-inet6.t	2006-02-21 11:31:46.000000000 +0000
+++ Net-DNS-0.56/t/11-inet6.t	2006-02-21 11:32:20.000000000 +0000
@@ -21,7 +21,7 @@
 
 
 SKIP: { skip "Socket6 and or IO::Socket::INET6 not loaded\n".
-	    "You will need to install these modules for IPv6 transport support", 9 unless $has_inet6;
+	    "You will need to install these modules for IPv6 transport support", 10 unless $has_inet6;
 
 	diag "";
 	diag "The libraries needed for IPv6 support have been found\n";
