Description: fix spelling mistakes: defaut -> default
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2018-04-15

--- a/lib/Compress/Raw/Lzma.pm
+++ b/lib/Compress/Raw/Lzma.pm
@@ -1099,7 +1099,7 @@
 the C<code> and C<flush> methods.
 
 Defaults to 0.
-(Note in versions of this module prior to 2.072 the defaut value was
+(Note in versions of this module prior to 2.072 the default value was
 incorrectly documented as 1).
 
 =item B<< BufSize => $number >>
@@ -1143,7 +1143,7 @@
 the C<code> and C<flush> methods.
 
 Defaults to 0.
-(Note in versions of this module prior to 2.072 the defaut value was
+(Note in versions of this module prior to 2.072 the default value was
 incorrectly documented as 1).
 
 =item B<< BufSize => $number >>
@@ -1200,7 +1200,7 @@
 the C<code> and C<flush> methods.
 
 Defaults to 0.
-(Note in versions of this module prior to 2.072 the defaut value was
+(Note in versions of this module prior to 2.072 the default value was
 incorrectly documented as 1).
 
 =item B<< BufSize => $number >>
@@ -1249,7 +1249,7 @@
 the C<code> and C<flush> methods.
 
 Defaults to 0.
-(Note in versions of this module prior to 2.072 the defaut value was
+(Note in versions of this module prior to 2.072 the default value was
 incorrectly documented as 1).
 
 =item B<< BufSize => $number >>
