--- Test-Strict-0.07/t/02fail.t	2005-03-21 00:00:05.000000000 +0100
+++ Test-Strict-0.08/t/02fail.t	2006-08-31 11:43:16.000000000 +0200
@@ -8,10 +8,18 @@
 ##
 
 use strict;
+BEGIN {
+  if ($^O =~ /win32/i) {
+    require Test::More;
+    Test::More->import(
+      skip_all => "Windows does not allow two processes to access the same file."
+    );
+  }
+}
+
 use Test::More tests => 8;
 use File::Temp qw( tempdir tempfile );
 
-
 my $perl  = $^X || 'perl';
 my $inc = join(' -I ', @INC) || '';
 $inc = "-I $inc" if $inc;
