--- cover.orig	2011-08-07 17:31:31.000000000 +0200
+++ cover	2011-08-07 17:36:51.000000000 +0200
@@ -254,7 +254,12 @@
         File::Find::find({ wanted => $gp, no_chdir => 1 }, ".");
         if (@gc)
         {
-            my $c = "gcov2perl -db $dbname @gc";
+            # Find the right gcov2perl based on this current script.
+            require Cwd;
+            my $cover = Cwd::abs_path($0);
+            my ($vol, $dir, $cover) = File::Spec->splitpath($cover);
+            my $gcov2perl = File::Spec->catpath($vol, $dir, 'gcov2perl');
+            my $c = "$gcov2perl -db $dbname @gc";
             print STDERR "cover: running $c\n";
             system $c;
         }
