Description: Fix spelling error
Author: Fabrizio Regalli <fabreg@fabreg.it>
Last-Update: 2011-09-07
Forwarded: no

--- a/lib/Kephra.pm
+++ b/lib/Kephra.pm
@@ -193,7 +193,7 @@
 
 =head1 NAME
 
-Kephra - crossplatform, GUI-Texteditor along Perl alike Paradigms 
+Kephra - crossplatform GUI Texteditor along Perl alike Paradigms
 
 =head1 SYNOPSIS
 
@@ -206,7 +206,7 @@
 
 =head2 Philosophy
 
-=over 4
+=over 3
 
 =item Main Goals
 
@@ -257,6 +257,8 @@
 
     That is the plan, but we are currently not nearly that far.
 
+=over 3
+
 =item Name
 
 Especially from the last item derives the name, which is old egyptian and means
@@ -271,7 +273,7 @@
 
 I believe that Kephra's agenda is very similar to Perl's. Its common wisdom
 that freedom means not only happiness but also life works most effective in
-freedom. So there should not only be more than one way to write a programm,
+freedom. So there should not only be more than one way to write a program,
 but also more than one way use an editor. You could:
 
 =over 4
--- a/lib/Kephra/App/StatusBar.pm
+++ b/lib/Kephra/App/StatusBar.pm
@@ -275,7 +275,7 @@
 
 =head1 NAME
 
-Kephra::App::StatusBar - 
+Kephra::App::StatusBar - Kephra::App::StatusBar
 
 =head1 DESCRIPTION
 
