<!DOCTYPE html>

<html lang="en">
  <head>
    <title>Bug #116683 for Bio-SamTools: Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <!-- The X-UA-Compatible <meta> tag above must be very early in <head> -->

    <script>
if (window.top !== window.self) {
    document.write = "";

    window.top.location = window.self.location;

    setTimeout(function(){
        document.body.innerHTML = "";
    }, 1);

    window.self.onload = function(){
        document.body.innerHTML = "";
    };
}
</script>



<link rel="shortcut icon" href="/NoAuth/images/favicon.png" type="image/png" />
<link rel="stylesheet" href="/NoAuth/css/aileron-squished-00f758259152f8b6670ee485f462e6b8.css" type="text/css" media="all" />
<link rel="stylesheet" href="/NoAuth/css/print.css" type="text/css" media="print" />




<script type="text/javascript" src="/NoAuth/js/squished-42167f2409ce368da2d31038e5cb2a57.js"></script>

<script type="text/javascript"><!--
	jQuery( loadTitleBoxStates );

--></script>


<!--[if lt IE 8]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie.css" type="text/css" media="all" />
<![endif]-->
<!--[if lt IE 7]>
<link rel="stylesheet" href="/NoAuth/css/aileron/msie6.css" type="text/css" media="all" />
<![endif]-->
<script type="text/javascript">
jQuery(document).ready(function(){
    jQuery.fn.supersubs.defaults = {
        maxWidth: 30,
        extraWidth: 2
    };

    jQuery("#app-nav.toplevel").addClass('sf-menu sf-js-enabled sf-shadow').supersubs().superfish({ speed: 'fast' });
    jQuery("#page-menu.toplevel").addClass('sf-menu sf-js-enabled').supersubs().superfish({ dropShadows: false, speed: 'fast' }).supposition();
});
</script>


<!-- Site CSS from theme editor -->
<style type="text/css" media="all" id="sitecss">
</style>

<meta name="google-site-verification" content="kD3-uEJK1AkTqaC-rckcSt_HPkAj674DPPuEN6i-y0g" />

</head>
  <body class="aileron sidebyside" id="comp-Public-Bug-Display">

<div id="logo">
<a href="https://rt.cpan.org/"><img
    src="/NoAuth/images/cpan.png"
    alt="CPAN"
    width="181"
    height="38" /></a>
    <span class="rtname">RT for rt.cpan.org</span>
</div>


<div id="quickbar">
  <div id="quick-personal">
    <span class="hide"><a href="#skipnav">Skip Menu</a> | </span>
</div>

</div>
<div id="header"><h1>Bug #116683 for Bio-SamTools: Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</h1></div>


<div id="main-navigation"><ul id="app-nav" class="toplevel">
 <li id="li-search_dist"><a id="search_dist" class="menu-item " href="/Public/">Search Distributions</a></li>
 <li id="li-browse_dist"><a id="browse_dist" class="menu-item " href="/Public/Dist/Browse.html">Browse Distributions</a></li>
 <li id="li-about" class="has-children"><a id="about" class="menu-item " href="/NoAuth/RT/CPAN/About.html">About rt.cpan.org</a>
 <ul>
  <li id="li-about-usage"><a id="about-usage" class="menu-item " href="/NoAuth/RT/CPAN/Usage.html">How do Iâ€¦?</a></li>
  <li id="li-about-about"><a id="about-about" class="menu-item " href="/NoAuth/RT/CPAN/About.html">About</a></li>
 </ul>
 </li>
 <li id="li-preferences"><a id="preferences" class="menu-item ">Welcome <span class="current-user">anonymous guest</span>.</a></li>
 <li id="li-login"><a id="login" class="menu-item " href="/NoAuth/Logout.html">Login as another user</a></li>
</ul></div>
<div id="page-navigation"><ul id="page-menu" class="toplevel">
 <li id="li-page-active_bugs"><a id="page-active_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Active;Name=Bio-SamTools">Active bugs</a></li>
 <li id="li-page-resolved_bugs"><a id="page-resolved_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Resolved;Name=Bio-SamTools">Resolved bugs</a></li>
 <li id="li-page-rejected_bugs"><a id="page-rejected_bugs" class="menu-item " href="/Public/Dist/Display.html?Status=Rejected;Name=Bio-SamTools">Rejected bugs</a></li>
 <li id="li-page-report"><a id="page-report" class="menu-item " href="/Public/Bug/Report.html?Queue=Bio-SamTools">Report a new bug</a></li>
</ul></div>
<div id="topactions"><form action="/Public/Search/Simple.html" id="simple-search">
  <input size="12" name="q" accesskey="0" class="field" value="" placeholder="Search..." />
</form>

<form action="/Public/Bug/Report.html" name="CreateTicketInQueue" id="CreateTicketInQueue" >
<input type="submit" class="button" value="New ticket in" />&nbsp;<label accesskey="9">
  <input name="Queue" size="25" placeholder="Module or distribution" value="" class="select-queue" data-autocomplete="Queues" data-autocomplete-autosubmit=1 data-autocomplete-params="right=CreateTicket" />

</label>

</form>

</div>
<div id="body">
<div id="announcement" style="padding: 1em; margin: 1em; border: solid black 1px; background: #eee">
<a href="https://log.perl.org/2020/12/rtcpanorg-sunset.html">rt.cpan.org will be shut down on March 1st, 2021.</a>
</div>

<a name="skipnav" id="skipnav" accesskey="8"></a>





<p>This queue is for tickets about the <a href="https://metacpan.org/release/Bio-SamTools">Bio-SamTools CPAN distribution</a>.</p>





<div class="">
  <div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DUmVwb3J0IGluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Report information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------UmVwb3J0IGluZm9ybWF0aW9u---0">


<table width="100%" class="ticket-summary">
<tr>
  <td valign="top" class="boxcontainer">
    <div class="ticket-info-basics">
  <div class="titlebox ticket-info-basics " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dbasics\x2D\x2D\x2D\x2DVGhlIEJhc2ljcw\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">The Basics</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-basics----VGhlIEJhc2ljcw__---0">


        <table>
  <tr class="id">
    <td class="label">Id:</td>
    <td class="value">116683</td>
  </tr>
  <tr class="status">
    <td class="label">Status:</td>
    <td class="value">new</td>
  </tr>
  <tr class="priority">
    <td class="label">Priority:</td>
    <td class="value">0/
</td>
  </tr>
  <tr class="queue">
    <td class="label">Queue:</td>
    <td class="value"><a href="/Dist/Display.html?Queue=Bio-SamTools">Bio-SamTools</a>
</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>


    <div class="ticket-info-people">
  <div class="titlebox ticket-info-people " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dpeople\x2D\x2D\x2D\x2DUGVvcGxl\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">People</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-people----UGVvcGxl---0">


        <table>
  <tr>
    <td class="label">Owner:</td>
    <td class="value">Nobody in particular
    
    </td>
  </tr>
  <tr>
    <td class="labeltop">Requestors:</td>
    <td class="value">
kevin.buckley.ecs.vuw.ac.nz [...] gmail.com

<br />

</td>
  </tr>
  <tr>
    <td class="labeltop">Cc:</td>
    <td class="value">

</td>
  </tr>
  <tr>
    <td class="labeltop">AdminCc:</td>
    <td class="value">

</td>
  </tr>
</table>

        <hr class="clear" />
  </div>
</div>




</div>



  </td>
  <td valign="top" class="boxcontainer">
    
    <div class="ticket-info-cfs">
  <div class="titlebox ticket-info-cfs " id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2Dticket\x2Dinfo\x2Dcfs\x2D\x2D\x2D\x2DQnVnIEluZm9ybWF0aW9u\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">Bug Information</span>
    <span class="right-empty">		    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html--ticket-info-cfs----QnVnIEluZm9ybWF0aW9u---0">

 
        <table>
  <tr id="CF-1-ShowRow">
    <td class="label">Severity:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-228092-ShowRow">
    <td class="label">Broken in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
  <tr id="CF-228093-ShowRow">
    <td class="label">Fixed in:</td>
    <td class="value">
<i>(no value)</i>
    </td>
  </tr>
</table>


        <hr class="clear" />
  </div>
</div>




</div>


    




  </td>
</tr>
</table>

    <hr class="clear" />
  </div>
</div>




</div>


<br />


<div class="history">
<div class="titlebox" id="">
  <div class="titlebox-title">
    <span class="widget"><a href="#" onclick="return rollup('TitleBox\x2D\x2D\x5FPublic\x5FBug\x5FDisplay\x5Fhtml\x2D\x2D\x2D\x2D\x2D\x2DSGlzdG9yeQ\x5F\x5F\x2D\x2D\x2D0');" title="Toggle visibility"></a></span>
    <span class="left">History</span>
    <span class="right">		<a href="#" data-direction="open" onclick="return toggle_all_folds(this, 'Show\x20all\x20quoted\x20text', 'Hide\x20all\x20quoted\x20text');">Show all quoted text</a> &mdash; <a href="/Public/Bug/Display.html?ShowHeaders=1;id=116683">Show full headers</a>    </span>
  </div>
  <div class="titlebox-content " id="TitleBox--_Public_Bug_Display_html------SGlzdG9yeQ__---0">


<div id="ticket-history">
<div class="ticket-transaction message Create odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1655326" href="/Public/Bug/Display.html?id=116683#txn-1655326">#</a>      
    </span>
    <span class="date">Wed&nbsp;Aug&nbsp;03&nbsp;01:40:10&nbsp;2016</span>
    <span class="description">kevin.buckley.ecs.vuw.ac.nz [...] gmail.com -  Ticket created</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Wed, 3 Aug 2016 17:39:59 +1200</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-SamTools [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Kevin Buckley &lt;kevin.buckley.ecs.vuw.ac.nz [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1655326/888380/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/888380">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 2.8k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Came to add Bio-SamTools-1.43 into a BioPerl installtion

cpan[1]&gt; install LDS/Bio-SamTools-1.43.tar.gz

I get told that

<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; This module requires samtools 0.1.10 or higher &#40;samtools.sourceforge.net&#41;.
&gt; Please enter the location of the bam.h and compiled libbam.a files:
</div>
so I supply the path to the directory in which I built SAMtools 1.3

path/to/my/samtools-1.3/

The Bio-SamTools-1.43 build then fails telling me that

In file included from c_bin/bam2bedgraph.c:2:
/path/to/my/samtools-1.3/sam.h:28:24: \
 error: htslib/sam.h: No such file or directory

and that&#39;s right, because

samtools-1.3/

has an

htslib-1.3/

subdirectory and not a plain, unversioned

htslib/

subdirectory .

Furthermore, inside the htslib-1.3 subdirectory there is no file

sam.h

but there is a file named

sam.h

in the top-level samtools-1.3 directory.

I therefore create a link

htslib -&gt; htslib-1.3/

and within htslib-1.3, another link

sam.h -&gt; ../sam.h

and the build gets further, before failing with this one

/path/to/my/samtools-1.3/bam.h:48:25:
error: htslib/bgzf.h: No such file or directory

and that&#39;s where I run out of ideas, because that header &#40;* see below&#41;
doesn&#39;t exist anywhere in samtools-1.3, let alone where Bio-SamTools
seems to think it should be.


I also tried looking in SAMtools 0.1.19 &#40;after all there are a multitude of
SAM/BAM/THANKYOU/MAM/tools out there in BioInfo-land&#41; but that
doesn&#39;t have an

htslib

subdirectory of any kind.


* But read on

I become aware that although there is no header file, bgzf.h, in samtools 1.3
there is a bgzf.c inside the

htslib-1.3/

subdirectory and I see that that #includes

htslib/bgzf.h


at which point I realise that  the

htslib-1.3/

subdirectory has it&#39;s own &#40;unversioned&#41;

htslib

subdirectory &#40;or subsubdirectory of samtools 1.3 if you prefer&#41;

and inside there I find all of these fine fellows

bgzf.h      hts.h            klist.h     regisubdirectory &#40;or
subsubdirectory dx.h             vcf_sweep.h
cram.h      kbitset.h        knetfile.h  sam.h                vcfutils.h
faidx.h     kfunc.h          kseq.h      synced_bcf_reader.h
hfile.h     khash.h          ksort.h     tbx.h
hts_defs.h  khash_str2int.h  kstring.h   vcf.h

so now I am assuming that subdirectory. or subsubdirectory, depending where
you start from is what  Bio-SamTools is probably looking for, so I undo all my
previous attempts to trick Bio-SamTools into finding what it wants to find by
creating a link from the top-level

samtools-1.3/

directory, where you&#39;ll recall, libbam.a is, as

htslib -&gt; htslib-1.3/htslib/


And, after all that, Bio-SamTools tells me

lib/Bio/DB/Sam.xs:29:19: error: khash.h: No such file or directory

but, as we saw above

khash.h:

is there for all to see.

I know I must have missed something obvious: but what ?


Any clues welcome.

Kevin M. Buckley

eScience Consultant
School of Engineering and Computer Science
Victoria University of Wellington
New Zealand
</div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1655557" href="/Public/Bug/Display.html?id=116683#txn-1655557">#</a>      
    </span>
    <span class="date">Wed&nbsp;Aug&nbsp;03&nbsp;19:47:41&nbsp;2016</span>
    <span class="description">kevin.buckley.ecs.vuw.ac.nz [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> Re: [rt.cpan.org #116683] Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Thu, 4 Aug 2016 11:47:27 +1200</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-SamTools [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Kevin Buckley &lt;kevin.buckley.ecs.vuw.ac.nz [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1655557/888514/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/888514">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 1k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">Got a bit further

If I link three flles from within the

htslib-1.3/htslib/

directory up into the top build directpory for samtools-1.3

$ ln -s htslib-1.3/htslib/khash.h .
$ ln -s htslib-1.3/htslib/faidx.h .
$ ln -s htslib-1.3/htslib/hts_defs.h .

then I get to defeat the fact that the way the #includes are defined
differently within

lib/Bio/DB/Sam.xs

#include &#34;bam.h&#34;
#include &#34;khash.h&#34;
#include &#34;faidx.h&#34;

and in the samtools-1.3/sam.h

#include &#34;htslib/sam.h&#34;

and i ovecome the fact that the Bio-SamTools

Build.PL

doesn&#39;t  generate a correct set of -I/path/to/include/dirs.


However, I now see that there&#39;s probably some even greater mis-match
between what Bio-SamTools-1.43 and samtools-1.3 think about how the
internal structures are defined:

lib/Bio/DB/Sam.xs: In function &#39;XS_Bio__DB__Bam__Alignment_l_aux&#39;:
lib/Bio/DB/Sam.xs:575: error: &#39;bam1_t&#39; has no member named &#39;l_aux&#39;
lib/Bio/DB/Sam.xs:576: error: &#39;bam1_t&#39; has no member named &#39;l_aux&#39;

Has anyone out there compiled Bio-SamTools-1.43 against a SMAtools
version and, if so, which version ?
</div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond odd">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1655586" href="/Public/Bug/Display.html?id=116683#txn-1655586">#</a>      
    </span>
    <span class="date">Wed&nbsp;Aug&nbsp;03&nbsp;22:40:16&nbsp;2016</span>
    <span class="description">kevin.buckley.ecs.vuw.ac.nz [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> [rt.cpan.org #116683] Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Thu, 4 Aug 2016 14:40:06 +1200</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-SamTools [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Kevin Buckley &lt;kevin.buckley.ecs.vuw.ac.nz [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1655586/888529/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/888529">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 1.1k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">It appears I have posted this reply with out the RT bit in the subect line
and started another issue - apologies.

To save anyone merging the other strand, here you go again


Bit more info gleaned about this

In the

bam.h

file that&#39;s in the top build directory for samtools-1.3, I read

---8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;----
  @discussion Notes:

   1. qname is zero tailing and core.l_qname includes the tailing &#39;\0&#39;.
   2. l_qseq is calculated from the total length of an alignment block
      on reading or from CIGAR.
   3. cigar data is encoded 4 bytes per CIGAR operation.
   4. seq is nybble-encoded according to bam_nt16_table.
 */
// typedef struct { ... } bam1_t;
// NOTE htslib version doesn&#39;t have l_aux; use bam_get_l_aux&#40;b&#41; instead
#ifndef SAMTOOLS_HTSLIB_SUPPRESS_HACKS
// NOTE htslib also renames data_len to l_data; this macro may help or hinder

---8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;---------8&lt;----

so the missing member really isn&#39;t there by design.

So which version of SAMtools should Bio-SamTools-1.43 be compiled against ?
</div></div>
    </div>
</div>
</div>
<div class="ticket-transaction message Correspond even">
<div class="ticket-transaction">
  <div class="metadata">
    <span class="type">
      <a name="txn-1657514" href="/Public/Bug/Display.html?id=116683#txn-1657514">#</a>      
    </span>
    <span class="date">Wed&nbsp;Aug&nbsp;10&nbsp;23:37:43&nbsp;2016</span>
    <span class="description">kevin.buckley.ecs.vuw.ac.nz [...] gmail.com -  Correspondence added</span>
    <span class="time-taken"></span>    <span class="actions hidden"></span>
  </div>
    <div class="content">
      
<table>
  <tr>
    <td align="right" class="message-header-key">Subject:</td>
    <td class="message-header-value"> [rt.cpan.org #116683] Bio-SamTools-1.43 is confused about what various SAMtools build dirs supply where - and so am I</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">Date:</td>
    <td class="message-header-value"> Thu, 11 Aug 2016 15:37:30 +1200</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">To:</td>
    <td class="message-header-value"> bug-Bio-SamTools [...] rt.cpan.org</td>
  </tr>
  <tr>
    <td align="right" class="message-header-key">From:</td>
    <td class="message-header-value"> Kevin Buckley &lt;kevin.buckley.ecs.vuw.ac.nz [...] gmail.com&gt;</td>
  </tr>
</table>
<div class="downloadattachment">
<a href="/Ticket/Attachment/1657514/889580/">Download &#40;untitled&#41;</a> / <a href="/Ticket/Attachment/WithHeaders/889580">with headers</a>
<br />
<span class="downloadcontenttype">text/plain 6.2k</span>
</div>
<div class="messagebody">
<div class="message-stanza plain-text-white-space">I do hope that someone is using this software to study worms, as
this issue seems to have been one massive can of them, however I
think I have accounted for all of them now.


Based on what I&#39;d seen so far, I decided to go looking in the past
versions of SAMtools, in case it was possible to determine which
one of them might match the perculiar layout of header files that
Bio-SamTools clearly expected to be there in the SAMtools it was
pointed to.

Here&#39;s what I found, amongst the ones I had lying around from
previous encounters into the SAMtools universe


SAMtools 1.2

   * doesn&#39;t have a plain htslib subdir but htslib-1.2.1
   * bam.h has the note about htslib version doesn&#39;t have l_aux

SAMtools 1.1 &#40;Github doesn&#39;t bundle htslib&#41;

   * doesn&#39;t have an htslib subdir AT ALL
   * bam.h has the note about htslib version doesn&#39;t have l_aux
   * Doesn&#39;t build: Makefile:86: ../htslib/htslib.mk: No such file or directory

SAMtools 1.1 &#40;Sourceforge&#41;

   * doesn&#39;t have a plain htslib subdir but htslib-1.1
   * bam.h has the note about htslib version doesn&#39;t have l_aux

SAMtools 1.0 &#40;Sourceforge&#41;

   * doesn&#39;t have a plain htslib subdir but htslib-1.0
   * bam.h has the note about htslib version doesn&#39;t have l_aux

SAMtools 0.1.19 &#40;Sourceforge&#41;

   * doesn&#39;t have an htslib subdir AT ALL
   * bam.h DOES HAVE a bam1_t type which has an l_aux
   * Might have all the required header files


So now I go and build a SAMtools 0.1.19 and point Bio-SamTools at that
and get a bit further, but it fails trying to do this

gcc -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions \
 -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic \
 -o blib/arch/auto/Bio/DB/Sam/Sam.so lib/Bio/DB/Sam.o c_bin/bam2bedgraph.o \
 -L/path/to/samtools-0.1.19 -lbam \
 -lpthread -lz

/usr/bin/ld: /path/to/samtools-0.1.19/libbam.a&#40;bgzf.o&#41;: \
 relocation R_X86_64_32 against `.rodata.str1.1&#39; can not be used when making \
 a shared object; recompile with -fPIC

/path/to/samtools-0.1.19/libbam.a: could not read symbols: Bad value


However, that

 -lbam

in the compilation looked to me as though it was trying to link in a
dynamic library and not the static one it was pointed to as part of
the configuration - surely not ?

So I thought to go and create a libbam.so dynamic library, using all
of the object files that SAMtools 0.1.19 produces in creating the
static library.

After the main build doing its:

ar -csru libbam.a bgzf.o kstring.o bam_aux.o bam.o bam_import.o sam.o \
  bam_index.o bam_pileup.o bam_lpileup.o bam_md.o razf.o faidx.o bedidx.o \
  knetfile.o bam_sort.o sam_header.o bam_reheader.o kprobaln.o bam_cat.o


I go and do a

gcc -shared -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions \
  -fstack-protector --param=ssp-buffer-size=4 -m64 -mtune=generic  \
  -o libbam.so \
  bgzf.o kstring.o bam_aux.o bam.o bam_import.o sam.o \
  bam_index.o bam_pileup.o bam_lpileup.o bam_md.o razf.o faidx.o bedidx.o \
  knetfile.o bam_sort.o sam_header.o bam_reheader.o kprobaln.o bam_cat.o



and I find I can get the Bio-SamTools module to build, however it fails
to run any of its tests, notionally because the libbam.so isn&#39;t available
to it at run time.


I say &#34;notionally&#34; because it was that point, having seemingly built the
Bio-SamTools module but failed the tests, that I decided to take a look
at the Bio-SamTools test source lying around within the Perl build
directorires, and what do you think I saw ...

There&#39;s a README File and it says, right at the top,

---8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;---
This is a Perl interface to the SAMtools sequence alignment
interface. It ONLY works on versions of Samtools up to 0.1.17. It does
not work on version 1.0 or higher due to major changes in the library
structure.
---8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;------8&lt;---

so there&#39;s the answer to my question from the previous posts: I should be
building against 0.1.17.

Seems a shame that no-one could be bothered to tell me that but hey-ho,
no matter.


Armed with that knowledge and the knowledge that Bio-SamTools is going to
try and link against a non-existant dynamic libray that I&#39;ll need to
create for myself, I added, to my list of potential SAMtools versions


SAMtools 0.1.17 &#40;Sourceforge&#41;

   * doesn&#39;t have an htslib subdir AT ALL
   * bam.h DOES HAVE a bam1_t type which has an l_aux
   * Might have all the required header files

and proceed as follows, noting that I&#39;ll still need to create a
link to get around the way that the htslib headers are referred
to in the various parts of Bio-SamTools


tar xf amtools-0.1.17.tar.bz2
cd samtools-0.1.17
cp Makefile Makefile.orig
vim Makefile
make
ln -s . htslib


where the Makefile gets altered as follows

2c2
&lt; CFLAGS=               -g -Wall -O2 #-m64 #-arch ppc
---
<span
    class="message-stanza-folder closed"
    onclick="fold_message_stanza(this, 'Show\x20quoted\x20text', 'Hide\x20quoted\x20text');">Show quoted text</span><br /><div class="message-stanza closed">&gt; CFLAGS=               -g -Wall -O2 -fPIC #-m64 #-arch ppc
</div>

I then build the shared library as shown above, adding in the -fPIC,
and make it available, to the yet to be built Bio-SamTools module,
by poking it into LD_LIBRARY_PATH

export LD_LIBRARY_PATH=/path/to/samtools-0.1.17


The build of Bio-SamTools module no runs through and we see the
following files being installed created into the Perl heirachy
&#40;paths truncated  to avoid wrapping&#41;

Installing .../lib/perl5/x86_64-linux-thread-multi/auto/Bio/DB/Sam/Sam.so
Installing .../lib/perl5/x86_64-linux-thread-multi/auto/Bio/DB/Sam/Sam.bs
Installing .../bin/bam2bedgraph
Installing .../lib/perl5/x86_64-linux-thread-multi/Bio/DB/Sam.pm
Installing .../lib/perl5/x86_64-linux-thread-multi/Bio/DB/Sam/Segment.pm
Installing .../lib/perl5/x86_64-linux-thread-multi/Bio/DB/Sam/Constants.pm


This &#40;OK, make that &#34;All this&#34;&#41; seems to have done the trick.

I hope documenting this might be useful to someone, although, given
the lack of feedback, it&#39;s quite possible that no-one, bar the user
I was looking to install it for on my resources, is even using it
anymore?


Footnote:

I suppose it&#39;s possible that, even though the README says that it only
works with 0.1.17, there&#39;s clearly enough similarity between 0.1.17 and
0.1.19 failing to build that Bio-SamTools-1.43 might even compile aganst
that later version, as long as the hoop-jumping above id performed, however
I&#39;m quitting whilst I am ahead, and will leave that as an exercise for
the reader,
should this thread ever have one, I guess.
</div></div>
    </div>
</div>
</div>
</div>
    <hr class="clear" />
  </div>
</div>




</div>
 


  <hr class="clear" />
</div>
<p id="sponsors">
  This service is sponsored and maintained by
  <a href="http://bestpractical.com">Best Practical Solutions</a> and runs on
  <a href="http://noc.perl.org">Perl.org</a> infrastructure.
</p>

<p id="contact-info">
  Please report any issues with rt.cpan.org to <a href="mailto:rt-cpan-admin@bestpractical.com">rt-cpan-admin@bestpractical.com</a>.
</p>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA\x2D937849\x2D12');
  ga('send', 'pageview');
</script>

<div id="footer">
  <p id="time"><span>Time to display: 0.197522</span></p>
  <p id="bpscredits"><span>&#187;&#124;&#171; RT 4.0.18 Copyright 1996-2013 <a href="http://www.bestpractical.com?rt=4.0.18">Best Practical Solutions, LLC</a>.
</span></p>
</div>
  </body>
</html>
