--- /usr/lib/perl5/site_perl/5.8.8/Spreadsheet/ParseExcel/Utility.pm.orig	2009-01-23 19:41:30.000000000 -0500
+++ /usr/lib/perl5/site_perl/5.8.8/Spreadsheet/ParseExcel/Utility.pm	2009-05-01 13:42:25.000000000 -0400
@@ -772,6 +772,16 @@
                         ),
                         $data
                     );
+		    if ($data =~ /^${number_result}5/) {
+                        $number_result = sprintf(
+                            (
+                                defined($after_decimal)
+                                ? "%0${str_length}.${after_decimal}f"
+                                : "%0${str_length}f"
+                            ),
+                            "${data}1"
+                        );
+		    }
                 }
                 else {
                     $number_result = sprintf( "%0${str_length}.0f", $data );
