diff --git a/OpenSSH.pm b/OpenSSH.pm
index e2da06c..efa49ee 100644
--- a/OpenSSH.pm
+++ b/OpenSSH.pm
@@ -441,7 +441,7 @@ sub _connect {
         _load_module('IO::Pty');
         $self->{_mpty} = $mpty = IO::Pty->new;
 	push @master_opts, (-o => 'NumberOfPasswordPrompts=1',
-			    -o => 'PreferredAuthentications=password');
+			    -o => 'PreferredAuthentications=password,keyboard-interactive');
     }
 
     my @call = $self->_make_call(\@master_opts);
