diff -ur HTML-TagCloud-0.38/Makefile.PL HTML-TagCloud-0.38+no_installdirs/Makefile.PL
--- HTML-TagCloud-0.38/Makefile.PL	2013-04-20 07:04:45.000000000 +0200
+++ HTML-TagCloud-0.38+no_installdirs/Makefile.PL	2014-02-05 01:44:25.254127150 +0100
@@ -3,7 +3,6 @@
 WriteMakefile
 (
           'PL_FILES' => {},
-          'INSTALLDIRS' => 'site',
           'NAME' => 'HTML::TagCloud',
           'EXE_FILES' => [],
           'VERSION_FROM' => 'lib/HTML/TagCloud.pm',
