Description: Fixing man page
Author: Fabrizio Regalli <fabreg@fabreg.it>
Last-Update: 2011-07-07
Forwarded: no

--- a/lib/Validation/Class.pm
+++ b/lib/Validation/Class.pm
@@ -1732,7 +1732,7 @@
         my ($name, $email, $login, $password) =
             $self->get_params(qw/name email login password/);
         
-        # you should note that if the params dont exist they will return undef
+        # you should note that if the params don't exist they will return undef
         # ... meaning you should check that it exists before checking its value
         # e.g.
         
