--- /usr/local/lib/perl5/site_perl/5.12.1/Data/Dump.pm	2010-06-12 11:05:28.000000000 +0200
+++ Data/Dump.pm	2010-08-05 13:53:43.403968203 +0200
@@ -505,7 +505,7 @@
       {
 	  $require{"MIME::Base64"}++;
 	  return "MIME::Base64::decode(\"" .
-	             MIME::Base64::encode($_[0],"") .
+	             MIME::Base64::encode( ( my $copy = $_[0] ), "" ) .
 		 "\")";
       }
       return "pack(\"H*\",\"" . unpack("H*", $_[0]) . "\")";
