--- PhantomJS.pm	2014-11-07 13:38:41.605751267 -0700
+++ PhantomJS-new.pm	2014-11-07 13:39:30.925742547 -0700
@@ -2637,7 +2637,7 @@
 
     if( scalar keys %$rect ) {
 
-        $self->viewport_size( $ rect );
+        $self->eval_in_phantomjs( 'this.clipRect= arguments[0]', $rect );
     };
 
     return $self->render_content( format => 'png' );
@@ -2656,11 +2656,10 @@
     my( $self, $new )= @_;
 
     $self->eval_in_phantomjs( <<'JS', $new );
-        var viewportSize= this.clipRect;
         if( arguments[0]) {
-            this.clipRect= arguments[0];
+            this.viewportSize= arguments[0];
         };
-        viewportSize
+        return this.viewportSize;
 JS
 };
 
