Description: fix some minor typos
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-08-12

--- a/bin/efa
+++ b/bin/efa
@@ -352,7 +352,7 @@
 B<efa> has a builtin list of EFA entry points which can be used with the
 B<-A> and B<-D> options. You can also specify a custom URL using B<-u>.
 However, the default EFA service is sufficient in most cases (even ICE
-connections all over germany).
+connections all over Germany).
 
 =head1 OPTIONS
 
--- a/lib/Travel/Routing/DE/EFA.pm
+++ b/lib/Travel/Routing/DE/EFA.pm
@@ -768,7 +768,7 @@
 		[ 'http://www.linzag.at/static/XSLT_TRIP_REQUEST2', 'Linz AG' ],
 		[
 			'http://212.114.197.7/vgnExt_oeffi/XML_TRIP_REQUEST2',
-			'Verkehrsverbund Grossraum Nuernberb'
+			'Verkehrsverbund Grossraum Nuernberg'
 		],
 		[
 			'http://efa.vrr.de/vrr/XSLT_TRIP_REQUEST2',
