Description: Fix pod2man errors
Author: Scott Kitterman <scott@kitterman.com>
Origin: vendor
Forwarded: will be 
Last-Update: 2016-12-22

--- libmail-rbl-perl-1.10.orig/RBL.pm
+++ libmail-rbl-perl-1.10/RBL.pm
@@ -123,6 +123,8 @@ sub check ($$)
 
 =pod
 
+=over
+
 =item C<-E<gt>check_rhsbl($host)>
 
 Analogous to C<-E<gt>check()>, but  queries RHSBLs instead of IP-based
@@ -131,6 +133,8 @@ B<http://www.rfc-ignorant.org/>.
 
 Results and return values are the same as C<-E<gt>check()>.
 
+=back
+
 =cut
 
 sub check_rhsbl ($$)
