Index: Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Turtles.pm
===================================================================
--- Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Turtles.pm	(revisión: 5883)
+++ Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Turtles.pm	(copia de trabajo)
@@ -25,7 +25,7 @@
         grep { $_ =~ $filter }
         map { /^expr?_command_(\w+)/ ? $1 : () }
         map { $_->{name} }
-        $self->meta->compute_all_applicable_methods
+        $self->meta->get_all_methods
       ),
     );
   } else {
Index: Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Methods.pm
===================================================================
--- Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Methods.pm	(revisión: 5883)
+++ Devel-REPL/lib/Devel/REPL/Plugin/CompletionDriver/Methods.pm	(copia de trabajo)
@@ -58,7 +58,7 @@
   return $orig->(@_),
          grep { $_ =~ $re }
          map  { $_->{name} }
-         $metaclass->compute_all_applicable_methods;
+         $metaclass->get_all_methods;
 };
 
 1;
