--- Makefile.PL.dist	2005-10-21 00:01:37.000000000 +0300
+++ Makefile.PL	2009-02-12 21:19:15.500000000 +0200
@@ -3,12 +3,10 @@
 author('Ryan King <rking@sharpsaw.org');
 abstract('Cliboard - Copy and Paste with any OS');
 license('perl');
-version_from('lib/Clipboard.pm');
-requires(qw(Spiffy 0));
-# This conditional requires is kind of wacky. It works from the command-line
-# "perl Makefile.PL" invocation just fine, but I am afeard that there's some
-# automated script out there that won't like it - cpan/cpanp?  We'll find out.
-requires('Win32::Clipboard') if $^O =~ /^win|cygwin/i;
+version_from 'lib/Clipboard.pm';
+perl_version 5.006;
+requires 'Spiffy';
+requires 'Win32::Clipboard' if $^O =~ /mswin|cygwin/i;
 install_script("scripts/clip$_") for qw'join edit browse accumulate filter';
 check_nmake();
 WriteAll();
