--- bin/perlcritic	(revision 57937)
+++ bin/perlcritic	(local)
@@ -456,7 +456,7 @@
 
     my %color_of = (
         $SEVERITY_HIGHEST => 'bold red',
-        $SEVERITY_HIGH    => 'yellow',
+        $SEVERITY_HIGH    => 'magenta',
     );
     return map { _colorize( "$_", $color_of{$_->severity()} ) } @violations;
 
@@ -916,7 +916,7 @@
 =item C<--color>
 
 This option is on when outputting to a tty.  When set, Severity 5 and 4 are
-colored red and yellow, respectively.  Colorization only happens if
+colored red and magenta, respectively.  Colorization only happens if
 L<Term::ANSIColor|Term::ANSIColor> is installed and it only works on
 non-Windows environments.  Negate this switch to disable color.  You can set
 the default value for this option in your F<.perlcriticrc> file.
