=== modified file 'bin/pmat-show-sv'
--- bin/pmat-show-sv	2014-01-15 17:27:58 +0000
+++ bin/pmat-show-sv	2014-01-21 00:13:05 +0000
@@ -64,6 +64,7 @@
 
          my @pads = $sv->pads;
          foreach my $depth ( 0 .. $#pads ) {
+            next unless $pads[$depth];
             say "  pad[$depth]=" . $pads[$depth]->desc_addr;
          }
 

=== modified file 'lib/Devel/MAT/SV.pm'
--- lib/Devel/MAT/SV.pm	2014-01-15 18:21:16 +0000
+++ lib/Devel/MAT/SV.pm	2014-01-21 00:13:05 +0000
@@ -1231,8 +1231,10 @@
    bless $padnames, "Devel::MAT::SV::PADNAMES";
    $padnames->_set_padcv_at( $self->addr );
 
-   bless $_, "Devel::MAT::SV::PAD" for @pads;
-   $_->_set_padcv_at( $self->addr ) for @pads;
+   foreach my $pad ( @pads ) {
+      bless $pad, "Devel::MAT::SV::PAD";
+      $pad->_set_padcv_at( $self->addr );
+   }
 
    $self->{pads} = \@pads;
 

