diff --git a/utils/Math/PariBuild.pm b/utils/Math/PariBuild.pm
index da269a9..fc09e87 100644
--- a/utils/Math/PariBuild.pm
+++ b/utils/Math/PariBuild.pm
@@ -309,7 +309,7 @@ sub finish_download_pari ($$$$;$) {
 sub download_pari {
   my ($srcfile, $force) = (shift, shift);
   my $host = 'megrez.math.u-bordeaux.fr';
-  my $dir  = '/pub/pari/unix/';
+  my $dir  = '/pub/pari/';
   my($ftp, $ua, $base_url);

   print "Did not find GP/PARI build directory around.\n" unless defined $srcfile;
