diff -u -r Test-Harness-3.09/lib/TAP/Parser/Source/Perl.pm Test-Harness-3.09-new/lib/TAP/Parser/Source/Perl.pm
--- Test-Harness-3.09/lib/TAP/Parser/Source/Perl.pm	2008-02-09 17:25:10.000000000 -0600
+++ Test-Harness-3.09-new/lib/TAP/Parser/Source/Perl.pm	2008-02-18 16:12:34.400911200 -0600
@@ -4,6 +4,7 @@
 use Config;
 use vars qw($VERSION @ISA);
 
+use constant IS_CYGWIN => ( $^O eq 'cygwin' );
 use constant IS_WIN32 => ( $^O =~ /^(MS)?Win32$/ );
 use constant IS_VMS => ( $^O eq 'VMS' );
 
@@ -257,8 +258,9 @@
 
     # Quote the argument if there's any whitespace in it, or if
     # we're VMS, since VMS requires all parms quoted.  Also, don't quote
-    # it if it's already quoted.
+    # it if it's already quoted, or if the platform is cygwin.
     for (@switches) {
+        next if IS_CYGWIN;
         $_ = qq["$_"] if ( ( /\s/ || IS_VMS ) && !/^".*"$/ );
     }
 
