From 9d64754383777bb0a0aa633bd82e80526a9306dc Mon Sep 17 00:00:00 2001
From: Marius Gavrilescu <marius@ieval.ro>
Date: Mon, 12 Jan 2015 09:46:44 +0200
Subject: [PATCH 2/2] Localize $_ where needed in ::Style::Stream

---
 Parser/Style/Stream.pm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Parser/Style/Stream.pm b/Parser/Style/Stream.pm
index 2907e5f..7bbdaab 100644
--- a/Parser/Style/Stream.pm
+++ b/Parser/Style/Stream.pm
@@ -16,6 +16,7 @@ sub Init {
 
 sub Start {
   no strict 'refs';
+  local $_;
   my $expat = shift;
   my $type = shift;
   
@@ -38,6 +39,7 @@ sub Start {
 
 sub End {
   no strict 'refs';
+  local $_;
   my $expat = shift;
   my $type = shift;
   
@@ -63,6 +65,7 @@ sub Char {
 
 sub Proc {
   no strict 'refs';
+  local $_;
   my $expat = shift;
   my $target = shift;
   my $text = shift;
@@ -89,6 +92,7 @@ sub Final {
 
 sub doText {
   no strict 'refs';
+  local $_;
   my $expat = shift;
   $_ = $expat->{Text};
   
-- 
2.1.4

