diff -r ad8f8a761f61 Makefile.PL
--- a/Makefile.PL	Mon Mar 10 13:14:19 2014 +0100
+++ b/Makefile.PL	Mon Mar 10 15:36:43 2014 +0100
@@ -309,6 +309,7 @@
             # Put several options.
             $config_LIBS_alternatives = [
                 map { "$l $_" }
+                q/ -llibxml2/,
                 q/ -lxml2 -lzlib/,
                 q/ -llibxml2 -lzlib -llibgettextlib.dll/
             ];
