--- canvas-primitives.pl.orig	2004-03-21 07:16:47.000000000 +0300
+++ canvas-primitives.pl	2007-01-20 02:37:23.000000000 +0300
@@ -167,7 +167,7 @@
 					'Gnome2::Canvas::Rect',
 					x1 => 90.0,
 					y1 => 40.0,
-					x2 => 180.0,
+					x2 => 33000.0, # that triggers a bug, rectangle is broken!
 					y2 => 100.0,
 					fill_color => "mediumseagreen",
 					fill_stipple => $stipple,
@@ -557,9 +557,14 @@
 	return TRUE;
 }
 
+use Gnome2;
+
 sub create {
 	my $aa = shift;
 
+	printf "Gtk2::VERSION = %s, Glib::VERSION = %s\nGnome2::VERSION = %s, Gnome2::Canvas::VERSION = %s\n",
+	  $Gtk2::VERSION, $Glib::VERSION, $Gnome2::VERSION, $Gnome2::Canvas::VERSION;
+
 	my $vbox = Gtk2::VBox->new (FALSE, 4);
 	$vbox->set_border_width (4);
 	$vbox->show;
@@ -647,7 +652,7 @@
 	$frame->show;
 
 	$canvas->set_size_request (600, 450);
-	$canvas->set_scroll_region (0, 0, 600, 450);
+	$canvas->set_scroll_region (0, 0, 600000, 6000);
 	$frame->add ($canvas);
 	$canvas->show;
 
