Description: Spelling fix in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2019-02-04

--- a/lib/Test/MemoryGrowth.pm
+++ b/lib/Test/MemoryGrowth.pm
@@ -87,7 +87,7 @@
 A second failure case comes from the fact that memory usage is taken from the
 Operating System's measure of the process's Virtual Memory size, so as to be
 able to detect memory usage growth in C libraries or XS-level wrapping code,
-as well as Perl functions. Because Perl does not agressively return unused
+as well as Perl functions. Because Perl does not aggressively return unused
 memory to the Operating System, it is possible that a piece of code could use
 un-allocated but un-reclaimed memory to grow into; resulting in an increase in
 its requirements despite not requesting extra memory from the Operating
