--- DBIx-Class-Fixtures-1.001020/lib/DBIx/Class/Fixtures.pm	2013-07-20 12:40:41.000000000 -0700
+++ DBIx-Class-Fixtures-1.001020.fixed/lib/DBIx/Class/Fixtures.pm	2013-10-16 12:54:35.006532618 -0700
@@ -12,8 +12,7 @@
 use Data::Dump::Streamer;
 use Data::Visitor::Callback;
 use File::Path;
-use File::Copy::Recursive qw/dircopy/;
-use File::Copy qw/move/;
+use File::Copy::Recursive qw/dircopy rmove/;
 use Hash::Merge qw( merge );
 use Data::Dumper;
 use Class::C3::Componentised;
@@ -718,7 +717,7 @@
   }
 
   $self->msg("- moving temp dir to $output_dir");
-  move($_, dir($output_dir, $_->relative($_->parent)->stringify)) 
+  rmove($_, dir($output_dir, $_->relative($_->parent)->stringify)) 
     for $tmp_output_dir->children;
 
   if (-e $output_dir) {
