diff -Naur t/01live.t t/01live.t
--- t/01live.t	2005-08-18 18:48:30.000000000 +0100
+++ t/01live.t	2006-07-06 11:00:43.000000000 +0100
@@ -1,5 +1,6 @@
 use Test::More;
 BEGIN {
+    delete @ENV{ qw( http_proxy HTTP_PROXY ) };
     if (eval { require LWP::Simple }) {
 	plan tests => 5;
     } else {
diff -Naur t/04unhandlederrors.t t/04unhandlederrors.t
--- t/04unhandlederrors.t	2005-07-25 23:41:07.000000000 +0100
+++ t/04unhandlederrors.t	2006-07-06 11:00:58.000000000 +0100
@@ -1,5 +1,6 @@
 use Test::More;
 BEGIN {
+    delete @ENV{ qw( http_proxy HTTP_PROXY ) };
     if (eval { require LWP::Simple }) {
 	plan tests => 5;
     } else {
diff -Naur t/05handlederrors.t t/05handlederrors.t
--- t/05handlederrors.t	2005-07-25 23:41:07.000000000 +0100
+++ t/05handlederrors.t	2006-07-06 11:01:06.000000000 +0100
@@ -1,5 +1,6 @@
 use Test::More;
 BEGIN {
+    delete @ENV{ qw( http_proxy HTTP_PROXY ) };
     if (eval { require LWP::Simple }) {
 	plan tests => 5;
     } else {
