--- HiRes.t~	Sat Nov  5 03:25:20 2005
+++ HiRes.t	Sat Nov  5 15:36:18 2005
@@ -353,7 +353,7 @@
     }
 
     $t0 = gettimeofday();
-    $a = abs(usleep($sleep * 1E6) / ($sleep * 1E6) - 1.0);
+    $a = abs(Time::HiRes::usleep($sleep * 1E6) / ($sleep * 1E6) - 1.0);
     $td = gettimeofday() - $t0;
     $ratio = 1.0 + $a;
 
@@ -398,7 +398,7 @@
 print $@ =~ /::sleep\(-1\): negative time not invented yet/ ?
     "ok 24\n" : "not ok 24\n";
 
-eval { usleep(-2) };
+eval { Time::HiRes::usleep(-2) };
 print $@ =~ /::usleep\(-2\): negative time not invented yet/ ?
     "ok 25\n" : "not ok 25\n";
 
