--- a/lib/LaTeX/Driver.pm
+++ b/lib/LaTeX/Driver.pm
@@ -1090,7 +1090,7 @@
 
 =item no source specified
 
-The C<source> paramater should be specified as the name of a LaTeX
+The C<source> parameter should be specified as the name of a LaTeX
 source file or it should be a reference to a scalar variable holding
 the LaTeX source document.
 
@@ -1226,7 +1226,7 @@
 
 =item *
 
-Investigate pre- and post-processors and other auxilliary programs.
+Investigate pre- and post-processors and other auxiliary programs.
 
 =back
 
