Description: spelling fix
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-02-24

--- a/lib/Net/Dict.pod
+++ b/lib/Net/Dict.pod
@@ -217,7 +217,7 @@
 Specify the dictionaries that will be
 searched during the successive define() or match() calls.
 Defaults to '*'.
-No existance checks are performed by this interface, so you'd better make
+No existence checks are performed by this interface, so you'd better make
 sure the dictionaries you specify are on the server (e.g. by calling
 dbs()).
 
