--- GraphViz.pm	2010-12-27 02:49:29.695248480 +0100
+++ /usr/share/perl5/SQL/Translator/Producer/GraphViz.pm	2010-12-27 02:51:16.405873459 +0100
@@ -595,8 +595,8 @@
                 my $table2 = $tables[ $j ];
                 next if $done{ $table1 }{ $table2 };
                 $gv->add_edge(
-                    $table2,
-                    $table1,
+                    qq["$table2"],
+                    qq["$table1"],
                     arrowhead => $optional_constraints{$bi} ? 'empty' : 'normal',
                 );
                 $done{ $table1 }{ $table2 } = 1;
