Index: activitymail
===================================================================
--- activitymail	(revision 2498)
+++ activitymail	(working copy)
@@ -45,6 +45,10 @@
 ##############################################################################
 # Set up the variables and constants we'll need later.
 ##############################################################################
+if ($opt_w) {
+    require URI::Escape;
+    URI::Escape->import('uri_escape');
+}
 
 # Set constants.
 use constant DEBUG => $opt_D ? 1 : 0;
@@ -733,7 +737,7 @@
                 $nm .= " (r$new)" if $opt_V;
             }
             $msg .= "$nm\n";
-            $msg .= "        ($url)\n" if $opt_w;
+            $msg .= '        (' . uri_escape($url) . ")\n" if $opt_w;
         }
     }
     return \$msg;
