--- a/TFTP.pm	2007-07-18 13:32:43.000000000 +0200
+++ b/TFTP.pm	2010-04-30 22:32:29.000000000 +0200
@@ -129,7 +129,11 @@
 
     my $file = $local;
     unless(ref($local)) {
-	unlink($file);
+	my $retval = unlink($file);
+        if ($retval < 1) {
+                $self->{'error'} = "Can not unlink $file: $!";
+                return undef;
+        }
 	$local = IO::File->new($file,O_WRONLY|O_TRUNC|O_CREAT);
     }
 
