Description: qw) in list context is an error now
Origin: vendor
Bug-Debian: http://bugs.debian.org/711608
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-06-08

--- a/t/01-main.t
+++ b/t/01-main.t
@@ -28,7 +28,7 @@
 package main;
 use Test::More tests => 10;
 
-foreach my $class qw(WithTrait WithMeta)
+foreach my $class (qw(WithTrait WithMeta))
 {
 	my $foo = $class->new(x => 'x');
 	is($foo->x, 'x', "$class x works");
