--- RecursiveUpdate.pm	2011-04-01 03:56:16.000000000 +0000
+++ RecursiveUpdate.new.pm	2011-04-01 03:55:10.000000000 +0000
@@ -450,7 +450,7 @@
             unless (
                !$sub_object
             && !$updates
-            && ( exists $info->{attrs}{join_type}
+            && !( exists $info->{attrs}{join_type}
                 && $info->{attrs}{join_type} eq 'LEFT' )
             );
     }
