diff -ruN Image-Info-1.18-orig/Changes Image-Info-1.18/Changes
--- Image-Info-1.18-orig/Changes	2006-03-03 15:21:35.000000000 +0000
+++ Image-Info-1.18/Changes	2006-03-03 19:49:55.000000000 +0000
@@ -3,7 +3,7 @@
 2006-03-03 v1.18 Tels 61 tests
   * fix bug #6558: Image::Info 1.16 -> fails two tests (Thanx jpo!)
     + make string.t use strict, Test::More and simplify it
-    + let string.t skip tests if Image::XPM, Image::XBM or XML::Simple
+    + let string.t skip tests if Image::Xpm, Image::Xbm or XML::Simple
       are not installed (these would formerly silently fail)
     + Image::Info::XPM and Image::Info::XBM now handle in-memory files better
     + fix also pod_cov.t warnings (was "Image::Info.pm" vs "Image::Info")
diff -ruN Image-Info-1.18-orig/lib/Image/Info/XBM.pm Image-Info-1.18/lib/Image/Info/XBM.pm
--- Image-Info-1.18-orig/lib/Image/Info/XBM.pm	2006-03-03 15:08:04.000000000 +0000
+++ Image-Info-1.18/lib/Image/Info/XBM.pm	2006-03-03 19:52:45.000000000 +0000
@@ -10,8 +10,8 @@
 	$info->push_info(0, "Warn", shift);
     };
 
-    my $i = Image::Xbm->new();
-    # loading the file as a seperate step avoids a "-r" test, this would
+    my $i = Image::Xbm->new(-width => 0, -height => 0);
+    # loading the file as a separate step avoids a "-r" test, this would
     # file with in-memory strings (aka fake files)
     $i->load($source);
 
diff -ruN Image-Info-1.18-orig/lib/Image/Info/XPM.pm Image-Info-1.18/lib/Image/Info/XPM.pm
--- Image-Info-1.18-orig/lib/Image/Info/XPM.pm	2006-03-03 14:21:45.000000000 +0000
+++ Image-Info-1.18/lib/Image/Info/XPM.pm	2006-03-03 19:52:54.000000000 +0000
@@ -13,8 +13,8 @@
 	$info->push_info(0, "Warn", shift);
     };
 
-    my $i = Image::Xpm->new()
-    # loading the file as a seperate step avoids a "-r" test, this would
+    my $i = Image::Xpm->new(-width => 0, -height => 0);
+    # loading the file as a separate step avoids a "-r" test, this would
     # file with in-memory strings (aka fake files)
     $i->load($source);
 
diff -ruN Image-Info-1.18-orig/Makefile.PL Image-Info-1.18/Makefile.PL
--- Image-Info-1.18-orig/Makefile.PL	2006-03-03 13:20:21.000000000 +0000
+++ Image-Info-1.18/Makefile.PL	2006-03-03 19:50:11.000000000 +0000
@@ -93,8 +93,8 @@
 
 requires	'IO::String'	=> 1.03 if $] < 5.008;
 
-recommends	'Image::XPM'	=> 0,
-		'Image::XBM'	=> 0,
+recommends	'Image::Xpm'	=> 0,
+		'Image::Xbm'	=> 0,
 		'XML::Simple'	=> 0;
 
 build_requires  'Test::More'	=> 0.42;
diff -ruN Image-Info-1.18-orig/t/string.t Image-Info-1.18/t/string.t
--- Image-Info-1.18-orig/t/string.t	2006-03-03 10:59:03.000000000 +0000
+++ Image-Info-1.18/t/string.t	2006-03-03 19:50:32.000000000 +0000
@@ -19,8 +19,8 @@
 
 my $requires = 
   {
-  xpm => 'Image::XPM',
-  xbm => 'Image::XBM',
+  xpm => 'Image::Xpm',
+  xbm => 'Image::Xbm',
   svg => 'XML::Simple',
   };
 
