--- Tcl-1.05-UjbRn3/Makefile.PL	2016-06-29 00:48:16.000000000 +0800
+++ Tcl-1.05-SJ/Makefile.PL	2018-06-16 23:39:09.770237300 +0800
@@ -199,6 +199,16 @@
 	@extraargs = (dynamic_lib => {OTHERLDFLAGS => $libpath});
 	$libpath = "";
     }
+} else {
+    # MakeMaker LIBS argument only workable for string with 
+    # the format "-ltclstub" or "-L/lib/path". When apply --tclconfig option,
+    # the $libpath would be something like "-Wl,-ltclstub.a"; or something like
+    # "-Wl,/usr/lib/libtclstub.a" when apply --buildspec
+    # We need other method to specify lib path
+    if ($libpath =~ /^-Wl,/) {
+        @extraargs = (dynamic_lib => {OTHERLDFLAGS => $libpath});
+        $libpath = "";
+    }
 }
 
 #print <<"#EOS";
