--- timeout.pl~	2020-09-22 13:23:03.308904610 -0400
+++ timeout.pl	2020-09-22 14:01:20.685688625 -0400
@@ -14,7 +14,8 @@ use strict;
 use warnings;
 use Net::Telnet;
 use Getopt::Long;
-use subs 'tell'; # Override CORE::tell with Main::tell functions
+use v5.16;	       # For fc
+use subs 'tell';       # Override CORE::tell with Main::tell functions
 
 my $name = 'TimeoutBot';
 my $host = 'freechess.org';
@@ -126,7 +127,9 @@ while (1) {
     chomp;
     s/^\r//;
 
-    if (/^([[:alpha:]]+).*? tells you:\s+(.*)$/) {
+    if (/^([[:alpha:]]+).*? tells you:\s+(.*)$/ &&
+	(fc $1) ne (fc $name)) {
 	tell $1, qq/Hi $1. You told me "$2"./;
+	$last_tell_time = time;
     }
 }
