--- SOAP-Lite-0.68/lib/XMLRPC/Lite.pm.nil_value	2004-11-14 14:30:50.000000000 -0500
+++ SOAP-Lite-0.68/lib/XMLRPC/Lite.pm	2006-07-18 13:06:53.000000000 -0400
@@ -286,6 +286,8 @@
     return +{map {$self->decode_object($_)} @{$children || []}};
   } elsif ($name =~ /^(?:param|fault)$/) {
     return scalar(($self->decode_object($children->[0]))[1]);
+  } elsif ($name =~ /^(?:nil)$/) {
+    return undef;
   } else {
     die "wrong element '$name'\n";
   }
