--- lib/CatalystX/ListFramework/Builder/Controller/AJAX.pm	2008-12-26 09:53:11.000000000 -0600
+++ /usr/lib/perl5/site_perl/5.8.8/CatalystX/ListFramework/Builder/Controller/AJAX.pm	2008-12-26 12:26:28.000000000 -0600
@@ -338,7 +338,10 @@
 
         # update or create the row; could this use a magic DBIC method?
         my $pk = $lf->{table_info}->{$model}->{pk};
-        my $row = eval { $c->model($model)->find( $data->{ $pk } ) };
+        my $row = undef;
+        if (defined($data->{ $pk })) {
+            $row = eval { $c->model($model)->find( $data->{ $pk } ) };
+        }
         $row = ( (blessed $row)
             ? $row->set_columns( $data )
             : $c->model($model)->new_result( $data ) );
