--- os/darwin.h.orig	2008-01-29 07:42:47.000000000 +0200
+++ os/darwin.h	2008-01-29 07:42:57.000000000 +0200
@@ -60,7 +60,6 @@
 #include <sys/resource.h>
 #include <sys/stat.h>
 #include <sys/sysctl.h>
-#include <kvm.h>
 #include <unistd.h>
 
 
@@ -108,7 +107,7 @@
 		struct policy_fifo_info fifo;
 	} schedinfo;
 	int	invalid_tinfo;
-	int	thread_count;
+	mach_msg_type_number_t thread_count;
 	thread_port_array_t thread_list;
 	thread_values_t *thval;
 	int	invalid_thinfo;
