Index: lib/Palm/Keyring.pm
===================================================================
RCS file: /cvs/scripts/misc/Palm/lib/Palm/Keyring.pm,v
retrieving revision 1.51
diff -u -r1.51 Keyring.pm
--- lib/Palm/Keyring.pm	13 Sep 2007 15:46:09 -0000	1.51
+++ lib/Palm/Keyring.pm	4 Dec 2007 03:20:23 -0000
@@ -633,6 +633,7 @@
     foreach my $k (keys %{ $new }) {
         $plaintext .= _pack_field($new->{$k});
     }
+    $plaintext .= chr(0xff) x 2;
 
     my $encrypted;
     if ($c->{name} eq 'None') {
