=== modified file 'lib/IO/Async/Loop.pm'
--- old/lib/IO/Async/Loop.pm	2019-08-19 14:29:17 +0000
+++ new/lib/IO/Async/Loop.pm	2019-11-23 17:39:12 +0000
@@ -209,7 +209,7 @@
 The constructed object is cached, and will be returned again by a subsequent
 call. The cache will also be set by a constructor on a specific subclass. This
 behaviour makes it possible to simply use the normal constructor in a module
-that wishes to interract with the main program's Loop, such as an integration
+that wishes to interact with the main program's Loop, such as an integration
 module for another event system.
 
 For example, the following two C<$loop> variables will refer to the same
@@ -1345,7 +1345,7 @@
 
 =back
 
-This method should not be used in new code; intead use L</run_process>
+This method should not be used in new code; instead use L</run_process>
 directly.
 
 =cut

