--- GnuPG.pm	2006/09/06 12:59:08	1.1
+++ GnuPG.pm	2006/09/06 12:59:23
@@ -394,8 +394,9 @@
   $self->_set_options($gnupg);
   # how we create some handles to interact with GnuPG
   my $input   = IO::Handle->new();
+  my $output   = IO::Handle->new();
   my $error   = IO::Handle->new();
-  my $handles = GnuPG::Handles->new( stderr => $error, stdin  => $input );
+  my $handles = GnuPG::Handles->new( stderr => $error, stdout => $output, stdin  => $input );
 
   my ($sigfh, $sigfile)
     = File::Temp::tempfile('mgsXXXXXXXX',
