Index: Changes
===================================================================
--- Changes	(revision 901)
+++ Changes	(working copy)
@@ -1323,6 +1323,8 @@
 - eliminate many -Wall warnings
 - update README to match unbuggy giflib
 - document index parameter of Imager::Font->new()
+- change faxable output to use a more fax natural PHOTOMETRIC_MINISWHITE, 
+  since T.4 normally works that way, and MINISBLACK confuses some readers.
 
 =================================================================
 
Index: tiff.c
===================================================================
--- tiff.c	(revision 900)
+++ tiff.c	(working copy)
@@ -580,7 +580,7 @@
     { mm_log((1, "i_writetiff_wiol_faxable: TIFFSetField bitpersample=1\n")); return 0; }
   if (!TIFFSetField(tif, TIFFTAG_PLANARCONFIG, PLANARCONFIG_CONTIG))
     { mm_log((1, "i_writetiff_wiol_faxable: TIFFSetField planarconfig\n")); return 0; }
-  if (!TIFFSetField(tif, TIFFTAG_PHOTOMETRIC, PHOTOMETRIC_MINISBLACK))
+  if (!TIFFSetField(tif, TIFFTAG_PHOTOMETRIC, PHOTOMETRIC_MINISWHITE))
     { mm_log((1, "i_writetiff_wiol_faxable: TIFFSetField photometric=%d\n", PHOTOMETRIC_MINISBLACK)); return 0; }
   if (!TIFFSetField(tif, TIFFTAG_COMPRESSION, 3))
     { mm_log((1, "i_writetiff_wiol_faxable: TIFFSetField compression=3\n")); return 0; }
@@ -620,7 +620,7 @@
       bits = width-x; if(bits>8) bits=8;
       i_gsamp(im, x, x+8, y, luma, &luma_chan, 1);
       for(bitpos=0;bitpos<bits;bitpos++) {
-	linebuf[linebufpos] |= ((luma[bitpos]>=128)?bitval:0);
+	linebuf[linebufpos] |= ((luma[bitpos] < 128) ? bitval : 0);
 	bitval >>= 1;
       }
       linebufpos++;
