diff -u old/Base.pm new/Base.pm
--- old/Base.pm	2008-10-20 21:46:17.000000000 -0400
+++ new/Base.pm	2008-10-20 21:53:37.000000000 -0400
@@ -56,11 +56,11 @@
 # in the URL.  These are recorded below, along with their corresponding
 # field names.
 
-@FIELDS = qw/symbol name last date time net p_change volume bid ask
-             close open day_range year_range eps pe div_date div div_yield
-	     cap ex_div avg_vol currency/;
+@FIELDS = qw/symbol name last net p_change volume bid ask
+             close open day_range year_range eps pe div div_yield
+	     cap avg_vol currency time date div_date ex_div/;
 
-@FIELD_ENCODING = qw/s n l1 d1 t1 c1 p2 v b a p o m w e r r1 d y j1 q a2 c4/;
+@FIELD_ENCODING = qw/s n l1 c1 p2 v b a p o m w e r d y j1 a2 c4 t1 d1 r1 q/;
 
 # This returns a list of labels that are provided, so that code
 # that make use of this module can know what it's dealing with.
diff -u old/Europe.pm new/Europe.pm
--- old/Europe.pm	2008-10-20 21:46:07.000000000 -0400
+++ new/Europe.pm	2008-10-20 21:43:47.000000000 -0400
@@ -50,9 +50,9 @@
 # Another solution might be to change Base.pm FIELDS labels to this
 # string + div_date and ex_div. Code would be nicier, but this will
 # need more testing for other yahoo modules and can be done later.
-our @YH_EUROPE_FIELDS = qw/symbol name last time date net p_change volume bid ask
+our @YH_EUROPE_FIELDS = qw/symbol name last net p_change volume bid ask
                            close open day_range year_range eps pe div div_yield
-                           cap avg_vol currency/;
+                           cap avg_vol currency time date/;
 
 sub methods {return (europe => \&yahoo_europe,yahoo_europe => \&yahoo_europe)};
 
