--- Util.pm.orig	2014-03-20 10:04:35.000000000 +0100
+++ Util.pm	2014-03-20 10:05:31.000000000 +0100
@@ -135,7 +135,8 @@
         $value =~ s/\.$//;
         return $value;
     } else {
-        my $value = sprintf('%.'.abs($ad).'f',$f);
+        my $format = '%.'.abs($ad).'f';
+        my $value = sprintf($format,$f);
         # Remove trailing zeros
         $value =~ s/(\.\d*?)0+$/$1/;
         $value =~ s/\.$//;
