--- stack.t.dist	Mon Mar  6 22:56:25 2006
+++ stack.t	Mon Nov  6 10:47:12 2006
@@ -4,6 +4,7 @@
 use lib 'lib';
 use Test::More tests => 29;
 use Devel::ebug;
+use File::Spec;
 
 my $ebug = Devel::ebug->new;
 $ebug->program("t/calc.pl");
@@ -18,10 +19,9 @@
 is(scalar(@trace), 1);
 
 # use YAML; warn Dump \@trace;
-
 my $trace = $trace[0];
 is($trace->package   , "main");
-is($trace->filename  , "t/calc.pl");
+is($trace->filename  , File::Spec->catfile('t','calc.pl'), "trace is on correct file name");
 is($trace->subroutine, "main::add");
 is($trace->wantarray , 0);
 is($trace->line      , 5);
