=== modified file 'Makefile.PL'
--- Makefile.PL	2018-01-08 20:07:01 +0000
+++ Makefile.PL	2018-01-11 22:47:53 +0000
@@ -80,7 +80,7 @@
     # -Dfoo                func()         $Config{key}
     HAS_GETADDRINFO  => [ "getaddrinfo",  "d_getaddrinfo" ],
     HAS_GETNAMEINFO  => [ "getnameinfo",  "d_getnameinfo" ],
-    HAS_GAI_STRERROR => [ "gai_strerror" ],
+    HAS_GAI_STRERROR => [ "gai_strerror", "d_gai_strerror" ],
     HAS_INET_ATON    => [ "inet_aton",    "d_inetaton" ],
     HAS_INETNTOP     => [ "inet_ntop",    "d_inetntop" ],
     HAS_INETPTON     => [ "inet_pton",    "d_inetpton" ],

=== modified file 'Socket.xs'
--- Socket.xs	2018-01-08 22:43:12 +0000
+++ Socket.xs	2018-01-11 22:39:17 +0000
@@ -826,8 +826,9 @@
 	sun_ad.sun_family = AF_UNIX;
 	pathname_pv = SvPV(pathname,len);
 	if (len > sizeof(sun_ad.sun_path)) {
-	    warn("Path length (%d) is longer than maximum supported length"
-	         " (%d) and will be truncated", len, sizeof(sun_ad.sun_path));
+	    warn("Path length (%" UVuf ") is longer than maximum supported length"
+	         " (%" UVuf ") and will be truncated",
+	         (UV)len, (UV)sizeof(sun_ad.sun_path));
 	    len = sizeof(sun_ad.sun_path);
 	}
 #  ifdef OS2	/* Name should start with \socket\ and contain backslashes! */

