From f38b4d739d831e3f1b1794708547fe13320f9460 Mon Sep 17 00:00:00 2001
From: Marius Gavrilescu <marius@ieval.ro>
Date: Fri, 24 Apr 2015 14:48:43 +0300
Subject: [PATCH] POD: Image::Magic -> Image::Magick

---
 lib/SVG.pm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/SVG.pm b/lib/SVG.pm
index d28e1de..c67a956 100644
--- a/lib/SVG.pm
+++ b/lib/SVG.pm
@@ -91,7 +91,7 @@ L<SVG::Extension>
 
 =head2 Converting SVG to PNG and other raster image formats
 
-The B<convert> command of L<http://www.imagemagick.org/> (also via L<Image::Magic> ) can convert SVG files to PNG
+The B<convert> command of L<http://www.imagemagick.org/> (also via L<Image::Magick> ) can convert SVG files to PNG
 and other formats.
 
 L<Image::LibRSVG> can convert SVG to other format.
-- 
2.1.4

