=== modified file 'lib/Socket/GetAddrInfo/Emul.pm'
--- lib/Socket/GetAddrInfo/Emul.pm	2011-02-24 17:50:39 +0000
+++ lib/Socket/GetAddrInfo/Emul.pm	2011-04-19 18:30:37 +0000
@@ -270,7 +270,7 @@
       $service = "$port";
    }
    else {
-      my $protname = $flag_dgram ? "udp" : "";
+      my $protname = $flag_dgram ? "udp" : "tcp";
       $service = getservbyport( $port, $protname );
       if( !defined $service ) {
          $service = "$port";

