diff -urN Data-Serializer-0.59.orig//t/serializer-testlib Data-Serializer-0.59/t/serializer-testlib
--- Data-Serializer-0.59.orig//t/serializer-testlib	2011-06-16 22:37:04.000000000 +0200
+++ Data-Serializer-0.59/t/serializer-testlib	2011-08-05 20:06:01.789100294 +0200
@@ -303,7 +303,7 @@
                 'compressppmd'    => [qw (Compress::PPMd)],
                 'encoding'       => [qw (MIME::Base64)],
                 'md5'       => [qw (Crypt::CBC Crypt::Blowfish Digest::MD5)],
-                'sha1'     	 => [qw (Crypt::CBC Crypt::Blowfish Digest::SHA1)],
+                'sha1'     	 => [qw (Crypt::CBC Crypt::Blowfish Digest::SHA)],
                 'sha-256'     	 => [qw (Crypt::CBC Crypt::Blowfish Digest::SHA)],
                 );
 
