diff --git i/t/entry.t w/t/entry.t
index 465a455..bafa29d 100755
--- i/t/entry.t
+++ w/t/entry.t
@@ -89,8 +89,10 @@ $mw->option("add", "*Entry.font", "Helvetica -12");
 my $e = $mw->Entry(qw(-bd 2 -relief sunken))->pack;
 $mw->update;
 
-if (!$Xft) { # XXX Is this condition necessary?
+#if (!$Xft)
+{ # XXX Is this condition necessary?
     my %fa = $mw->fontActual($e->cget(-font));
+    $fa{-family} = lc $fa{-family} if $fa{-family}; # normalize
     my %expected = (
 		    "-weight" => "normal",
 		    "-underline" => 0,
@@ -105,6 +107,8 @@ if (!$Xft) { # XXX Is this condition necessary?
 	    last;
 	}
     }
+require Data::Dumper; diag(Data::Dumper->new([\%fa, \%expected, $do_font_test],[qw()])->Indent(1)->Useqq(1)->Dump); # XXX
+
 }
 
 my $i;
