diff -ru Net-DNS-1.06-orig/lib/Net/DNS/RR/TXT.pm Net-DNS-1.06/lib/Net/DNS/RR/TXT.pm
--- Net-DNS-1.06-orig/lib/Net/DNS/RR/TXT.pm	2016-05-27 17:04:36.000000000 +0200
+++ Net-DNS-1.06/lib/Net/DNS/RR/TXT.pm	2016-12-05 13:29:41.000000000 +0100
@@ -73,7 +73,7 @@
 
 	return ( map $_->value, @$txtdata ) if wantarray;
 
-	join ' ', map $_->value, @$txtdata if defined wantarray;
+	join '', map $_->value, @$txtdata if defined wantarray;
 }
 
 
diff -ru Net-DNS-1.06-orig/t/07-zonefile.t Net-DNS-1.06/t/07-zonefile.t
--- Net-DNS-1.06-orig/t/07-zonefile.t	2016-05-27 17:04:36.000000000 +0200
+++ Net-DNS-1.06/t/07-zonefile.t	2016-12-05 13:32:22.000000000 +0100
@@ -331,7 +331,7 @@
 	is( $zonefile->read->name, 'hosta.example', 'name of simple RR propagated from previous RR' );
 	my $multilineRR = $zonefile->read;
 	is( $multilineRR->name,	   'hosta.example',		'name of multiline RR propagated from previous RR' );
-	is( $multilineRR->txtdata, 'multiline resource record', 'multiline RR correctly reassembled' );
+	is( $multilineRR->txtdata, 'multilineresourcerecord', 'multiline RR correctly reassembled' );
 	my $following = $zonefile->read;
 	is( $following->name,	      'hosta.example', 'name of following RR as expected' );
 	is( $following->txtdata,      'string',	       'superfluous brackets ignored' );
