diff --git a/lib/XML/Compile/SOAP11/Operation.pm b/lib/XML/Compile/SOAP11/Operation.pm
index f236142..05196d7 100644
--- a/lib/XML/Compile/SOAP11/Operation.pm
+++ b/lib/XML/Compile/SOAP11/Operation.pm
@@ -322,15 +322,16 @@ sub explain($$$@)
 
   BODY_PART:
     foreach my $part ( @{$def->{body}{parts} || []} )
     {   my $name = $part->{name};
         my ($kind, $value) = $part->{type} ? (type => $part->{type})
           : (element => $part->{element});
 
-        my $type = $schema->prefixed($value) || $value;
+        my $type = $schema->prefixFor($value) ?
+	  $schema->prefixed($value) : $value ;
         push @main, ''
           , "# Body part '$name' is $kind $type"
           , ($kind eq 'type' && $recurse ? "# See fake element '$name'" : ())
           , "my \$$name = {};";
         push @struct, "    $name => \$$name,";
 
         $recurse or next BODY_PART;
