Description: Fix typo "interfece" -> "interface"
Origin: vendor
Bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=538116
Forwarded: no
Author: Jonathan Yu <frequency@cpan.org>
--- a/lib/WebService/YouTube.pm
+++ b/lib/WebService/YouTube.pm
@@ -34,7 +34,7 @@
 
 =head1 NAME
 
-WebService::YouTube - Perl interfece to YouTube
+WebService::YouTube - Perl interface to YouTube
 
 =head1 VERSION
 
--- a/lib/WebService/YouTube/Feeds.pm
+++ b/lib/WebService/YouTube/Feeds.pm
@@ -172,7 +172,7 @@
 
 =head1 NAME
 
-WebService::YouTube::Feeds - Perl interfece to YouTube RSS Feeds
+WebService::YouTube::Feeds - Perl interface to YouTube RSS Feeds
 
 =head1 VERSION
 
--- a/lib/WebService/YouTube/Videos.pm
+++ b/lib/WebService/YouTube/Videos.pm
@@ -155,7 +155,7 @@
 
 =head1 NAME
 
-WebService::YouTube::Videos - Perl interfece to youtube.videos.*
+WebService::YouTube::Videos - Perl interface to youtube.videos.*
 
 =head1 VERSION
 
