From 8c217ba881b2e72b622aacacfcffab525125b6f3 Mon Sep 17 00:00:00 2001
From: Reini Urban <rurban@cpanel.net>
Date: Fri, 4 Apr 2014 12:22:17 -0500
Subject: [PATCH 3/5] flock only on supported platforms

see perldoc perlport
---
 Inline.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git Inline.pm Inline.pm
index dc0d504..760f4f0 100644
--- Inline.pm
+++ Inline.pm
@@ -859,7 +859,7 @@ sub create_config_file {
 
     my $file = File::Spec->catfile($ARGV[0], $configuration_file);
     open CONFIG, "> $file" or croak M24_open_for_output_failed($file);
-    flock(CONFIG, LOCK_EX);
+    flock(CONFIG, LOCK_EX) if $^O !~ /^VMS|riscos|VOS$/;
     print CONFIG Inline::denter->new()
       ->indent(*version => $Inline::VERSION,
 	       *languages => \%languages,
@@ -867,7 +867,7 @@ sub create_config_file {
 	       *modules => \%modules,
 	       *suffixes => \%suffixes,
 	      );
-    flock(CONFIG, LOCK_UN);
+    flock(CONFIG, LOCK_UN) if $^O !~ /^VMS|riscos|VOS$/;
     close CONFIG;
     exit 0;
 }
-- 
1.9.1

