--- a/lib/Proc/Daemon.pod
+++ b/lib/Proc/Daemon.pod
@@ -132,21 +132,21 @@
 =item child_STDIN
 
 Defines the path to STDIN of your daemon. Defaults to C</dev/null>. Default
-Mode is '<' (read). You can set your prefered Mode as if you use Perls C<open>
+Mode is '<' (read). You can set your preferred Mode as if you use Perls C<open>
 with a two-argument form.
 
 
 =item child_STDOUT
 
 Defines the path where the output of your daemon will go. Defaults to
-C</dev/null>. Default Mode is '+>' (write/read). You can set your prefered Mode
+C</dev/null>. Default Mode is '+>' (write/read). You can set your preferred Mode
 as if you use Perls C<open> with a two-argument form.
 
 
 =item child_STDERR
 
 Defines the path where the error output of your daemon will go. Defaults to
-C</dev/null>. Default Mode is '+>' (write/read). You can set your prefered Mode
+C</dev/null>. Default Mode is '+>' (write/read). You can set your preferred Mode
 as if you use Perls C<open> with a two-argument form.
 
 
@@ -397,7 +397,7 @@
 This module requires the C<POSIX> module to be installed.
 
 The C<Proc::ProcessTable> module is not essentially required but it can be
-usefull if it is installed (see above).
+useful if it is installed (see above).
 
 
 
