diff -ur Nagios-Object-0.08.orig/lib/Nagios/Object/Config.pm Nagios-Object-0.08/lib/Nagios/Object/Config.pm
--- Nagios-Object-0.08.orig/lib/Nagios/Object/Config.pm 2006-01-12 17:48:19.000000000 +1300
+++ Nagios-Object-0.08/lib/Nagios/Object/Config.pm      2006-08-09 14:27:41.000000000 +1200
@@ -122,6 +122,7 @@
 # TODO: add checks for undefined values where prohibited in %nagios_setup
 sub parse {
     my( $self, $filename ) = @_;
+    carp "$filename doesn't exist, ignoring" unless ( $filename && -f $filename );
     croak "cannot read file '$filename': $!" unless ( $filename && -r $filename );

     $Nagios::Object::pre_link = 1;
