--- Manager.pm.orig	2006-07-28 22:18:38.000000000 +0300
+++ Manager.pm	2006-07-28 22:19:42.000000000 +0300
@@ -295,7 +295,7 @@
 
     my $replies;
     if(my $irt  = $head->get('in-reply-to'))
-    {   for($irt =~ m/\<([^>]*)\>/)
+    {   for($irt =~ m/\<\s*([^>]+)\s*\>/)
         {   my $msgid = $1;
             $msgid    =~ s/\s+//g;
             $replies  = $self->{MBTM_ids}{$msgid} || $self->createDummy($msgid);
@@ -304,7 +304,7 @@
 
     my @refs;
     if(my $refs = $head->get('references'))
-    {   while($refs =~ s/\<([^>]*)\>//s)
+    {   while($refs =~ s/\<\s*([^>]+)\s*\>//s)
         {   my $msgid = $1;
             $msgid    =~ s/\s//gs;
             push @refs, $self->{MBTM_ids}{$msgid} || $self->createDummy($msgid);
