--- lib/Module/CoreList.pm.orig	Sat Jul 10 18:02:04 2004
+++ lib/Module/CoreList.pm	Sat Jul 10 18:05:11 2004
@@ -2307,6 +2307,7 @@
         'Filter::Simple'        => '0.78', #./lib/Filter/Simple.pm
         'Filter::Util::Call'    => '1.0601', #./lib/Filter/Util/Call.pm
         'FindBin'               => '1.43', #./lib/FindBin.pm
+        'FindExt'               => '1.00', #./win32/FindExt.pm
         'Getopt::Long'          => '2.34', #./lib/Getopt/Long.pm
         'Getopt::Std'           => '1.04', #./lib/Getopt/Std.pm
         'Hash::Util'            => '0.05', #./lib/Hash/Util.pm
@@ -2653,6 +2654,7 @@
         'Filter::Simple' => 0.78, #Filter\Simple.pm
         'Filter::Util::Call' => 1.0601, #Filter\Util\Call.pm
         'FindBin' => 1.43,      #FindBin.pm
+        'FindExt' => '1.00',    #FindExt.pm
         'Getopt::Long' => 2.34, #Getopt\Long.pm
         'Getopt::Std' => 1.04,  #Getopt\Std.pm
         'Hash::Util' => 0.05,   #Hash\Util.pm
@@ -2993,6 +2995,7 @@
         'Filter::Simple'        => '0.78',  #lib/Filter/Simple.pm
         'Filter::Util::Call'    => '1.0601',  #lib/Filter/Util/Call.pm
         'FindBin'               => '1.43',  #lib/FindBin.pm
+        'FindExt'               => '1.00',  #win32/FindExt.pm
         'Getopt::Long'          => '2.34',  #lib/Getopt/Long.pm
         'Getopt::Std'           => '1.04',  #lib/Getopt/Std.pm
         'Hash::Util'            => '0.05',  #lib/Hash/Util.pm
