--- a/lib/Config/Std.pm	2011-04-13 01:16:53.000000000 +0000
+++ b/lib/Config/Std.pm	2013-08-23 09:57:00.839769846 +0000
@@ -218,7 +218,7 @@
         return unless $is_first{ident $self};
 
         my $first = 1;
-        for my $key ( grep {!$updated_ref->{$_}} keys %{$hash_ref}) {
+        for my $key ( sort grep {!$updated_ref->{$_}} keys %{$hash_ref}) {
             my $value = $hash_ref->{$key};
             my $separate = ref $value || $value =~ m/\n./xms;
             $self->ensure_gap() if ($first ? $post_gap : $inter_gap)
@@ -323,7 +323,7 @@
             my $block = Config::Std::Block->new({name=>$block_name});
             my $subhash = $hash_ref->{$block_name};
             my $first = 1;
-            for my $key ( keys %{$subhash} ) {
+            for my $key ( sort keys %{$subhash} ) {
                 if (!defined $subhash->{$key}) {
                     croak "Can't save undefined value for key {'$block_name'}{'$key'} (only scalars or array refs)";
                 }
--- a/t/00write.t	2011-04-13 01:16:53.000000000 +0000
+++ b/t/00write.t	2013-08-23 09:56:20.102550718 +0000
@@ -42,8 +42,8 @@
 
 __DATA__
 
-def 3: def val 3
 def 2: def val 2
+def 3: def val 3
 
 def 3 ml: def val 3
         : across several
