Description: fix a spelling mistake in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-05-06

--- a/lib/SQL/Statement/Functions.pm
+++ b/lib/SQL/Statement/Functions.pm
@@ -479,7 +479,7 @@
 =head3 DECODE
 
  # purpose   : compare the first argument against
- #             succeding arguments at position 1 + 2N
+ #             succeeding arguments at position 1 + 2N
  #             (N = 0 to (# of arguments - 2)/2), and if equal,
  #				return the value of the argument at 1 + 2N + 1; if no
  #             arguments are equal, the last argument value is returned
