--- a/lib/Net/EPP/Simple.pm
+++ b/lib/Net/EPP/Simple.pm
@@ -1273,7 +1273,7 @@
 	my $hello = Net::EPP::Frame::Hello->new;
 	my $response = $self->request($hello);
 
-	return (UNIVERSAL::isa($response, 'XML::LibXML::Document') ? 1 : undef);
+    return ($response->isa('XML::LibXML::Document') ? 1 : undef);
 }
 
 sub _request {
@@ -1330,10 +1330,10 @@
 	$Error		= '';
 	$Message	= '';
 
-	$frame->clTRID->appendText(sha1_hex(ref($self).time().$$)) if (UNIVERSAL::isa($frame, 'Net::EPP::Frame::Command'));
+	$frame->clTRID->appendText(sha1_hex(ref($self).time().$$)) if ($frame->isa('Net::EPP::Frame::Command'));
 
 	$self->debug(sprintf('sending a %s to the server', ref($frame)));
-	if (UNIVERSAL::isa($frame, 'XML::LibXML::Document')) {
+	if ($frame->isa('XML::LibXML::Document')) {
 		map { $self->debug('C: '.$_) } split(/\n/, $frame->toString(1));
 
 	} else {
@@ -1343,7 +1343,7 @@
 
 	my $response = $self->SUPER::request($frame);
 
-	map { $self->debug('S: '.$_) } split(/\n/, $response->toString(1)) if (UNIVERSAL::isa($response, 'XML::LibXML::Document'));
+	map { $self->debug('S: '.$_) } split(/\n/, $response->toString(1)) if ($response->isa('XML::LibXML::Document'));
 
 	return $response;
 }
