Index: POE-Component-Server-Syslog/lib/POE/Filter/Syslog.pm
===================================================================
RCS file: /cvsroot/upstream/POE-Component-Server-Syslog/lib/POE/Filter/Syslog.pm,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 Syslog.pm
--- POE-Component-Server-Syslog/lib/POE/Filter/Syslog.pm	13 Mar 2005 21:08:42 -0000	1.1.1.1
+++ POE-Component-Server-Syslog/lib/POE/Filter/Syslog.pm	14 Nov 2005 17:40:12 -0000
@@ -42,7 +42,7 @@
 	my @found;
 	if($stream and length $stream) {
 
-		while ( $stream =~ s/$SYSLOG_REGEXP//x ) {
+		while ( $stream =~ s/$SYSLOG_REGEXP//sx ) {
 			my $time = $2 && parsedate("$2 $3 $4:$5:$6");
 			$time ||= time();
 
@@ -64,7 +64,7 @@
 	my $self = shift;
 	my $found = 0;
 	if($self->{buffer} and length $self->{buffer}) {
-		if ( $self->{buffer} =~ s/$SYSLOG_REGEXP//x ) {
+		if ( $self->{buffer} =~ s/$SYSLOG_REGEXP//xs ) {
 			my $time = $2 && parsedate("$2 $3 $4:$5:$6");
 			my $msg  = {
 				time     => $time,
