--- Geo-OSM-MapFeatures-0.04/lib/Geo/OSM/MapFeatures/Feature/Value/NumWithUnit.pm	2009-04-17 09:37:29.000000000 +0100
+++ Geo-OSM-MapFeatures-0.04.new/lib/Geo/OSM/MapFeatures/Feature/Value/NumWithUnit.pm	2009-06-13 20:49:56.858979949 +0100
@@ -41,7 +41,7 @@
     my $value = shift;
     my %classargs = %{ shift() };
 
-    $self->units($classargs{list});
+    $self->units($classargs{units});
 
     return $self;
 }
