From 499d04560d67ad7c93654557b6e30f8163a077d0 Mon Sep 17 00:00:00 2001
From: Marius Gavrilescu <marius@ieval.ro>
Date: Fri, 24 Apr 2015 14:40:51 +0300
Subject: [PATCH] Fix insertSibling{After,Before} methods

---
 lib/SVG/DOM.pm  | 4 ++--
 t/16-siblings.t | 4 +++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/SVG/DOM.pm b/lib/SVG/DOM.pm
index 9a59c55..d87984e 100644
--- a/lib/SVG/DOM.pm
+++ b/lib/SVG/DOM.pm
@@ -432,7 +432,7 @@ sub insertAfter {
 # sub insertSiblingAfter (Not in W3C DOM)
 sub insertSiblingAfter {
     my ( $self, $newChild ) = @_;
-    return $self->parent->insertAfter( $newChild, $self ) if $self->parent;
+    return $self->getParent->insertAfter( $newChild, $self ) if $self->getParent;
     return 0;
 }
 
@@ -440,7 +440,7 @@ sub insertSiblingAfter {
 # sub insertSiblingBefore (Not in W3C DOM)
 sub insertSiblingBefore {
     my ( $self, $newChild ) = @_;
-    return $self->parent->insertBefore( $newChild, $self ) if $self->parent;
+    return $self->getParent->insertBefore( $newChild, $self ) if $self->getParent;
     return 0;
 }
 
diff --git a/t/16-siblings.t b/t/16-siblings.t
index 9e7b748..1cc31cc 100755
--- a/t/16-siblings.t
+++ b/t/16-siblings.t
@@ -1,7 +1,7 @@
 use strict;
 use warnings;
 
-use Test::More tests => 3;
+use Test::More tests => 4;
 use SVG;
 
 # test: getFirstChild, getLastChild, getParent, getChildren
@@ -15,3 +15,5 @@ ok( $child1->hasSiblings(), "hasSiblings" );
 is( $child1->getNextSibling(),     $child2, "getNextSibling" );
 is( $child2->getPreviousSibling(), $child1, "getPreviousSibling" );
 
+$child2->insertSiblingAfter($child1);
+is( $child2->getNextSibling(), $child1, "insertSiblingAfter" );
-- 
2.1.4

