diff --git a/lib/XML/SAX/Machine.pm b/lib/XML/SAX/Machine.pm
index e50ea75..1917676 100644
--- a/lib/XML/SAX/Machine.pm
+++ b/lib/XML/SAX/Machine.pm
@@ -1468,13 +1468,8 @@ sub new {
     my $proto = shift;
     my $class = ref $proto || $proto;
 
-    my $self = do {
-        no strict 'refs';
-        bless [ \%{ "${class}::FIELDS" } ];
-    };
-
-    ## Pseudo hashes are badly borken in older Perls.  Should use something
-    ## else, really.
+    my $self = bless {}, $class;
+    
     my %options = @_ ;
     $self->{$_} = $options{$_} for keys %options;
 
