Description: fix some spelling mistaked
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2011-07-09

--- a/htdocs/Mason.html
+++ b/htdocs/Mason.html
@@ -65,7 +65,7 @@
 web-pages, while smaller components typically return HTML snippets for
 embedding in top-level components. This object-like architecture
 greatly simplifies site maintenance: change a shared component, and
-you instantly changed all dependant pages that refer to it across a
+you instantly changed all dependent pages that refer to it across a
 site (or across many virtual sites).</p>
 <p>Mason's component syntax lets designers separate a web page into
 programmatic and design elements. This means the esoteric Perl bits
--- a/lib/HTML/Mason.pm
+++ b/lib/HTML/Mason.pm
@@ -49,7 +49,7 @@
 web-pages, while smaller components typically return HTML snippets for
 embedding in top-level components. This object-like architecture
 greatly simplifies site maintenance: change a shared component, and
-you instantly changed all dependant pages that refer to it across a
+you instantly changed all dependent pages that refer to it across a
 site (or across many virtual sites).
 
 Mason's component syntax lets designers separate a web page into
--- a/htdocs/Admin.html
+++ b/htdocs/Admin.html
@@ -772,7 +772,7 @@
 
 <dd>
 <p>By default, <code>Cache::FileCache</code> is the subclass used for data caching,
-although this may be overriden by the developer. <code>Cache::FileCache</code>
+although this may be overridden by the developer. <code>Cache::FileCache</code>
 creates a separate subdirectory for every component that uses caching,
 and one file some number of levels underneath that subdirectory for
 each cached item.  The root of the cache tree is
--- a/htdocs/Params.html
+++ b/htdocs/Params.html
@@ -567,7 +567,7 @@
 <pre>
     data_cache_api      =&gt; 'CHI',
     data_cache_defaults =&gt; {driver =&gt; 'FastMmap'},</pre>
-<p>These settings are overriden by options given to particular
+<p>These settings are overridden by options given to particular
 <code>$m-&gt;cache</code> calls.</p>
 <p>
 </p>
--- a/htdocs/Request.html
+++ b/htdocs/Request.html
@@ -128,7 +128,7 @@
 <pre>
     data_cache_api      =&gt; 'CHI',
     data_cache_defaults =&gt; {driver =&gt; 'FastMmap'},</pre>
-<p>These settings are overriden by options given to particular
+<p>These settings are overridden by options given to particular
 <A HREF="#cache"><code>$m-&gt;cache</code></a> calls.</p>
 </dd>
 <dt><strong><A NAME="dhandler_name" class="item">dhandler_name</a></strong></dt>
--- a/lib/HTML/Mason/Admin.pod
+++ b/lib/HTML/Mason/Admin.pod
@@ -707,7 +707,7 @@
 =item Cache files
 
 By default, C<Cache::FileCache> is the subclass used for data caching,
-although this may be overriden by the developer. C<Cache::FileCache>
+although this may be overridden by the developer. C<Cache::FileCache>
 creates a separate subdirectory for every component that uses caching,
 and one file some number of levels underneath that subdirectory for
 each cached item.  The root of the cache tree is
--- a/lib/HTML/Mason/ApacheHandler.pm
+++ b/lib/HTML/Mason/ApacheHandler.pm
@@ -1037,7 +1037,7 @@
             }
 
             # Call $r->print (using the real Apache method, not our
-            # overriden method).
+            # overridden method).
             $r->$final_output_method( grep {defined} @_ );
             $r->rflush;
         };
--- a/lib/HTML/Mason/Params.pod
+++ b/lib/HTML/Mason/Params.pod
@@ -546,7 +546,7 @@
     data_cache_api      => 'CHI',
     data_cache_defaults => {driver => 'FastMmap'},
 
-These settings are overriden by options given to particular
+These settings are overridden by options given to particular
 C<$m-E<gt>cache> calls.
 
 
--- a/lib/HTML/Mason/Request.pm
+++ b/lib/HTML/Mason/Request.pm
@@ -1680,7 +1680,7 @@
     data_cache_api      => 'CHI',
     data_cache_defaults => {driver => 'FastMmap'},
 
-These settings are overriden by options given to particular
+These settings are overridden by options given to particular
 C<$m-E<gt>cache> calls.
 
 =item dhandler_name
