diff -urN POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber/J14.pm POE-Component-Jabber-3.00/lib/POE/Component/Jabber/J14.pm
--- POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber/J14.pm	2009-03-22 09:50:33.000000000 +0100
+++ POE-Component-Jabber-3.00/lib/POE/Component/Jabber/J14.pm	2011-05-07 19:47:35.930251676 +0200
@@ -8,7 +8,7 @@
 use POE::Filter::XML;
 use POE::Filter::XML::Node;
 use POE::Filter::XML::NS qw/ :JABBER :IQ /;
-use Digest::SHA1 qw/ sha1_hex /;
+use Digest::SHA qw/ sha1_hex /;
 
 use base('POE::Component::Jabber::Protocol');
 
diff -urN POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber/Legacy.pm POE-Component-Jabber-3.00/lib/POE/Component/Jabber/Legacy.pm
--- POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber/Legacy.pm	2009-03-22 09:50:33.000000000 +0100
+++ POE-Component-Jabber-3.00/lib/POE/Component/Jabber/Legacy.pm	2011-05-07 19:47:35.930251676 +0200
@@ -8,7 +8,7 @@
 use POE::Filter::XML;
 use POE::Filter::XML::Node;
 use POE::Filter::XML::NS qw/ :JABBER :IQ /;
-use Digest::SHA1 qw/ sha1_hex /;
+use Digest::SHA qw/ sha1_hex /;
 
 use base('POE::Component::Jabber::Protocol');
 
diff -urN POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber.pm POE-Component-Jabber-3.00/lib/POE/Component/Jabber.pm
--- POE-Component-Jabber-3.00.orig/lib/POE/Component/Jabber.pm	2009-03-22 09:50:33.000000000 +0100
+++ POE-Component-Jabber-3.00/lib/POE/Component/Jabber.pm	2011-05-07 19:47:35.930251676 +0200
@@ -410,7 +410,7 @@
     $self->[+_pcj_sid] = 0;
     $self->[+_pcj_pending] = {};
     $self->[+_pcj_init_finished] = 0;
-    $self->[+_pcj_id] ||= Digest::SHA1->new();
+    $self->[+_pcj_id] ||= Digest::SHA->new();
     $self->[+_pcj_id]->add(time().rand().$$.rand().$^T.rand());
     $self->[+_pcj_wheel] = undef;
     $self->[+_pcj_sfwheel] = undef;
