diff --recursive -U3 Devel-Cover-0.64/lib/Devel/Cover/Test.pm Devel-Cover-0.64-new/lib/Devel/Cover/Test.pm
--- Devel-Cover-0.64/lib/Devel/Cover/Test.pm	2008-04-10 12:57:39.000000000 +0100
+++ Devel-Cover-0.64-new/lib/Devel/Cover/Test.pm	2008-09-10 16:20:13.183516000 +0100
@@ -64,7 +64,7 @@
         close T or die "Cannot close $test: $!";
     }
 
-    $self->{select}         ||= "-select $self->{test}";
+    $self->{select}         ||= '-select '.$self->test_file;
     $self->{test_parameters}  = "$self->{select}"
                               . " -ignore blib Devel/Cover $self->{ignore}"
                               . " -merge 0 -coverage $self->{criteria} "
