*** Fedora.pm-orig	2015-11-09 17:22:50.781997994 -0500
--- Fedora.pm	2015-11-09 18:38:13.942997821 -0500
***************
*** 45,55 ****
--- 45,63 ----
  Requires:  perl(:MODULE_COMPAT_%(eval "`%{__perl} -V:version`"; echo $version))
  [% IF status.is_noarch %]BuildArch:  noarch[% END %]
  [% brs = buildreqs; FOREACH br = brs.keys.sort -%]
+ [% IF br == 'perl' %]
+ Requires: perl >= [% brs.$br %]
+ [% ELSE %]
  Requires: perl([% br %])[% IF (brs.$br != 0) %] >= [% brs.$br %][% END %]
+ [% END %]
  [% END -%]
  BuildRequires: perl(ExtUtils::MakeMaker)
  [% FOREACH br = brs.keys.sort -%]
+ [% IF br == 'perl' %]
+ BuildRequires: perl >= [% brs.$br %]
+ [% ELSE %]
  BuildRequires: perl([% br %])[% IF (brs.$br != 0) %] >= [% brs.$br %][% END %]
+ [% END %]
  [% END -%]
  
  
