--- USB-orig.pm	Mon Apr 16 10:52:54 2007
+++ USB.pm	Mon Apr 16 00:50:56 2007
@@ -622,6 +622,8 @@
     int i = 0;
     int retval = 0;
 
+    Inline_Stack_Vars;
+
     if(debugLevel)
     {
         printf( "libusb_control_msg( %#04x, %#04x, %#04x, %#04x, %p, %d, %d )\n",
@@ -634,8 +636,6 @@
     {
         printf( "\t => %d\n",retval );
     }
-
-    Inline_Stack_Vars;
 
     /* quiet compiler warnings. */
     (void)i;
