diff -r -u Parse-MediaWikiDump-0.31/lib/Parse/MediaWikiDump.pm Parse-MediaWikiDump-0.31.patch/lib/Parse/MediaWikiDump.pm
--- Parse-MediaWikiDump-0.31/lib/Parse/MediaWikiDump.pm	2006-01-10 20:43:18.000000000 +0100
+++ Parse-MediaWikiDump-0.31.patch/lib/Parse/MediaWikiDump.pm	2006-01-11 20:34:43.000000000 +0100
@@ -687,7 +687,7 @@
 		}
 
 		if ($ignore_ws_only) {
-			return 1 if $chars =~ m/^\s+$/m;
+			return 1 if $chars =~ m/^[ \t\r\n]+$/m;
 		}
 
 		push(@$buffer, [T_TEXT, \$chars]);
