Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-08-12

--- a/lib/File/BOM.pm
+++ b/lib/File/BOM.pm
@@ -234,7 +234,7 @@
 Opens my_file.txt for reading in an appropriate encoding found from the BOM in
 that file, or as a UTF-8 file if none is found.
 
-In the absense of a $default_mode argument, the following 2 calls should be equivalent:
+In the absence of a $default_mode argument, the following 2 calls should be equivalent:
 
     open_bom(FH, 'no_bom.txt');
 
