diff -ur ZMQ-LibZMQ4-0.01/Makefile.PL ZMQ-LibZMQ4-0.01.1/Makefile.PL
--- ZMQ-LibZMQ4-0.01/Makefile.PL	2014-09-19 18:02:12.000000000 +0200
+++ ZMQ-LibZMQ4-0.01.1/Makefile.PL	2017-09-11 17:26:49.621635138 +0200
@@ -4,7 +4,7 @@
 # Otherwise, do the usual.
 BEGIN {
     if (-e '../.git') {
-        do 'tools/check_mi_mods.pl';
+        do './tools/check_mi_mods.pl';
     } else {
         unshift @INC, 'inc';
         require inc::Module::Install;
@@ -17,7 +17,7 @@
 all_from 'lib/ZMQ/LibZMQ4.pm';
 
 # Detect libzmq, and display it for sanity
-do 'tools/detect_zmq.pl';
+do './tools/detect_zmq.pl';
 print "Detected the following ZMQ settings:\n";
 foreach my $env (qw(ZMQ_HOME ZMQ_H ZMQ_INCLUDES ZMQ_LIBS ZMQ_TRACE)) {
     printf " + %s = %s\n", $env, exists $ENV{$env} ? $ENV{$env} : "(null)";
@@ -192,7 +192,7 @@
 homepage "https://github.com/lestrrat/p5-ZMQ";
 bugtracker "https://github.com/lestrrat/p5-ZMQ/issues";
 
-do 'tools/genfiles.pl';
+do './tools/genfiles.pl';
 WriteAll;
 
 if ($^O eq 'darwin' && -f "Makefile") {
