--- /Users/chris/perl/System/Library/Perl/5.8.6/Pod/Usage.pm	2005-09-05 15:11:25.000000000 -0500
+++ lib/Pod/Usage.pm	2006-10-25 11:30:15.000000000 -0500
@@ -601,10 +601,14 @@
         $$self{USAGE_SKIPPING} = 1;
         my $heading = $$self{USAGE_HEAD1};
         $heading .= '/' . $$self{USAGE_HEAD2} if defined $$self{USAGE_HEAD2};
-        for (@{ $$self{USAGE_SELECT} }) {
-            if ($heading =~ /^$_\s*$/) {
-                $$self{USAGE_SKIPPING} = 0;
-                last;
+        if (!$$self{USAGE_SELECT} || !@{ $$self{USAGE_SELECT} }) {
+            $$self{USAGE_SKIPPING} = 0;
+        } else {
+            for (@{ $$self{USAGE_SELECT} }) {
+                if ($heading =~ /^$_\s*$/) {
+                    $$self{USAGE_SKIPPING} = 0;
+                    last;
+                }
             }
         }
 
