--- C:/Users/jose/CPAN/SQL-Translator-Parser-DBI-PostgreSQL/PostgreSQL.pm	Thu Nov 01 21:28:38 2012
+++ C:/Users/jose/CPAN/SQL-Translator-Parser-DBI-PostgreSQL/PostgreSQL_patched.pm	Thu Nov 01 21:28:57 2012
@@ -150,12 +150,16 @@
                      or !defined($$indexhash{'indkey'}) );
 
             my $type;
+            my @column_ids = split /\s+/, $$indexhash{'indkey'};
+
             if      ($$indexhash{'indisprimary'}) {
                 $type = UNIQUE; #PRIMARY_KEY;
 
                 #tell sqlt that this is the primary key:
-                my $col_name=$column_names[($$indexhash{'indkey'} - 1)];
-                $table->get_field($col_name)->{is_primary_key}=1;
+				foreach my $column_id (@column_ids) {
+                  my $col_name=$column_names[($column_id - 1)];
+                  $table->get_field($col_name)->{is_primary_key}=1;
+                }
 
             } elsif ($$indexhash{'indisunique'}) {
                 $type = UNIQUE;
@@ -163,8 +167,7 @@
                 $type = NORMAL;
             }
 
-
-            my @column_ids = split /\s+/, $$indexhash{'indkey'};
+            
             my @columns = split /\s+/, $$indexhash{'attname'};
 
             $table->add_index(
