Description: fix a spelling mistake in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-05-06

--- a/lib/Tree/Simple/Visitor/CreateDirectoryTree.pm
+++ b/lib/Tree/Simple/Visitor/CreateDirectoryTree.pm
@@ -91,7 +91,7 @@
   use Tree::Simple::Visitor::CreateDirectoryTree;
 
   # create a Tree::Simple object which
-  # represents a directory heirarchy
+  # represents a directory hierarchy
   my $tree = Tree::Simple->new("www/")
                     ->addChildren(
                         Tree::Simple->new("conf/")
