--- lib/XML/RAI/Object.pm.orig	2008-04-07 09:38:15.000000000 -0700
+++ lib/XML/RAI/Object.pm	2008-04-07 09:38:39.000000000 -0700
@@ -11,6 +11,7 @@
 
 use Date::Parse 2.26;
 use Date::Format 2.22;
+use Scalar::Util qw(weaken);
 
 sub new {
     my $class = shift;
@@ -26,6 +27,8 @@
     while ($_[0]->{__RAI}->can('parent')) {
         $_[0]->{__RAI} = $_[0]->{__RAI}->parent;
     }
+    weaken($_[0]->{__parent});
+    weaken($_[0]->{__RAI});
 }
 
 sub src    { $_[0]->{__source} }
