diff -Naur t/cart_item_create.t ../Handel-1.00003/t/cart_item_create.t
--- t/cart_item_create.t	2007-06-29 22:26:27.000000000 -0400
+++ ../Handel-1.00003/t/cart_item_create.t	2007-08-21 13:39:05.000000000 -0400
@@ -39,6 +39,8 @@
     ## test for Handel::Exception::Argument where first param is not a hashref
     {
         try {
+
+            local $ENV{'LANG'} = 'en';
             my $item = $subclass->create(sku => 'FOO');
 
             fail('no exception thrown');
diff -Naur t/cart_items.t ../Handel-1.00003/t/cart_items.t
--- t/cart_items.t	2007-06-29 22:26:27.000000000 -0400
+++ ../Handel-1.00003/t/cart_items.t	2007-08-21 13:39:05.000000000 -0400
@@ -96,6 +96,7 @@
             local $ENV{'HandelMaxQuantityAction'} = 'Exception';
 
             try {
+                local $ENV{'LANG'} = 'en';
                 $item2->quantity(6);
 
                 fail('no exception thrown');
diff -Naur t/checkout_plugins.t ../Handel-1.00003/t/checkout_plugins.t
--- t/checkout_plugins.t	2007-06-29 22:26:26.000000000 -0400
+++ ../Handel-1.00003/t/checkout_plugins.t	2007-08-21 13:39:05.000000000 -0400
@@ -38,6 +38,7 @@
         local $SIG{'__WARN__'} = sub {
             $warning = shift;
         };
+        local $ENV{'LANG'} = 'en';
         is(Handel::Checkout::Plugin->register, undef, 'register does nothing');
         like($warning, qr/plugin .* defined register/i, 'warning was set');
     };
diff -Naur t/compat_cart_item_new.t ../Handel-1.00003/t/compat_cart_item_new.t
--- t/compat_cart_item_new.t	2007-06-29 22:26:26.000000000 -0400
+++ ../Handel-1.00003/t/compat_cart_item_new.t	2007-08-21 13:39:05.000000000 -0400
@@ -18,11 +18,13 @@
     use_ok('Handel::Subclassing::CartItem');
     use_ok('Handel::Constraints', 'constraint_uuid');
     use_ok('Handel::Exception', ':try');
-
+    
+    local $ENV{'LANG'} = 'en';
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
     use_ok('Handel::Compat');
+
 };
 
 
diff -Naur t/compat_cart_items.t ../Handel-1.00003/t/compat_cart_items.t
--- t/compat_cart_items.t	2007-06-29 22:26:28.000000000 -0400
+++ ../Handel-1.00003/t/compat_cart_items.t	2007-08-21 13:39:05.000000000 -0400
@@ -20,6 +20,8 @@
     use_ok('Handel::Constants', qw(:cart :returnas));
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_cart_load.t ../Handel-1.00003/t/compat_cart_load.t
--- t/compat_cart_load.t	2007-06-29 22:26:27.000000000 -0400
+++ ../Handel-1.00003/t/compat_cart_load.t	2007-08-21 13:39:05.000000000 -0400
@@ -20,6 +20,8 @@
     use_ok('Handel::Constants', qw(:cart :returnas));
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_cart_new.t ../Handel-1.00003/t/compat_cart_new.t
--- t/compat_cart_new.t	2007-06-29 22:26:26.000000000 -0400
+++ ../Handel-1.00003/t/compat_cart_new.t	2007-08-21 13:39:05.000000000 -0400
@@ -21,6 +21,8 @@
     use_ok('Handel::Constraints', 'constraint_uuid');
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_order_item_new.t ../Handel-1.00003/t/compat_order_item_new.t
--- t/compat_order_item_new.t	2007-06-29 22:26:26.000000000 -0400
+++ ../Handel-1.00003/t/compat_order_item_new.t	2007-08-21 13:39:05.000000000 -0400
@@ -22,6 +22,8 @@
     use_ok('Handel::Constraints', 'constraint_uuid');
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_order_items.t ../Handel-1.00003/t/compat_order_items.t
--- t/compat_order_items.t	2007-06-29 22:26:20.000000000 -0400
+++ ../Handel-1.00003/t/compat_order_items.t	2007-08-21 13:39:05.000000000 -0400
@@ -20,6 +20,8 @@
     use_ok('Handel::Constants', qw(:order :returnas));
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_order_load.t ../Handel-1.00003/t/compat_order_load.t
--- t/compat_order_load.t	2007-06-29 22:26:27.000000000 -0400
+++ ../Handel-1.00003/t/compat_order_load.t	2007-08-21 13:39:05.000000000 -0400
@@ -20,6 +20,8 @@
     use_ok('Handel::Constants', qw(:order :returnas));
     use_ok('Handel::Exception', ':try');
 
+    local $ENV{'LANG'} = 'en';
+
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/compat_order_new.t ../Handel-1.00003/t/compat_order_new.t
--- t/compat_order_new.t	2007-06-29 22:26:27.000000000 -0400
+++ ../Handel-1.00003/t/compat_order_new.t	2007-08-21 13:39:05.000000000 -0400
@@ -55,6 +55,8 @@
 use_ok('Handel::Subclassing::OrderOnly');
 use_ok('Handel::Subclassing::Cart');
 
+local $ENV{'LANG'} = 'en';
+
 local $SIG{__WARN__} = sub {
     like(shift, qr/deprecated/);
 };
diff -Naur t/compat.t ../Handel-1.00003/t/compat.t
--- t/compat.t	2007-06-29 22:26:25.000000000 -0400
+++ ../Handel-1.00003/t/compat.t	2007-08-21 13:39:05.000000000 -0400
@@ -7,6 +7,7 @@
     use lib 't/lib';
     use Handel::Test tests => 15;
 
+    local $ENV{'LANG'} = 'en'; 
     local $SIG{__WARN__} = sub {
         like(shift, qr/deprecated/);
     };
diff -Naur t/storage_primary_columns.t ../Handel-1.00003/t/storage_primary_columns.t
--- t/storage_primary_columns.t	2007-06-29 22:26:20.000000000 -0400
+++ ../Handel-1.00003/t/storage_primary_columns.t	2007-08-21 13:39:05.000000000 -0400
@@ -29,6 +29,8 @@
 ## throw exception when primary column doesn't exists in columns
 {
     try {
+
+        local $ENV{'LANG'} = 'en';
         $storage->primary_columns(qw/bar quix/);
 
         fail('no exception thrown');
