--- PDF-Tiny-0.07/lib/PDF/Tiny.pm	2017-03-07 01:43:12.000000000 +0100
+++ PDF-Tiny-0.07/lib/PDF/Tiny.pm	2017-04-02 00:56:17.278564836 +0200
@@ -773,10 +773,12 @@ sub decode_stream :lvalue{
  my $self = shift;
  my $stream = $self->get_obj(@_);
  my @filters = $self->get_obj($stream, "/Filter");
- if ($filters[0][0] eq 'array') {
-  @filters = map $self->get_obj($filters[0],$_)->[1],0..$#{$filters[0][1]};
+ if (@filters) {
+   if ($filters[0][0] eq 'array') {
+    @filters = map $self->get_obj($filters[0],$_)->[1],0..$#{$filters[0][1]};
+   }
+   else { @filters = $filters[0][1] }
  }
- else { @filters = $filters[0][1] }
  my @params = $self->get_obj($stream, "/DecodeParms")
            || $self->get_obj($stream, "/DP"); # unofficial but Acrobat sup-
  if (@params) {                               # ports it
