*** Crypto.pm	Thu Aug 28 08:04:46 2008
--- Crypto.pm.patched	Thu Aug 28 08:04:33 2008
***************
*** 518,524 ****
          $lib_dir = $dir;
      }
      elsif ($self->is_win32()) {
!         if (-d ($dir = catdir($prefix_dir, 'out32')) and
              ($lib_file, $lib_name) = $self->probe_for_lib_file($dir))
          {
              $lib_dir = $dir;
--- 518,529 ----
          $lib_dir = $dir;
      }
      elsif ($self->is_win32()) {
!         if (-d ($dir = catdir($prefix_dir, 'lib/MinGW')) and  
!             ($lib_file, $lib_name) = $self->probe_for_lib_file($dir))    
!         {
!             $lib_dir = $dir;
!         }				
!         elsif (-d ($dir = catdir($prefix_dir, 'out32')) and
              ($lib_file, $lib_name) = $self->probe_for_lib_file($dir))
          {
              $lib_dir = $dir;
