Index: lib/Net/LDAP/Util.pm
===================================================================
--- lib/Net/LDAP/Util.pm	(revision 46796)
+++ lib/Net/LDAP/Util.pm	(working copy)
@@ -309,6 +309,8 @@
             #escape leading and trailing whitespace
             $val =~ s/(^\s+|\s+$)/
               "\\20" x length $1/xeg; 
+            #compact multiple spaces
+            $val =~ s/\s+/ /g;
           }
           
           # case fold attribute type and create return value
