--- Packet.pm.orig	2009-01-26 09:31:42.000000000 -0800
+++ Packet.pm	2009-04-08 15:08:39.000000000 -0700
@@ -597,8 +597,13 @@
 
 	while ( $offset < $packetlen ) {
 		unless ( my $length = unpack("\@$offset C", $$packet) ) {
+                    # See RFC 2181.
+                    if (!$name) {
+                        return('.', ++$offset);
+                    } else {
 			$name =~ s/\.$//o;
 			return ($name, ++$offset);
+                    }
 
 		} elsif ( ($length & 0xc0) == 0xc0 ) {		# pointer
 			my $point = 0x3fff & unpack("\@$offset n", $$packet);
