Description: Fix spelling errors in manpage

--- a/lib/IO/Socket/IP.pm
+++ b/lib/IO/Socket/IP.pm
@@ -261,7 +261,7 @@
 
 If neither C<Type> nor C<Proto> hints are provided, a default of
 C<SOCK_STREAM> and C<IPPROTO_TCP> respectively will be set, to maintain
-compatibilty with C<IO::Socket::INET>.
+compatibility with C<IO::Socket::INET>.
 
 If the constructor fails, it will set C<$@> to an appropriate error message;
 this may be from C<$!> or it may be some other string; not every failure
