--- PodcastFetch.pm.org	2009-01-11 09:02:28.000000000 +0100
+++ PodcastFetch.pm	2009-01-11 09:01:54.000000000 +0100
@@ -259,6 +259,7 @@
   my $self = bless {},ref $class || $class;
   $self->base($args{-base}       || '/tmp/podcasts');
   $self->subdir($args{-subdir});
+  $args{-rss} =~ s!\\!/!g if( $args{-rss} =~ m!^file://[A-Z]:!i );
   $self->rss($args{-rss}         || croak 'please provide -rss argument');
   $self->max($args{-max}                             );
   $self->timeout($args{-timeout} || 30               );
