diff -ruN Image-Info-1.20-beta/lib/Image/Info/TIFF.pm Image-Info-1.20/lib/Image/Info/TIFF.pm
--- Image-Info-1.20-beta/lib/Image/Info/TIFF.pm	2006-03-05 09:10:51.000000000 +0000
+++ Image-Info-1.20/lib/Image/Info/TIFF.pm	2006-03-11 21:39:28.000000000 +0000
@@ -247,8 +247,8 @@
 MAGIC: /^MM\x00\x2a/
 MAGIC: /^II\x2a\x00/
 
-The C<TIFF> spec can be found at (requires a login):
-L<http://partners.adobe.com/asn/developer/PDFS/TN/TIFF6.pdf>
+The C<TIFF> spec can be found at:
+L<http://partners.adobe.com/public/developer/tiff/>
 
 The EXIF spec can be found at:
 L<http://www.exif.org/>
