diff -u Class-Throwable-0.10/t/10_Class_Throwable_test.t Class-Throwable-0.10_01/t/10_Class_Throwable_test.t
--- Class-Throwable-0.10/t/10_Class_Throwable_test.t	2004-12-12 16:44:18.000000000 +0100
+++ Class-Throwable-0.10_01/t/10_Class_Throwable_test.t	2009-05-10 10:30:54.359375000 +0200
@@ -10,7 +10,7 @@
 }
 
 my $path_seperator = "/";
-$path_seperator = "\\" if $^O eq 'MSWin32';
+# MSWin32 perl works fine with "/"
 $path_seperator = ":"  if $^O eq 'MacOS';
 
 can_ok("Class::Throwable", 'throw');
diff -u Class-Throwable-0.10/t/20_Class_Throwable_subException_test.t Class-Throwable-0.10_01/t/20_Class_Throwable_subException_test.t
--- Class-Throwable-0.10/t/20_Class_Throwable_subException_test.t	2004-12-12 16:44:18.000000000 +0100
+++ Class-Throwable-0.10_01/t/20_Class_Throwable_subException_test.t	2009-05-10 10:30:51.125000000 +0200
@@ -10,7 +10,7 @@
 }
 
 my $path_seperator = "/";
-$path_seperator = "\\" if $^O eq 'MSWin32';
+# MSWin32 perl works fine with "/"
 $path_seperator = ":"  if $^O eq 'MacOS';
 
 eval {
diff -u Class-Throwable-0.10/t/35_Class_Throwable_sub_class_test.t Class-Throwable-0.10_01/t/35_Class_Throwable_sub_class_test.t
--- Class-Throwable-0.10/t/35_Class_Throwable_sub_class_test.t	2004-12-12 16:44:18.000000000 +0100
+++ Class-Throwable-0.10_01/t/35_Class_Throwable_sub_class_test.t	2009-05-10 10:30:47.375000000 +0200
@@ -12,7 +12,7 @@
 }
 
 my $path_seperator = "/";
-$path_seperator = "\\" if $^O eq 'MSWin32';
+# MSWin32 perl works fine with "/"
 $path_seperator = ":"  if $^O eq 'MacOS';
 
 eval {
