--- a/lib/Config/Model/Cookbook/CreateModelFromDoc.pod
+++ b/lib/Config/Model/Cookbook/CreateModelFromDoc.pod
@@ -2,7 +2,7 @@
 
 =head1 NAME
 
-Create a configuration model for application documentation
+Config::Model::Cookbook::CreateModelFromDoc - Create a configuration model for application documentation
 
 =head1 VERSION
 
@@ -273,4 +273,4 @@
 
 
 
-=head1 The resulting editor
\ No newline at end of file
+=head1 The resulting editor
--- a/lib/Config/Model/Manual/ModelCreationIntroduction.pod
+++ b/lib/Config/Model/Manual/ModelCreationIntroduction.pod
@@ -2,7 +2,7 @@
 
 =head1 NAME
 
-Introduction to model creation with Config::Model
+Config::Model::Manual::ModelCreationIntroduction - Introduction to model creation with Config::Model
 
 =head1 VERSION
 
@@ -1004,4 +1004,4 @@
 
 =head1 AUTHORS
 
-Dominique Dumont <ddumont at cpan.org>
\ No newline at end of file
+Dominique Dumont <ddumont at cpan.org>
