--- 05-steps.t.dist	2004-10-07 09:04:12.000000000 +0300
+++ 05-steps.t	2012-03-30 21:42:37.843750000 +0300
@@ -65,6 +65,8 @@
 	  [fin(sub { -f }, '.')],
 	  'all files even after messing with _ pseudo handle');
 
+SKIP: {
+skip 'user/nouser/etc. not supported on Windows', 8 if $^O eq 'MSWin32';
 is_deeply([File::Finder->user($<)->in(qw(.))],
 	  [fin(sub { -o }, '.')],
 	  'owned');
@@ -96,6 +98,7 @@
 is_deeply([File::Finder->not->nogroup->in(qw(.))],
 	  [fin(sub { defined getgrgid((stat)[5]) }, '.')],
 	  'not nogroup');
+}
 
 is_deeply([File::Finder->links('-2')->in(qw(.))],
 	  [fin(sub { (stat)[3] < 2 }, '.')],
