--- Makefile.PL.OLD	2015-04-10 21:17:48.000000000 +0200
+++ Makefile.PL	2015-06-10 17:34:24.740302300 +0200
@@ -2062,8 +2062,8 @@
 	my $t = $self->SUPER::install(@_);
 	my $n = $t =~ s[
 		(pure_\w+_install.*?)                       # 1
-		(INST_ARCHLIB\)\s+)\$\(DEST(\w+)\)(.*?)     # 2,3,4
-		(INST_BIN\)\s+)\$\(DEST(\w+)\)(.*?)         # 5,6,7
+		(INST_ARCHLIB\)["\s]+)\$\(DEST(\w+)\)(.*?)     # 2,3,4
+		(INST_BIN\)["\s]+)\$\(DEST(\w+)\)(.*?)         # 5,6,7
 		(.*?)                                       # 8
 		\n\n
 	][
