--- Multiplex.pm	2005-10-24 17:57:54.000000000 +0200
+++ Multiplex.pm.new	2006-05-15 18:19:20.415682270 +0200
@@ -148,7 +148,7 @@
 	# the above condition is wrong, because then _any_ prepare()
 	# will definitely go to second condition.
 
-	if  ($parent_handle->{'mx_master_id'} && &DBD::Multiplex::mx_is_modify_statement(\$statement)) {
+	if  ($parent_handle->{'mx_master_id'} && ( &DBD::Multiplex::mx_is_modify_statement(\$statement) or not $parent_handle->{AutoCommit} )) {
 		
 		# TK Note:
 		# Loop to find the master handle.
