Description: fix a small typo in mimeopen manpage
Bug-Debian: http://bugs.debian.org/728614
Author: Jonathan Dowland <jmtd@debian.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-11-03

--- a/mimeopen
+++ b/mimeopen
@@ -266,7 +266,7 @@
 default desktop application. If no default application is configured the
 user is prompted with an "open with" menu in the terminal.
 
-To use this script you need the freedestop mime-info database and the
+To use this script you need the freedesktop mime-info database and the
 freedesktop desktop-file-utils package. See L<File::MimeInfo::Applications(3)>
 for more details.
 
