Description: fix spelling mistakes (or unusual variants …)
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-10-29

--- a/lib/Image/ExifTool/WritePNG.pl
+++ b/lib/Image/ExifTool/WritePNG.pl
@@ -366,7 +366,7 @@
 Compress::Zlib is required to write compressed text.
 
 Existing text tags are always rewritten in their original form (compressed
-zTXt, uncompressed tEXt or internation iTXt), so pre-existing compressed
+zTXt, uncompressed tEXt or international iTXt), so pre-existing compressed
 information can only be modified if Compress::Zlib is available.
 
 Newly created textual information is written in uncompressed tEXt form by
--- a/html/TagNames/Minolta.html
+++ b/html/TagNames/Minolta.html
@@ -635,7 +635,7 @@
 models (and sometimes even between different firmware versions), so this
 information may not be as reliable as it should be.  Because of this, tags
 in the following tables are set to lower priority to prevent them from
-superceding the values of same-named tags in other locations when duplicate
+superseding the values of same-named tags in other locations when duplicate
 tags are disabled.</p>
 <blockquote>
 <table class=frame><tr><td>
--- a/lib/Image/ExifTool/Minolta.pm
+++ b/lib/Image/ExifTool/Minolta.pm
@@ -1014,7 +1014,7 @@
         models (and sometimes even between different firmware versions), so this
         information may not be as reliable as it should be.  Because of this, tags
         in the following tables are set to lower priority to prevent them from
-        superceding the values of same-named tags in other locations when duplicate
+        superseding the values of same-named tags in other locations when duplicate
         tags are disabled.
     },
     1 => {
--- a/lib/Image/ExifTool/TagNames.pod
+++ b/lib/Image/ExifTool/TagNames.pod
@@ -8943,7 +8943,7 @@
 models (and sometimes even between different firmware versions), so this
 information may not be as reliable as it should be.  Because of this, tags
 in the following tables are set to lower priority to prevent them from
-superceding the values of same-named tags in other locations when duplicate
+superseding the values of same-named tags in other locations when duplicate
 tags are disabled.
 
   Index4   Tag Name                             Writable
