Description: spelling fixes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-11-26

--- a/lib/Net/SIP/Endpoint.pm
+++ b/lib/Net/SIP/Endpoint.pm
@@ -134,7 +134,7 @@
 #   $callback: [ \&sub,@arg ] which will be called if the layer receives
 #     responses important to the upper layer (e.g 180 Ringing, 200 Ok,
 #     401/407 Authorization required...)
-#     if callback is ommitted the callback from the context is used,
+#     if callback is omitted the callback from the context is used,
 #     if callback is set it will be the new callback for the context
 #   $body: optional Body, either scalar or smth with method as_string
 #     (like Net::SIP::SDP)
--- a/lib/Net/SIP/Simple.pm
+++ b/lib/Net/SIP/Simple.pm
@@ -251,7 +251,7 @@
 ###########################################################################
 # mainloop
 # Args: (;$timeout,@stopvar)
-#  $timeout: timeout, undef for no timeout. argument can be ommitted
+#  $timeout: timeout, undef for no timeout. argument can be omitted
 #  @stopvar: @array of Scalar-REF, loop stops if one scalar is true
 # Returns: NONE
 ###########################################################################
--- a/lib/Net/SIP/Util.pod
+++ b/lib/Net/SIP/Util.pod
@@ -247,7 +247,7 @@
 If the first argument is a hash ref it will be treated as a hash ref as returned
 by B<ip_string2parts>. In this case also C<use_host> can be given which prefers
 C<host> to C<addr> for stringification. Also, C<default_port> can be used to
-give a port number which will be treated as default and ommitted from string.
+give a port number which will be treated as default and omitted from string.
 
 =item ip_sockaddr2parts(SOCKADDR, [FAMILY]) -> (IP, PORT, FAMILY)
 
