--- Readonly.old	2008-07-22 15:25:14.000000000 -0700
+++ Readonly.pm	2008-07-22 15:46:46.000000000 -0700
@@ -345,6 +345,12 @@
         croak "Readonly scalar must have only one value" if @_ > 2;
 
         my $tieobj = eval {tie ${$_[0]}, 'Readonly::Scalar', $_[1]};
+        if ( ref $_[1] eq 'HASH' ) {
+            Hash %{$_[1]}, %{$_[1]};
+        }
+        elsif ( ref $_[1] eq 'ARRAY' ) {
+            Array @{$_[1]}, @{$_[1]};
+        }
         # Tie may have failed because user tried to tie a constant, or we screwed up somehow.
         if ($@)
         {
