Description: fix a spelling mistake in the POD
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-12-04

--- a/lib/XML/Simple.pm
+++ b/lib/XML/Simple.pm
@@ -2221,7 +2221,7 @@
 
 =head2 ContentKey => 'keyname' I<# in+out - seldom used>
 
-When text content is parsed to a hash value, this option let's you specify a
+When text content is parsed to a hash value, this option lets you specify a
 name for the hash key to override the default 'content'.  So for example:
 
   XMLin('<opt one="1">Text</opt>', ContentKey => 'text')
