--- Parser.pm	2011-05-24 06:43:26.000000000 +0100
+++ Parser.pm	2012-05-24 13:01:22.000000000 +0100
@@ -433,16 +433,13 @@
     my $service = $tag->first_child('service[@name]');
     my $service_hashref;
     $service_hashref->{port} = $portid;
+    $service_hashref->{proto} = $tag->{att}->{protocol};	
 
     if ( defined $service ) {
         $service_hashref->{name}      = $service->{att}->{name} || 'unknown';
         $service_hashref->{version}   = $service->{att}->{version};
         $service_hashref->{product}   = $service->{att}->{product};
         $service_hashref->{extrainfo} = $service->{att}->{extrainfo};
-        $service_hashref->{proto} =
-             $service->{att}->{proto}
-          || $service->{att}->{protocol}
-          || 'unknown';
         $service_hashref->{rpcnum}      = $service->{att}->{rpcnum};
         $service_hashref->{tunnel}      = $service->{att}->{tunnel};
         $service_hashref->{method}      = $service->{att}->{method};
