Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-12-15

--- a/README
+++ b/README
@@ -75,7 +75,7 @@
 
     To make up for this, if you have DateTime installed, any DateTime::Tiny
     module can be inflated into the equivalent DateTime as needing, loading
-    DateTime on the fly if necesary.
+    DateTime on the fly if necessary.
 
     This is somewhat similar to DateTime::LazyInit, but unlike that module
     DateTime::Tiny is not modifiable.
--- a/lib/DateTime/Tiny.pm
+++ b/lib/DateTime/Tiny.pm
@@ -82,7 +82,7 @@
 
 To make up for this, B<if> you have L<DateTime> installed, any
 B<DateTime::Tiny> module can be inflated into the equivalent L<DateTime>
-as needing, loading L<DateTime> on the fly if necesary.
+as needing, loading L<DateTime> on the fly if necessary.
 
 This is somewhat similar to DateTime::LazyInit, but unlike that module
 B<DateTime::Tiny> is not modifiable.
