diff -Nur MIME-Fast-1.6-orig/cinclude/gmime-newfunc.c MIME-Fast-1.6/cinclude/gmime-newfunc.c
--- MIME-Fast-1.6-orig/cinclude/gmime-newfunc.c	2003-06-16 18:04:46.000000000 +0200
+++ MIME-Fast-1.6/cinclude/gmime-newfunc.c	2011-08-09 09:01:59.596859554 +0200
@@ -29,10 +29,10 @@
         		enc = g_mime_part_get_encoding(mime_part);
         		switch (enc) {
         		  case GMIME_PART_ENCODING_BASE64:
-        		    lsize = BASE64_ENCODE_LEN(lsize);
+        		    lsize = GMIME_BASE64_ENCODE_LEN(lsize);
         		    break;
         		  case GMIME_PART_ENCODING_QUOTEDPRINTABLE:
-        		    lsize = QP_ENCODE_LEN(lsize);
+        		    lsize = GMIME_QP_ENCODE_LEN(lsize);
         		    break;
         		}
         	    }
diff -Nur MIME-Fast-1.6-orig/gmime-stream-perlio.h MIME-Fast-1.6/gmime-stream-perlio.h
--- MIME-Fast-1.6-orig/gmime-stream-perlio.h	2004-12-16 21:28:51.000000000 +0100
+++ MIME-Fast-1.6/gmime-stream-perlio.h	2011-08-09 09:01:59.596859554 +0200
@@ -36,11 +36,11 @@
 #include <gmime/gmime-stream.h>
 
 #define GMIME_TYPE_STREAM_PERLIO            (g_mime_stream_perlio_get_type ())
-#define GMIME_STREAM_PERLIO(obj)            (GMIME_CHECK_CAST ((obj), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIO))
-#define GMIME_STREAM_PERLIO_CLASS(klass)    (GMIME_CHECK_CLASS_CAST ((klass), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIOClass))
-#define GMIME_IS_STREAM_PERLIO(obj)         (GMIME_CHECK_TYPE ((obj), GMIME_TYPE_STREAM_PERLIO))
-#define GMIME_IS_STREAM_PERLIO_CLASS(klass) (GMIME_CHECK_CLASS_TYPE ((klass), GMIME_TYPE_STREAM_PERLIO))
-#define GMIME_STREAM_PERLIO_GET_CLASS(obj)  (GMIME_CHECK_GET_CLASS ((obj), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIOClass))
+#define GMIME_STREAM_PERLIO(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIO))
+#define GMIME_STREAM_PERLIO_CLASS(klass)    (G_TYPE_CHECK_CLASS_CAST ((klass), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIOClass))
+#define GMIME_IS_STREAM_PERLIO(obj)         (G_TYPE_CHECK_INSTANCE_TYPE ((obj), GMIME_TYPE_STREAM_PERLIO))
+#define GMIME_IS_STREAM_PERLIO_CLASS(klass) (G_TYPE_CHECK_CLASS_TYPE ((klass), GMIME_TYPE_STREAM_PERLIO))
+#define GMIME_STREAM_PERLIO_GET_CLASS(obj)  (G_TYPE_INSTANCE_GET_CLASS ((obj), GMIME_TYPE_STREAM_PERLIO, GMimeStreamPerlIOClass))
 
 typedef struct _GMimeStreamPerlIO GMimeStreamPerlIO;
 typedef struct _GMimeStreamPerlIOClass GMimeStreamPerlIOClass;
diff -Nur MIME-Fast-1.6-orig/t/parser.t MIME-Fast-1.6/t/parser.t
--- MIME-Fast-1.6-orig/t/parser.t	2004-12-17 12:57:59.000000000 +0100
+++ MIME-Fast-1.6/t/parser.t	2011-08-09 09:02:28.616890870 +0200
@@ -140,6 +140,7 @@
 
 my $content = "Content-Type: text/html";
 $param->write_to_string(1, $content);
+$content =~ s/[\n]*$//s;
 
 is($content,'Content-Type: text/html; charset=iso8859-2', 'Param::write_to_string works');
 
