diff -urN sqlfairy-0.11007.orig//META.yml sqlfairy-0.11007/META.yml
--- sqlfairy-0.11007.orig//META.yml	2010-11-30 23:24:03.000000000 +0100
+++ sqlfairy-0.11007/META.yml	2011-05-06 14:51:40.297253964 +0200
@@ -242,7 +242,7 @@
   Class::Data::Inheritable: 0.02
   Class::MakeMethods: 0
   DBI: 0
-  Digest::SHA1: 2
+  Digest::SHA: 0
   File::ShareDir: 1
   File::Spec: 0
   IO::Dir: 0
diff -urN sqlfairy-0.11007.orig//Makefile.PL sqlfairy-0.11007/Makefile.PL
--- sqlfairy-0.11007.orig//Makefile.PL	2010-11-30 18:26:37.000000000 +0100
+++ sqlfairy-0.11007/Makefile.PL	2011-05-06 14:51:57.609253166 +0200
@@ -10,7 +10,7 @@
     'Class::Base'              => 0,
     'Class::Data::Inheritable' => 0.02,
     'Class::MakeMethods'       => 0,
-    'Digest::SHA1'             => 2.00,
+    'Digest::SHA'              => 0,
     'Carp::Clan'               => 0,
     'IO::Dir'                  => 0,
     'IO::Scalar'               => 2.110,
diff -urN sqlfairy-0.11007.orig//lib/SQL/Translator/Utils.pm sqlfairy-0.11007/lib/SQL/Translator/Utils.pm
--- sqlfairy-0.11007.orig//lib/SQL/Translator/Utils.pm	2010-11-30 18:26:37.000000000 +0100
+++ sqlfairy-0.11007/lib/SQL/Translator/Utils.pm	2011-05-06 14:52:39.025251318 +0200
@@ -21,7 +21,7 @@
 use strict;
 use base qw(Exporter);
 use vars qw($VERSION $DEFAULT_COMMENT @EXPORT_OK);
-use Digest::SHA1 qw( sha1_hex );
+use Digest::SHA qw( sha1_hex );
 use Exporter;
 
 $VERSION = '1.59';
