--- Image/Info/JPEG.pm.orig	2005-02-01 17:34:31.000000000 +0100
+++ Image/Info/JPEG.pm	2005-02-01 17:35:26.000000000 +0100
@@ -257,13 +257,14 @@
 	# information of this (thumbnail) image file...
 	if (my($ipos) = $info->get_info($i, "JPEGInterchangeFormat", 1)) {
 	    my($ilen) = $info->get_info($i, "JPEGInterchangeFormatLngth", 1);
-	    die unless $ilen;
+	    if ($ilen) {
 	    my $jdata = substr($data, $ipos, $ilen);
 	    #$info->push_info($i, "JPEGImage" => $jdata);
 
 	    with_io_string {
 		_process_file($info, $_, $i);
 	    } $jdata;
+ 	    }
 	}
 
 	# Turn XResolution/YResolution into 'resolution'
