From 0ea9ab23ba2ff5f3bf6b927487818c3880c6c28d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Thu, 8 Nov 2012 08:36:12 +0100
Subject: [PATCH 1/2] Correct s390(x) preprocessor tests

---
 CPU.xs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CPU.xs b/CPU.xs
index 0d7a9a3..9563830 100644
--- a/CPU.xs
+++ b/CPU.xs
@@ -269,7 +269,7 @@ int proc_cpuinfo_clock (void) {
     return (0);
 }
 
-#ifdef __s390__ || __s390x__
+#if defined __s390__ || defined __s390x__
 
 /* Return machine value from s390 processor line, NULL if not found */
 char *processor_machine_field (char *processor) {
@@ -372,7 +372,7 @@ CODE:
     char *value = NULL;
     int retcode = 0;
 #ifdef __linux__
-#ifdef __s390__ || __s390x__
+#if defined __s390__ || defined __s390x__
     value = processor_machine_field (proc_cpuinfo_field ("processor") );
 #endif
     if (!value) value = proc_cpuinfo_field ("model name");
-- 
1.7.11.7

