--- URI/_punycode.pm.orig	2013-06-09 10:14:14.000000000 +0400
+++ URI/_punycode.pm.new	2013-06-11 11:49:19.000000000 +0400
@@ -86,7 +86,11 @@
 	warn join " ", map sprintf('%04x', $_), @output if $DEBUG;
 	$i++;
     }
-    return join '', map chr, @output;
+    my $uri	= join '', map chr, @output;
+    use Encode;
+    my $octets     = encode('UTF-8', $uri, Encode::FB_CROAK);
+    $uri = decode('UTF-8', $octets,     Encode::FB_CROAK);
+    return $uri;
 }
 
 sub encode_punycode {
