diff --git a/lib/Domain/PublicSuffix.pm b/lib/Domain/PublicSuffix.pm
index 25d7f39..904c290 100755
--- a/lib/Domain/PublicSuffix.pm
+++ b/lib/Domain/PublicSuffix.pm
@@ -45,7 +45,7 @@ The Mozilla PublicSuffix file is an open source, fully documented format that
 shows absolute root TLDs, primarily for Mozilla's browser products to be able
 to determine how far a cookie's security boundaries go.
 
-This module will attempt to search etc directories in /usr, /usr/local, and
+This module will attempt to search etc directories in /usr/share/publicsuffix, /usr, /usr/local, and
 /opt/local for the effective_tld_names.dat file. If a file is not found, a
 default file is loaded from Domain::PublicSuffix::Default, which is current at
 the time of the module's release. You can override the data file path by
@@ -237,6 +237,7 @@ sub _parse_data_file {
 		
 	} else {
 		my @paths = (
+			File::Spec->catdir(File::Spec->rootdir, qw/ usr share publicsuffix /),
 			File::Spec->catdir(File::Spec->rootdir, qw/ etc /),
 			File::Spec->catdir(File::Spec->rootdir, qw/ usr etc /),
 			File::Spec->catdir(File::Spec->rootdir, qw/ usr local etc /),
@@ -409,4 +410,4 @@ under the same terms as Perl itself.
 
 =cut
 
-1;
\ No newline at end of file
+1;
