diff --git a/lib/Date/Manip/Zones.pm b/lib/Date/Manip/Zones.pm
index 707289f..433ca9f 100644
--- a/lib/Date/Manip/Zones.pm
+++ b/lib/Date/Manip/Zones.pm
@@ -1901,13 +1901,13 @@ $LastYear      = '2065';
                      'america/maceio',
                      'america/recife',
                      'america/belem' ],
-  'brt'         => [ 'america/araguaina',
+  'brt'         => [ 'america/sao_paulo',
+                     'america/araguaina',
                      'america/bahia',
-                     'america/belem',
                      'america/fortaleza',
                      'america/maceio',
                      'america/recife',
-                     'america/sao_paulo',
+                     'america/belem',
                      'america/santarem' ],
   'bst'         => [ 'europe/london',
                      'america/adak',
diff --git a/t/tz.define_abbrev.t b/t/tz.define_abbrev.t
index a38475b..4641c28 100755
--- a/t/tz.define_abbrev.t
+++ b/t/tz.define_abbrev.t
@@ -26,13 +26,13 @@ $obj->config("forcedate","now,America/New_York");
 $tests="
 
 BRT reset =>
+   America/Sao_Paulo
    America/Araguaina
    America/Bahia
-   America/Belem
    America/Fortaleza
    America/Maceio
    America/Recife
-   America/Sao_Paulo
+   America/Belem
    America/Santarem
 
 BRT
