Description: spelling fixes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-07-02

--- a/Org.pm
+++ b/Org.pm
@@ -381,7 +381,7 @@
 
 =item I<-Morphology>
 
-Provide morphology informations.
+Provide morphology information.
 
 Default: standard.
 
--- a/leo
+++ b/leo
@@ -307,7 +307,7 @@
 Translate a term from german to english or vice versa.
 
   -s, --spelltolerance=on|off                   allow spelling errors
-  -m, --morphology=none|forcedAll               provide morphology informations
+  -m, --morphology=none|forcedAll               provide morphology information
   -c, --chartolerance=fuzzy|exact               allow umlaut alternatives
   -l, --language=en|fr|de2(en|fr)|(en|fr)2de    translation direction
   -n, --noescapechars                           dont use escapes for highlighting
@@ -377,7 +377,7 @@
 
 =item I<-m --morphology>
 
-Provide morphology informations.
+Provide morphology information.
 
 Possible values: B<standard>, B<none> or B<forcedAll>.
 
@@ -548,7 +548,7 @@
  export http_proxy=http://max:34dwe2@172.16.120.120:3128
 
 As security is always important, I have to warn you, that
-other users on the same machine can read your enviroment
+other users on the same machine can read your environment
 using the 'ps -e ..' command, so this is not recommended.
 
 The most secure way for proxy authentication is just to
