--- /opt/local/lib/perl5/site_perl/5.8.8/Catalyst/DispatchType/Chained.pm.org	2009-05-14 11:54:08.000000000 +0900
+++ /opt/local/lib/perl5/site_perl/5.8.8/Catalyst/DispatchType/Chained.pm	2009-05-14 11:54:54.000000000 +0900
@@ -46,7 +46,7 @@
 
     my $column_width = Catalyst::Utils::term_width() - 35 - 9;
     my $paths = Text::SimpleTable->new(
-       [ 35, 'Path Spec' ], [ 36, 'Private' ], [ $column_width, 'Private' ]
+       [ 35, 'Path Spec' ], [ $column_width, 'Private' ]
     );
 
     ENDPOINT: foreach my $endpoint (
