commit 78d8f19e600bfc5c2250b1b818038315d884d21b
Author: Sébastien Aperghis-Tramoni <sebastien.aperghis-tramoni@diabolocom.com>
Date:   Fri Apr 19 19:30:33 2013 +0200

    new files

diff --git a/MANIFEST b/MANIFEST
index 4c38892..4bd55be 100644
--- a/MANIFEST
+++ b/MANIFEST
@@ -24,6 +24,9 @@ README
 t/00_compile.t
 t/01_basic.t
 t/02_method.t
+t/10_setup_pcap.t
 t/97_kwalitee.t
 t/98_pod-coverage.t
 t/99_pod.t
+t/samples/ping-ietf-20pk-be.dmp
+t/samples/ping-ietf-20pk-le.dmp
