--- 03compare_ok.t.dist	Mon Apr 30 19:28:36 2007
+++ 03compare_ok.t	Mon Apr 30 19:26:46 2007
@@ -6,11 +6,17 @@
 use Test::Files;
 
 my $test_file    = File::Spec->catfile( 't', '03compare_ok.t'   );
+$test_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $missing_file = File::Spec->catfile( 't', 'missing'          );
+$missing_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $pass_file    = File::Spec->catfile( 't', 'ok_pass.dat'      );
+$pass_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $absent_file  = File::Spec->catfile( 't', 'absent'           );
+$absent_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $same_file    = File::Spec->catfile( 't', 'ok_pass.same.dat' );
+$same_file=~s#\\#/#g if $^O eq 'MSWin32';
 my $diff_file    = File::Spec->catfile( 't', 'ok_pass.diff.dat' );
+$diff_file=~s#\\#/#g if $^O eq 'MSWin32';
 
 #-----------------------------------------------------------------
 # Compare two files with the same content.
