--- HTML-Form-6.03/lib/HTML/Form.pm	2012-03-30 05:04:43.000000000 -0600
+++ HTML-Form-6.03-rb-fix/lib/HTML/Form.pm	2012-06-04 16:36:30.148610000 -0600
@@ -1168,9 +1168,17 @@
     else {
 	$self->{option_disabled}++
 	    if $type eq "radio" && delete $self->{disabled};
-	$self->{menu} = [
-            {value => $value, name => $value_name},
-        ];
+
+	if ($type eq "radio") {
+	    $self->{menu} = [
+                {value => $value, id => delete $self->{id}},
+	    ];
+	}
+	else {
+	    $self->{menu} = [
+                {value => $value, name => $value_name},
+	    ];
+	}
 	my $checked = $self->{checked} || $self->{option_selected};
 	delete $self->{checked};
 	delete $self->{option_selected};
