--- a/Build.PL	(revision 41217)
+++ b/Build.PL	(working copy)
@@ -10,7 +10,7 @@
         'Data::Page'    => 0,
         'DBI'           => 0,
         'List::Util'    => 0,
-        'SQL::Abstract' => 1.17,
+        'SQL::Abstract' => 1.55,
         'DBD::SQLite'   => 1.08
     },
     dist_abstract      => 'lib/Class/DBI/Sweet.pm',
--- a/t/04join.t
+++ b/t/04join.t
@@ -29,7 +29,7 @@ my ($sql) = $cds[0]->_search({ 'artist.n
 
 is($sql->{from}, "cd me, artist artist", "FROM statement ok");
 
-is($sql->{where}, "( artist.name = ? ) AND me.artist = artist.artistid",
+is($sql->{where}, "artist.name = ? AND me.artist = artist.artistid",
     "WHERE clause ok");
 
 cmp_ok($cds[2]->year, '==', 2001, "Last CD returned correctly");
