--- Protocol.pm 2007-03-29 14:24:35.000000000 +0200
+++ Protocol_fix.pm     2010-09-24 16:56:53.234080033 +0200
@@ -3233,7 +3233,7 @@
     $self->{DEBUG}->Log1("callbackIQ: type($type) ns($ns)");
 
     if (exists($self->{CB}->{IQns}->{$ns})
-        && (ref($self->{CB}->{IQns}->{$ns}) != 'HASH' )
+        && (ref($self->{CB}->{IQns}->{$ns}) ne 'HASH' )
         )
     {
         $self->{DEBUG}->Log1("callbackIQ: goto user function( $self->{CB}->{IQns}->{$ns} )");
