Description: spelling fixes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2018-06-19
Forwarded: 
Bug: 

--- a/bin/findorule
+++ b/bin/findorule
@@ -90,7 +90,7 @@
 =head1 DESCRIPTION
 
 C<findorule> mostly borrows the interface from GNU find(1) to provide a
-command-line interface onto the File::Find::Object::Rule heirarchy of modules.
+command-line interface onto the File::Find::Object::Rule hierarchy of modules.
 
 The syntax for expressions is the rule name, preceded by a dash,
 followed by an optional argument.  If the argument is an opening
@@ -105,7 +105,7 @@
 
  find -file -name foo -bar
 
-files named C<foo>, that have pubs (for this is what our ficticious
+files named C<foo>, that have pubs (for this is what our fictitious
 C<bar> clause specifies), below the current directory.
 
  find -file -name ( -bar )
--- a/lib/File/Find/Object/Rule/Extending.pod
+++ b/lib/File/Find/Object/Rule/Extending.pod
@@ -35,7 +35,7 @@
 =head1 DESCRIPTION
 
 L<File::Find::Object::Rule> inherits L<File::Find::Rule>'s extensibility. It
-is now possibile to extend it, using the following conventions.
+is now possible to extend it, using the following conventions.
 
 =head2 Declare your package
 
--- a/t/findorule.t
+++ b/t/findorule.t
@@ -122,7 +122,7 @@
 
 
 {
-    local $TODO = "Win32 cmd.exe hurts my brane"
+    local $TODO = "Win32 cmd.exe hurts my brain"
       if ($^O =~ m/Win32/ || $^O eq 'dos');
 
     # TEST
--- a/t/sample-data/to-copy-from/findorule.txt
+++ b/t/sample-data/to-copy-from/findorule.txt
@@ -25,7 +25,7 @@
 
 
 {
-    local $TODO = "Win32 cmd.exe hurts my brane"
+    local $TODO = "Win32 cmd.exe hurts my brain"
       if ($^O =~ m/Win32/ || $^O eq 'dos');
 
     is_deeply(run 't -file -name \( foobar \*.t \)',
