commit af485314e6212bffc0bfca358d5a8943fce946e0
Author: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date:   Mon Dec 20 23:17:49 2010 -0500

    added digests from SHA2 family

diff --git a/lib/Crypt/X509.pm b/lib/Crypt/X509.pm
index ef229ab..86d521e 100644
--- a/lib/Crypt/X509.pm
+++ b/lib/Crypt/X509.pm
@@ -18,7 +18,11 @@ my %oid2enchash = (
 					'1.2.840.113549.1.1.3' => { 'enc' => 'RSA', 'hash' => 'MD4' },
 					'1.2.840.113549.1.1.4' => { 'enc' => 'RSA', 'hash' => 'MD5' },
 					'1.2.840.113549.1.1.5' => { 'enc' => 'RSA', 'hash' => 'SHA1' },
-					'1.2.840.113549.1.1.6' => { 'enc' => 'OAEP' }
+					'1.2.840.113549.1.1.6' => { 'enc' => 'OAEP' },
+					'1.2.840.113549.1.1.11' => { 'enc' => 'RSA', 'hash' => 'SHA256' },
+					'1.2.840.113549.1.1.12' => { 'enc' => 'RSA', 'hash' => 'SHA384' },
+					'1.2.840.113549.1.1.13' => { 'enc' => 'RSA', 'hash' => 'SHA512' },
+					'1.2.840.113549.1.1.14' => { 'enc' => 'RSA', 'hash' => 'SHA224' },
 );
 my %oid2attr = (
 				 "2.5.4.3"                    => "CN",
