diff --git Alias.xs Alias.xs
index f619da5..57afd50 100644
--- Alias.xs
+++ Alias.xs
@@ -25,13 +25,6 @@
 				PERL_SUBVERSION)
 #endif
 
-#if defined(USE_DTRACE) && defined(PERL_CORE)
-#undef ENTRY_PROBE
-#undef RETURN_PROBE
-#define ENTRY_PROBE(func, file, line, stash)
-#define RETURN_PROBE(func, file, line, stash)
-#endif
-
 #if defined(PERL_CORE) && defined(MULTIPLICITY) && \
 		(PERL_COMBI_VERSION < 5013006)
 #undef PL_sv_placeholder
