Index: Bio/Root/Build.pm
===================================================================
--- Bio/Root/Build.pm	(revision 16179)
+++ Bio/Root/Build.pm	(working copy)
@@ -209,6 +209,8 @@
         $final =~ s/^/bp_/ unless $final =~ /^bp/; # add the "bp" prefix
         $final = File::Spec->catfile($script_dir, $final);
         $self->log_info("$result -> $final\n");
+        unlink $final;
+        warn "[WARNING] Deleting '$final' failed!\n" if (-e $final);
         File::Copy::move($result, $final) or die "Can't rename '$result' to '$final': $!";
     }
 }
