--- 0.18/Utils.pod	2010-03-02 17:53:58.000000000 +0100
+++ HR/Utils.pod	2010-03-02 18:16:14.000000000 +0100
@@ -148,11 +148,11 @@
 
 =back
 
-The module it uses for lookup (L<Locales::Language>) is only required when this method is called. Make sure you have the latest verison of L<Locales> as 0.04 (i.e. L<Locales::Base> 0.03) is buggy!
+The module it uses for lookup (L<Locales>) is only required when this method is called. Make sure you have at least version 0.15 of L<Locales> as 0.04 (i.e. L<Locales::Base> 0.03) is buggy!
 
-The module it uses for lookup (L<Locales::Language>) is currently limited to two character codes but we try to handle it gracefully here.
+The module it uses for lookup (L<Locales>) is currently limited to two character codes but we try to handle it gracefully here.
 
-In array context it will build and return an additional hashref with the same keys whose values are the language name in the langueage itself.
+In array context it will build and return an additional hashref with the same keys whose values are the language name in the language itself.
 
 Does not ensure that the tags are loadable, to do that see below.
 
@@ -737,9 +737,9 @@
 
 =head1 SEE ALSO
 
-L<Locale::Maketext>, L<Locales::Language>, L<Locale::Maketext::Pseudo>
+L<Locale::Maketext>, L<Locales>, L<Locale::Maketext::Pseudo>
 
-If you use L</$lh->lang_names_hashref()> or L</$lh->loadable_lang_names_hashref()> make sure you have the latest verison of L<Locales> as 0.04 (i.e. L<Locales::Base> 0.03) is buggy!
+If you use L</$lh->lang_names_hashref()> or L</$lh->loadable_lang_names_hashref()> make sure you have at least version 0.15 of L<Locales> as 0.04 (i.e. L<Locales::Base> 0.03) is buggy!
 
 =head1 TODO
 
@@ -776,4 +776,4 @@
 it under the same terms as Perl itself, either Perl version 5.8.6 or,
 at your option, any later version of Perl 5 you may have available.
 
-=cut
\ Kein Zeilenumbruch am Dateiende.
+=cut
