Usage statement

From:  <>


---

 bin/rtldapimport |   11 +++++++++--
 1 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/bin/rtldapimport b/bin/rtldapimport
index 651110e..f4cd3fc 100644
--- a/bin/rtldapimport
+++ b/bin/rtldapimport
@@ -11,9 +11,16 @@ RT::Init;
 
 use RT::Extension::LDAPImport;
 
-my $debug;
+my ($debug, $help);
 use Getopt::Long;
-GetOptions ( "debug" => \$debug );
+GetOptions ( "debug" => \$debug, "help" => \$help );
+if ($help) {  
+   print "$0: [--debug] [--import] [--ldapdirs=file] [--help]\n";
+   print "   --help           This usage statement.\n";
+   print "   --debug          Enable debugging.\n";
+   exit 0;
+}   
+
 
 my $importer = RT::Extension::LDAPImport->new;
 $importer->screendebug(1) if $debug;
