Description: Fix spelling errors
Author: Fabrizio Regalli <fabreg@fabreg.it>
Forwarded: no

--- a/lib/Log/Report/Lexicon/Index.pod
+++ b/lib/Log/Report/Lexicon/Index.pod
@@ -71,7 +71,7 @@
 The EXTENSION filter can be used to reduce the filenames further, for
 instance to select only C<po> or only C<mo> files, and ignore readme's.
 Use an string, without dot and interpreted case-insensitive, or a
-regular expresion.
+regular expression.
 
 example: 
 
--- a/lib/Log/Report/Lexicon/PO.pod
+++ b/lib/Log/Report/Lexicon/PO.pod
@@ -29,7 +29,7 @@
 
 =item automatic => PARAGRAPH
 
-Automaticly added comments.
+Automatically added comments.
 See L<addAutomatic()|Log::Report::Lexicon::PO/"Attributes">.
 
 =item comment => PARAGRAPH
--- a/lib/Log/Report/Message.pod
+++ b/lib/Log/Report/Message.pod
@@ -70,8 +70,8 @@
 category of the message.
 
 One message can be part of multiple classes.  The STRING is used as
-comma- and/or blank seperated list of class tokens, the ARRAY lists all
-tokens seperately. See L<classes()|Log::Report::Message/"Accessors">.
+comma- and/or blank separated list of class tokens, the ARRAY lists all
+tokens separately. See L<classes()|Log::Report::Message/"Accessors">.
 
 =item _classes => STRING|ARRAY
 
