--- dumpvar_epic.pm.orig	2008-02-18 16:41:35.000000000 +0000
+++ dumpvar_epic.pm	2008-02-18 16:42:27.000000000 +0000
@@ -165,15 +165,16 @@
     }
     elsif (ref($tmp) eq 'SCALAR')
     {
-        if (defined($$tmp))
+        my $value = $$tmp; # Copy value to avoid modifying its representation
+        if (defined($value))
         {
-            if (length($$tmp) > MAX_SCALAR_LENGTH)
+            if (length($value) > MAX_SCALAR_LENGTH)
             {
-                $val = '\''.substr($$tmp, 0, MAX_SCALAR_LENGTH).'\'';
+                $val = '\''.substr($value, 0, MAX_SCALAR_LENGTH).'\'';
             }
             else
             {
-                $val = '\''.$$tmp.'\'';
+                $val = '\''.$value.'\'';
             }
         }
         else { $val = 'undef'; }
