--- 01live.t	2008-06-05 21:41:15.000000000 +0000
+++ 01live.new	2008-12-27 13:33:13.166097600 +0000
@@ -140,11 +140,12 @@
 
 sub run_server_tests {
     my $class = shift;
+    $PORT += 1 if $class eq 'SlowServer';
     my $s = $class->new($PORT);
     is($s->port(),$PORT,"Constructor set port correctly");
 
     my $pid=$s->background();
-    select(undef,undef,undef,0.2); # wait a sec
+    sleep 3; # wait a sec
 
     like($pid, '/^-?\d+$/', 'pid is numeric');
 
