Author: Dominik George <nik@naturalnet.de>
Description: Makes $tempdir defined in initial setup
Forwarded: no
--- a/rssdrop
+++ b/rssdrop
@@ -30,6 +30,10 @@ my @getopt =
 GetOptions(\%opt, @getopt) or do { print STDERR usage(); exit 1; };
 do { print usage(); exit 0; } if $opt{help};
 
+my $tempdir_xxx = sprintf '%s/tmp.%s.XXXXXX'
+  => $opt{prefix}, strftime('%F', localtime);
+my $tempdir = tempdir($tempdir_xxx);
+
 if (!-e "$opt{prefix}/config") {
   print STDERR "entering one-time setup...\n";
   setup();
@@ -38,9 +42,6 @@ if (!-e "$opt{prefix}/config") {
 my $config = read_config("$opt{prefix}/config");
 my $mdu = mdu_init();
 my $ua = LWP::UserAgent->new;
-my $tempdir_xxx = sprintf '%s/tmp.%s.XXXXXX'
-  => $opt{prefix}, strftime('%F', localtime);
-my $tempdir = tempdir($tempdir_xxx);
 
 $config->{mailfolder} = $opt{mailfolder} if $opt{mailfolder};
 
