--- /usr/lib/perl5/site_perl/5.10/SVN/Dumpfile.pm.orig	2009-04-22 15:01:02.763220400 -0400
+++ /usr/lib/perl5/site_perl/5.10/SVN/Dumpfile.pm	2009-04-22 15:03:25.669470400 -0400
@@ -273,6 +273,7 @@
     }
 
     if ( $self->{'SVN-fs-dump-format-version'} > 1 ) {
+        my $pos = $fh->tell;
         while ( ( $line = $fh->getline ) =~ /^$/ ) { }
         if ( $line =~ /^UUID: (.*)$/ ) {
             $self->{'UUID'} = $1;
@@ -280,6 +281,8 @@
         else {
             carp "Error: Dumpfile looks invalid. Couldn't find valid ",
                 "'UUID' header.\n";
+            # reset filehandle position
+            $fh->seek($pos, 0);
         }
 
         # read blank line after UUID:
