diff -ur libnet-3.02/lib/Net/Cmd.pm libnet-3.02+dataend-fix/lib/Net/Cmd.pm
--- libnet-3.02/lib/Net/Cmd.pm	2014-10-11 00:59:32.000000000 +0200
+++ libnet-3.02+dataend-fix/lib/Net/Cmd.pm	2014-11-21 17:30:57.000000000 +0100
@@ -564,19 +564,7 @@
 
   $tosend .= ".\015\012";
 
-  local $SIG{PIPE} = 'IGNORE' unless $^O eq 'MacOS';
-
-  $cmd->debug_print(1, ".\n")
-    if ($cmd->debug);
-
-  my $len = length $tosend;
-  my $w = syswrite($cmd, $tosend, $len);
-  unless (defined($w) && $w == $len)
-  {
-    $cmd->close;
-    $cmd->_set_status_closed;
-    return 0;
-  }
+  return 0 if !$cmd->rawdatasend($tosend);
 
   delete ${*$cmd}{'net_cmd_last_ch'};
 
