--- lib/ExtUtils/MM_Unix.pm	2004-05-30 22:41:33.785420688 +0200
+++ lib/ExtUtils/MM_Unix.pm	2004-05-30 22:41:16.901987360 +0200
@@ -2735,7 +2735,7 @@
 $(MAP_TARGET) :: static $(MAKE_APERL_FILE)
 	$(MAKE) -f $(MAKE_APERL_FILE) $@
 
-$(MAKE_APERL_FILE) : $(FIRST_MAKEFILE)
+$(MAKE_APERL_FILE) : $(FIRST_MAKEFILE) pm_to_blib
 	$(NOECHO) $(ECHO) Writing \"$(MAKE_APERL_FILE)\" for this $(MAP_TARGET)
 	$(NOECHO) $(PERLRUNINST) \
 		Makefile.PL DIR=}, $dir, q{ \
@@ -3473,7 +3473,7 @@
 all :: $target
 	\$(NOECHO) \$(NOOP)
 
-$target :: $plfile
+$target :: $plfile pm_to_blib
 	\$(PERLRUNINST) $plfile $target
 ";
 	}
