Description: fix a spelling mistake
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-08-25

--- a/README
+++ b/README
@@ -243,7 +243,7 @@
         You can also set aliases with the "SET_VAR" tag. See "SET_VAR"
 
         To use template parameters with a "$" at the beginning (which is not
-        officially suppported, but some are abviously using it), you can
+        officially supported, but some are abviously using it), you can
         set:
 
             local $HTML::Template::Compiled::Compiler::DISABLE_NEW_ALIAS = 1;
--- a/lib/HTML/Template/Compiled.pm
+++ b/lib/HTML/Template/Compiled.pm
@@ -1828,7 +1828,7 @@
 You can also set aliases with the C<SET_VAR> tag. See L<"SET_VAR">
 
 To use template parameters with a C<$> at the beginning (which is not
-officially suppported, but some are abviously using it), you can set:
+officially supported, but some are abviously using it), you can set:
 
     local $HTML::Template::Compiled::Compiler::DISABLE_NEW_ALIAS = 1;
 
