--- TCP.pm-orig	2009-07-13 19:39:37.000000000 -0800
+++ TCP.pm-mod	2009-07-13 19:44:57.000000000 -0800
@@ -910,8 +910,8 @@
 named callbacks.  Its syntax is that of POE:Session->create()'s
 inline_states parameter.
 
-Remember: These InlineStates handlers will be added to the main
-listening session, not to every connection.  A yield() in a connection
+Remember: These InlineStates handlers will be added to the client 
+sessions, not to the main listening session.  A yield() in the listener
 will not reach these handlers.
 
 If POE::Kernel::ASSERT_USAGE is enabled, the constructor will croak() if it
@@ -924,8 +924,8 @@
 and the events they will handle.  The arrayref must follow the syntax
 for POE::Session->create()'s object_states parameter.
 
-Remember: These ObjectStates handlers will be added to the main
-listening session, not to every connection.  A yield() in a connection
+Remember: These ObjectStates handlers will be added to the client 
+sessions, not to the main listening session.  A yield() in the listener
 will not reach these handlers.
 
 If POE::Kernel::ASSERT_USAGE is enabled, the constructor will croak() if it
@@ -939,8 +939,8 @@
 follow the syntax for POE::Session->create()'s package_states
 parameter.
 
-Remember: These PackageStates handlers will be added to the main
-listening session, not to every connection.  A yield() in a connection
+Remember: These PackageStates handlers will be added to the client 
+sessions, not to the main listening session.  A yield() in the listener
 will not reach these handlers.
 
 If POE::Kernel::ASSERT_USAGE is enabled, the constructor will croak() if it
