Description: fix spelling mistake and POD syntax issues
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-07-09

--- a/lib/Net/CLI/Interact/Transport/Net_OpenSSH.pm
+++ b/lib/Net/CLI/Interact/Transport/Net_OpenSSH.pm
@@ -78,6 +78,8 @@
 __END__
 =pod
 
+=encoding UTF-8
+
 =head1 NAME
 
 Net::CLI::Interact::Transport::Net_OpenSSH - Net::OpenSSH based CLI connection
@@ -96,7 +98,7 @@
 approach is not well suited (i.e. network equipment running custom
 administration shells) and still use the capability of Net::OpenSSH to
 run several sessions over one single SSH connection, including
-accesing SCP and SFTP services.
+accessing SCP and SFTP services.
 
 Note that this transport is not supported on Windows as Net::OpenSSH
 is not supported there either.
@@ -132,8 +134,6 @@
 The default is to pass nothing which on conforming SSH implementations
 starts the shell configured for the user.
 
-=back
-
 Examples:
 
   # interact with default user shell:
