--- libmail-gnupg-perl-0.08.orig/GnuPG.pm
+++ libmail-gnupg-perl-0.08/GnuPG.pm
@@ -338,6 +338,12 @@
   The message can either be in RFC compliant-ish multipart/signed
   format, or just a single part ascii armored message.
 
+  Note that MIME-encoded data should be supplied unmodified inside
+  the MIME::Entity input message, otherwise the signature will be 
+  broken. Since MIME-tools version 5.419, this can be achieved with
+  the C<decode_bodies> method of MIME::Parser. See the MIME::Parser
+  documentation for more information.
+
  Output:
   On error:
     Exit code of gpg.  (0 on success)
