--- lib/Date/Manip/TZ.pod.old	2015-06-01 10:31:50.000000000 -0300
+++ lib/Date/Manip/TZ.pod	2015-06-15 15:55:01.000000000 -0300
@@ -602,6 +602,11 @@
                              be the first non-blank non-comment
                              line in the file.
 
+   tzdata                    Determine which /usr/share/zoneinfo
+                             file /etc/localtime represents, and
+                             derive the time zone from the name
+                             of that file.
+
    command    COMMAND        Runs a command which produces
                              a time zone as the output.
 
@@ -634,6 +639,7 @@
    file     /etc/timezone
    file     /etc/sysconfig/clock
    file     /etc/default/init
+   tzinfo
    command  "/bin/date +%Z"
    command  "/usr/bin/date +%Z"
    command  "/usr/local/bin/date +%Z"
