--- Pro.xs.92	2009-11-09 22:10:56 -0500
+++ Pro.xs.92.mod	2009-11-10 20:53:40 -0500
@@ -573,12 +573,12 @@
   callback_state->force_untaint=get_integer_from_hash(aTHX_ SelfHash,"force_untaint");
   /* end setting callback_state */
 
-  if (get_integer_from_hash(aTHX_ SelfHash,"__use_perl_find_file")) {
+/*  if (get_integer_from_hash(aTHX_ SelfHash,"__use_perl_find_file")) { */
     tmplpro_set_option_FindFileFuncPtr(param, &get_filepath);
-  } else {
+/*  } else {
     tmplpro_set_option_path(param, get_array_of_strings_from_hash(aTHX_ SelfHash, "path", callback_state));
     tmplpro_set_option_FindFileFuncPtr(param, NULL);
-  }
+  } */
   return param;
 }
 
