--- a/t/1.t
+++ b/t/1.t
@@ -10,5 +10,5 @@
 use Math::Random::MT;
 ok($loaded = 1);
 ok(my $gen = Math::Random::MT->new(5489));
-ok(0.814723691903055, $gen->rand());
-ok(0.135477004107088, $gen->rand());
+ok(abs($gen->rand() - 0.814723691903055) < 1e-14);
+ok(abs($gen->rand() - 0.135477004107088) < 1e-14);
--- a/t/4.t
+++ b/t/4.t
@@ -13,5 +13,5 @@
 use Math::Random::MT qw(srand rand);
 ok($loaded = 1);
 srand(5489);
-ok(0.814723691903055, rand());
-ok(0.135477004107088, rand());
+ok(abs( 0.814723691903055 - rand()) < 1e-14);
+ok(abs( 0.135477004107088 - rand()) < 1e-14);
--- a/t/5.t
+++ b/t/5.t
@@ -14,4 +14,4 @@
 my $gen;
 ok($loaded = 1);
 ok( $gen = Math::Random::MT->new(1, 2, 3, 4) );
-ok( $gen->rand(1), 0.67886575916782 );
+ok(abs($gen->rand(1) - 0.67886575916782) < 1e-14);
