Description: fix spelling mistakes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-11-01

--- a/lib/XML/Compile/SOAP/Operation.pod
+++ b/lib/XML/Compile/SOAP/Operation.pod
@@ -42,7 +42,7 @@
 
 =item action => STRING
 
-Some string which is refering to the action which is taken.  For SOAP
+Some string which is referring to the action which is taken.  For SOAP
 protocols, this defines the soapAction header.
 
 =item endpoints => ADDRESS|ARRAY
--- a/lib/XML/Compile/SOAP11/Operation.pod
+++ b/lib/XML/Compile/SOAP11/Operation.pod
@@ -192,7 +192,7 @@
 type, as pure Perl HASH structure.  An 'request-response' operation
 will return then answer, or C<undef> in case of failure.  An 'one-way'
 operation with return C<undef> in case of failure, and a true value
-when successfull.
+when successful.
 
 You B<cannot> pass options for L<XML::Compile::Schema::compile()|XML::Compile::Schema/"Compilers">, like
 C<<sloppy_integers => 0>>, hooks or typemaps this way. Provide these to
