diff -Naur Net-LDAP-SimpleServer-0.0.17.orig/lib/Net/LDAP/SimpleServer/ProtocolHandler.pm Net-LDAP-SimpleServer-0.0.17/lib/Net/LDAP/SimpleServer/ProtocolHandler.pm
--- Net-LDAP-SimpleServer-0.0.17.orig/lib/Net/LDAP/SimpleServer/ProtocolHandler.pm	2014-12-10 09:26:19.440177243 -0800
+++ Net-LDAP-SimpleServer-0.0.17/lib/Net/LDAP/SimpleServer/ProtocolHandler.pm	2014-12-10 09:27:10.318335835 -0800
@@ -40,7 +40,7 @@
 sub new {
     my $class  = shift;
     my $params = shift || croak 'Must pass parameters!';
-    my $self   = $class->SUPER::new( $params->{input}, $params->{output} );
+    my $self   = $class->SUPER::new( $params->{sock} );
 
     croak 'Parameter must be a HASHREF' unless reftype($params) eq 'HASH';
     croak 'Must pass option {store}' unless exists $params->{store};
diff -Naur Net-LDAP-SimpleServer-0.0.17.orig/lib/Net/LDAP/SimpleServer.pm Net-LDAP-SimpleServer-0.0.17/lib/Net/LDAP/SimpleServer.pm
--- Net-LDAP-SimpleServer-0.0.17.orig/lib/Net/LDAP/SimpleServer.pm	2014-12-10 09:26:19.440177243 -0800
+++ Net-LDAP-SimpleServer-0.0.17/lib/Net/LDAP/SimpleServer.pm	2014-12-10 09:27:10.318335835 -0800
@@ -112,6 +112,7 @@
     }
     $params->{input}  = *STDIN{IO};
     $params->{output} = *STDOUT{IO};
+    $params->{sock} = $self->{server}->{client};
     my $handler = Net::LDAP::SimpleServer::ProtocolHandler->new($params);
 
     until ( $handler->handle ) {
