--- Image.pm.bk	2008-11-14 19:58:45.000000000 -0800
+++ Image.pm	2008-11-14 20:09:44.000000000 -0800
@@ -129,6 +129,12 @@
         $nw = $ow * ( $max_height / $oh );
     }
 
+    if (($nh <= $oh) && ($nw <= $ow)) {
+        #warn "Image does not need to be resized";
+        seek( $fh, 0, 0 );
+        return $fh;
+    }
+
     $result = $image->Resize( width => $nw, height => $nh, @the_rest );
     if ("$result") { # quotes are there as per the Image::Magick examples
         #warn "$result";
