diff -ruN SQL-Statement-1.32/Changes SQL-Statement-1.33/Changes
--- SQL-Statement-1.32/Changes	2011-01-19 00:09:19.000000000 -0600
+++ SQL-Statement-1.33/Changes	2011-02-02 17:07:48.000000000 -0600
@@ -1,5 +1,12 @@
 Changes log for Perl extension SQL::Statement
 
+Version 1.33, released February 1, 2001
+
+Remove circular dependency on DBD::AnyData as per rt 65041
+
+Also noted DBD::CSV was a circular dependency, added info to rt, 
+and addressed in this dev version
+
 Version 1.32, released January 19th, 2011
 ----------------------------------------------
 
diff -ruN SQL-Statement-1.32/META.yml SQL-Statement-1.33/META.yml
--- SQL-Statement-1.32/META.yml	2011-01-19 00:12:13.000000000 -0600
+++ SQL-Statement-1.33/META.yml	2011-02-02 17:08:34.000000000 -0600
@@ -1,6 +1,6 @@
 --- #YAML:1.0
 name:               SQL-Statement
-version:            1.32
+version:            1.33
 abstract:           SQL parsing and processing engine
 author:
     - Jeff Zucker <jeff@vpservices.com>, Jens Rehsack <rehsack@cpan.org>
@@ -9,8 +9,6 @@
 configure_requires:
     ExtUtils::MakeMaker:  0
 build_requires:
-    DBD::AnyData:         0.110
-    DBD::CSV:             0.30
     DBD::DBM:             0.06
     DBD::File:            0.40
     DBD::SQLite:          0
diff -ruN SQL-Statement-1.32/Makefile.PL SQL-Statement-1.33/Makefile.PL
--- SQL-Statement-1.32/Makefile.PL	2010-12-31 06:36:49.000000000 -0600
+++ SQL-Statement-1.33/Makefile.PL	2011-02-02 16:46:02.000000000 -0600
@@ -64,8 +64,9 @@
         # means: DBI >= 1.616
         'DBI::DBD::SqlEngine' => '0.03',
         'DBD::File'           => '0.40',
-        'DBD::AnyData'        => '0.110',
-        'DBD::CSV'            => '0.30',
+# rt65041 we only need this for tests
+#        'DBD::AnyData'        => '0.110',
+#        'DBD::CSV'            => '0.30',
         'DBD::DBM'            => '0.06',
         'DBD::SQLite'         => '0',
         'MLDBM'               => '0',
diff -ruN SQL-Statement-1.32/lib/SQL/Statement.pm SQL-Statement-1.33/lib/SQL/Statement.pm
--- SQL-Statement-1.32/lib/SQL/Statement.pm	2011-01-15 05:13:02.000000000 -0600
+++ SQL-Statement-1.33/lib/SQL/Statement.pm	2011-02-02 17:07:48.000000000 -0600
@@ -30,7 +30,7 @@
 
 #use locale;
 
-$VERSION = '1.32';
+$VERSION = '1.33';
 
 sub new
 {
