--- t/31sessions.old	2009-01-11 15:13:01.000000000 -0500
+++ t/31sessions.t	2009-03-02 09:50:39.000000000 -0500
@@ -277,7 +277,7 @@
 for my $si (0..$#SESSIONS) {
     my $session = $SESSIONS[$si];
     my $system = $session->{system} || $default_system;
-    # warn "# DEBUG: name[$session->{name}]system[$system]";
+    warn "# DEBUG: name[$session->{name}]system[$system]";
     ok($session->{name}, "opening new session $session->{name}");
     open SYSTEM, "| $system 2> $devnull" or die "Could not open '| $system': $!";
     for (my $i = 0; 2*$i < $#{$session->{pairs}}; $i++) {
@@ -295,6 +295,8 @@
         my($expect) = $session->{pairs}[2*$i+1];
         my($actual) = $chunks[$i+1];
         $actual =~ s{t\\00}{t/00}g if ($^O eq 'MSWin32');
+	# $VERBOSE = $command eq 'test CPAN::Test::Dummy::Perl5::Make::Features';
+	$VERBOSE = 1;
         diag("cmd[$command]expect[$expect]actual[$actual]") if $VERBOSE;
         my $success = like($actual,"/$expect/","cmd[$command]");
         if (!$success) {
